<div class="container">

<table style="width: 100%;"><tr>
<td>BRCA_Data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>BRCA TCGA Dataset</h2>

<h3>Description</h3>

<p>These data were obtained from the data freeze for The Cancer Genome Atlas flagship BRCA publication (Cancer Genome Atlas Network, 2013), and processed as described in Lock and Dunson, 2013.  Gene expression, methylation, and miRNA data are provided for 348 BRCA tumor samples.       
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(BRCA_data)</code></pre>


<h3>Format</h3>

<p>This dataset is a list of three entries for three different molecular sources:
</p>

<ul>
<li>
<p> Data[[1]] (Data$Expression): gene expression matrix for 654 genes (rows) and 348 samples (columns)
</p>
</li>
<li>
<p> Data[[2]] (Data$Methylation): DNA methylation matrix for 574 cg sites (rows) and 348 samples (columns)
</p>
</li>
<li>
<p> Data[[3]] (Data$miRNA): miRNA expression matrix for 423 cg sites (rows) and 348 samples (columns).
</p>
</li>
</ul>
<p>The 348 columns are shared by the data sources (here, they correspond to tumor samples)</p>


<h3>References</h3>

<p>Cancer Genome Atlas Network. 2012. ”Comprehensive Molecular Portraits of Human Breast Tumours.” Nature 490 (7418): 61-70.
</p>
<p>Lock, E.F. and Dunson, D.B. 2013. ”Bayesian Consensus Clustering.” Bioinformatics 29 (20): 2610-16.
</p>
<p>O'Connell, M. J., &amp; Lock, E.F. (2016). R.JIVE for Exploration of Multi-Source Molecular Data. Bioinformatics advance access: 10.1093/bioinformatics/btw324.
</p>


</div>