<div class="container">

<table style="width: 100%;"><tr>
<td>Ltree-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The Ltree Class</h2>

<h3>Description</h3>

<p>The Ltree class is the central object storing all information generated during lineage tree inference by the StemID algorithm.
It comprises a number of slots for a variety of objects.
</p>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>An Ltree object.</p>
</td>
</tr></table>
<h3>Slots</h3>


<dl>
<dt><code>sc</code></dt>
<dd>
<p>An <code>SCseq</code> object with the RaceID3 analysis of the single-cell RNA-seq data for which a lineage tree should be derived.</p>
</dd>
<dt><code>ldata</code></dt>
<dd>
<p>List object storing information on the clustering partition, the distance matrix, and the cluster centers in dimensionally-reduced
input space and in two-dimensional t-sne space. Elements:
<code>lp</code>: vector with the filtered partition into clusters after discarding clusters with cthr cells or less.
<code>pdi</code>:matrix with the coordinates of all cells in the embedded space. Clusters with <code>cthr</code> transcripts or less were discarded (see function <code>projcells</code>).
Rows are medoids and columns are coordinates.
<code>cn</code>: data.frame with the coordinates of the cluster medoids in the embedded space. Clusters with <code>cthr</code> transcripts or less were discarded.
Rows are medoids and columns are coordinates.
<code>m</code>: vector with the numbers of the clusters which survived the filtering.
<code>pdil</code>:    data.frame with coordinates of cells in the two-dimensional t-SNE representation computed by RaceID3. Clusters with <code>cthr</code> transcripts or less were
discarded. Rows are cells and columns are coordinates.
<code>cnl</code>:    data.frame with the coordinates of the cluster medoids in the two-dimensional t-SNE representation computed by RaceID3. Clusters with <code>cthr</code>
transcripts or less were discarded. Rows are medoids and columns are coordinates.</p>
</dd>
<dt><code>entropy</code></dt>
<dd>
<p>Vector with transcriptome entropy computed for each cell.</p>
</dd>
<dt><code>trproj</code></dt>
<dd>
<p>List containing two data.frames. Elements:
<code>res</code>:    data.frame with three columns for each cell. The first column <code>o</code> shows the cluster of a cell,
the second column <code>l</code> shows the cluster number for the link the cell is assigned to, and the third column <code>h</code> shows the projection as a fraction of the length
of the inter-cluster link. Parallel projections are positive, while anti-parallel projections are negative.
<code>rma</code>: data.frame with all projection coordinates for each cell. Rows are cells and columns are clusters. Projections are given as a fraction of the length of the
inter-cluster link. Parallel projections are positive, while anti-parallel projections are negative. The column corresponding to the originating cluster of a cell
shows <code>NA</code>.</p>
</dd>
<dt><code>par</code></dt>
<dd>
<p>List of parameters used for the StemID2 analysis.</p>
</dd>
<dt><code>prback</code></dt>
<dd>
<p>data.frame of the same structure as the <code>trproj$res</code>. In case randomizations are used to compute significant projections, the projections of all
<code>pdishuff</code> randomizations are appended to this data.frame and therefore the number of rows corresponds to the number of cells multiplied by <code>pdishuf</code>. See
function <code>projback</code>.</p>
</dd>
<dt><code>prbacka</code></dt>
<dd>
<p>data.frame reporting the aggregated results of the randomizations with four columns. Column <code>n</code> denotes the number of the randomization sample,
column <code>o</code> and <code>l</code> contain the numbers of the originating and the terminal cluster, respectively, for each inter-cluster link and column <code>count</code> shows
the number of cells assigned to this link in randomization sample <code>n</code>. The discrete distribution for the computation of the link p-value is given by the data
contained in this object (if <code>nmode=FALSE</code>).</p>
</dd>
<dt><code>ltcoord</code></dt>
<dd>
<p>Matrix storing projection coordinates of all cells in the two-dimensional t-SNE space, used for visualization.</p>
</dd>
<dt><code>prtree</code></dt>
<dd>
<p>List with two elements. The first element <code>l</code> stores a list with the projection coordinates for each link. The name of each element identifies the
link and is composed of two cluster numbers separated by a dot. The second element <code>n</code> is a list of the same structure and contains the cell names corresponding
to the projection coordinates stored in <code>l</code>.</p>
</dd>
<dt><code>cdata</code></dt>
<dd>
<p>list of data.frames, each with cluster ids as rows and columns:
<code>counts</code> data.frame indicating the number of cells on the links connecting the cluster of origin (rows) to other clusters (columns).
<code>counts.br</code> data.frame containing the cell counts on cluster connections averaged across the randomized background samples (if <code>nmode = FALSE</code>) or as derived
from sampling statistics (if <code>nmode = TRUE</code>).
<code>pv.e</code> matrix of enrichment p-values estimated from sampling statistics (if <code>nmode = TRUE</code>); entries are 0 if the observed number of cells on the respective
link exceeds the <code>(1 â€“ pethr)</code>-quantile of the randomized background distribution and 0.5 otherwise (if <code>nmode = FALSE</code>).
<code>pv.d</code> matrix of depletion p-values estimated from sampling statistics (if <code>nmode = TRUE</code>); entries are 0 if the observed number of cells on the respective
link is lower than the <code>pethr</code>-quantile of the randomized background distribution and 0.5 otherwise (if <code>nmode = FALSE</code>).
<code>pvn.e</code> matrix of enrichment p-values estimated from sampling statistics (if <code>nmode = TRUE</code>); 1- quantile, with the quantile estimated from the number of cells on a link as derived from the randomized background distribution (if <code>nmode = FALSE</code>). 
<code>pvn.d</code> matrix of depletion p-values estimated from sampling statistics (if <code>nmode = TRUE</code>); quantile estimated from the number of cells on a link as derived from the randomized background distribution (if <code>nmode = FALSE</code>).</p>
</dd>
</dl>
</div>