<div class="container">

<table style="width: 100%;"><tr>
<td>plotexpmap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Highlighting gene expression in a dimensional reduction representation</h2>

<h3>Description</h3>

<p>This functions highlights gene expression in a two-dimensional t-SNE map, UMAP, or a Fruchterman-Rheingold graph layout
of the singe-cell transcriptome data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotexpmap(
  object,
  g,
  n = NULL,
  logsc = FALSE,
  imputed = FALSE,
  fr = FALSE,
  um = FALSE,
  cells = NULL,
  cex = 0.5,
  map = TRUE,
  leg = TRUE,
  noise = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p><code>SCseq</code> class object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>g</code></td>
<td>
<p>Individual gene name or vector with a group of gene names corresponding to a subset of valid row names of the <code>ndata</code> slot
of the <code>SCseq</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>String of characters representing the title of the plot. Default is <code>NULL</code> and the first element of <code>g</code> is chosen.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logsc</code></td>
<td>
<p>logical. If <code>TRUE</code>, then gene expression values are log2-transformed after adding a pseudo-count of 0.1. Default is <code>FALSE</code>
and untransformed values are shown.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>imputed</code></td>
<td>
<p>logical. If <code>TRUE</code> and imputing was done by calling <code>compdist</code> with <code>knn &gt; 0</code>, then imputed expression values are shown. If <code>FALSE</code>, then raw counts are shown. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fr</code></td>
<td>
<p>logical. If <code>TRUE</code> then plot Fruchterman-Rheingold layout. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>um</code></td>
<td>
<p>logical. If <code>TRUE</code> then plot umap dimensional reduction representation. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cells</code></td>
<td>
<p>Vector of valid cell names corresponding to column names of slot <code>ndata</code> of the <code>SCseq</code> object. Gene expression is ony shown for
this subset.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex</code></td>
<td>
<p>size of data points. Default value is 0.5.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>logical. If <code>TRUE</code> then data points are shown. Default value is <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>leg</code></td>
<td>
<p>logical. If <code>TRUE</code> then the legend is shown. Default value is <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>noise</code></td>
<td>
<p>logical. If <code>TRUE</code> then display local gene expression variability instead of gene expression (requires VarID analysis)/ Default value is <code>FALSE</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>None
</p>


</div>