<div class="container">

<table style="width: 100%;"><tr>
<td>SCseq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The SCseq Class</h2>

<h3>Description</h3>

<p>The SCseq class is the central object storing all information generated during cell type identification with the RaceID3 algorithm.
It comprises a number of slots for a variety of objects.
</p>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>An SCseq object.</p>
</td>
</tr></table>
<h3>Slots</h3>


<dl>
<dt><code>expdata</code></dt>
<dd>
<p>The raw expression data matrix with cells as columns and genes as rows in sparse matrix format.</p>
</dd>
<dt><code>ndata</code></dt>
<dd>
<p>Filtered data with expression normalized to one for each cell.</p>
</dd>
<dt><code>noise</code></dt>
<dd>
<p>Matrix with local gene expression noise estimates (used for VarID analysis)</p>
</dd>
<dt><code>counts</code></dt>
<dd>
<p>Vector with total transcript counts for each cell in <code>ndata</code> remaining after filtering.</p>
</dd>
<dt><code>genes</code></dt>
<dd>
<p>Vector with gene names of all genes in <code>ndata</code> remaining after filtering.</p>
</dd>
<dt><code>dimRed</code></dt>
<dd>
<p>list object object storing information on a feature matrix obtained by dimensional reduction, batch effect correction etc.
Component <code>x</code> stores the actual feature matrix.</p>
</dd>
<dt><code>distances</code></dt>
<dd>
<p>distance (or dis-similarity) matrix computed by RaceID3.</p>
</dd>
<dt><code>imputed</code></dt>
<dd>
<p>list with two matrices computed for imputing gene expression. The first matrix <code>nn</code> contains the cell indices of the <code>knn</code> nearest neighbours,
the second matrix contains the probabilities at which each cell contributes to thye imputed gene expression value for the cell correponding to the columns.</p>
</dd>
<dt><code>tsne</code></dt>
<dd>
<p>data.frame with coordinates of two-dimensional tsne layout computed by RaceID3.</p>
</dd>
<dt><code>fr</code></dt>
<dd>
<p>data.frame with coordinates of two-dimensional Fruchterman-Rheingold graphlayout computed by RaceID3.</p>
</dd>
<dt><code>umap</code></dt>
<dd>
<p>data.frame with coordinates of two-dimensional umap representation computed by RaceID3.</p>
</dd>
<dt><code>cluster</code></dt>
<dd>
<p>list storing information on the initial clustering step of the RaceID3 algorithm</p>
</dd>
<dt><code>background</code></dt>
<dd>
<p>list storing the polynomial fit for the background model of gene expression variability computed by RaceID3,
which is used for outlier identification.</p>
</dd>
<dt><code>out</code></dt>
<dd>
<p>list storing information on outlier cells used for the prediction of rare cell types by RaceID3</p>
</dd>
<dt><code>cpart</code></dt>
<dd>
<p>vector containing the final clustering (i.e. cell type) partition computed by RaceID3</p>
</dd>
<dt><code>fcol</code></dt>
<dd>
<p>vector contaning the colour scheme for the RaceID3 clusters</p>
</dd>
<dt><code>medoids</code></dt>
<dd>
<p>vector containing the cell ids for th cluster medoids</p>
</dd>
<dt><code>filterpar</code></dt>
<dd>
<p>list containing the parameters used for cell and gene filterung</p>
</dd>
<dt><code>clusterpar</code></dt>
<dd>
<p>list containing the parameters used for clustering</p>
</dd>
<dt><code>outlierpar</code></dt>
<dd>
<p>list containing the parameters used for outlier identification</p>
</dd>
</dl>
</div>