<div class="container">

<table style="width: 100%;"><tr>
<td>clustdiffgenes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Inference of differentially expressed genes in a cluster</h2>

<h3>Description</h3>

<p>This functions computes differentially expressed genes in a (set of) cluster(s) by comparing to all remaining cells outside of the cluster (or a given background set of clusters) based on a negative binomial
model of gene expression
</p>


<h3>Usage</h3>

<pre><code class="language-R">clustdiffgenes(object, cl, bgr = NULL, pvalue = 0.01)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p><code>SCseq</code> class object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cl</code></td>
<td>
<p>A valid set of cluster numbers from the final cluster partition stored in the <code>cpart</code> slot of the <code>SCseq</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bgr</code></td>
<td>
<p>Ordered vector of cluster numbers to be used as background set. If <code>NULL</code> then all clusters not in <code>cl</code> are used as background set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvalue</code></td>
<td>
<p>Positive real number smaller than one. This is the p-value cutoff for the inference of differential gene expression. Default is 0.01.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of two components. The first component <code>dg</code> contains a a data.frame of differentially expressed genes ordered by p-value in increasing order, with four columns:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>mean.ncl</code></td>
<td>
<p>mean expression across cells outside of cluster <code>cl</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mean.cl</code></td>
<td>
<p>mean expression across cells within cluster <code>cl</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fc</code></td>
<td>
<p>fold-change of mean expression in cluster <code>cl</code> versus the remaining cells.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pv</code></td>
<td>
<p>inferred p-value for differential expression.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>padj</code></td>
<td>
<p>Benjamini-Hochberg corrected FDR.</p>
</td>
</tr>
</table>
<p>The second component <code>de</code> contains the conventional output of <code>diffexpnb</code>, where set B corresponds to all clusters in <code>cl</code> and B to the background set (all clusters in <code>bgr</code> or not in <code>cl</code>). This component can be used for plotting by <code>plotdiffgenesnb</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">sc &lt;- SCseq(intestinalDataSmall)
sc &lt;- filterdata(sc)
sc &lt;- compdist(sc)
sc &lt;- clustexp(sc)
sc &lt;- findoutliers(sc)
x &lt;- clustdiffgenes(sc,1)
head(x$dg[x$dg$fc&gt;1,])
</code></pre>


</div>