<div class="container">

<table style="width: 100%;"><tr>
<td>fitBackVar</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function for computing a background model of gene expression variability</h2>

<h3>Description</h3>

<p>This funtion fits a second order polynomial to the variance-mean dependence across all genes in log space.
</p>


<h3>Usage</h3>

<pre><code class="language-R">fitBackVar(x, mthr = -1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Matrix of transcript counts with genes as rows and cells as columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mthr</code></td>
<td>
<p>Real number. Threshold of log2 mean expression. Genes with mean expression <code>&lt; mthr</code> are discarded prior to fitting the polynomial. Default is -1.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>List object of four components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>fit</code></td>
<td>
<p>model fit as returned by the <code>lm</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genes</code></td>
<td>
<p>genes with expression variance greater than the polynomial fit.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>
<p>mean expression of all genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v</code></td>
<td>
<p>expression variance of all genes</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">bg &lt;- fitBackVar(intestinalDataSmall)
</code></pre>


</div>