<div class="container">

<table style="width: 100%;"><tr>
<td>compTBNoise</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function for fitting a negative binomial noise model of technical and biological variability across cells in pruned k-nearest neighbourhoods.</h2>

<h3>Description</h3>

<p>This function fits negative binomial models to transcript counts of pruned k-nearest neighbourhoods inferred by <code>pruneKnn</code> thereby deconvoluting variability into sampling noise, global cell-to-cell variability of transcript counts, and residual variability, which corresponds to biological noise.
</p>


<h3>Usage</h3>

<pre><code class="language-R">compTBNoise(
  res,
  expData,
  pvalue = 0.01,
  genes = NULL,
  minN = 5,
  no_cores = NULL,
  gamma = 0.5,
  x0 = 0,
  lower = 0,
  upper = 100
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>res</code></td>
<td>
<p>List object with k nearest neighbour information returned by <code>pruneKnn</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>expData</code></td>
<td>
<p>Matrix of gene expression values with genes as rows and cells as columns. These values have to correspond to unique molecular identifier counts.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvalue</code></td>
<td>
<p>Positive real number between 0 and 1. All nearest neighbours with link probability <code>&lt; pvalue</code> are discarded. Default is 0.01.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genes</code></td>
<td>
<p>Vector of gene names corresponding to a subset of rownames of <code>expData</code>. Only for these genes local gene expression variability is computed. Default is <code>NULL</code> and values for all genes are returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minN</code></td>
<td>
<p>Positive integer number. Noise inference is only done for k-nearest neighbourhoods with at least <code>minN</code> neighbours remaining after pruning.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>no_cores</code></td>
<td>
<p>Positive integer number. Number of cores for multithreading. If set to <code>NULL</code> then the number of available cores minus two is used. Default is <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gamma</code></td>
<td>
<p>Positive real number. Scale paramter of the cauchy prior. Default is 0.5.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x0</code></td>
<td>
<p>Real number greater or equal to zero. Location parameter of the cauchy prior.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lower</code></td>
<td>
<p>Real number greater or equal to zero. Lower bound for the maximum a posterior inference of the biological noise. Default is 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>upper</code></td>
<td>
<p>Real number greater or equal to zero. Upper bound for the maximum a posterior inference of the biological noise. Default is 100.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>List object of three components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>mu</code></td>
<td>
<p>Vector of mean expression for all k-nearest neighbourhoods. Componenets are set to <code>NA</code> if less than <code>minN</code> neighbours are present in pruned neighbourhood. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rt</code></td>
<td>
<p>Vector of dispersion parameters capturing global cell-to-cell variability of transcript counts for all k-nearest neighbourhoods. Componenets are set to <code>NA</code> if less than <code>minN</code> neighbours are present in pruned neighbourhood.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>epsilon</code></td>
<td>
<p>Matrix of biological noise estimates for all genes across for all k-nearest neighbourhoods. Componenets are set to <code>NA</code> if less than <code>minN</code> neighbours present in pruned neighbourhood.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pars</code></td>
<td>
<p>List of parameters.</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
res &lt;- pruneKnn(intestinalDataSmall,knn=10,alpha=1,no_cores=1,FSelect=FALSE)
noise &lt;- compTBNoise(res,intestinalDataSmall,pvalue=0.01,genes = NULL,no_cores=1)

## End(Not run)
</code></pre>


</div>