<div class="container">

<table style="width: 100%;"><tr>
<td>extractCounts</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function for filtering count data</h2>

<h3>Description</h3>

<p>This function discards lowly expressed genes from a count matrix stored in an <code>SCseq</code> object, and returns (normalized or non-normalized) gene expression or noise values.
</p>


<h3>Usage</h3>

<pre><code class="language-R">extractCounts(
  object,
  minexpr = 5,
  minnumber = 5,
  noise = FALSE,
  pt = NULL,
  n = NULL,
  g = NULL,
  norm = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p><code>SCseq</code> class object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minexpr</code></td>
<td>
<p>Integer number greater or equal to zero. Minimum expression of a gene in at least <code>minnumber</code> cells to not be discarded. Default is 5.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minnumber</code></td>
<td>
<p>Integer number greater or equal to zero. Minimum number of cells required to have at least <code>minexpr</code> transcript counts for a gene to not be discarded. Default is 5.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>noise</code></td>
<td>
<p>logical. If <code>TRUE</code>, then noise (in <code>object@noise</code>) is returned for the filtered genes and cells. Default is <code>FALSE</code> and gene expression counts are returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pt</code></td>
<td>
<p>List object returned by function <code>pseudoTime</code>. If given, then feature matrix is returned for cells in <code>pt$ord</code> and ordered by pseudo-time. Default is NULL and feature matrix is returned for all cells in <code>object$ndata</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>Vector of valid column names corresponding to a subset of valid column names of the <code>object@ndata</code>. Default is <code>NULL</code> filtering is done on all cells in <code>object@ndata</code>. Only considered if <code>pt</code> is NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>g</code></td>
<td>
<p>Vector of gene IDs (valid row names of <code>object@ndata</code>). If given, then all genes not in <code>g</code> are discarded prior to filtering. Default is NULL and filtering is done on all genes in <code>object@ndata</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>norm</code></td>
<td>
<p>logical. If <code>TRUE</code>, then transcipt counts are normalized to the minimum number of total transcript counts across all cells in the feature matrix.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Filtered expression matrix.
</p>


</div>