<div class="container">

<table style="width: 100%;"><tr>
<td>testPrior</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Posterior check of the model</h2>

<h3>Description</h3>

<p>This functions compares variance estimates obtained from the maximum a posterior estimate with a given prior to the data. The ratio between the predicted variance and the actual variance for a random subset of genes is computed across all pruned k nearest neighbourhoods.
</p>


<h3>Usage</h3>

<pre><code class="language-R">testPrior(
  res,
  expData,
  gamma = c(0.2, 0.5, 1, 5, 1000),
  rseed = 12345,
  ngenes = 200,
  pvalue = 0.01,
  minN = 5,
  no_cores = NULL,
  x0 = 0,
  lower = 0,
  upper = 100
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>res</code></td>
<td>
<p>List object with k nearest neighbour information returned by <code>pruneKnn</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>expData</code></td>
<td>
<p>Matrix of gene expression values with genes as rows and cells as columns. These values have to correspond to unique molecular identifier counts.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gamma</code></td>
<td>
<p>Vector of <code>gamma</code>-values to test for the Cauchy prior distribution. Default is <code>c(0.2,0.5,1,5,1000)</code>. Large values correspond to weak priors (<code>gamma=1000</code> corresponds to a maximum likelihood estimate).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rseed</code></td>
<td>
<p>Integer number. Random seed to enforce reproducible gene sampling. Default is 12345.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ngenes</code></td>
<td>
<p>Positive integer number. Randomly sampled number of genes (from rownames of <code>expData</code>) used for noise estimation. Genes are sampled uniformly across the entire expression range. Default is 200.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvalue</code></td>
<td>
<p>Input parameter for <code>compTBNoise</code>. See <code>help(compTBNoise)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minN</code></td>
<td>
<p>Input parameter for <code>compTBNoise</code>. See <code>help(compTBNoise)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>no_cores</code></td>
<td>
<p>Input parameter for <code>compTBNoise</code>. See <code>help(compTBNoise)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x0</code></td>
<td>
<p>Input parameter for <code>compTBNoise</code>. See <code>help(compTBNoise)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lower</code></td>
<td>
<p>Input parameter for <code>compTBNoise</code>. See <code>help(compTBNoise)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>upper</code></td>
<td>
<p>Input parameter for <code>compTBNoise</code>. See <code>help(compTBNoise)</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>List of three components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>pp.var.ratio</code></td>
<td>
<p>List of vectors for each gamma value of ratios between predicted and actual variances across all sampled genes and neighbourhoods.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>noise</code></td>
<td>
<p>List of noise objects obtained from <code>compTBNoise</code> for each gamma value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tc</code></td>
<td>
<p>Vector of total transcript counts for all cells</p>
</td>
</tr>
</table>
</div>