<div class="container">

<table style="width: 100%;"><tr>
<td>modify.data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function to modify genotype and phenotype data to match</h2>

<h3>Description</h3>

<p>Function to modify genotype and phenotype data to match
</p>


<h3>Usage</h3>

<pre><code class="language-R">modify.data(
  pheno.mat,
  geno.mat,
  pheno.labels = NULL,
  geno.names = NULL,
  map = NULL,
  return.ZETA = TRUE,
  return.GWAS.format = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pheno.mat</code></td>
<td>
<p>A <code class="reqn">n _ 1 \times p</code> matrix of phenotype data. rownames(pheno.mat) should be genotype (line; accesion; variety) names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno.mat</code></td>
<td>
<p>A <code class="reqn">n _ 2 \times m</code> matrix of marker genotype data. rownames(geno.mat) should be genotype (line; accesion; variety) names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno.labels</code></td>
<td>
<p>A vector of genotype (line; accesion; variety) names which correpond to phenotypic values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno.names</code></td>
<td>
<p>A vector of genotype (line; accesion; variety) names for marker genotype data (duplication is not recommended).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>Data frame with the marker names in the first column. The second and third columns contain the chromosome and map position.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.ZETA</code></td>
<td>
<p>If this argument is TRUE, the list for mixed model equation (ZETA) will be returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.GWAS.format</code></td>
<td>
<p>If this argument is TRUE, phenotype and genotype data for GWAS will be returned.</p>
</td>
</tr>
</table>
<h3>Value</h3>


<dl>
<dt>$geno.modi</dt>
<dd>
<p>The modified marker genotype data.</p>
</dd>
<dt>$pheno.modi</dt>
<dd>
<p>The modified phenotype data.</p>
</dd>
<dt>$ZETA</dt>
<dd>
<p>The list for mixed model equation (ZETA).</p>
</dd>
<dt>$pheno.GWAS</dt>
<dd>
<p>GWAS formatted phenotype data.</p>
</dd>
<dt>$geno.GWAS</dt>
<dd>
<p>GWAS formatted marker genotype data.</p>
</dd>
</dl>
</div>