<div class="container">

<table style="width: 100%;"><tr>
<td>CalcThreshold</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function to calculate threshold for GWAS</h2>

<h3>Description</h3>

<p>Calculate thresholds for the given GWAS (genome-wide association studies) result by the Benjamini-Hochberg method or Bonferroni method.
</p>


<h3>Usage</h3>

<pre><code class="language-R">CalcThreshold(input, sig.level = 0.05, method = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>input</code></td>
<td>
<p>Data frame of GWAS results where the first column is the marker names,
the second and third column is the chromosome amd map position, and the forth column is -log10(p) for each marker.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sig.level</code></td>
<td>
<p>Significance level for the threshold. The default is 0.05. You can also assign vector of sinificance levels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Three methods are offered:
</p>
<p>"BH": Benjamini-Hochberg method. To control FDR, use this method.
</p>
<p>"Bonf": Bonferroni method. To perform simple correction of multiple testing, use this method.
</p>
<p>"Sidak": Sidak method.
</p>
<p>You can also assign two of them by 'method = c("BH", "Bonf")'</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The value of the threshold. If there is no threshold, it returns NA.
</p>


<h3>References</h3>

<p>Benjamini, Y. and Hochberg, Y. (1995) Controlling the false discovery rate:
a practical and powerful approach to multiple testing. J R Stat Soc. 57(1): 289-300.
</p>
<p>Storey, J.D. and Tibshirani, R. (2003) Statistical significance for genomewide studies. Proc Natl Acad Sci. 100(16): 9440-9445.
</p>


</div>