<div class="container">

<table style="width: 100%;"><tr>
<td>genetrait</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate pseudo phenotypic values</h2>

<h3>Description</h3>

<p>This function generates pseudo phenotypic values according to the following formula.
</p>
<p style="text-align: center;"><code class="reqn">y = X \beta + Z u + e</code>
</p>

<p>where effects of major genes are regarded as fixed effects <code class="reqn">\beta</code> and
polygenetic effects are regarded as random effects <code class="reqn">u</code>.
The variances of <code class="reqn">u</code> and <code class="reqn">e</code> are automatically determined by the heritability.
</p>


<h3>Usage</h3>

<pre><code class="language-R">genetrait(
  x,
  sample.sets = NULL,
  candidate = NULL,
  pos = NULL,
  x.par = NULL,
  ZETA = NULL,
  x2 = NULL,
  num.qtn = 3,
  weight = c(2, 1, 1),
  qtn.effect = rep("A", num.qtn),
  prop = 1,
  polygene.weight = 1,
  polygene = TRUE,
  h2 = 0.6,
  h.correction = FALSE,
  seed = NULL,
  plot = TRUE,
  saveAt = NULL,
  subpop = NULL,
  return.all = FALSE,
  seed.env = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A <code class="reqn">n \times m</code> genotype matrix where <code class="reqn">n</code> is sample size and <code class="reqn">m</code> is the number of markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.sets</code></td>
<td>
<p>A n.sample x n.mark genotype matrix. Markers with fixed effects (QTNs) are chosen from sample.sets.
If sample.sets = NULL, sample.sets = x.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>candidate</code></td>
<td>
<p>If you want to fix QTN postitions, please set the number where SNPs to be fixed are located in your data (so not position).
If candidate = NULL, QTNs were randomly sampled from sample.sets or x.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>
<p>A n.mark x 1 vector. Cumulative position (over chromosomes) of each marker.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x.par</code></td>
<td>
<p>If you don't want to match the sampling population and the genotype data to QTN effects, then use this argument as the latter.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ZETA</code></td>
<td>
<p>A list of covariance (relationship) matrix (K: <code class="reqn">m \times m</code>) and its design matrix (Z: <code class="reqn">n \times m</code>) of random effects.
Please set names of list "Z" and "K"! You can use more than one kernel matrix.
For example,
</p>
<p>ZETA = list(A = list(Z = Z.A, K = K.A), D = list(Z = Z.D, K = K.D))
</p>

<dl>
<dt>Z.A, Z.D</dt>
<dd>
<p>Design matrix (<code class="reqn">n \times m</code>) for the random effects. So, in many cases, you can use the identity matrix.</p>
</dd>
<dt>K.A, K.D</dt>
<dd>
<p>Different kernels which express some relationships between lines.</p>
</dd>
</dl>
<p>For example, K.A is additive relationship matrix for the covariance between lines, and K.D is dominance relationship matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x2</code></td>
<td>
<p>A genotype matrix to calculate additive relationship matrix when Z.ETA = NULL.
If Z.ETA = NULL &amp; x2 = NULL, calcGRM(x) will be calculated as kernel matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.qtn</code></td>
<td>
<p>The number of QTNs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weight</code></td>
<td>
<p>The weights for each QTN by their standard deviations. Negative value is also allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>qtn.effect</code></td>
<td>
<p>Additive of dominance for each marker effect. This argument should be the same length as num.qtn.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prop</code></td>
<td>
<p>The proportion of effects of QTNs to polygenetic effects.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>polygene.weight</code></td>
<td>
<p>If there are multiple kernels, this argument determines the weights of each kernel effect.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>polygene</code></td>
<td>
<p>If polygene = FALSE, pseudo phenotypes with only QTN effects will be generated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>h2</code></td>
<td>
<p>The wide-sense heritability for generating phenotypes. 0 &lt;= h2 &lt; 1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>h.correction</code></td>
<td>
<p>If TRUE, this function will generate phenotypes to match the genomic heritability and "h2".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>If seed is not NULL, some fixed phenotypic values will be generated according to set.seed(seed)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>If TRUE, boxplot for generated phenotypic values will be drawn.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>saveAt</code></td>
<td>
<p>When drawing any plot, you can save plots in png format. In saveAt, you should substitute the name you want to save.
When saveAt = NULL, the plot is not saved.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subpop</code></td>
<td>
<p>If there is subpopulation structure, you can draw boxpots divide by subpopulations.
n.sample x n.subpop matrix. Please indicate the subpopulation information by (0, 1) for each element.
(0 means that line doen't belong to that subpopulation, and 1 means that line belongs to that subpopulation)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.all</code></td>
<td>
<p>If FALSE, only returns generated phenotypic values.
If TRUE, this function will return other information such as positions of candidate QTNs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed.env</code></td>
<td>
<p>If TRUE, this function will generate different environment effects every time.</p>
</td>
</tr>
</table>
<h3>Value</h3>


<dl>
<dt>trait</dt>
<dd>
<p>Generated phenotypic values</p>
</dd>
<dt>u</dt>
<dd>
<p>Generated genotyope values</p>
</dd>
<dt>e</dt>
<dd>
<p>Generated environmental effects</p>
</dd>
<dt>candidate</dt>
<dd>
<p>The numbers where QTNs are located in your data (so not position).</p>
</dd>
<dt>qtn.position</dt>
<dd>
<p>QTN positions</p>
</dd>
<dt>heritability</dt>
<dd>
<p>Genomic heritability for generated phenotypic values.</p>
</dd>
</dl>
</div>