<div class="container">

<table style="width: 100%;"><tr>
<td>calcGRM</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function to calculate genomic relationship matrix (GRM)</h2>

<h3>Description</h3>

<p>Function to calculate genomic relationship matrix (GRM)
</p>


<h3>Usage</h3>

<pre><code class="language-R">calcGRM(
  genoMat,
  methodGRM = "addNOIA",
  subpop = NULL,
  kernel.h = "tuned",
  returnWMat = FALSE,
  probaa = NULL,
  probAa = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genoMat</code></td>
<td>
<p>A <code class="reqn">N \times M</code> matrix of marker genotype</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>methodGRM</code></td>
<td>
<p>Method to calculate genomic relationship matrix (GRM). We offer the following methods;
"addNOIA", "domNOIA", "A.mat", "linear", "gaussian", "exponential", "correlation".
For NOIA methods, please refer to Vitezica et al. 2017.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subpop</code></td>
<td>
<p>Sub-population names corresponding to each individual.
By utilizing 'subpop' argument, you can consider the difference of allele frequencies
between sub-populations when computing the genomic relationship matrix. This argument is only valid when NOIA methods are selected.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kernel.h</code></td>
<td>
<p>The hyper parameter for gaussian or exponential kernel.
If kernel.h = "tuned", this hyper parameter is calculated as the median of off-diagonals of distance matrix of genotype data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>returnWMat</code></td>
<td>
<p>If this argument is TRUE, we will return W matrix instead of GRM.
Here, W satisfies <code class="reqn">GRM = W W ^ {T}</code>. W corresponds to H matix in Vitezica et al. 2017.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>probaa</code></td>
<td>
<p>Probability of being homozygous for the reference allele for each marker.
If NULL (default), it will be calculated from genoMat.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>probAa</code></td>
<td>
<p>Probability of being heterozygous for the reference and alternative alleles for each marker
If NULL (default), it will be calculated from genoMat.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>genomic relationship matrix (GRM)
</p>


<h3>References</h3>

<p>Vitezica, Z.G., Legarra, A., Toro, M.A. and Varona, L. (2017) Orthogonal Estimates of Variances for Additive, Dominance, and Epistatic Effects in Populations. Genetics. 206(3): 1297-1307.
</p>
<p>Endelman, J.B. and Jannink, J.L. (2012) Shrinkage Estimation of the Realized Relationship Matrix. G3 Genes, Genomes, Genet. 2(11): 1405-1413.
</p>


</div>