<div class="container">

<table style="width: 100%;"><tr>
<td>adjustGRM</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function to adjust genomic relationship matrix (GRM) with subpopulations</h2>

<h3>Description</h3>

<p>Function to adjust genomic relationship matrix (GRM) with subpopulations
</p>


<h3>Usage</h3>

<pre><code class="language-R">adjustGRM(
  y,
  X = NULL,
  ZETA,
  subpopInfo = NULL,
  nSubpop = 5,
  nPcsFindCluster = 10,
  include.epistasis = FALSE,
  package.MM = "gaston"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>A <code class="reqn">n \times 1</code> vector. A vector of phenotypic values should be used. NA is allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>A <code class="reqn">n \times p</code> matrix. You should assign mean vector (rep(1, n)) and covariates. NA is not allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ZETA</code></td>
<td>
<p>A list of variance matrices and its design matrices of random effects.
You can use only one kernel matrix for this function.
For example, ZETA = list(A = list(Z = Z.A, K = K.A)) (A for additive)
Please set names of lists "Z" and "K"!</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subpopInfo</code></td>
<td>
<p>The information on group memberships (e.g., subgroups for the population) will be required.
You can set a vector of group names (or clustering ids) for each genotype as this argument. This vector should be factor.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nSubpop</code></td>
<td>
<p>When 'subpopInfo = NULL', 'subpopInfo' will be automatically determined by using <code>find.clusters</code> function.
You should specify the number of groups by this argument to decide 'subpopInfo'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nPcsFindCluster</code></td>
<td>
<p>Number of principal components to be used for 'adegenet::find.clusters'.
This argument is used inly when 'subpopInfo' is 'NULL'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>include.epistasis</code></td>
<td>
<p>Whether or not including the genome-wide epistastic effects into the model
to adjust ZETA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>package.MM</code></td>
<td>
<p>The package name to be used when solving mixed-effects model. We only offer the following three packages:
"RAINBOWR", "MM4LMM" and "gaston". Default package is 'gaston'.
See more details at <code>EM3.general</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p> A List of
</p>
<dl>
<dt>$ZETAAdjust</dt>
<dd>
<p>Adjusted ZETA including only one kernel.</p>
</dd>
<dt>$subpopInfo</dt>
<dd>
<p>A vector of 'subpopInfo' used in this function.</p>
</dd>
<dt>$covariates</dt>
<dd>
<p>A matrix of covariates used in the mixed effects model.</p>
</dd>
</dl>
<p>#' </p>
<dl>
<dt>$nullModel</dt>
<dd>
<p>Results of mixed-effects model for multiple kernels.</p>
</dd>
<dt>$nSubpop</dt>
<dd>
<p>'nSubpop' used in this function.</p>
</dd>
<dt>$include.epistasis</dt>
<dd>
<p>'include.epistasis' used in this function.</p>
</dd>
</dl>
<h3>References</h3>

<p>Rio S, Mary-Huard T, Moreau L, Bauland C, Palaffre C, et al. (2020)
Disentangling group specific QTL allele effects from
genetic background epistasis using admixed individuals in GWAS:
An application to maize flowering. PLOS Genetics 16(3): e1008241.
</p>


</div>