<div class="container">

<table style="width: 100%;"><tr>
<td>plotHaploNetwork</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function to plot haplotype network from the estimated results</h2>

<h3>Description</h3>

<p>Function to plot haplotype network from the estimated results
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotHaploNetwork(
  estNetworkRes,
  traitName = NULL,
  blockName = NULL,
  plotNetwork = TRUE,
  subpopInfo = estNetworkRes$subpopInfo,
  saveName = NULL,
  saveStyle = "png",
  plotWhichMDS = 1:2,
  colConnection = c("grey40", "grey60"),
  ltyConnection = c("solid", "dashed"),
  lwdConnection = c(1.5, 0.8),
  pchBase = c(1, 16),
  colCompBase = c(2, 4),
  colHaploBase = c(3, 5, 6),
  cexMax = 2,
  cexMin = 0.7,
  ggPlotNetwork = FALSE,
  cexMaxForGG = 0.025,
  cexMinForGG = 0.008,
  alphaBase = c(0.9, 0.3)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>estNetworkRes</code></td>
<td>
<p>The estimated results of haplotype network by 'estNetwork' function for one</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>traitName</code></td>
<td>
<p>Name of trait of interest. This will be used in the title of the plots.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blockName</code></td>
<td>
<p>You can specify the haplotype block (or gene set, SNP-set) of interest by the name of haplotype block in 'geno'.
This will be used in the title of the plots.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plotNetwork</code></td>
<td>
<p>If TRUE, the function will return the plot of haplotype network.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subpopInfo</code></td>
<td>
<p>The information of subpopulations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>saveName</code></td>
<td>
<p>When drawing any plot, you can save plots in png format. In saveName, you should substitute the name you want to save.
When saveName = NULL, the plot is not saved.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>saveStyle</code></td>
<td>
<p>This argument specifies how to save the plot of phylogenetic tree.
The function offers 'png', 'pdf', 'jpg', and 'tiff'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plotWhichMDS</code></td>
<td>
<p>We will show the MDS (multi-dimensional scaling) plot,
and this argument is a vector of two integers specifying that will define which MDS dimension will be plotted.
The first and second integers correspond to the horizontal and vertical axes, respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colConnection</code></td>
<td>
<p>A vector of two integers or characters specifying the colors of connection between nodes for the original and complemented haplotypes, respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ltyConnection</code></td>
<td>
<p>A vector of two characters specifying the line types of connection between nodes for the original and complemented haplotypes, respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lwdConnection</code></td>
<td>
<p>A vector of two integers specifying the line widths of connection between nodes for the original and complemented haplotypes, respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pchBase</code></td>
<td>
<p>A vector of two integers specifying the plot types for the positive and negative genotypic values respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colCompBase</code></td>
<td>
<p>A vector of two integers or characters specifying color of complemented haplotypes for the positive and negative genotypic values respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colHaploBase</code></td>
<td>
<p>A vector of integers or characters specifying color of original haplotypes for the positive and negative genotypic values respectively.
The length of the vector should equal to the number of subpopulations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cexMax</code></td>
<td>
<p>A numeric specifying the maximum point size of the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cexMin</code></td>
<td>
<p>A numeric specifying the minimum point size of the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ggPlotNetwork</code></td>
<td>
<p>If TRUE, the function will return the ggplot version of haplotype network.
It offers the precise information on subgroups for each haplotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cexMaxForGG</code></td>
<td>
<p>A numeric specifying the maximum point size of the plot for the ggplot version of haplotype network,
relative to the range of x and y-axes (0 &lt; cexMaxForGG &lt;= 1).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cexMinForGG</code></td>
<td>
<p>A numeric specifying the minimum point size of the plot for the ggplot version of haplotype network,
relative to the range of x and y-axes (0 &lt; cexMaxForGG &lt;= 1).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alphaBase</code></td>
<td>
<p>alpha (parameter that indicates the opacity of a geom) for original haplotype with positive / negative effects.
alpha for complemented haplotype will be same as the alpha for original haplotype with negative effects.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Draw plot of haplotype network.
</p>


</div>