<div class="container">

<table style="width: 100%;"><tr>
<td>convertBlockList</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function to convert haplotype block list from PLINK to RAINBOWR format</h2>

<h3>Description</h3>

<p>Function to convert haplotype block list from PLINK to RAINBOWR format
</p>


<h3>Usage</h3>

<pre><code class="language-R">convertBlockList(
  fileNameBlocksDetPlink,
  map,
  blockNamesHead = "haploblock_",
  imputeOneSNP = FALSE,
  insertZeros = FALSE,
  n.core = 1,
  parallel.method = "mclapply",
  count = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>fileNameBlocksDetPlink</code></td>
<td>
<p>File name of the haplotype block list generated by PLINK (See reference).
The file names must contain ".blocks.det" in the tail.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>Data frame with the marker names in the first column.
The second and third columns contain the chromosome and map position.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blockNamesHead</code></td>
<td>
<p>You can specify the header of block names for the returned data.frame.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>imputeOneSNP</code></td>
<td>
<p>As default, blocks including only one SNP will be discarded from the returned data.
If you want to include them when creating haplotype-block list for RAINBOWR,
please set 'imputeOneSNP = TRUE'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>insertZeros</code></td>
<td>
<p>When naming blocks, whether or not inserting zeros to the name of blocks.
For example, if there are 1,000 blocks in total, the function will name the block 1 as
"block_1" when 'insertZeros = FALSE' and "block_0001" when 'insertZeros = TRUE'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.core</code></td>
<td>
<p>Setting n.core &gt; 1 will enable parallel execution on a machine with multiple cores.
This argument is not valid when 'parallel.method = "furrr"'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel.method</code></td>
<td>
<p>Method for parallel computation. We offer three methods, "mclapply", "furrr", and "foreach".
</p>
<p>When 'parallel.method = "mclapply"', we utilize <code>pbmclapply</code> function in the 'pbmcapply' package
with 'count = TRUE' and <code>mclapply</code> function in the 'parallel' package with 'count = FALSE'.
</p>
<p>When 'parallel.method = "furrr"', we utilize <code>future_map</code> function in the 'furrr' package.
With 'count = TRUE', we also utilize <code>progressor</code> function in the 'progressr' package to show the progress bar,
so please install the 'progressr' package from github (<a href="https://github.com/HenrikBengtsson/progressr">https://github.com/HenrikBengtsson/progressr</a>).
For 'parallel.method = "furrr"', you can perform multi-thread parallelization by
sharing memories, which results in saving your memory, but quite slower compared to 'parallel.method = "mclapply"'.
</p>
<p>When 'parallel.method = "foreach"', we utilize <code>foreach</code> function in the 'foreach' package
with the utilization of <code>makeCluster</code> function in 'parallel' package,
and <code>registerDoParallel</code> function in 'doParallel' package.
With 'count = TRUE', we also utilize <code>setTxtProgressBar</code> and
<code>txtProgressBar</code> functions in the 'utils' package to show the progress bar.
</p>
<p>We recommend that you use the option 'parallel.method = "mclapply"', but for Windows users,
this parallelization method is not supported. So, if you are Windows user,
we recommend that you use the option 'parallel.method = "foreach"'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>count</code></td>
<td>
<p>When count is TRUE, you can know how far RGWAS has ended with percent display.</p>
</td>
</tr>
</table>
<h3>Value</h3>


<p>A data.frame object of
</p>
<dl>
<dt>$block</dt>
<dd>
<p>Block names for SNP-set methods in RAINBOWR</p>
</dd>
<dt>$marker</dt>
<dd>
<p>Marker names in each block for SNP-set methods in RAINBOWR</p>
</dd>
</dl>
<p>Purcell, S. and Chang, C. (2018). PLINK 1.9, www.cog-genomics.org/plink/1.9/.
Chang CC, Chow CC, Tellier LCAM, Vattikuti S, Purcell SM, Lee JJ (2015) Second-generation PLINK: rising to the challenge of larger and richer datasets. GigaScience, 4.
Gaunt T, Rodr√≠guez S, Day I (2007) Cubic exact solutions for the estimation of pairwise haplotype frequencies: implications for linkage disequilibrium analyses and a web tool 'CubeX'. BMC Bioinformatics, 8.
Taliun D, Gamper J, Pattaro C (2014) Efficient haplotype block recognition of very long and dense genetic sequences. BMC Bioinformatics, 15.
</p>


</div>