<div class="container">

<table style="width: 100%;"><tr>
<td>adjustedCADD.annotation.SNVs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
SNVs annotation with adjusted CADD scores
</h2>

<h3>Description</h3>

<p>Annotate SNVs with the adjusted CADD scores (CADD PHRED scores for coding, regulatory and intergenic regions)
</p>


<h3>Usage</h3>

<pre><code class="language-R">adjustedCADD.annotation.SNVs(x, variant.scores = NULL, 
			     cores = 10, verbose = T, path.data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A bed.matrix annotated with CADD regions using <code>set.CADDregions</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variant.scores</code></td>
<td>
<p> A dataframe containing the ADJUSTED CADD scores of the SNVs (Optional, useful to gain in computation time if the adjusted CADD scores of variants in the study are available)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cores</code></td>
<td>
<p> How many cores to use, set at 10 by default</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> Whether to display information about the function actions </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path.data</code></td>
<td>
<p> The repository where data for RAVA-FIRST are or will be downloaded from https://lysine.univ-brest.fr/RAVA-FIRST/</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>SNVs are directly annotated with the adjusted CADD scores in the function using the file "AdjustedCADD_v1.4_202108.tsv.gz" downloaded from https://lysine.univ-brest.fr/RAVA-FIRST/ in the repository of the package Ravages or the scores of variants can be provided to <code>variant.scores</code> to gain in computation time (this file should contain 5 columns: the chromosome ('chr'), position ('pos'), reference allele ('A1'), alternative allele ('A2') and adjusted CADD scores ('adjCADD').  
</p>
<p>Those adjusted scores are used in the <code>RAVA.FIRST()</code> pipeline to filter rare variants.  
</p>
<p>As this function can take time when a large number of SNVs are present, it is recommended to use this function chromosome by chromosome for large datasets or to fitler the bed matrix before the annotation.
</p>


<h3>Value</h3>

<p>The bed matrix x with adjusted CADD scores in <code>adjCADD</code>.
</p>


<h3>Source</h3>

<p>https://lysine.univ-brest.fr/RAVA-FIRST/</p>


<h3>See Also</h3>

<p><code>adjustedCADD.annotation, adjustedCADD.annotation.indels, RAVA.FIRST</code>, <code>filter.adjustedCADD</code> 
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Import 1000Genome data from region around LCT gene
#x &lt;- as.bed.matrix(LCT.gen, LCT.fam, LCT.bim)

#Annotate variants with adjusted CADD score
#x &lt;- adjustedCADD.annotation.SNVs(x)
</code></pre>


</div>