<div class="container">

<table style="width: 100%;"><tr>
<td>burden</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Linear, logistic or multinomial regression on a genetic score
</h2>

<h3>Description</h3>

<p>Performs burden tests on categorical or continuous phenotypes
</p>


<h3>Usage</h3>

<pre><code class="language-R">burden(x, NullObject, genomic.region = x@snps$genomic.region, burden,
       maf.threshold = 0.5, get.effect.size = FALSE, alpha = 0.05, cores = 10,
       verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A bed matrix, only needed if <code>burden="CAST"</code> or <code>burden="WSS"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>NullObject</code></td>
<td>
<p> A list returned from <code>NullObject.parameters</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomic.region</code></td>
<td>
<p> A factor containing the genomic region of each SNP, <code>x@snps$genomic.region</code> by default, only needed if <code>burden</code>="CAST" or <code>burden</code>="WSS"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>burden</code></td>
<td>
<p>"CAST" or "WSS" to directly compute the CAST or the WSS genetic score, or a matrix with one row per individual and one column per <code>genomic.region</code> if another genetic score is wanted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf.threshold</code></td>
<td>
<p> The MAF threshold to use for the definition of a rare variant in the CAST score. Set at 0.5 by default</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>get.effect.size</code></td>
<td>
<p> TRUE/FALSE: whether to return effect sizes of the tested <code>genomic.region</code> (OR for categorical phenotypes, betas for continuous phenotypes) </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p> The alpha threshold to use for the OR confidence interval</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cores</code></td>
<td>
<p> How many cores to use, set at 10 by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> Whether to display information about the function actions </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function will return results from the regression of the phenotype on the genetic score for each genomic region.
</p>
<p>If only two groups of individuals are present, a classical logistic regression is performed.
If more than two groups of individuals are present, a non-ordinal multinomial regression is performed,
comparing each group of individuals to the reference group indicated by the argument <code>ref.level</code> in <code>NullObject.parameters</code>.
The choice of the reference group won't affect the p-values, but only the Odds Ratios.
In both types of regression, the p-value is estimated using the Likelihood Ratio test and the function <code>burden.mlogit</code>.  
</p>
<p>If the phenotype is continuous, a linear regression is performed using the function <code>burden.continuous</code>.  
</p>
<p>The type of phenotype is determined from <code>NullObject$pheno.type</code>.  
</p>
<p>If another genetic score than CAST or WSS is wanted, a matrix with one row per individual and one column per <code>genomic.region</code>
containing this score should be given to <code>burden</code>. In this situation, no bed matrix <code>x</code> is needed.
</p>


<h3>Value</h3>

<p>A dataframe with one row per genomic region and at least two columns:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>
<p> The p.value of the regression</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is.err</code></td>
<td>
<p> 0/1: whether there was a convergence problem with the regression</p>
</td>
</tr>
</table>
<p>If <code>NullObject$pheno.type = "categorical"</code> and <code>get.OR.value=TRUE</code>, additional columns are present:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>OR/beta</code></td>
<td>
<p> The OR/beta value(s) associated to the regression. For categorical phenotypes, if there are more than two groups, there will be one OR value per group compared to the reference group</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>l.lower</code></td>
<td>
<p> The lower bound of the confidence interval of each OR/beta</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>l.upper</code></td>
<td>
<p> The upper bound of the confidence interval of each OR/beta</p>
</td>
</tr>
</table>
<h3>References</h3>

<p> Bocher O, et al. DOI: 10.1002/gepi.22210. <em>Rare variant association testing for multicategory phenotype.</em> Genet.Epidemiol. 2019;43:646â€“656.</p>


<h3>See Also</h3>

 <p><code>NullObject.parameters</code>, <code>burden.continuous</code>, <code>burden.mlogit</code>,  <code>CAST</code>, <code>WSS</code>, <code>burden.weighted.matrix</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">#Import data in a bed matrix
x &lt;- as.bed.matrix(x=LCT.matrix.bed, fam=LCT.matrix.fam, bim=LCT.snps)

#Add population
x@ped[,c("pop", "superpop")] &lt;- LCT.matrix.pop1000G[,c("population", "super.population")]

#Select EUR superpopulation
x &lt;- select.inds(x, superpop=="EUR")
x@ped$pop &lt;- droplevels(x@ped$pop)

#Group variants within known genes
x &lt;- set.genomic.region(x)

#Filter of rare variants: only non-monomorphic variants with
#a MAF lower than 2.5%
#keeping only genomic regions with at least 10 SNPs
x1 &lt;- filter.rare.variants(x, filter = "whole", maf.threshold = 0.025, min.nb.snps = 10)

#run null model, using the 1000Genome population as "outcome"
x1.H0 &lt;- NullObject.parameters(pheno = x1@ped$pop, ref.level = "CEU",
                               RVAT = "burden", pheno.type = "categorical")

#run burden test WSS
burden(x1, NullObject = x1.H0, burden = "WSS", get.effect.size=TRUE, cores = 1)


</code></pre>


</div>