<div class="container">

<table style="width: 100%;"><tr>
<td>genotypic.freq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Genotypic frequencies calculation for data simulations
</h2>

<h3>Description</h3>

<p>Calculates the three genotypic frequencies in the controls group and each group of cases based on MAF in the general population and GRR values
</p>


<h3>Usage</h3>

<pre><code class="language-R">genotypic.freq(genes.maf = Kryukov, GRR.het, GRR.homo.alt, prev, 
               genetic.model = c("general", "multiplicative", 
                                 "dominant", "recessive"), select.gene,
               selected.controls = T)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genes.maf</code></td>
<td>
<p> A file containing the MAF in the general population (column maf) for variants with their associated gene (column gene), by default the file <code>Kryukov</code> is used</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GRR.het</code></td>
<td>
<p> A matrix giving the GRR of the heterozygous genotype compared to the homozygous reference genotype with one row per cases group and one column per variant </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GRR.homo.alt</code></td>
<td>
<p> A matrix giving the GRR of the homozygous alternative genotype compared to the homozygous reference genotype with one row per cases group and one column per variant, only need if <code>genetic.model="general"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prev</code></td>
<td>
<p> A vector containing the prevalence of each group of cases</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genetic.model</code></td>
<td>
<p> The genetic model of the disease </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>select.gene</code></td>
<td>
<p> Which gene to choose from <code>genes.maf$gene</code> if multiple genes are present. If missing, only the first level is kept.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>selected.controls</code></td>
<td>
<p> Whether controls are selected controls (by default) or controls from the general population</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is used to simulate genetic data.
</p>
<p>The genetic model of the disease needs to be specified to <code>genetic.model</code>:
</p>
<p>If <code>genetic.model="general"</code>, there is no link between the GRR associated 
to the heterozygous genotype and the GRR associated to the homozygous alternative 
genotype. Therefore, the user has to give two matrices of GRR, one for each of these genotypes.
</p>
<p>If <code>genetic.model="multiplicative"</code>, we assume that the GRR associated to the homozygous 
alternative genotype is the square of the GRR associated to the heterozygous genotype.
</p>
<p>If <code>genetic.model="dominant"</code>, we assume that the GRR associated to the heterozygous genotype 
and the GRR associated to the homozygous alternative genotype are equal.
</p>
<p>If <code>genetic.model="recessive"</code>, we assume that the GRR associated to the heterozygous genotype 
is equal to 1: the GRR given is the one associated to the homozygous alternative genotype.  
</p>
<p><code>prev</code> corresponds to the proportion of each sub-group of cases in the population. 
It is used only to calculate the MAF in the controls group.  
</p>
<p>If <code>selected.controls</code> = T, genotypic frequencies in the control group are computed from genotypic frequencies in the cases groups and the prevalence of the disease. 
If FALSE, genotypic frequencies in the control group are computed from allelic frequencies under Hardy-Weinberg equilibrium.  
</p>
<p>The dataframes <code>Kryukov</code> or <code>GnomADgenes</code> available with the package Ravages can be used for the argument <code>genes.maf</code>.
</p>


<h3>Value</h3>

<p>A matrix of MAF values with one column per variant and one row per group (the first one being the controls group)
</p>


<h3>See Also</h3>

 <p><code>GRR.matrix</code>, <code>rbm.GRR</code>, <code>GnomADgenes</code>, <code>Kryukov</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">#Construction of the GRR matrix using the formula from SKAT
#to compute the GRR (higher weights to rarer variants)
#GRR in the second group are twice as high as in the first group
GRR.del &lt;- GRR.matrix(GRR = "SKAT", GRR.multiplicative.factor=2,
                      select.gene="R1")

#Calculation of frequency in the three groups of individuals 
#under a multilpicative model of the disease
geno.freq.groups &lt;- genotypic.freq(genes.maf = Kryukov, GRR.het = GRR.del, 
                                   prev = c(0.001, 0.001), select.gene="R1", 
                                   genetic.model = "multiplicative")
</code></pre>


</div>