<div class="container">

<table style="width: 100%;"><tr>
<td>Jaccard</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Jaccard index
</h2>

<h3>Description</h3>

<p>Calculates the Jaccard index for each pair of individuals using a bed.matrix
</p>


<h3>Usage</h3>

<pre><code class="language-R">Jaccard(x, maf.threshold = 0.01)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A bed.matrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf.threshold</code></td>
<td>
<p>The MAF used for the definition of a rare variant, set at 0.01 by default</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The individuals carrying no rare variants will have a null Jaccard index with all the individuals including themselves.
</p>


<h3>Value</h3>

<p>A squared matrix giving the Jaccard index for each pair of individuals
</p>


<h3>References</h3>

<p> Jaccard, P. (1908) <em>Nouvelles researches sur la distribution florale</em>, Bulletin de la Société vaudoise des sciences naturelles, <b>44, 223-270</b> </p>


<h3>Examples</h3>

<pre><code class="language-R">
#Simulation of genetic data with GRR values according to the SKAT formula
GRR.del &lt;- GRR.matrix(GRR = "SKAT", genes.maf = Kryukov, 
                      n.case.groups = 2, select.gene = "R1",
                      GRR.multiplicative.factor=2)

#Simulation of one group of 1,000 controls and two groups of 500 cases, 
#50% of causal variants, 5 genomic regions are simulated.
x &lt;- rbm.GRR(genes.maf=Kryukov, size = c(1000, 500, 500),
             prev = c(0.001, 0.001), select.gene = "R1", 
             GRR.matrix.del = GRR.del, p.causal = 0.5,
             genetic.model = "multiplicative", replicates = 5)
                           
#Calculate the Jaccard matrix
J &lt;- Jaccard(x, maf.threshold = 0.01)

</code></pre>


</div>