<div class="container">

<table style="width: 100%;"><tr>
<td>set.genomic.region</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Variants annotation based on gene positions </h2>

<h3>Description</h3>

<p>Attributes regions to variants based on given region positions
</p>


<h3>Usage</h3>

<pre><code class="language-R">set.genomic.region(x, regions = genes.b37, flank.width = 0L, split = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A bed.matrix </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>regions</code></td>
<td>

<p>A dataframe in bed format (start is 0-based and end is 1-based) containing the fields : <code>Chr</code> (the chromosome of the gene),
<code>Start</code> (the start position of the gene, 0-based), <code>End</code> (the end position of the gene, 1-based), and
<code>Name</code> (the name of the gene - a factor),
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>flank.width</code></td>
<td>
<p> An integer: width of the flanking regions in base pairs downstream and upstream the regions. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>split</code></td>
<td>
<p> Whether to split variants attributed to multiple regions by duplicating this variants, set at TRUE by default</p>
</td>
</tr>
</table>
<h3>Details</h3>

 
<p>Warnings: <code>regions$Name</code> should be a factor containing UNIQUE names of the regions, ORDERED in the genome order. 
</p>
<p>We provide two data sets of autosomal humain genes, <code>genes.b37</code> and <code>genes.b38</code>.  
</p>
<p>If <code>x@snps$chr</code> is not a vector of integers, it should be a factor with same levels as <code>regions$Chr</code>.
</p>
<p>If <code>flank.width</code> is null, only the variants having their position between the <code>regions$Start</code> and the <code>regions$End</code> of a gene will be attributed to the corresponding gene.  
When two regions overlap, variants in the overlapping zone will be assigned to those two regions, separated by a comma.
</p>
<p>If <code>flank.width</code> is a positive number, variants <code>flank.width</code> downstream or upstream a gene will be annotated annotated to this gene. You can use <code>flank.width = Inf</code>
to have each variant attributed to the nearest gene.  
</p>
<p>If a variant is attributed to multiple genomic regions, it will be duplicated in the bed matrix with one row per genomic region if <code>split = TRUE</code>. Variants will have new IDs being CHR:POS:A1:A2:genomic.region.  
</p>


<h3>Value</h3>

<p>The same bed matrix as x with an additional column <code>x@snps$genomic.region</code> containing the annotation of each variant.
</p>


<h3>See Also</h3>

<p><code>genes.b37</code>, <code>genes.b38</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Import 1000Genome data from region around LCT gene
x &lt;- as.bed.matrix(LCT.gen, LCT.fam, LCT.bim)

#Group variants within known genes
x &lt;- set.genomic.region(x)

#Group variants within know genes +/- 500bp
x &lt;- set.genomic.region(x, flank.width=500)
</code></pre>


</div>