<div class="container">

<table style="width: 100%;"><tr>
<td>bed.matrix.split.genomic.region</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Bed matrix for variants associated to multiple genomic regions</h2>

<h3>Description</h3>

<p>Creates a new bed matrix with variants associated to multiple genomic regions being duplicated
</p>


<h3>Usage</h3>

<pre><code class="language-R">bed.matrix.split.genomic.region(x, changeID=TRUE, genomic.region=NULL, 
                                split.pattern=",")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A bed.matrix </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>changeID</code></td>
<td>
<p> TRUE/FALSE: whether to change the variants ID by including the gene name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomic.region</code></td>
<td>
<p> A vector containing the genomic region of each variant </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>split.pattern</code></td>
<td>
<p> The character separating the genomic regions</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If <code>changeID=TRUE</code>, variants will have new IDs being CHR:POS:A1:A2:genomic.region.  
</p>
<p>The genomic region(s) associated to each varaint should be in <code>x@snps$genomic.region</code> or given as a vector to <code>genomic.region</code>. If both are present, <code>genomic.region</code> is used.
</p>


<h3>Value</h3>

<p>A bed matrix with variants assigned to multiple genomic regions being duplicated and the corresponding genomic regions separated and transformed into factors.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Example bed matrix with 4 variants
  x.ex &lt;- as.bed.matrix(x=matrix(0, ncol=4, nrow=10), 
                       bim=data.frame(chr=1:4, id=paste("rs", 1:4, sep=""), dist = rep(0,4), 
                                      pos=c(150,150,200,250), A1=rep("A", 4), A2=rep("T", 4)))

#Example genes dataframe
  genes.ex &lt;- data.frame(Chr=c(1,1,3,4), Start=c(10,110,190,220), End=c(170,180,250,260), 
                         Gene_Name=factor(letters[1:4]))

#Attribute genomic regions
  x.ex &lt;- set.genomic.region(x.ex, regions = genes.ex)

#Split genomic regions
  x.ex.split &lt;- bed.matrix.split.genomic.region(x.ex, split.pattern = ",")
</code></pre>


</div>