<div class="container">

<table style="width: 100%;"><tr>
<td>LCT.matrix</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>LCT genotypes matrix</h2>

<h3>Description</h3>

<p>These data contain the genotype matrix corresponding to haplotypes present in <code>LCT.haplotypes</code> from the 1000 Genomes data for a ~300kb segment containing the Lactase gene.  
Information about individuals is present in <code>LCT.matrix.fam</code>, and information about population (population and super population) is present in <code>LCT.matrix.pop1000G</code>, in a format needed to generate a bedmatrix.  
<code>LCT.snps</code> from <code>LCT.haplotypes</code> can be used as the corresponding bim file of this genotypes matrix.  
</p>


<h3>Usage</h3>

<pre><code class="language-R"> data(LCT.matrix) </code></pre>


<h3>Format</h3>

<p>Three data objects are present in <code>LCT.haplotypes</code>:
</p>

<dl>
<dt><code>LCT.matrix.bed</code></dt>
<dd>
<p> The matrix of genotypes</p>
</dd>
<dt><code>LCT.matrix.fam</code></dt>
<dd>
<p> The corresponding fam file</p>
</dd>
<dt><code>LCT.matrix.pop1000G</code></dt>
<dd>
<p> A data frame with population information for individuals (population, superpopulation)</p>
</dd>
</dl>
<h3>Source</h3>

<p> Data were obtained from the 1000 Genomes Project. </p>


<h3>References</h3>

<p> McVean et al, 2012, <em>An integrated map of genetic variation from 1,092 human genomes</em>, Nature <b>491, 56-65</b> doi:10.1038/nature11632 </p>


<h3>See Also</h3>

 <p><code>LCT.haplotypes</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">#Import data in a bed matrix
x &lt;- as.bed.matrix(x=LCT.matrix.bed, fam=LCT.matrix.fam, bim=LCT.snps)
#Add population
x@ped[,c("pop", "superpop")] &lt;- LCT.matrix.pop1000G[,c("population", "super.population")]

</code></pre>


</div>