<div class="container">

<table style="width: 100%;"><tr>
<td>GnomADgenes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>GnomADgenes dataset</h2>

<h3>Description</h3>

<p>This dataframe contains variants from the GnomAD database with MAF values in the Non-Finnish European (NFE)
and their consequences from VEP with each associated gene in build version 37.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(GnomADgenes)</code></pre>


<h3>Format</h3>

<p>GnomADgenes is a dataframe with five columns:
</p>

<dl>
<dt>chr</dt>
<dd>
<p> The chromosome of the variant</p>
</dd>
<dt>pos</dt>
<dd>
<p> The position of the variant</p>
</dd>
<dt>consequence</dt>
<dd>
<p> The functionnal consequence of the variant predicted by Variant Effect Predictor (VEP)</p>
</dd>
<dt>gene</dt>
<dd>
<p> The gene associated to each variant predicted by VEP</p>
</dd>
<dt>maf</dt>
<dd>
<p> The MAF of the variant in the NFE population</p>
</dd>
</dl>
<h3>Source</h3>

<p>The data were obtained from the GnomAD website (see http://gnomad.broadinstitute.org/) and the VEP website (see https://www.ensembl.org/info/docs/tools/vep/).</p>


</div>