<div class="container">

<table style="width: 100%;"><tr>
<td>WSS</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> WSS genetic score</h2>

<h3>Description</h3>

<p>Caluclates the WSS genetic score
</p>


<h3>Usage</h3>

<pre><code class="language-R">WSS(x, genomic.region = x@snps$genomic.region)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A bed.matrix </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomic.region</code></td>
<td>
<p> A factor containing the genomic region of each variant </p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A matrix containing the WSS genetic score with one row per individual and one column per <code>genomic.region</code>
</p>


<h3>References</h3>

<p> Madsen E and Browning S. <em>A Groupwise Association Test for Rare Mutations Using a Weighted Sum Statistic.</em> PLoS Genet. 2009 </p>


<h3>See Also</h3>

 <p><code>CAST</code>, <code>burden.weighted.matrix</code>, <code>burden.mlogit</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">#Import data in a bed matrix
x &lt;- as.bed.matrix(x=LCT.matrix.bed, fam=LCT.matrix.fam, bim=LCT.snps)

# Group variants within known genes
x &lt;- set.genomic.region(x)

# Filter variants with maf (computed on whole sample) &lt; 0.025
# keeping only genomic region with at least 10 SNPs
x1 &lt;- filter.rare.variants(x, filter = "whole", maf.threshold = 0.025, min.nb.snps = 10)

# Compute burden score WSS
score.WSS &lt;- WSS(x1)
</code></pre>


</div>