<div class="container">

<table style="width: 100%;"><tr>
<td>rbm.haplos.power</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Power of RVAT based on simulations with haplotypes
</h2>

<h3>Description</h3>

<p>Computes the power of the tests CAST, WSS and SKAT based on simulations with haplotypes
</p>


<h3>Usage</h3>

<pre><code class="language-R">  rbm.haplos.power(haplos, freqs, weights = "SKAT", 
                   max.maf.causal = 0.01, maf.filter = max.maf.causal, p.causal = 0.5, 
                   p.protect = 0, h2 = c(0.01, 0.01), prev = c(1, 0.01),
                   normal.approx = TRUE, size = c(500, 500), verbose = TRUE,
                   alpha = 2.5e-6, RVAT = c("CAST", "WSS", "SKAT"),
                   SKAT.method = c("permutations", "theoretical"),
                   simus.haplos = c("freqs", "liability"), 
                   replicates = 1000, rep.by.causal = 50, cores = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>haplos</code></td>
<td>
<p> A matrix of haplotypes with one row per haplotype and one column per variant</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freqs</code></td>
<td>
<p> A matrix of haplotypes frequencies in each group of individuals, only needed if <code>simus.haplos = "freqs"</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p> How to weight rare variants (if "constant", all variants have the same weight, if "SKAT", the rarest variants have the highest weights: weights = -0.4*log10(MAF) )</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.maf.causal</code></td>
<td>
<p> The maf threshold to consider a rare variant  (set at 0.01 by default). Only variants with a MAF upper than this threshold will be kept to compute RVAT power. If <code>simus.haplos="liability"</code>, variants with a MAF upper this threshold will have a weight of 0</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf.filter</code></td>
<td>
<p> The MAF filter to apply after the simulations to select rare variants to keep for RVAT power analysis. By default corresponds to <code>max.maf.causal</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.causal</code></td>
<td>
<p> The percentage of causal variants, only needed if <code>simus.haplos = "liability"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.protect</code></td>
<td>
<p> The proportion of protective variants among causal variants, only needed if <code>simus.haplos = "liability"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>h2</code></td>
<td>
<p> The variance explained by the gene, only needed if <code>simus.haplos = "liability"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prev</code></td>
<td>
<p> A vector with the prevalence in each group of individuals, only needed if <code>simus.haplos = "liability"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normal.approx</code></td>
<td>
<p> TRUE/FALSE: whether to use the normal approximation to compute thresholds. Set at TRUE by default, only needed if <code>simus.haplos = "liability"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>size</code></td>
<td>
<p> The sizes of each group of individuals</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> Whether to display information about the function actions </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p> The significance level to compute the power</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>RVAT</code></td>
<td>
<p> On which RVAT among 'CAST', 'WSS' and 'SKAT' to compute power (only needed if <code>power.type="simulations"</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SKAT.method</code></td>
<td>
<p> Which method to use to compute SKAT ppower, i.e. permutations or theoretical moments (cf <code>SKAT</code> documentation)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simus.haplos</code></td>
<td>
<p> Which method to simulate the data, if <code>simus.haplos="freqs"</code>, rbm.haplos.freqs() is used, otherwise <code>rbm.haplos.thresholds()</code> is used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>replicates</code></td>
<td>
<p> The number of simulations to perform to estimate the power</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rep.by.causal</code></td>
<td>
<p> The number of time causal variants will be sampled</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cores</code></td>
<td>
<p> How many cores to use for moments computation, set at 10 by default </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Simulations are perfromed accordingly to <code>rbm.haplos.thresholds()</code> or <code>rbm.haplos.freqs()</code>. Please refer to the corresponding manuals for more details on the simulation procedures.  
Variants are filtered after the simulations to keep only the rare ones, defined by <code>maf.filter</code>. By defaut, it corresponds to <code>max.maf.causal</code> is used. To disable this filter, set <code>maf.filter</code> at 0.5.
</p>


<h3>Value</h3>

<p>Power values of <code>RVAT</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
  #Simulations of 5 groups of individuals with haplotypes frequencies
  #from the 5 EUR populations

  #Load LCT dataset for haplotype matrix
  data(LCT.haplotypes)
  #Haplotypes for the variants in the LCT gene in the EUR population
  LCT.gene.hap &lt;- LCT.hap[which(LCT.sample$super.population=="EUR"),
                          which(LCT.snps$pos&gt;=136545410 &amp; LCT.snps$pos&lt;=136594750)]

  #Individuals from EUR
  LCT.sample.EUR &lt;- subset(LCT.sample, super.population=="EUR")
  #Matrix of haplotypic frequencies
  LCT.freqs &lt;- sapply(unique(LCT.sample.EUR$population), function(z)
                      ifelse(LCT.sample.EUR$population==z,
                             1/table(LCT.sample.EUR$population)[z], 0))

  #Simulation of genetic data for five groups of 50 individuals
  rbm.haplos.power(haplos=LCT.gene.hap, freqs=LCT.freqs, size=rep(50,5), 
                   replicates=5, rep.by.causal = 5,  RVAT = "CAST", 
                   alpha = c(0.001,2.5e-6), cores = 1)

</code></pre>


</div>