<div class="container">

<table style="width: 100%;"><tr>
<td>set.CADDregions</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Variants annotation based on 'CADD regions' and genomic categories </h2>

<h3>Description</h3>

<p>Attributes CADD regions and genomic categories to variants based on their positions
</p>


<h3>Usage</h3>

<pre><code class="language-R">set.CADDregions(x, verbose = T, path.data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A bed.matrix </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> Whether to display information about the function actions </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path.data</code></td>
<td>
<p> The repository where data for RAVA-FIRST are or will be downloaded from https://lysine.univ-brest.fr/RAVA-FIRST/</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>To attribute variants to CADD regions and genomic categories, the files "CADDRegions.2021.hg19.bed.gz" and "FunctionalAreas.hg19.bed.gz" will be downloaded from https://lysine.univ-brest.fr/RAVA-FIRST/ in the repository of the package Ravages.  
CADD regions are non-overlapping regions that have been defined in the whole genome to perform rare variant association tests in the <code>RAVA.FIRST()</code> pipeline.  
It is recommended to use this function chromosome by chromosome for large datasets for time and memory managment.
</p>


<h3>Value</h3>

<p>The same bed matrix as x with three additional columns :
</p>
<table>
<tr style="vertical-align: top;">
<td><code>genomic.region</code></td>
<td>
<p> The CADD region of each variant</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SubRegion</code></td>
<td>
<p> The genomic category of each variant among 'Coding', 'Regulatory' or 'Intergenic'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adjCADD.Median</code></td>
<td>
<p> The median of adjusted CADD of variants observed at least to times in GnomAD genomes r2.0.1</p>
</td>
</tr>
</table>
<h3>Source</h3>

<p>https://lysine.univ-brest.fr/RAVA-FIRST/</p>


<h3>See Also</h3>

<p><code>RAVA.FIRST</code>, <code>filter.adjustedCADD</code>, <code>burden.subscores</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Import 1000Genome data from region around LCT gene
#x &lt;- as.bed.matrix(LCT.gen, LCT.fam, LCT.bim)

#Group variants within CADD regions and genomic categories
#x &lt;- set.CADDregions(x)
#table(x@snps$genomic.region) #CADD regions
#table(x@snps$SubRegion) #Genomic categories
</code></pre>


</div>