<div class="container">

<table style="width: 100%;"><tr>
<td>genes.positions</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Genes positions</h2>

<h3>Description</h3>

<p>Positions of human genes in bed format (Start is 0-based and End is 1-based). These data were downloaded from Biomart on the Ensembl website with the GRCh37 and GRCh38 versions. Only genes present in GnomAD were kept.
</p>
<p>Data contain the <code>Chr</code>, the <code>Start</code> position, the <code>End</code> position and the <code>Name</code> of all the genes in chromosomes 1 to 22 representing 19375 and 18278 genes in the two GRCh versions respectively.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(genes.b37)
data(genes.b38) </code></pre>


<h3>Format</h3>

<p>The data contain one dataframe with four columns:
</p>

<dl>
<dt><code>Chr</code></dt>
<dd>
<p>The chromosome of the gene</p>
</dd>
<dt><code>Start</code></dt>
<dd>
<p>The start position of the gene (0-based)</p>
</dd>
<dt><code>End</code></dt>
<dd>
<p>The end position of the gene (1-based)</p>
</dd>
<dt><code>Name</code></dt>
<dd>
<p>The name of the gene</p>
</dd>
</dl>
<h3>Source</h3>

<p>The data were obtained from the Ensembl website.</p>


<h3>References</h3>

<p> RJ Kinsella et al, 2011, <em>Ensembl BioMarts: a hub for data retrieval across taxonomic space</em>, Database. doi:10.1093/database/bar030;   
</p>
<p>AD Yates et al, 2020, <em>Ensembl 2020</em>, Nucleic Acide Research. doi:10.1093/nar/gkz966</p>


<h3>See Also</h3>

 <p><code>set.genomic.region</code> </p>


</div>