<div class="container">

<table style="width: 100%;"><tr>
<td>adjustedCADD.annotation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
SNVs and Indels annotation with adjusted CADD scores
</h2>

<h3>Description</h3>

<p>Annotate SNVs and Indels with the adjusted CADD scores (CADD PHRED scores for coding, regulatory and intergenic regions)
</p>


<h3>Usage</h3>

<pre><code class="language-R">adjustedCADD.annotation(x, SNVs.scores = NULL, indels.scores = NULL,
			cores = 10, verbose = T, path.data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A bed.matrix annotated with CADD regions using <code>set.CADDregions</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SNVs.scores</code></td>
<td>
<p> A dataframe containing the ADJUSTED CADD scores of the SNVs (Optional, useful to gain in computation time if the adjusted CADD scores of variants in the study are available)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>indels.scores</code></td>
<td>
<p> A dataframe containing the CADD PHREDv1.4 scores of the indels - Compulsory if indels are present in <code>x</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cores</code></td>
<td>
<p> How many cores to use, set at 10 by default</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> Whether to display information about the function actions </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path.data</code></td>
<td>
<p> The repository where data for RAVA-FIRST are or will be downloaded from https://lysine.univ-brest.fr/RAVA-FIRST/</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function calls <code>adjustedCADD.annotation.SNVs</code> and <code>adjustedCADD.annotation.indels</code>. See the help of those two functions for more details.  
</p>


<h3>Value</h3>

<p>The bed matrix x with adjusted CADD scores in <code>adjCADD</code>.
</p>


<h3>Source</h3>

<p>https://lysine.univ-brest.fr/RAVA-FIRST/</p>


<h3>See Also</h3>

<p><code>adjustedCADD.annotation.SNVs, adjustedCADD.annotation.indels, RAVA.FIRST</code>, <code>filter.adjustedCADD</code> 
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Import 1000Genome data from region around LCT gene
#x &lt;- as.bed.matrix(LCT.gen, LCT.fam, LCT.bim)

#Annotate variants with adjusted CADD score
#x &lt;- adjustedCADD.annotation(x)
</code></pre>


</div>