<div class="container">

<table style="width: 100%;"><tr>
<td>LCT.haplotypes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>LCT haplotypes data set</h2>

<h3>Description</h3>

<p>These data contain the haplotype matrix <code>LCT.hap</code> (5008 haplotypes) of the 2004 individuals from the 1000 Genomes data for a ~300kb segment containing the Lactase gene.  
Information about individuals (sex, population and super population) is present in <code>LCT.sample</code>, and information about snps is available in <code>LCT.snps</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R"> data(LCT.haplotypes) </code></pre>


<h3>Format</h3>

<p>Three data objects are present in <code>LCT.haplotypes</code>:
</p>

<dl>
<dt><code>LCT.hap</code></dt>
<dd>
<p> A matrix of haplotypes </p>
</dd>
<dt><code>LCT.sample</code></dt>
<dd>
<p> A data frame with information on individuals (sex, population, super.population)</p>
</dd>
<dt><code>LCT.snps</code></dt>
<dd>
<p> A data frame with information on snps (chr, id, dist, pos, A1, A2) </p>
</dd>
</dl>
<h3>Source</h3>

<p> Data were obtained from the 1000 Genomes Project. </p>


<h3>References</h3>

<p> McVean et al, 2012, <em>An integrated map of genetic variation from 1,092 human genomes</em>, Nature <b>491, 56-65</b> doi:10.1038/nature11632 </p>


<h3>See Also</h3>

 <p><code>LCT.matrix</code> </p>


</div>