<div class="container">

<table style="width: 100%;"><tr>
<td>CAST</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Cohort Allelic Sum Test</h2>

<h3>Description</h3>

<p> Calculates the CAST genetic score </p>


<h3>Usage</h3>

<pre><code class="language-R">CAST(x, genomic.region = x@snps$genomic.region, maf.threshold = 0.5, 
     flip.rare.alleles = T)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A bed.matrix </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomic.region</code></td>
<td>
<p> A factor defining the genomic region of each variant </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf.threshold</code></td>
<td>
<p> The MAF used for the definition of a rare variant, set at 0.5 by default, i.e. all variants are kept</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>flip.rare.alleles</code></td>
<td>
<p> Whether to flip the A1/A2 alleles if the A1 allele is rare, set at T by default</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>By default, CAST counts if an individual carries at least one rare allele in the genomic region. If <code>flip.rare.alleles = F</code> and the reference allele A1 is rare, the alles A1 and A2 won't be flipped and CAST will count the number of alternative alleles A2.
</p>


<h3>Value</h3>

<p>A matrix containing the CAST genetic score with one row per individual and one column per <code>genomic.region</code>
</p>


<h3>References</h3>

<p> Morgenthaler S and Thilly WG. <em>A strategy to discover genes that carry multi-allelic or mono-allelic risk for common diseases: a cohort allelic sums test (CAST).</em> Mutat Res. 2007</p>


<h3>See Also</h3>

 <p><code>WSS</code>, <code>burden.weighted.matrix</code>, <code>burden.mlogit</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">#Import data in a bed matrix
x &lt;- as.bed.matrix(x=LCT.matrix.bed, fam=LCT.matrix.fam, bim=LCT.snps)

# Group variants within known genes
x &lt;- set.genomic.region(x)

# Filter variants with maf (computed on whole sample) &lt; 0.025
# keeping only genomic region with at least 10 SNPs
x1 &lt;- filter.rare.variants(x, filter = "whole", maf.threshold = 0.025, min.nb.snps = 10)

# Compute burden score CAST
score.CAST &lt;- CAST(x1, maf.threshold=0.025)
</code></pre>


</div>