<div class="container">

<table style="width: 100%;"><tr>
<td>filter.rare.variants</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Rare variants filtering
</h2>

<h3>Description</h3>

<p>Filter rare variants based on a MAF threshold and a given number of SNP or a given cumulative MAF per genomic region
</p>


<h3>Usage</h3>

<pre><code class="language-R">filter.rare.variants(x, ref.level = NULL, filter=c("whole", "controls", "any"), 
                     maf.threshold=0.01, min.nb.snps = 2, min.cumulative.maf = NULL, 
                     group = NULL, genomic.region = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A bed.matrix </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref.level</code></td>
<td>
<p> The level corresponding to the controls group, only needed if <code>filter=="controls"</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter</code></td>
<td>
<p> On which group the filter will be applied </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf.threshold</code></td>
<td>
<p> The MAF threshold used to define a rare variant, set at 0.01 by default</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.nb.snps</code></td>
<td>
<p> The minimum number of variants needed to keep a genomic region, set at 2 by default</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.cumulative.maf</code></td>
<td>
<p> The minimum cumulative maf of variants needed to keep a genomic region</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group</code></td>
<td>
<p> A factor indicating the group of each individual, only needed if <code>filter = "controls"</code> or <code>filter = "any"</code>. If missing, <code>x@ped$pheno</code> is taken</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomic.region</code></td>
<td>
<p> An optional factor containing the genomic region of each variant, only needed if <code>min.nb.snps</code> or <code>min.cumulative.maf</code> is specified and if <code>x@snps$genomic.region</code> doesn't exist</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>To use this function, a factor 'genomic.region' should be present in the slot <code>x@snps</code>.  
</p>
<p>If <code>filter="whole"</code>, only the variants having a MAF lower than the threshold in the entire sample are kept.
</p>
<p>If <code>filter="controls"</code>, only the variants having a MAF lower than the threshold in the controls group are kept.
</p>
<p>If <code>filter="any"</code>, only the variants having a MAF lower than the threshold in any of the groups are kept.
</p>


<h3>Value</h3>

<p>A bed.matrix with filtered variants
</p>


<h3>Examples</h3>

<pre><code class="language-R">
#Import 1000Genome data from region around LCT gene
x &lt;- as.bed.matrix(LCT.gen, LCT.fam, LCT.bim)

#Group variants within known genes
x &lt;- set.genomic.region(x)
table(x@snps$genomic.region, useNA="ifany")
                             
#Filter of rare variants: only non-monomorphic variants with 
#a MAF lower than 2.5%
#keeping only genomic regions with at least 10 SNPs
x1 &lt;- filter.rare.variants(x, filter = "whole", maf.threshold = 0.025, min.nb.snps = 10)
table(x1@snps$genomic.region, useNA="ifany")

#Keep only variants with a MAF&lt;2%
#and regions with a cumulative MAF&gt;10%
filter.rare.variants(x, filter = "whole", maf.threshold = 0.02, min.nb.snps = 1, 
                     min.cumulative.maf=0.2)

</code></pre>


</div>