<div class="container">

<table style="width: 100%;"><tr>
<td>burden.weighted.matrix</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Score matrix for burden tests</h2>

<h3>Description</h3>

<p>Computes the score matrix for burden tests based on variants' weights
</p>


<h3>Usage</h3>

<pre><code class="language-R">burden.weighted.matrix(x, weights, genomic.region = x@snps$genomic.region)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A bed.matrix </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p> A vector containing the weight of each variant</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomic.region</code></td>
<td>
<p> A factorcontaining the genomic region of each variant </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For variant <em>i</em> and individual <em>j</em>, the genetic score will be computed as weight of variant <em>i</em> * number of minor alleles for individual <em>j</em>. This function returns a weighted score of rare alleles in the genomic region: if the reference allele is rare, it will be counted in the score instead of the atlernative allele.
</p>


<h3>Value</h3>

<p>A matrix containing the computed genetic score with one row per individual and one column per <code>genomic.region</code>.
</p>


<h3>See Also</h3>

 <p><code>CAST</code>, <code>WSS</code>, <code>burden.mlogit</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">#Import data in a bed matrix
x &lt;- as.bed.matrix(x=LCT.matrix.bed, fam=LCT.matrix.fam, bim=LCT.snps)

# Group variants within known genes
x &lt;- set.genomic.region(x)

# Filter variants with maf (computed on whole sample) &lt; 0.025
# keeping only genomic region with at least 10 SNPs
x1 &lt;- filter.rare.variants(x, filter = "whole", maf.threshold = 0.025, min.nb.snps = 10)

#Compute burden score with weights = 1-maf
score.burden &lt;- burden.weighted.matrix(x1, weights=1-x1@snps$maf)
</code></pre>


</div>