<div class="container">

<table style="width: 100%;"><tr>
<td>rbm.GRR</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Simulation of genetic data using GRR values
</h2>

<h3>Description</h3>

<p>Generates a simulated bed.matrix with genotypes for cases and controls based on GRR values
</p>


<h3>Usage</h3>

<pre><code class="language-R">rbm.GRR(genes.maf = Kryukov, size, prev, replicates, 
        GRR.matrix.del, GRR.matrix.pro = NULL, 
        p.causal = 0.5, p.protect = 0, same.variant = FALSE, 
        genetic.model=c("general", "multiplicative", "dominant", "recessive"), 
        select.gene, selected.controls = T, max.maf.causal = 0.01)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genes.maf</code></td>
<td>
<p> A dataframe containing at least the MAF in the general population (column maf) for variants with their associated gene (column gene), by default the file <code>Kryukov</code> is used</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>size</code></td>
<td>
<p> A vector containing the size of each group (the first one being the control group)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prev</code></td>
<td>
<p> A vector containing the prevalence of each group of cases</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>replicates</code></td>
<td>
<p> The number of simulations to perform </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GRR.matrix.del</code></td>
<td>
<p> A list containing the GRR matrix associated to the heterozygous genotype compared to the homozygous reference genotype as if all variants are deleterious. An additional GRR matrix associated to the homozygous for the alternate allele is needed if <code>genetic.genetic.model="general"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GRR.matrix.pro</code></td>
<td>
<p> The same argument as <code>GRR.matrix.del</code> but for protective variants</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.causal</code></td>
<td>
<p> The proportion of causal variants in cases</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.protect</code></td>
<td>
<p> The proportion of protective variants in cases among causal variants</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>same.variant</code></td>
<td>
<p> TRUE/FALSE: whether the causal variants are the same in the different groups of cases</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genetic.model</code></td>
<td>
<p> The genetic model of the disease </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>select.gene</code></td>
<td>
<p> Which gene to choose from <code>genes.maf$gene</code> if multiple genes are present. If missing, only the first level is kept.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>selected.controls</code></td>
<td>
<p> Whether controls are selected controls (by default) or controls from the general population</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.maf.causal</code></td>
<td>
<p> Only variants with a MAF lower than this threshold can be sampled as causal variants. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The genetic model of the disease needs to be specified in this function.
</p>
<p>If <code>genetic.model="general"</code>, there is no link between the GRR for the heterozygous genotype and the GRR for the homozygous alternative genotype. 
Therefore, the user has to give two matrices of GRR, one for the heterozygous genotype, the other for the homozygous alternative genotype.
</p>
<p>If <code>genetic.model="multiplicative"</code>, we assume that the the GRR for the homozygous alternative genotype is the square of the GRR for the heterozygous genotype. 
</p>
<p>If <code>genetic.model="dominant"</code>, we assume that the GRR for the heterozygous genotype and the GRR for the homozygous alternative genotype are equal.
</p>
<p>If <code>genetic.model="recessive"</code>, we assume that the GRR for the heterozygous genotype is equal to 1: the GRR given is the one associated to the homozygous alternative genotype.
</p>
<p><code>GRR.matrix.del</code> contains GRR values as if all variants are deleterious. These values will be used only for the proportion <code>p.causal</code> of variants that will be sampled as causal.  
</p>
<p>If <code>selected.controls</code> = T, genotypic frequencies in the control group are computed from genotypic frequencies in the cases groups and the prevalence of the disease.
If FALSE, genotypic frequencies in the control group are computed from allelic frequencies under Hardy-Weinberg equilibrium.  
</p>
<p>The files <code>Kryukov</code> or <code>GnomADgenes</code> available with the package Ravages can be used as the argument <code>genes.maf</code>.
</p>
<p>If <code>GRR.matrix.del</code> (or <code>GRR.matrix.pro</code>) has been generated using the function <code>GRR.matrix</code>, the arguments <code>genes.maf</code> and <code>select.gene</code> should have
the same value as in <code>GRR.matrix</code>.
</p>
<p>Only non-monomorphic variants are kept for the simulations.  
</p>
<p>Causal variants that have been sampled in each group of individuals are indicated in <code>x@ped$Causal</code>.
</p>


<h3>Value</h3>

<p>A bed.matrix with as much columns (variants) as <code>replicates</code>*number of variants.
The field <code>x@snps$genomic.region</code> contains the replicate number and the field <code>x@ped$pheno</code> contrains the group of each individual, "0" being the controls group.
</p>


<h3>See Also</h3>

 <p><code>GRR.matrix</code>, <code>Kryukov</code>, <code>GnomADgenes</code>, <code>rbm.GRR.power</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">#GRR values calculated with the SKAT formula
GRR.del &lt;- GRR.matrix(GRR = "SKAT", genes.maf = Kryukov, 
                      n.case.groups = 2, select.gene = "R1",
                      GRR.multiplicative.factor=2)
                              
#Simulation of one group of 1,000 controls and two groups of 500 cases, 
#each one with a prevalence of 0.001
#with 50% of causal variants, 5 genomic regions are simulated.
x &lt;- rbm.GRR(genes.maf = Kryukov, size = c(1000, 500, 500), 
             prev = c(0.001, 0.001), GRR.matrix.del = GRR.del, 
             p.causal = 0.5, p.protect = 0, select.gene="R1",
             same.variant = FALSE, 
             genetic.model = "multiplicative", replicates = 5)
</code></pre>


</div>