<div class="container">

<table style="width: 100%;"><tr>
<td>rbm.haplos.freqs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Simulation of genetic data based on haplotypic frequencies
</h2>

<h3>Description</h3>

<p>Simulates genetic data with respect to allele frequency spectrum and linkage disequilibrium pattern observed on given haplotypes and their frequencies
</p>


<h3>Usage</h3>

<pre><code class="language-R">  rbm.haplos.freqs(haplos, freqs, size, replicates)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>haplos</code></td>
<td>
<p> A matrix of haplotypes with one row per haplotype and one column per variant</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freqs</code></td>
<td>
<p> A matrix of haplotypes frequencies in each group of individuals</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>size</code></td>
<td>
<p> The sizes of each group of individuals</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>replicates</code></td>
<td>
<p> The number of simulations to perform</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Simulations are performed to respect linkage disequilibrium pattern and allelic frequency spectrum in each group of individuals
The phenotypic values will be the colnames of <code>freqs</code> and stored in <code>@ped$pheno</code>. The simulation number will be in <code>@snps$genomic.region</code>.
</p>


<h3>Value</h3>

<table><tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A bed matrix with simulated genotypes</p>
</td>
</tr></table>
<h3>Examples</h3>

<pre><code class="language-R">  #Simulations of 5 groups of individuals with haplotypes frequencies
  #from the 5 EUR populations

  #Load LCT dataset for haplotype matrix
  data(LCT.haplotypes)
  #Haplotypes for the variants in the LCT gene in the EUR population
  LCT.gene.hap &lt;- LCT.hap[which(LCT.sample$super.population=="EUR"),
                         which(LCT.snps$pos&gt;=136545410 &amp; LCT.snps$pos&lt;=136594750)]

  #Individuals from EUR
  LCT.sample.EUR &lt;- subset(LCT.sample, super.population=="EUR")
  #Matrix of haplotypic frequencies
  LCT.freqs &lt;- sapply(unique(LCT.sample.EUR$population), function(z) 
                      ifelse(LCT.sample.EUR$population==z, 
                             1/table(LCT.sample.EUR$population)[z], 0))

  #Simulation of genetic data for five groups of 50 individuals
  x &lt;- rbm.haplos.freqs(haplos=LCT.gene.hap, freqs=LCT.freqs, size=rep(50,5), replicates=5)
</code></pre>


</div>