<div class="container">

<table style="width: 100%;"><tr>
<td>set.genomic.region.subregion</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Variants annotation based on regions and subregions positions </h2>

<h3>Description</h3>

<p>Attributes regions and subregions to variants based on given positions
</p>


<h3>Usage</h3>

<pre><code class="language-R">set.genomic.region.subregion(x, regions, subregions, split = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A bed.matrix </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>regions</code></td>
<td>

<p>A dataframe in bed format (start is 0-based and end is 1-based) containing the regions with the fields : <code>Chr</code> (the chromosome of the gene),
<code>Start</code> (the start position of the gene, 0-based), <code>End</code> (the end position of the gene, 1-based), and
<code>Name</code> (the name of the gene - a factor),
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subregions</code></td>
<td>
<p> A dataframe containing the subregions in the same format as <code>regions</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>split</code></td>
<td>
<p> Whether to split variants attributed to multiple regions by duplicating this variants, set at TRUE by default</p>
</td>
</tr>
</table>
<h3>Details</h3>

 
<p>Warnings: <code>regions$Name</code> and <code>subregions$Name</code> should be factors containing UNIQUE names of the regions, ORDERED in the genome order. 
</p>
<p>If <code>x@snps$chr</code> is not a vector of integers, it should be a factor with same levels as <code>regions$Chr</code>.
</p>
<p>If a variant is attributed to multiple genomic regions, it will be duplicated in the bed matrix with one row per genomic region if <code>split = TRUE</code>.  
</p>
<p>This function can be applied before using <code>burden.subscores</code> to perform a functionally-informed burden tests with sub-scores for each <code>SubRegion</code> within each <code>genomic.region</code>. 
</p>


<h3>Value</h3>

<p>The same bed matrix as x with two additional columns: <code>x@snps$genomic.region</code> containing the annotation of the <code>regions</code> and <code>x@snps$SubRegion</code> containing the annotation of the <code>subregions</code>.
</p>


<h3>See Also</h3>

<p><code>set.genomic.region</code>, <code>burden.subscores</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Import 1000Genome data from region around LCT gene
x &lt;- as.bed.matrix(LCT.gen, LCT.fam, LCT.bim)

#Group variants within known genes and 
#Within coding and regulatory regions
x &lt;- set.genomic.region.subregion(x, 
 regions = genes.b37, subregions = subregions.LCT)
</code></pre>


</div>