<div class="container">

<table style="width: 100%;"><tr>
<td>multinomial.asso.freq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Single variant association test with categorical phenotype
</h2>

<h3>Description</h3>

<p>Performs an association test between categorical phenotypes and single variants
</p>


<h3>Usage</h3>

<pre><code class="language-R">multinomial.asso.freq(x, pheno = x@ped$pheno, ref.level,
                      test = c("Genotypic", "Allelic"), get.effect.size = F,
                      min.maf.threshold = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A bed matrix, only needed if <code>burden="CAST"</code> or <code>burden="WSS"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p> The phenotype of each individual: a factor if <code>pheno.type = "categorical"</code>, and a numeric vector if <code>pheno.type = "continuous"</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref.level</code></td>
<td>
<p> The reference group of individuals for the estimation of the effect size, only needed if <code>get.effect.size = T</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test</code></td>
<td>
<p> Whether to perform the test on the three genotypes ("Genotypic") or on the two alleles ("Allelic")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>get.effect.size</code></td>
<td>
<p> TRUE/FALSE: whether to return effect sizes of the variants (OR) </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.maf.threshold</code></td>
<td>
<p> MAF threshold used to define a frequent variant to apply single-variant test</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This association test is based on a chi-square with the following number of df:  
If <code>test = "Genotypic"</code>, (number of groups of individuals - 1)* 2  
If <code>test = "Allelic"</code>, (number of groups of individuals - 1)  
</p>


<h3>Value</h3>

<p>A dataframe with one row per variant and three columns: the chromosome, position and p-value of each variant.  
If <code>get.effect.size = T</code>, a list with <code>Asso</code> containing the previous dataframe and <code>OR</code> containing the OR in each group for each variant.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Import data in a bed matrix
x &lt;- as.bed.matrix(x=LCT.matrix.bed, fam=LCT.matrix.fam, bim=LCT.snps)

#Add population
x@ped[,c("pop", "superpop")] &lt;- LCT.matrix.pop1000G[,c("population", "super.population")]

#Select EUR superpopulation
x &lt;- select.inds(x, superpop=="EUR")
x@ped$pop &lt;- droplevels(x@ped$pop)

#Perform association test
x.freq.asso &lt;- multinomial.asso.freq(x, test = "Genotypic",
                                     pheno = x@ped$pop)
</code></pre>


</div>