<div class="container">

<table style="width: 100%;"><tr>
<td>SKAT.bootstrap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Multi group SKAT test using bootstrap sampling </h2>

<h3>Description</h3>

<p> Peforms SKAT on two or more groups of individuals using bootstrap sampling</p>


<h3>Usage</h3>

<pre><code class="language-R">  SKAT.bootstrap(x, NullObject, genomic.region = x@snps$genomic.region,
                 weights = (1-x@snps$maf)**24, maf.threshold = 0.5,
                 perm.target = 100, perm.max = 5e4, debug = FALSE,
                 estimation.pvalue = "kurtosis")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A bed.matrix </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>NullObject</code></td>
<td>
<p> A list returned from <code>NullObject.parameters</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomic.region</code></td>
<td>
<p> A factor defining the genomic region of each variant </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p> A vector with the weight of each variant. By default, the weight of each variant is inversely proportionnal to its MAF, as it was computed in the original SKAT method</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf.threshold</code></td>
<td>
<p> The MAF above which variants are removed (default is to keep all variants)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perm.target</code></td>
<td>
<p> The number of times to exceed the observed statistics. If not reached, <code>perm.max</code> permutations will be used </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perm.max</code></td>
<td>
<p> The maximum number of permutations to perform to estimate the p-value, will be used if <code>perm.target</code> is not reached</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>debug</code></td>
<td>
<p> Whether to print details about the permutations (mean, standard deviation, skewness, kurtosis), FALSE by default</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estimation.pvalue</code></td>
<td>
<p> Whether to use the skewness ("skewness") or the kurtosis ("kurtosis") for the chi-square approximation </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>P-values estimation is based on bootstrap sampling and a sequential procedure: permutated statistics are computed and each one is compared to the observed statistics.
The boostrap progam stops when either <code>perm.target</code> or <code>perm.max</code> is reached.
P-values are then computed using a mixed procedure:
</p>
<p>if <code>perm.target</code> is reached, the p-value is computed as : <code>perm.target</code> divided by the number of permutations used to reach <code>perm.target</code>;
</p>
<p>if <code>perm.max</code> is reached, p-values are approximated using a chi-square distributions based on the first three moments if <code>estimation.pvalue = "skewness"</code>, or on statistics' moments 1, 2 and 4 if <code>estimation.pvalue = "kurtosis"</code>.  
</p>
<p>If <code>debug=TRUE</code>, more informations about the estimated statistics moments are given.  
</p>
<p>This function is used by <code>SKAT</code> when the sample size is smaller than 2000 and covariates are present.
</p>
<p>All missing genotypes are imputed by the mean genotype.
</p>


<h3>Value</h3>

<p>A data frame containing for each genomic:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>stat</code></td>
<td>
<p> The observed statistics</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>
 <p><code>p.perm</code> if <code>perm.target</code> is reached, <code>p.chi2</code> if <code>perm.max</code> is reached. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.perm</code></td>
<td>
<p> The p-value computed by permutations: number of times permutated is greater than observed statistics divided by the total number of permutations performed</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.chi2</code></td>
<td>
<p> The p-value computed by the chi-square approximation using the SKAT small sample procedure</p>
</td>
</tr>
</table>
<p>If <code>debug=TRUE</code>, other informations are given about the moments estimation:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>nb.gep</code></td>
<td>
<p> The number of times a permutated statistics is equal or greater than the observed statistics <code>stat</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nb.eq</code></td>
<td>
<p> The number of times a permutated statistics is equal to the observed statistics <code>stat</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nb.perms</code></td>
<td>
<p> The total number of simulations performed </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mean</code></td>
<td>
<p> The mean of the permutated statistics</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma</code></td>
<td>
<p> The standard deviation of the permutated statistics</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>skewness</code></td>
<td>
<p> The skweness of the permutated statistics</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kurtosis</code></td>
<td>
<p> The kurtosis of the permutated statistics</p>
</td>
</tr>
</table>
<h3>References</h3>

<p> Wu et al. 2011, <em>Rare-variant association testing for sequencing data with the sequence kernel association test</em>, American Journal of Human Genetics <b>82-93</b> doi:10.1016/j.ajhg.2011.05.029;  
</p>
<p>Lee et al. 2012, <em>Optimal Unified Approach for Rare-Variant Association Testing with Application to Small-Sample Case-Control Whole-Exome Sequencing Studies</em>, American Journal of Human Genetics, doi:10.1016/j.ajhg.2012.06.007;
</p>


<h3>See Also</h3>

 <p><code>NullObject.parameters</code>, <code>SKAT</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">
#Import data in a bed matrix
x &lt;- as.bed.matrix(x=LCT.matrix.bed, fam=LCT.matrix.fam, bim=LCT.snps)
#Add population
x@ped[,c("pop", "superpop")] &lt;- LCT.matrix.pop1000G[,c("population", "super.population")]

#Select EUR superpopulation
x &lt;- select.inds(x, superpop=="EUR")
x@ped$pop &lt;- droplevels(x@ped$pop)

#Group variants within known genes
x &lt;- set.genomic.region(x)

#Filter of rare variants: only non-monomorphic variants with
#a MAF lower than 1%
#keeping only genomic regions with at least 10 SNPs
x1 &lt;- filter.rare.variants(x, filter = "whole", maf.threshold = 0.01, min.nb.snps = 10)

#Simulation of a covariate + Sex as a covariate
sex &lt;- x1@ped$sex
set.seed(1) ; u &lt;- runif(nrow(x1))
covar &lt;- cbind(sex, u)

#run SKAT using the 1000 genome EUR populations as "outcome"
#The maximum number of permutations used is 100,
#and the target number is 10, please increase
#both values for a more accurate estimation of the p-values
#Fit Null model with covariates
x1.H0 &lt;- NullObject.parameters(x1@ped$pop, data = covar, RVAT = "SKAT", pheno.type = "categorical")

SKAT.bootstrap(x1, x1.H0, perm.target = 10, perm.max = 100)

</code></pre>


</div>