<div class="container">

<table style="width: 100%;"><tr>
<td>am_simulate</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate genotype/phenotype data under equilibrium univariate AM.</h2>

<h3>Description</h3>

<p>Simulate genotype/phenotype data under equilibrium univariate AM.
</p>


<h3>Usage</h3>

<pre><code class="language-R">am_simulate(h2_0, r, m, n, afs = NULL, min_MAF = 0.1, haplotypes = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>h2_0</code></td>
<td>
<p>generation zero (panmictic) heritability</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r</code></td>
<td>
<p>cross-mate phenotypic correlation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>
<p>number of biallelic causal variants</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>sample size</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>afs</code></td>
<td>
<p>(optional). Allele frequencies to use. If not provided, <code>m</code> will be drawn
uniformly from the interval [<code>min_MAF</code>, 1-<code>min_MAF</code>]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_MAF</code></td>
<td>
<p>(optional) minimum minor allele frequency for causal variants.
Ignored if if <code>afs</code> is not NULL. Defaults to 0.1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>haplotypes</code></td>
<td>
<p>logical. If TRUE, includes (phased) haploid genotypes in output.
Defaults to FALSE</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list including the following objects:
</p>

<ul>
<li> <p><code>y</code>: phenotype vector
</p>
</li>
<li> <p><code>g</code>: heritable component of the phenotype vector
</p>
</li>
<li> <p><code>X</code>: matrix of diploid genotypes
</p>
</li>
<li> <p><code>AF</code>: vector of allele frequencies
</p>
</li>
<li> <p><code>beta_std</code>: standardized genetic effects
</p>
</li>
<li> <p><code>beta_raw</code>: unstandardized genetic effects
</p>
</li>
<li> <p><code>H</code>: matrix of haploid genotypes (returned only if <code>haplotypes</code>=TRUE)
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">set.seed(1)
h2_0 = .5; m = 200; n = 1000; r =.5

## simulate genotype/phenotype data
sim_dat &lt;- am_simulate(h2_0, r, m, n)
str(sim_dat)

## empirical h2 vs expected equilibrium h2
(emp_h2 &lt;- var(sim_dat$g)/var(sim_dat$y))
h2_eq(r, h2_0)
</code></pre>


</div>