<div class="container">

<table style="width: 100%;"><tr>
<td>rba_reactome_pathways_low</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get lower level pathways Containing a 'Physical Entity' or Event</h2>

<h3>Description</h3>

<p>Use this function to search the event hierarchy and retrieve a list of
all lower level pathways (non TopLevelPathway class) that contain
a given 'Physical Entity' or Event. See "Arguments section" on how to
modify your search.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_reactome_pathways_low(
  entity_id,
  with_diagram = FALSE,
  all_forms = FALSE,
  species = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>entity_id</code></td>
<td>
<p>The entity that should exist in the pathways.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>with_diagram</code></td>
<td>
<p>Logical: only include pathways with diagram?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>all_forms</code></td>
<td>
<p>Logical: should other variants of your supplied entity_id
be considered? (e.g. same molecule but in different compartment,
secretory form etc.) see <code>rba_reactome_participants</code>'s
"Details section" to learn more about how Reactome classifies molecules.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>species</code></td>
<td>
<p>(optional) Numeric or Character: confine your search to a
specific species by providing it's NCBI Taxonomy identifier
(Human Taxonomy ID is 9606) or species name (e.g. "Homo sapiens").
See <code>rba_reactome_species</code> or
<a href="https://reactome.org/content/schema/objects/Species/">Reactome
Data Schema: Entries: Species</a>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Data frame where each row is a pathway that contains your supplied
entity and columns are pertinent information.
</p>


<h3>Corresponding API Resources</h3>

<p>"GET https://reactome.org/ContentService/data/pathways/low/entity/{id}"
<br> "GET https://reactome.org/ContentService/data/pathways/low/diagram/
entity/{id}"
<br> "GET https://reactome.org/ContentService/data/pathways/low/diagram/
entity/{id}/allForms"
</p>


<h3>References</h3>


<ul>
<li>
<p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter Dâ€™Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li>
<p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Other "Reactome Content Service - Pathway Related Queries": 
<code>rba_reactome_pathways_events()</code>,
<code>rba_reactome_pathways_top()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_reactome_pathways_low(entity_id = "R-HSA-199420")


rba_reactome_pathways_low(entity_id = "R-HSA-199420", with_diagram = TRUE)


rba_reactome_pathways_low(entity_id = "R-HSA-199420", with_diagram = TRUE,
    all_forms = TRUE)


</code></pre>


</div>