<div class="container">

<table style="width: 100%;"><tr>
<td>rba_uniprot_uniparc_search</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Search UniParc Entries</h2>

<h3>Description</h3>

<p>Use this function to search
<a href="https://www.uniprot.org/help/uniparc">UniProt Archive (UniParc)</a>
entries.You may also refine your search with modifiers such as sequence
length, taxon id etc. See "Arguments section" for more information.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_uniprot_uniparc_search(
  upi = NULL,
  accession = NULL,
  db_type = NULL,
  db_id = NULL,
  gene = NULL,
  protein = NULL,
  taxid = NULL,
  organism = NULL,
  sequence_checksum = NULL,
  ipr = NULL,
  signature_db = NULL,
  signature_id = NULL,
  upid = NULL,
  seq_length = NULL,
  rf_dd_type = NULL,
  rf_db_id = NULL,
  rf_active = NULL,
  rf_tax_id = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>upi</code></td>
<td>
<p>unique UniParc Identifier(s). You can supply up to 100 IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>db_type</code></td>
<td>
<p><a href="https://www.uniprot.org/database/">cross-reference</a>
(external database) name.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>db_id</code></td>
<td>
<p>Protein ID in the cross-reference (external) database.
You can supply up to 100 IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene</code></td>
<td>
<p><a href="https://www.uniprot.org/help/gene_name">UniProt gene
name(s)</a>. You can supply up to 20 gene names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>protein</code></td>
<td>
<p><a href="https://www.uniprot.org/help/protein_names">UniProt
protein name</a>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.
You can supply up to 20 taxon IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>organism</code></td>
<td>
<p><a href="https://www.uniprot.org/taxonomy/">Organism name</a>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sequence_checksum</code></td>
<td>
<p>Sequence CRC64 checksum.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ipr</code></td>
<td>
<p><a href="https://www.ebi.ac.uk/interpro/about/interpro/">InterPro
identifier(s)</a>. You can supply up to 20 IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>signature_db</code></td>
<td>
<p>InterPro's
<a href="https://interpro-documentation.readthedocs.io/en/latest/databases.html">signature
database</a>. You can supply up to 13 of the following values:
<br> "CATH", "CDD", "HAMAP", "MobiDB Lite", "Panther", "Pfam", "PIRSF",
"PRINTS", "Prosite", "SFLD", "SMART", "SUPERFAMILY" and/or "TIGRfams"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>signature_id</code></td>
<td>
<p>Signature ID in the InterPro's
<a href="https://interpro-documentation.readthedocs.io/en/latest/databases.html">signature
database</a>. You can supply up to 20 IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>upid</code></td>
<td>
<p><a href="https://www.uniprot.org/help/proteome_id">UniProt Proteome
identifier (UPID)</a>. You can supply up to 100 UPIDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq_length</code></td>
<td>
<p>An exact sequence length (e.g. 150) or a range of sequence
lengths (e.g. "130-158").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rf_dd_type</code></td>
<td>
<p>Filter the content of the each UniParc entry by
<a href="https://www.uniprot.org/database/">cross-reference</a> names. You can
supply multiple values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rf_db_id</code></td>
<td>
<p>Filter the content of the each UniParc entry by protein
identifiers in any cross-reference database. You can supply multiple
values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rf_active</code></td>
<td>
<p>(logical ) Filter the content of each UniParc entry based on
active status on source database:</p>

<ul>
<li>
<p> NULL: (default) don't filter contents based on active status.
</p>
</li>
<li>
<p> TRUE: only return contents which are still active.
</p>
</li>
<li>
<p> FALSE: Only return contents which are not active.</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rf_tax_id</code></td>
<td>
<p>(Numeric) Filter the content of each UniParc entry by
NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>. You can
supply multiple values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
</p>


<h3>Value</h3>

<p>A List where each element corresponds to one UniParc entry returned
by your search query. The element itself is a sub-list containing sequence
information and reference entries.
</p>


<h3>Corresponding API Resources</h3>

<p>"GET https://ebi.ac.uk/proteins/api/uniparc"
</p>


<h3>References</h3>


<ul>
<li>
<p> The UniProt Consortium, UniProt: the universal protein
knowledgebase in 2021, Nucleic Acids Research, Volume 49, Issue D1,
8 January 2021, Pages D480–D489, https://doi.org/10.1093/nar/gkaa1100
</p>
</li>
<li>
<p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Other "UniProt - UniParc": 
<code>rba_uniprot_uniparc()</code>,
<code>rba_uniprot_uniparc_bestguess()</code>,
<code>rba_uniprot_uniparc_sequence()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_uniprot_uniparc_search(upi = "UPI00000000C9")


rba_uniprot_uniparc_search(accession = "P30914")


rba_uniprot_uniparc_search(accession = "P30914", rf_active = TRUE)


rba_uniprot_uniparc_search(taxid = "694009", protein = "Nucleoprotein")


</code></pre>


</div>