<div class="container">

<table style="width: 100%;"><tr>
<td>rba_jaspar_matrix_search</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Search matrix profiles available in JASPAR</h2>

<h3>Description</h3>

<p>You can use this function to list the JASPAR matrix
profiles that match your search query, or run the function without any
arguments to return a list of every matrix profile available in the
latest release.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_jaspar_matrix_search(
  term = NULL,
  tf_name = NULL,
  tf_class = NULL,
  tf_family = NULL,
  tax_group = NULL,
  tax_id = NULL,
  data_type = NULL,
  collection = NULL,
  release = 2024,
  only_last_version = FALSE,
  order = NULL,
  page_size = 1000,
  page = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>term</code></td>
<td>
<p>Character: A search term.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tf_name</code></td>
<td>
<p>Character: Transcription factor names (Case-sensitive).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tf_class</code></td>
<td>
<p>Character: Transcription factor class</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tf_family</code></td>
<td>
<p>Character: Transcription factor family</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tax_group</code></td>
<td>
<p>Character: Taxonomic group. Use
<code>rba_jaspar_taxons</code> to get a list of supported Taxonomic
groups.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tax_id</code></td>
<td>
<p>Numeric: NCBI taxonomic Identifier of species. Use
<code>rba_jaspar_species</code> to get a list of supported Species.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data_type</code></td>
<td>
<p>Character: Type of the data (i.e The Methodology used
for matrix construction). For example: "ChIP-seq", "PBM"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>collection</code></td>
<td>
<p>Character: JASPAR matrix profile collection name. USE
<code>rba_jaspar_collections</code> to get a list of collection names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>release</code></td>
<td>
<p>Numeric: (default = 2024) Which JASPAR database release
to use? Available options are: 2024, 2022, 2020, 2018, 2016, and 2014.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>only_last_version</code></td>
<td>
<p>Logical: (default = FALSE) If TRUE, only the
latest version of a matrix profile will be returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>order</code></td>
<td>
<p>Character: A character string or a vector of character strings
of field names that will be used to order the results.
<br> Providing multiple field names is supported. You can also use prefix
"-" before a field name to indicate reverse ordering.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>page_size</code></td>
<td>
<p>Numeric: (default = 1000) This resource returns paginated
results. What is the maximum numbers of results that you want to retrieve
per a page? Accepted values are between 1 and 1000.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>page</code></td>
<td>
<p>Numeric: Which page of the results to retrieve? The accepted
values depend on the page size and number of results.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
<br> The results are paginated. You can control the page's size number
with the function's arguments. Also, you can use <code>rba_pages</code>
to automatically iterate over multiple pages.
</p>


<h3>Value</h3>

<p>A list that contains a data frame of matrix profiles' information.
</p>


<h3>Corresponding API Resources</h3>

<p>"GET https://jaspar.elixir.no/api/v1/api/v1/matrix/"
</p>


<h3>References</h3>


<ul>
<li>
<p> Rauluseviciute I, Riudavets-Puig R, Blanc-Mathieu R,
Castro-Mondragon JA, Ferenc K, Kumar V, Lemma RB, Lucas J, Ch√®neby J,
Baranasic D, Khan A, Fornes O, Gundersen S, Johansen M, Hovig E, Lenhard
B, Sandelin A, Wasserman WW, Parcy F, Mathelier A JASPAR 2024:
20th anniversary of the open-access database of transcription factor
binding profiles Nucleic Acids Res. in_press; doi: 10.1093/nar/gkad1059
</p>
</li>
<li>
<p> Khan, A. and Mathelier, A. JASPAR RESTful API: accessing JASPAR data
from any programming language. Bioinformatics, 2017,
doi: 10.1093/bioinformatics/btx804
</p>
</li>
<li>
<p><a href="https://jaspar.elixir.no/api/v1/docs/">JASPAR API Documentation</a>
</p>
</li>
<li> <p><a href="https://jaspar.elixir.no/faq/">Citations note
on JASPAR website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Other "JASPAR": 
<code>rba_jaspar_collections()</code>,
<code>rba_jaspar_collections_matrices()</code>,
<code>rba_jaspar_matrix()</code>,
<code>rba_jaspar_matrix_versions()</code>,
<code>rba_jaspar_releases()</code>,
<code>rba_jaspar_sites()</code>,
<code>rba_jaspar_species()</code>,
<code>rba_jaspar_species_matrices()</code>,
<code>rba_jaspar_taxons()</code>,
<code>rba_jaspar_taxons_matrices()</code>,
<code>rba_jaspar_tffm()</code>,
<code>rba_jaspar_tffm_search()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_jaspar_matrix_search(term = "FOX")
rba_jaspar_matrix_search(tf_name = "FOXP3")
rba_jaspar_matrix_search(tf_name = "FOXP3", only_last_version = TRUE)
rba_jaspar_matrix_search(tf_class = "Zipper-Type")
rba_jaspar_matrix_search(tax_group = "insects")
rba_jaspar_matrix_search(page_size = 100)


</code></pre>


</div>