<div class="container">

<table style="width: 100%;"><tr>
<td>rba_panther_info</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get PANTHER database Information</h2>

<h3>Description</h3>

<p>Using this function you can retrieve a list of available organisms,
annotation datasets, families, and pathways which are supported in
PANTHER.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_panther_info(what, organism_chr_loc = FALSE, families_page = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>what</code></td>
<td>
<p>what information to retrieve? should be one of: </p>

<ul>
<li>
<p> "organisms": Retrieve supported organisms in PANTHER.
</p>
</li>
<li>
<p> "datasets": Retrieve available annotation datasets.
</p>
</li>
<li>
<p> "families" Retrieve available family IDs.
</p>
</li>
<li>
<p> "species_tree" Retrieve the PANThER's species tree.
</p>
</li>
<li>
<p> "pathways" Retrieve available pathway IDs.</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>organism_chr_loc</code></td>
<td>
<p>(Logical) (only when 'what = "organisms"')
If TRUE, only organisms with chromosome location will be returned.
If FALSE (default) every organisms will be returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>families_page</code></td>
<td>
<p>(Numeric) (only when 'what = "families"')
Family information is very long, so results are paginated. Use this
argument to define the page to retrieve.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>For families and species tree, a list and otherwise a data frame
with pertinent information.
</p>


<h3>Corresponding API Resources</h3>

<p>"GET https://www.pantherdb.org/services/oai/pantherdb/supportedgenomes"
<br> "GET https://www.pantherdb.org/services/oai/pantherdb/supportedannotdatasets"
<br> "GET https://www.pantherdb.org/services/oai/pantherdb/supportedpantherfamilies"
<br> "GET https://www.pantherdb.org/services/oai/pantherdb/supportedpantherpathways"
<br> "GET https://www.pantherdb.org/services/oai/pantherdb/speciestree"
</p>


<h3>References</h3>


<ul>
<li>
<p> Huaiyu Mi, Dustin Ebert, Anushya Muruganujan, Caitlin Mills,
Laurent-Philippe Albou, Tremayne Mushayamaha, Paul D Thomas, PANTHER
version 16: a revised family classification, tree-based classification
tool, enhancer regions and extensive API, Nucleic Acids Research,
Volume 49, Issue D1, 8 January 2021, Pages D394â€“D403,
https://doi.org/10.1093/nar/gkaa1106
</p>
</li>
<li> <p><a href="https://www.pantherdb.org/services/details.jsp">PANTHER
Services Details</a>
</p>
</li>
<li>
<p><a href="https://www.pantherdb.org/publications.jsp#HowToCitePANTHER">Citations
note on PANTHER website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Other "PANTHER": 
<code>rba_panther_enrich()</code>,
<code>rba_panther_family()</code>,
<code>rba_panther_homolog()</code>,
<code>rba_panther_mapping()</code>,
<code>rba_panther_ortholog()</code>,
<code>rba_panther_tree_grafter()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_panther_info(what = "organisms")


rba_panther_info(what = "families", families_page = 4)


</code></pre>


</div>