<div class="container">

<table style="width: 100%;"><tr>
<td>rba_string_interactions_network</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get STRING Network Interactions</h2>

<h3>Description</h3>

<p>This function will retrieve Sting interaction pairs among your input protein
ids, with the combined score and separate score for each STRING score
channels. You can further expand your network to a defined size by
providing "add_node" parameter.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_string_interactions_network(
  ids,
  species = NULL,
  required_score = NULL,
  add_nodes = NULL,
  network_type = "functional",
  use_query_labels = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p>Your protein IDs. It is strongly recommended to supply
STRING IDs. See <code>rba_string_map_ids</code> for more information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>species</code></td>
<td>
<p>Numeric: NCBI Taxonomy identifier; Human Taxonomy ID is 9606.
(Recommended, but optional if your input is less than 100 IDs.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>required_score</code></td>
<td>
<p>Numeric: A minimum of interaction score for an
interaction to be included in the image. if not supplied, the threshold
will be applied by STRING Based in the network. (low Confidence = 150,
Medium Confidence = 400, High Confidence = 700, Highest confidence = 900)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add_nodes</code></td>
<td>
<p>Numeric: Number of neighboring proteins to be added to the
network. If none supplied by the user, this argument value will depend
on the number of supplied "ids" argument:</p>

<ol>
<li>
<p> Single id: add_node will be set to 10 to retrieve the interaction
neighborhood  of you input protein.
</p>
</li>
<li>
<p> Multiple ids: add_node will be set to 0, thus the output will be the
interactions between your input proteins.</p>
</li>
</ol>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>network_type</code></td>
<td>
<p>should be one of:</p>

<ul>
<li>
<p> "functional": (default) The edge's indicate both physical and
functional associations.
</p>
</li>
<li>
<p> "physical": The edges indicate that two proteins have a physical
interaction or are parts of a complex.</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use_query_labels</code></td>
<td>
<p>Logical: (Default = FALSE) Use the names supplied
with the 'ids' argument as the nodes labels instead of STRING's default
ones.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Note that this function will return interactions between your set of
supplied proteins, or at most, expand the interaction network by the
given parameters. TO retrieve a list of all possible interacting proteins
with your given input, see
<code>rba_string_interaction_partners</code>.
</p>


<h3>Value</h3>

<p>A data frame which each row is a network interaction and the
columns contains interactor information and interaction scores: </p>

<ul>
<li>
<p> stringId_A: STRING identifier (protein A)
</p>
</li>
<li>
<p> stringId_B:STRING identifier (protein B)
</p>
</li>
<li>
<p> preferredName_A: common protein name (protein A)
</p>
</li>
<li>
<p> preferredName_B: common protein name (protein B)
</p>
</li>
<li>
<p> ncbiTaxonId: NCBI taxon identifier
</p>
</li>
<li>
<p> score: combined score
</p>
</li>
<li>
<p> nscore: gene neighborhood score
</p>
</li>
<li>
<p> fscore: gene fusion score
</p>
</li>
<li>
<p> pscore: phylogenetic profile score
</p>
</li>
<li>
<p> ascore: co-expression score
</p>
</li>
<li>
<p> escore: experimental score
</p>
</li>
<li>
<p> dscore: database score
</p>
</li>
<li>
<p> tscore: textmining score</p>
</li>
</ul>
<h3>Corresponding API Resources</h3>

<p>"POST https://string-db.org/api/{output-format}/network?identifiers=
{your_identifiers}&amp;{optional_parameters}"
</p>


<h3>References</h3>


<ul>
<li>
<p> Damian Szklarczyk, Rebecca Kirsch, Mikaela Koutrouli, Katerina
Nastou, Farrokh Mehryary, Radja Hachilif, Annika L Gable, Tao Fang,
Nadezhda T Doncheva, Sampo Pyysalo, Peer Bork, Lars J Jensen, Christian
von Mering, The STRING database in 2023: protein–protein association
networks and functional enrichment analyses for any sequenced genome of
interest, Nucleic Acids Research, Volume 51, Issue D1, 6 January 2023,
Pages D638–D646, https://doi.org/10.1093/nar/gkac1000
</p>
</li>
<li> <p><a href="https://string-db.org/help/api/">STRING API Documentation</a>
</p>
</li>
<li>
<p><a href="https://string-db.org/cgi/about?footer_active_subpage=references">
Citations note on STRING website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p><code>rba_string_map_ids, rba_string_interaction_partners</code>
</p>
<p>Other "STRING": 
<code>rba_string_annotations()</code>,
<code>rba_string_enrichment()</code>,
<code>rba_string_enrichment_ppi()</code>,
<code>rba_string_homology_inter()</code>,
<code>rba_string_homology_intra()</code>,
<code>rba_string_interaction_partners()</code>,
<code>rba_string_map_ids()</code>,
<code>rba_string_network_image()</code>,
<code>rba_string_version()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_string_interactions_network(ids = c("9606.ENSP00000269305",
    "9606.ENSP00000398698",
    "9606.ENSP00000275493"),
    network_type = "functional")


rba_string_interactions_network(ids = c("9606.ENSP00000269305",
    "9606.ENSP00000398698",
    "9606.ENSP00000275493"),
    species = 9606,
    add_nodes = 10)


</code></pre>


</div>