<div class="container">

<table style="width: 100%;"><tr>
<td>rba_string_homology_intra</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get Similarity Scores Hits of Proteins in a Species</h2>

<h3>Description</h3>

<p>Using this function, you can retrieve the Smith-Waterman bit scores among
proteins of the same species.
Bit Scores serve as similarity scores between protein sequence;
And, according to STRING documentations, as a proxy for protein homology.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_string_homology_intra(ids, species = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p>Your protein ID(s). It is strongly recommended to supply
STRING IDs. See <code>rba_string_map_ids</code> for more information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>species</code></td>
<td>
<p>Numeric: NCBI Taxonomy identifier; Human Taxonomy ID is 9606.
(Recommended, but optional if your input is less than 100 IDs.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Note that this function will retrieve similarity scores of different
proteins "within the same species". To Get a similarity scores of a given
protein and it's closets homologous proteins in other species, see
<code>rba_string_homology_inter</code>.
<br> Similarity matrix is imported -by STRING- from:
<a href="https://cube.univie.ac.at/resources/simap">Similarity Matrix of
Proteins (SIMAP)</a>
</p>


<h3>Value</h3>

<p>A data frame with bit scores between your supplied proteins and
their self-hit. To Reduce the transferred data, STRING returns only one
half of the similarity matrix; This will not pose a problem because
similarity matrix is symmetrical.
</p>


<h3>Corresponding API Resources</h3>

<p>"POST https://string-db.org/api/{output-format}/homology?identifiers=
{your_identifiers}"
</p>


<h3>References</h3>


<ul>
<li>
<p> Damian Szklarczyk, Rebecca Kirsch, Mikaela Koutrouli, Katerina
Nastou, Farrokh Mehryary, Radja Hachilif, Annika L Gable, Tao Fang,
Nadezhda T Doncheva, Sampo Pyysalo, Peer Bork, Lars J Jensen, Christian
von Mering, The STRING database in 2023: protein–protein association
networks and functional enrichment analyses for any sequenced genome of
interest, Nucleic Acids Research, Volume 51, Issue D1, 6 January 2023,
Pages D638–D646, https://doi.org/10.1093/nar/gkac1000
</p>
</li>
<li> <p><a href="https://string-db.org/help/api/">STRING API Documentation</a>
</p>
</li>
<li>
<p><a href="https://string-db.org/cgi/about?footer_active_subpage=references">
Citations note on STRING website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p><code>rba_string_map_ids, rba_string_homology_inter</code>
</p>
<p>Other "STRING": 
<code>rba_string_annotations()</code>,
<code>rba_string_enrichment()</code>,
<code>rba_string_enrichment_ppi()</code>,
<code>rba_string_homology_inter()</code>,
<code>rba_string_interaction_partners()</code>,
<code>rba_string_interactions_network()</code>,
<code>rba_string_map_ids()</code>,
<code>rba_string_network_image()</code>,
<code>rba_string_version()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_string_homology_intra(ids = c("CDK1", "CDK2"), species = 9606)


</code></pre>


</div>