<div class="container">

<table style="width: 100%;"><tr>
<td>rba_reactome_analysis_species</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compare Human Pathways with with Other Species</h2>

<h3>Description</h3>

<p>Use This function to Compare human's manually-curated pathways
and computationally inferred pathways (orthologous) in other species.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_reactome_analysis_species(
  species_dbid,
  sort_by = "ENTITIES_PVALUE",
  order = "ASC",
  resource = "TOTAL",
  p_value = 1,
  min = NULL,
  max = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>species_dbid</code></td>
<td>
<p>Numeric: Reactome DbId (e.g  Mus musculus is 48892) of
the species you want to compare with Homo sapiens. See
<code>rba_reactome_species</code> or
<a href="https://reactome.org/content/schema/objects/Species/">Reactome
Data Schema: Entries: Species</a>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sort_by</code></td>
<td>
<p>Sort the result based on what column? available choices
are: "NAME", "TOTAL_ENTITIES", "TOTAL_INTERACTORS", "TOTAL_REACTIONS",
"FOUND_ENTITIES", "FOUND_INTERACTORS", "FOUND_REACTIONS", "ENTITIES_RATIO",
"ENTITIES_PVALUE", "ENTITIES_FDR" or "REACTIONS_RATIO"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>order</code></td>
<td>
<p>Sort Order. Can be either "ASC" (default) or "DESC".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resource</code></td>
<td>
<p>Filter results based on the resource. Default is "TOTAL",
available choices are: "TOTAL", "UNIPROT", "ENSEMBL", "CHEBI", "IUPHAR",
"MIRBASE", "NCBI_PROTEIN", "EMBL", "COMPOUND", "ENTITIES_FDR" or
"PUBCHEM_COMPOUND".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_value</code></td>
<td>
<p>Set a P value threshold. Only results with P value equal to
or less than your supplied threshold will be returned. (default = 1,
Meaning no P value filtering)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min</code></td>
<td>
<p>(numeric) Minimum number of entities that a pathways should have
to be included in the results.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max</code></td>
<td>
<p>(numeric) Maximum number of entities that a pathways should have
to be included in the results.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Reactome incorporate manually curated human reactions and PANTHER's
protein homology data to Computationally infer events in other eukaryotic
species.
<br> In version 73 (11 June 2020), using an orthology-based approach,
Homo sapiens events was projected to 18,654 orthologous pathways (with
81,835 orthologous proteins) in 15 non-human species.
See <a href="https://reactome.org/documentation/inferred-events/">
Reactome Computationally Inferred Events</a> for more information.
</p>


<h3>Value</h3>

<p>List with the results of the comparison.
</p>


<h3>Corresponding API Resources</h3>

<p>"GET https://reactome.org/AnalysisService/species/homoSapiens/{species}"
</p>


<h3>References</h3>


<ul>
<li>
<p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter Dâ€™Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li>
<p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/AnalysisService/">Reactome Analysis
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite">Citations note on Reactome website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p><code>rba_reactome_orthology</code>
</p>
<p>Other "Reactome Analysis Service": 
<code>rba_reactome_analysis()</code>,
<code>rba_reactome_analysis_download()</code>,
<code>rba_reactome_analysis_import()</code>,
<code>rba_reactome_analysis_mapping()</code>,
<code>rba_reactome_analysis_pdf()</code>,
<code>rba_reactome_analysis_token()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_reactome_analysis_species(species_dbid = 48892)


</code></pre>


</div>