<div class="container">

<table style="width: 100%;"><tr>
<td>rba_uniprot_uniparc_sequence</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get UniParc Entries by Sequence</h2>

<h3>Description</h3>

<p>Retrieve UniParc Entry by providing an exact sequence. Note that partial
matches will not be accepted. You can also filter the returned content of
the returned UniParc entry. see "Argument" section for more details.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_uniprot_uniparc_sequence(
  sequence,
  rf_dd_type = NULL,
  rf_db_id = NULL,
  rf_active = NULL,
  rf_tax_id = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sequence</code></td>
<td>
<p>Exact UniParc protein sequence. Partial matches will not be
accepted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rf_dd_type</code></td>
<td>
<p>Filter the content of the UniParc entry by
<a href="https://www.uniprot.org/database/">cross-reference</a> names. You can
supply multiple values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rf_db_id</code></td>
<td>
<p>Filter the content of the UniParc entry by protein
identifiers in any cross-reference database. You can supply multiple
values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rf_active</code></td>
<td>
<p>(logical ) Filter the content of UniParc entry based on
active status on source database:</p>

<ul>
<li>
<p> NULL: (default) don't filter contents based on active status.
</p>
</li>
<li>
<p> TRUE: only return contents which are still active.
</p>
</li>
<li>
<p> FALSE: Only return contents which are not active.</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rf_tax_id</code></td>
<td>
<p>(Numeric) Filter the content of the UniParc entry by
NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>. You can
supply multiple values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list which correspond to a UniParc entry.
</p>


<h3>Corresponding API Resources</h3>

<p>"POST https://ebi.ac.uk/proteins/api/uniparc/sequence"
</p>


<h3>References</h3>


<ul>
<li>
<p> The UniProt Consortium, UniProt: the universal protein
knowledgebase in 2021, Nucleic Acids Research, Volume 49, Issue D1,
8 January 2021, Pages D480–D489, https://doi.org/10.1093/nar/gkaa1100
</p>
</li>
<li>
<p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Other "UniProt - UniParc": 
<code>rba_uniprot_uniparc()</code>,
<code>rba_uniprot_uniparc_bestguess()</code>,
<code>rba_uniprot_uniparc_search()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_uniprot_uniparc_sequence("GMRSCPRGCSQRGRCENGRCVCNPGYTGEDC")


</code></pre>


</div>