<div class="container">

<table style="width: 100%;"><tr>
<td>rba_reactome_analysis</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reactome Over-Representation or Expression Analysis</h2>

<h3>Description</h3>

<p>Using this function, you can perform Reactome Analysis In a convenient way.
The Analysis Type will be chosen depending on your supplied
input:</p>

<ol>
<li>
<p> If you supply a vector or a single-columned table,
"Over-Representation" analysis will be performed.
</p>
</li>
<li>
<p> If you supply a multi-column table, with the first column being
molecules identifiers and the rest being numeral expression values,
"Expression" analysis will be performed.</p>
</li>
</ol>
<p>See the details section for the accepted input types and format.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_reactome_analysis(
  input,
  input_format = NULL,
  projection = TRUE,
  interactors = FALSE,
  species = NULL,
  sort_by = "ENTITIES_PVALUE",
  order = "ASC",
  resource = "TOTAL",
  p_value = 1,
  include_disease = TRUE,
  min = NULL,
  max = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>input</code></td>
<td>
<p>A vector, data frame, matrix or a local file path or URL
that points to your data. See "Details section" for more information of
how to organize and supply your input.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>input_format</code></td>
<td>
<p>(Optional) This function will automatically identify
your supplied input's format. But in case of unexpected issues or if you
want to be explicit, set this argument to one of:</p>

<ul>
<li>
<p> "table": If you supplied a data frame or matrix as input.
</p>
</li>
<li>
<p> "vector": If you supplied a simple vector (numeric or character) as
input.
</p>
</li>
<li>
<p> "file": If you supplied a local file path pointing to a
correctly-formatted text file.
</p>
</li>
<li>
<p> "url": If you supplied a URL pointing to a correctly-formatted
text file.</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>projection</code></td>
<td>
<p>Logical (default = TRUE) Should non-human identifiers
be projected to their human equivalents? (using Reactome orthology data)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>interactors</code></td>
<td>
<p>Logical (default = FALSE) Should IntAct interaction data
be used to increase the analysis background?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>species</code></td>
<td>
<p>Numeric or Character: NCBI Taxonomy identifier (Human
is 9606), species name (e.g. "Homo sapiens") or Reactome DbId (e.g
Homo sapiens is 48887). See
<code>rba_reactome_species</code> or
<a href="https://reactome.org/content/schema/objects/Species/">Reactome
Data Schema: Entries: Species</a>. Note that you cannot supply the species
parameter when projection parameter is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sort_by</code></td>
<td>
<p>Sort the result based on what column? available choices
are: "NAME", "TOTAL_ENTITIES", "TOTAL_INTERACTORS", "TOTAL_REACTIONS",
"FOUND_ENTITIES", "FOUND_INTERACTORS", "FOUND_REACTIONS", "ENTITIES_RATIO",
"ENTITIES_PVALUE", "ENTITIES_FDR" or "REACTIONS_RATIO"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>order</code></td>
<td>
<p>Sort Order. Can be either "ASC" (default) or "DESC".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resource</code></td>
<td>
<p>Filter results based on the resource. Default is "TOTAL",
available choices are: "TOTAL", "UNIPROT", "ENSEMBL", "CHEBI", "IUPHAR",
"MIRBASE", "NCBI_PROTEIN", "EMBL", "COMPOUND", "ENTITIES_FDR" or
"PUBCHEM_COMPOUND".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_value</code></td>
<td>
<p>Set a P value threshold. Only results with P value equal to
or less than your supplied threshold will be returned. (default = 1,
Meaning no P value filtering)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>include_disease</code></td>
<td>
<p>Logical (default = TRUE) Should the disease pathways
be included in the results?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min</code></td>
<td>
<p>(numeric) Minimum number of entities that a pathways should have
to be included in the results.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max</code></td>
<td>
<p>(numeric) Maximum number of entities that a pathways should have
to be included in the results.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>You can supply your table or vector input in numerous formats:</p>

<ol>
<li>
<p> A R object which can be data frame, matrix or a simple vector.
</p>
</li>
<li>
<p> A path to a local text file in your device that contains the molecules
data. (The file should be formatted correctly, see below.)
</p>
</li>
<li>
<p> A URL pointing to a text file on the web that contains the molecules
data. (The file should be formatted correctly, see below.</p>
</li>
</ol>
<p>If you supply a text file (as a local file path or URL), it should be
in TSV (Tab-Separated Values) format; Column names should start with "#"
character. Note that if you are providing the file for
"Over-Representation" analysis (i.e. Single columned-data) this header
line is optional and will be used as your 'Sample Name', otherwise it is
required.
<br> Also, form the "summary" element in the function's output, you can see
how Reactome Interpreted your input and subsequently the type of analysis
that has been performed.
<br> There is no strict criteria about the type of your molecules Identifiers,
Reactome will Map the IDs to it's internal database entities.
Nevertheless, You can check if all your identifiers has been found in
"identifiersNotFound" element in the function's output.
<br> After Any Analysis, Reactome will associate a token to your analysis.
It can be later used to in function that requires the token (e.g to retrieve
the analysis results, download pdf).
<br> Note that Reactome will store your token for only 7 days. You can
download your full results with
<code>rba_reactome_analysis_download</code>, and re-import it anytime to
reactome (using <code>rba_reactome_analysis_import</code>) to generate
a new token.
</p>


<h3>Value</h3>

<p>List containing the results and information of your analysis. Note
that you can use the token returned in the "summary" sub-list of the
results (i.e. results$summary$token) to retrieve your results later or
in other Reactome analysis functions.
</p>


<h3>Corresponding API Resources</h3>

<p>"POST https://reactome.org/AnalysisService/identifiers/form"
<br> "POST https://reactome.org/AnalysisService/identifiers/url"
<br> "POST https://reactome.org/AnalysisService/identifiers/form/projection"
<br> "POST https://reactome.org/AnalysisService/identifiers/url/projection"
</p>


<h3>References</h3>


<ul>
<li>
<p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter Dâ€™Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li>
<p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/AnalysisService/">Reactome Analysis
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite">Citations note on Reactome website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Other "Reactome Analysis Service": 
<code>rba_reactome_analysis_download()</code>,
<code>rba_reactome_analysis_import()</code>,
<code>rba_reactome_analysis_mapping()</code>,
<code>rba_reactome_analysis_pdf()</code>,
<code>rba_reactome_analysis_species()</code>,
<code>rba_reactome_analysis_token()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
rba_reactome_analysis(input = c("p53", "BRCA1", "cdk2", "Q99835", "CDC42"))

## End(Not run)
## Not run: 
rba_reactome_analysis(input = "c:/rbioapi/genes.txt")

## End(Not run)
## Not run: 
rba_reactome_analysis(input = "https://qazwsx.com/genes.txt")

## End(Not run)

</code></pre>


</div>