<div class="container">

<table style="width: 100%;"><tr>
<td>rba_uniprot_features</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get UniProt protein sequence features by accession</h2>

<h3>Description</h3>

<p>Use this function to retrieve
<a href="https://www.uniprot.org/help/sequence_annotation">sequence annotations
(features)</a> of a protein by it's UniProt accession.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_uniprot_features(accession, types = NULL, categories = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>types</code></td>
<td>
<p><a href="https://www.uniprot.org/help/sequence_annotation">
Sequence annotation (Features)</a> types. accepted values
are: "INIT_MET", "SIGNAL", "PROPEP", "TRANSIT", "CHAIN", "PEPTIDE",
"TOPO_DOM", "TRANSMEM", "DOMAIN", "REPEAT", "CA_BIND", "ZN_FING",
"DNA_BIND", "NP_BIND", "REGION", "COILED", "MOTIF", "COMPBIAS",
"ACT_SITE", "METAL", "BINDING", "SITE", "NON_STD", "MOD_RES", "LIPID",
"CARBOHYD", "DISULFID", "CROSSLNK", "VAR_SEQ", "VARIANT", "MUTAGEN",
"UNSURE", "CONFLICT", "NON_CONS", "NON_TER", "HELIX", "TURN", "STRAND"
and/or "INTRAMEM". You can supply up to 20 types.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>categories</code></td>
<td>
<p><a href="https://www.uniprot.org/help/sequence_annotation">
Sequence annotation (Features)</a> categories (subsection). accepted values
are: "MOLECULE_PROCESSING", "TOPOLOGY", "SEQUENCE_INFORMATION",
"STRUCTURAL", "DOMAINS_AND_SITES", "PTM", "VARIANTS" and/or "MUTAGENESIS".
You can supply up to 8 categories.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list in which you can find all of your given protein's sequence
annotations in a sub-list named "features".
</p>


<h3>Corresponding API Resources</h3>

<p>"GET https://www.ebi.ac.uk/proteins/api/features/{accession}"
</p>


<h3>References</h3>


<ul>
<li>
<p> The UniProt Consortium, UniProt: the universal protein
knowledgebase in 2021, Nucleic Acids Research, Volume 49, Issue D1,
8 January 2021, Pages D480–D489, https://doi.org/10.1093/nar/gkaa1100
</p>
</li>
<li>
<p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Other "UniProt - Features": 
<code>rba_uniprot_features_search()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_uniprot_features("Q99616")


rba_uniprot_features(accession = "Q99616", types = "DISULFID")


</code></pre>


</div>