<div class="container">

<table style="width: 100%;"><tr>
<td>rba_panther_ortholog</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Search PANTHER for Orthologs of Gene(s)</h2>

<h3>Description</h3>

<p>Using this function you can search and retrieve orthologs of given gene(s),
and optionally return the corresponding position in the target organisms'
protein sequences.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_panther_ortholog(
  genes,
  organism,
  type = "all",
  target_organisms = NULL,
  seq_pos = NULL,
  include_msa = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genes</code></td>
<td>
<p>Character vector of genes identifiers with maximum length of
10 or only one if seq_pos is supplied. Can be any of: Ensemble gene ID,
Ensemble protein ID, Ensemble transcript ID, Entrez gene ID, gene symbol,
NCBI GI, HGNC ID, International protein index ID, NCBI UniGene ID,
UniProt accession and/or UniProt ID.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>organism</code></td>
<td>
<p>(numeric) NCBI taxon ID of the organism of your supplied
genes. run <code>rba_panther_info</code> with argument
'what = "organisms"' to get a list of PANTHER's supported organisms.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Ortholog types to return. either "all" (default) or "LDO" to
only return least diverged orthologs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>target_organisms</code></td>
<td>
<p>(numeric) NCBI taxon ID(s) to filter the results.
run <code>rba_panther_info</code> with argument 'what = "organisms"' to
get a list of PANTHER's supported organisms.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq_pos</code></td>
<td>
<p>(Numeric) A position in the protein's sequence of the
supplied gene. should be in the range of the protein's length.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>include_msa</code></td>
<td>
<p>(Logical) Only if a sequence position is supplied,
should MSA (Multiple Sequence Alignment) information be included in the
results?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame with Orthologs information.
</p>


<h3>Corresponding API Resources</h3>

<p>"POST https://www.pantherdb.org/services/oai/pantherdb/ortholog/matchortho"
<br> "POST https://www.pantherdb.org/services/oai/pantherdb/ortholog/homologpos"
</p>


<h3>References</h3>


<ul>
<li>
<p> Huaiyu Mi, Dustin Ebert, Anushya Muruganujan, Caitlin Mills,
Laurent-Philippe Albou, Tremayne Mushayamaha, Paul D Thomas, PANTHER
version 16: a revised family classification, tree-based classification
tool, enhancer regions and extensive API, Nucleic Acids Research,
Volume 49, Issue D1, 8 January 2021, Pages D394â€“D403,
https://doi.org/10.1093/nar/gkaa1106
</p>
</li>
<li> <p><a href="https://www.pantherdb.org/services/details.jsp">PANTHER
Services Details</a>
</p>
</li>
<li>
<p><a href="https://www.pantherdb.org/publications.jsp#HowToCitePANTHER">Citations
note on PANTHER website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Other "PANTHER": 
<code>rba_panther_enrich()</code>,
<code>rba_panther_family()</code>,
<code>rba_panther_homolog()</code>,
<code>rba_panther_info()</code>,
<code>rba_panther_mapping()</code>,
<code>rba_panther_tree_grafter()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_panther_ortholog("CD40", organism = 9606, type = "LDO")


</code></pre>


</div>