<div class="container">

<table style="width: 100%;"><tr>
<td>rba_uniprot_variation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get natural variants in UniProt by NIH-NCBI SNP database identifier</h2>

<h3>Description</h3>

<p>Retrieve natural variant annotations of a sequence using UniProt protein
accession, dbSNP or HGVS expression.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_uniprot_variation(
  id,
  id_type,
  source_type = NULL,
  consequence_type = NULL,
  wild_type = NULL,
  alternative_sequence = NULL,
  location = NULL,
  save_peff = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p>An ID which can be either a
<a href="https://www.uniprot.org/help/accession_numbers">UniProt primary or
secondary accession</a>, NIH-NCBI dbSNP ID or HGVS expression.
<a href="https://www.ncbi.nlm.nih.gov/snp/">NIH-NCBI dbSNP id</a> or
<a href="https://varnomen.hgvs.org/">HGVS Expression</a>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id_type</code></td>
<td>
<p>The type of supplied ID argument, one of:
<a href="https://www.uniprot.org/help/accession_numbers">"uniprot"</a>,
<a href="https://www.ncbi.nlm.nih.gov/snp/">"dbsnp"</a> or
<a href="https://varnomen.hgvs.org/">"hgvs"</a></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>source_type</code></td>
<td>
<p>Variation's source type. You can choose up to two of:
"UniProt", "large scale study" and/or "mixed".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>consequence_type</code></td>
<td>
<p>Variation's consequence type. You can choose up to
two of: "missense", "stop gained" or "stop lost".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>wild_type</code></td>
<td>
<p>Wild type amino acid. Accepted values are IUPAC
single-letter amino acid (e.g. D for    Aspartic acid) and "*" for stop
codon. You can supply up to 20 values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alternative_sequence</code></td>
<td>
<p>Alternative amino acid. Accepted values are IUPAC
single-letter amino acid (e.g. D for    Aspartic acid) and "*" for stop codon
and "-" for deletion. You can supply up to 20 values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>location</code></td>
<td>
<p>A valid amino acid range (e.g. 10-25) within the sequence
range where the variation occurs. You can supply up to 20 values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save_peff</code></td>
<td>
<p>Logical or Character:</p>

<ul>
<li>
<p> FALSE: (default) Do not save PEFF file, just return as a list object.
</p>
</li>
<li>
<p> TRUE: Save as PEFF file to an automatically-generated path.
</p>
</li>
<li>
<p> Character string: A valid file path to save the PEFF file.</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list where each element is a list that corresponds to a UniProt
protein entry.
</p>


<h3>Corresponding API Resources</h3>

<p>"GET https://www.ebi.ac.uk/proteins/api/variation/dbsnp/{dbid}"
<br> "GET https://www.ebi.ac.uk/proteins/api/variation/hgvs/{hgvs}"
<br> "GET https://www.ebi.ac.uk/proteins/api/variation/{accession}"
</p>


<h3>References</h3>


<ul>
<li>
<p> The UniProt Consortium, UniProt: the universal protein
knowledgebase in 2021, Nucleic Acids Research, Volume 49, Issue D1,
8 January 2021, Pages D480–D489, https://doi.org/10.1093/nar/gkaa1100
</p>
</li>
<li>
<p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Other "UniProt - Variation": 
<code>rba_uniprot_variation_search()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_uniprot_variation(id = "rs121434451", id_type = "dbsnp")


rba_uniprot_variation(id = "NC_000008.11:g.22119227C&gt;T", id_type = "hgvs")


rba_uniprot_variation(id = "O43593", id_type = "uniprot")


</code></pre>


</div>