<div class="container">

<table style="width: 100%;"><tr>
<td>sequences</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>DNA sequence associated with each taxonomic identifier.</h2>

<h3>Description</h3>

<p>DNA sequence associated with each taxonomic identifier.
</p>


<h3>Usage</h3>

<pre><code class="language-R">sequences(biom)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>biom</code></td>
<td>
<p>A <code>BIOM</code> object, as returned from read.biom.</p>
</td>
</tr></table>
<h3>Value</h3>

<p>A named character vector of sequences in <code>biom</code>. If this data
is not present, then returns <code>NULL</code>.
</p>


<h3>See Also</h3>

<p>Other accessor functions: 
<code>counts()</code>,
<code>info()</code>,
<code>metadata()</code>,
<code>nsamples()</code>,
<code>ntaxa()</code>,
<code>phylogeny()</code>,
<code>sample.names()</code>,
<code>taxa.names()</code>,
<code>taxa.ranks()</code>,
<code>taxonomy()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    
    sequences(biom)[1:4]
    
    # Write to a compressed fasta file in the temporary directory:
    seqs &lt;- sequences(biom)
    conn &lt;- bzfile(file.path(tempdir(), "Sequences.fa.bz2"), "w")
    cat(sprintf("&gt;%s\n%s", names(seqs), seqs), file=conn, sep="\n")
    close(conn)
    
    # You can also use the write.fasta function for this task:
    write.fasta(biom, file.path(tempdir(), "Sequences.fa.gz"))

</code></pre>


</div>