<div class="container">

<table style="width: 100%;"><tr>
<td>rba_string_enrichment</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Getting Functional Enrichment</h2>

<h3>Description</h3>

<p>STRING cross-reference the proteins with several databases (see "Details"
section). By providing your input set o proteins (and optionally
background or universe protein set), you can use this function to
perform enrichment test and retrieve a list of enriched terms in each
database, among with pertinent information for each term.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_string_enrichment(
  ids,
  species = NULL,
  background = NULL,
  split_df = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p>Your protein ID(s). It is strongly recommended to supply
STRING IDs. See <code>rba_string_map_ids</code> for more information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>species</code></td>
<td>
<p>Numeric: NCBI Taxonomy identifier; Human Taxonomy ID is 9606.
(Recommended, but optional if your input is less than 100 IDs.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>background</code></td>
<td>
<p>character vector: A set of STRING protein IDs
to be used as the statistical background (or universe) when computing
P-value for the terms. Only STRING IDs are acceptable. (See
<code>rba_string_map_ids</code> to map your IDs.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>split_df</code></td>
<td>
<p>(logical, default = TRUE), If TRUE, instead of one
data frame, results from different categories will be split into
multiple data frames based on their 'category'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>STRING currently maps to and retrieve enrichment results based on Gene
Ontology (GO), KEGG pathways, UniProt Keywords, PubMed publications, Pfam
domains, InterPro domains, and SMART domains.
<br> Note that this function will only return the enriched terms pertinent
to your proteins that have a p-value lesser than 0.1. To retrieve a full
list of the terms -unfiltered by enrichment p-values-, use
<code>rba_string_annotations</code>.
</p>


<h3>Value</h3>

<p>A list of data frames which every row is an enriched terms with p-value
smaller than 0.1 and the columns are the terms category, description,
number of genes, p-value, fdr and other pertinent information.
</p>


<h3>Corresponding API Resources</h3>

<p>"POST https://string-db.org/api/{output_format}/enrichment?identifiers=
{your_identifiers}&amp;{optional_parameters}"
</p>


<h3>References</h3>


<ul>
<li>
<p> Damian Szklarczyk, Rebecca Kirsch, Mikaela Koutrouli, Katerina
Nastou, Farrokh Mehryary, Radja Hachilif, Annika L Gable, Tao Fang,
Nadezhda T Doncheva, Sampo Pyysalo, Peer Bork, Lars J Jensen, Christian
von Mering, The STRING database in 2023: protein–protein association
networks and functional enrichment analyses for any sequenced genome of
interest, Nucleic Acids Research, Volume 51, Issue D1, 6 January 2023,
Pages D638–D646, https://doi.org/10.1093/nar/gkac1000
</p>
</li>
<li> <p><a href="https://string-db.org/help/api/">STRING API Documentation</a>
</p>
</li>
<li>
<p><a href="https://string-db.org/cgi/about?footer_active_subpage=references">
Citations note on STRING website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p><code>rba_string_map_ids, rba_string_annotations</code>
</p>
<p>Other "STRING": 
<code>rba_string_annotations()</code>,
<code>rba_string_enrichment_ppi()</code>,
<code>rba_string_homology_inter()</code>,
<code>rba_string_homology_intra()</code>,
<code>rba_string_interaction_partners()</code>,
<code>rba_string_interactions_network()</code>,
<code>rba_string_map_ids()</code>,
<code>rba_string_network_image()</code>,
<code>rba_string_version()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_string_enrichment(ids = c("TP53", "TNF", "EGFR"), species = 9606)


</code></pre>


</div>