<div class="container">

<table style="width: 100%;"><tr>
<td>rba_uniprot_proteomes_search</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Search Proteomes in UniProt</h2>

<h3>Description</h3>

<p>UniProt collects and annotates proteomes (Protein sets expressed in an
organism). Using this function you can search UniProt for available
proteomes. see <a href="https://www.uniprot.org/help/proteome">What are
proteomes?</a> for more information. You may also
refine your search with modifiers such as keyword, taxon id etc. See
"Arguments section" for more information.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_uniprot_proteomes_search(
  name = NULL,
  upid = NULL,
  taxid = NULL,
  keyword = NULL,
  xref = NULL,
  genome_acc = NULL,
  is_ref_proteome = NULL,
  is_redundant = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>a keyword in proteome's name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>upid</code></td>
<td>
<p><a href="https://www.uniprot.org/help/proteome_id">UniProt Proteome
identifier (UPID)</a>. You can supply up to 100 UPIDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.
You can supply up to 20 taxon IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keyword</code></td>
<td>
<p>Limit the search to entries that contain your supplied
keyword. see: <a href="https://www.uniprot.org/keywords/">UniProt Keywords</a></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xref</code></td>
<td>
<p>Proteome cross-references such as Genome assembly ID or
Biosample ID. You can supply up to 20 cross-reference IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome_acc</code></td>
<td>
<p>Genome accession associated with the proteome's components.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is_ref_proteome</code></td>
<td>
<p>(logical) If TRUE, only return reference proteomes; If
FALSE, only returns non-reference proteomes; If NULL (default), the results
will not be filtered by this criteria see
<a href="https://www.uniprot.org/help/reference_proteome">'What are reference
proteomes?'</a> for more information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is_redundant</code></td>
<td>
<p>(logical) If TRUE, only return redundant proteomes; If
FALSE, only returns non-redundant proteomes; If NULL (default), the results
will not be filtered by redundancy. see
<a href="https://www.uniprot.org/help/proteome_redundancy">'Reducing proteome
redundancy'</a> for more information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
</p>


<h3>Value</h3>

<p>A list where each element is a list that corresponds to a single
proteome (search hit) and contains informations pertinent to that proteome.
</p>


<h3>Corresponding API Resources</h3>

<p>"GET https://ebi.ac.uk/proteins/api/proteomes"
</p>


<h3>References</h3>


<ul>
<li>
<p> The UniProt Consortium, UniProt: the universal protein
knowledgebase in 2021, Nucleic Acids Research, Volume 49, Issue D1,
8 January 2021, Pages D480–D489, https://doi.org/10.1093/nar/gkaa1100
</p>
</li>
<li>
<p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Other "UniProt - Proteomes": 
<code>rba_uniprot_genecentric()</code>,
<code>rba_uniprot_genecentric_search()</code>,
<code>rba_uniprot_proteomes()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_uniprot_proteomes_search(name = "SARS-CoV")


rba_uniprot_proteomes_search(name = "SARS-CoV", is_ref_proteome = TRUE)


rba_uniprot_proteomes_search(name = "SARS-CoV", is_ref_proteome = TRUE)


rba_uniprot_proteomes_search(genome_acc = "AY274119")


</code></pre>


</div>