<div class="container">

<table style="width: 100%;"><tr>
<td>rba_uniprot_features_search</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>UniProt maintains <a href="https://www.uniprot.org/help/sequence_annotation">
sequence annotations (features)</a> that describe regions
in the protein sequence. Using this function, you can search and
retrieve UniProt proteins' sequence annotations (features). you may also
refine your search query with variety of modifiers.</h2>

<h3>Description</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
<br> UniProt Entries are grouped in two sections:</p>

<ol>
<li>
<p> Reviewed(Swiss-Prot): Manually annotated records with information
extracted from literature and curator-evaluated computational analysis.
</p>
</li>
<li>
<p> Unreviewed (TrEMBL): Computationally analyzed records that await
full manual annotation.</p>
</li>
</ol>
<h3>Usage</h3>

<pre><code class="language-R">rba_uniprot_features_search(
  accession = NULL,
  gene = NULL,
  exact_gene = NULL,
  protein = NULL,
  reviewed = NULL,
  organism = NULL,
  taxid = NULL,
  categories = NULL,
  types = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene</code></td>
<td>
<p><a href="https://www.uniprot.org/help/gene_name">UniProt gene
name(s)</a>. You can supply up to 20 gene names. e.g. if you supply
"CD40", "CD40 ligand" will also be included.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exact_gene</code></td>
<td>
<p><a href="https://www.uniprot.org/help/gene_name">UniProt
exact gene name(s)</a>. You can supply up to 20 exact gene names. e.g.
if you supply "CD40", "CD40 ligand" will not be included in the results.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>protein</code></td>
<td>
<p><a href="https://www.uniprot.org/help/protein_names">UniProt
protein name</a></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reviewed</code></td>
<td>
<p>Logical: If TRUE, only return
"UniProtKB/Swiss-Prot" (reviewed) entries; If FALSE, only return TrEMBL
(un-reviewed) entries.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>organism</code></td>
<td>
<p><a href="https://www.uniprot.org/taxonomy/">Organism name</a>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.
You can supply up to 20 taxon IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>categories</code></td>
<td>
<p><a href="https://www.uniprot.org/help/sequence_annotation">
Sequence annotation (Features)</a> categories (subsection). accepted values
are: "MOLECULE_PROCESSING", "TOPOLOGY", "SEQUENCE_INFORMATION",
"STRUCTURAL", "DOMAINS_AND_SITES", "PTM", "VARIANTS" and/or "MUTAGENESIS".
You can supply up to 8 categories.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>types</code></td>
<td>
<p><a href="https://www.uniprot.org/help/sequence_annotation">
Sequence annotation (Features)</a> types. accepted values
are: "INIT_MET", "SIGNAL", "PROPEP", "TRANSIT", "CHAIN", "PEPTIDE",
"TOPO_DOM", "TRANSMEM", "DOMAIN", "REPEAT", "CA_BIND", "ZN_FING",
"DNA_BIND", "NP_BIND", "REGION", "COILED", "MOTIF", "COMPBIAS",
"ACT_SITE", "METAL", "BINDING", "SITE", "NON_STD", "MOD_RES", "LIPID",
"CARBOHYD", "DISULFID", "CROSSLNK", "VAR_SEQ", "VARIANT", "MUTAGEN",
"UNSURE", "CONFLICT", "NON_CONS", "NON_TER", "HELIX", "TURN", "STRAND"
and/or "INTRAMEM". You can supply up to 20 types.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>List where each element corresponds to one UniProt entity returned
by your search query. The element itself is a sub-list containing all
information that UniProt has about that entity.
</p>


<h3>Corresponding API Resources</h3>

<p>"GET https://www.ebi.ac.uk/proteins/api/features"
</p>


<h3>References</h3>


<ul>
<li>
<p> The UniProt Consortium, UniProt: the universal protein
knowledgebase in 2021, Nucleic Acids Research, Volume 49, Issue D1,
8 January 2021, Pages D480–D489, https://doi.org/10.1093/nar/gkaa1100
</p>
</li>
<li>
<p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Other "UniProt - Features": 
<code>rba_uniprot_features()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_uniprot_features_search(accession = "Q99616")


rba_uniprot_features_search(gene = "cd40")


rba_uniprot_features_search(gene = "cd40 ligand")


rba_uniprot_features_search(gene = "cd40",  reviewed = TRUE)


rba_uniprot_features_search(accession = "Q99616",
    categories = c("MOLECULE_PROCESSING", "TOPOLOGY"))


rba_uniprot_features_search(accession = "Q99616", types = "DISULFID")


</code></pre>


</div>