<div class="container">

<table style="width: 100%;"><tr>
<td>rba_string_interaction_partners</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get All STRING Interaction Partners</h2>

<h3>Description</h3>

<p>This function will retrieve all the STRING interactions which include your
proteins as one party of the interaction. (e.g. interaction between your
proteins and every other STRING proteins.)
<br> Given the size of STRING database, this function could return a very
long results. See "Arguments" section for information on how to filter
the interactions.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_string_interaction_partners(
  ids,
  species = NULL,
  required_score = NULL,
  network_type = "functional",
  limit = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p>Your protein ID(s). It is strongly recommended to supply
STRING IDs. See <code>rba_string_map_ids</code> for more information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>species</code></td>
<td>
<p>Numeric: NCBI Taxonomy identifier; Human Taxonomy ID is 9606.
(Recommended, but optional if your input is less than 100 IDs.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>required_score</code></td>
<td>
<p>Numeric: A minimum of interaction score for an
interaction to be included in the image. if not supplied, the threshold
will be applied by STRING Based in the network. (low Confidence = 150,
Medium Confidence = 400, High Confidence = 700, Highest confidence = 900)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>network_type</code></td>
<td>
<p>should be one of:</p>

<ul><li>
<p> "functional": (default) The edge's indicate both physical and
functional associations.</p>
</li></ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>limit</code></td>
<td>
<p>Limit the number returned interaction partners per each of
your input proteins. (e.g. Number of the most confident interaction partner
to return per each input protein.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Note that this function will retrieve the interactions between your input
proteins and every other STRING proteins. To retrieve the interaction
among your input protein-set, see
<code>rba_string_interactions_network</code>.
</p>


<h3>Value</h3>

<p>A data frame which each row is a network interaction and the
columns contains interactor information and interaction scores.
</p>


<h3>Corresponding API Resources</h3>

<p>"POST https://string-db.org/api/{output-format}/
interaction_partners?identifiers={your_identifiers}&amp;{optional_parameters}"
</p>


<h3>References</h3>


<ul>
<li>
<p> Damian Szklarczyk, Rebecca Kirsch, Mikaela Koutrouli, Katerina
Nastou, Farrokh Mehryary, Radja Hachilif, Annika L Gable, Tao Fang,
Nadezhda T Doncheva, Sampo Pyysalo, Peer Bork, Lars J Jensen, Christian
von Mering, The STRING database in 2023: protein–protein association
networks and functional enrichment analyses for any sequenced genome of
interest, Nucleic Acids Research, Volume 51, Issue D1, 6 January 2023,
Pages D638–D646, https://doi.org/10.1093/nar/gkac1000
</p>
</li>
<li> <p><a href="https://string-db.org/help/api/">STRING API Documentation</a>
</p>
</li>
<li>
<p><a href="https://string-db.org/cgi/about?footer_active_subpage=references">
Citations note on STRING website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p><code>rba_string_map_ids, rba_string_interactions_network</code>
</p>
<p>Other "STRING": 
<code>rba_string_annotations()</code>,
<code>rba_string_enrichment()</code>,
<code>rba_string_enrichment_ppi()</code>,
<code>rba_string_homology_inter()</code>,
<code>rba_string_homology_intra()</code>,
<code>rba_string_interactions_network()</code>,
<code>rba_string_map_ids()</code>,
<code>rba_string_network_image()</code>,
<code>rba_string_version()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_string_interaction_partners(ids = c("9606.ENSP00000269305",
    "9606.ENSP00000398698",
    "9606.ENSP00000275493"),
    network_type = "functional")


    rba_string_interaction_partners(ids = "9606.ENSP00000269305",
    species = 9606,
    required_score = 700)


</code></pre>


</div>