<div class="container">

<table style="width: 100%;"><tr>
<td>rba_uniprot_variation_search</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Search UniProt Natural Variants</h2>

<h3>Description</h3>

<p>Using this function, you can search and retrieve
<a href="https://www.uniprot.org/help/variant">Natural variant(s)</a> that
has been annotated in the protein's sequences. You may also refine your
search with modifiers such as source type, disease etc. See
"Arguments section" for more information.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_uniprot_variation_search(
  accession = NULL,
  source_type = NULL,
  consequence_type = NULL,
  wild_type = NULL,
  alternative_sequence = NULL,
  location = NULL,
  disease = NULL,
  omim = NULL,
  evidence = NULL,
  taxid = NULL,
  db_type = NULL,
  db_id = NULL,
  save_peff = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>source_type</code></td>
<td>
<p>Variation's source type. You can choose up to two of:
"UniProt", "large scale study" and/or "mixed".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>consequence_type</code></td>
<td>
<p>Variation's consequence type. You can choose up to
two of: "missense", "stop gained" or "stop lost".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>wild_type</code></td>
<td>
<p>Wild type amino acid. Accepted values are IUPAC
single-letter amino acid (e.g. D for    Aspartic acid) and "*" for stop
codon. You can supply up to 20 values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alternative_sequence</code></td>
<td>
<p>Alternative amino acid. Accepted values are IUPAC
single-letter amino acid (e.g. D for    Aspartic acid) and "*" for stop codon
and "-" for deletion. You can supply up to 20 values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>location</code></td>
<td>
<p>A valid amino acid range (e.g. 10-25) within the sequence
range where the variation occurs. You can supply up to 20 values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>disease</code></td>
<td>
<p><a href="https://www.uniprot.org/diseases/">Human disease</a>
that are associated with a sequence variation. Accepted values are
disease name (e.g. Alzheimer disease 18), partial disease name
(Alzheimer) and/or disease acronym (e.g. AD). You can supply up to
20 values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>omim</code></td>
<td>
<p><a href="https://www.ncbi.nlm.nih.gov/omim">OMIM</a> ID that is
associated with a variation. You can supply up to 20 values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>evidence</code></td>
<td>
<p>Pubmed ID of the variation's
<a href="https://www.uniprot.org/citations/">citation</a> You can supply up
to 20 values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.
You can supply up to 20 taxon IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>db_type</code></td>
<td>
<p>cross-reference database of the variation. You can supply
up to two of the following:</p>

<ul>
<li>
<p> "dbSNP": <a href="https://www.ncbi.nlm.nih.gov/snp/">NIH-NCBI dbSNP
database</a>.
</p>
</li>
<li>
<p> "cosmic curate": <a href="https://cancer.sanger.ac.uk/cosmic/">COSMIC
(the Catalogue of Somatic Mutations in Cancer)</a>
</p>
</li>
<li>
<p> "ClinVar": <a href="https://www.ncbi.nlm.nih.gov/clinvar/">NIH-NCBI
ClinVar</a>
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>db_id</code></td>
<td>
<p>The variation ID in a Cross-reference (external) database.
You can supply up to 20 values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save_peff</code></td>
<td>
<p>Logical or Character:</p>

<ul>
<li>
<p> FALSE: (default) Do not save PEFF file, just return as a list object.
</p>
</li>
<li>
<p> TRUE: Save as PEFF file to an automatically-generated path.
</p>
</li>
<li>
<p> Character string: A valid file path to save the PEFF file.</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
</p>


<h3>Value</h3>

<p>List where each element corresponds to one UniProt entity returned
by your search query. The element itself is a sub-list containing all
information that UniProt has about that Variation.
</p>


<h3>Corresponding API Resources</h3>

<p>"GET https://www.ebi.ac.uk/proteins/api/variation"
</p>


<h3>References</h3>


<ul>
<li>
<p> The UniProt Consortium, UniProt: the universal protein
knowledgebase in 2021, Nucleic Acids Research, Volume 49, Issue D1,
8 January 2021, Pages D480–D489, https://doi.org/10.1093/nar/gkaa1100
</p>
</li>
<li>
<p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Other "UniProt - Variation": 
<code>rba_uniprot_variation()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_uniprot_variation_search(accession = "P05067")


rba_uniprot_variation_search(disease = "alzheimer disease, 18")


rba_uniprot_variation_search(disease = "alzheimer",
    wild_type = "A", alternative_sequence = "T")


</code></pre>


</div>