<div class="container">

<table style="width: 100%;"><tr>
<td>rba_string_network_image</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get STRING Network Image</h2>

<h3>Description</h3>

<p>Depending on that you supplied a single protein ID or more than one protein
ID, this function will produce a static image of the interaction networks
among your input proteins or/and with other proteins. See the
"Arguments" section to learn more about how you can modify the network
image.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_string_network_image(
  ids,
  image_format = "image",
  save_image = TRUE,
  species = NULL,
  add_color_nodes = NULL,
  add_white_nodes = NULL,
  required_score = NULL,
  network_flavor = "evidence",
  network_type = "functional",
  hide_node_labels = FALSE,
  use_query_labels = FALSE,
  hide_disconnected_nodes = FALSE,
  hide_structure_pics = FALSE,
  flat_nodes = FALSE,
  node_labels_center = FALSE,
  node_labels_font_size = 12,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p>Your protein ID(s). It is strongly recommended to supply
STRING IDs. See <code>rba_string_map_ids</code> for more information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>image_format</code></td>
<td>
<p>one of:</p>

<ul>
<li>
<p> "image": PNG image with normal resolution.
</p>
</li>
<li>
<p> "highres_image": High-resolution PNG image.
</p>
</li>
<li>
<p> "svg": Scalable Vector Graphics image.</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save_image</code></td>
<td>
<p>Logical or Character:</p>

<ul>
<li>
<p> TRUE: Save the image to an automatically-generated path.
</p>
</li>
<li>
<p> FALSE: Do not save the image, just return it as an R object.
</p>
</li>
<li>
<p> Character string: A valid file path to save the image to.</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>species</code></td>
<td>
<p>Numeric: NCBI Taxonomy identifier; Human Taxonomy ID is 9606.
(Recommended, but optional if your input is less than 100 IDs.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add_color_nodes</code></td>
<td>
<p>Numeric: The number of colored nodes (queried proteins
and first shell of interactors) to be added.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add_white_nodes</code></td>
<td>
<p>Numeric: The number of white nodes (second shell of
interactors) to be added after colored nodes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>required_score</code></td>
<td>
<p>Numeric (Between 0 to 1000): A minimum of interaction
score for an interaction to be included in the image. if not supplied, the
threshold will be applied by STRING Based in the network. (low Confidence
= 150, Medium Confidence = 400, High Confidence = 700, Highest
confidence = 900)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>network_flavor</code></td>
<td>
<p>The style of network edges, should be one of:</p>

<ul>
<li>
<p> "evidence": (default) Line's color is based on the type of evidences that
support the interaction.
</p>
</li>
<li>
<p> "confidence": Line's thickness is an indicator of the
interaction's confidence score.
</p>
</li>
<li>
<p> "action": Line's Shape is an indicator of the interaction's predicted
mode of actions.</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>network_type</code></td>
<td>
<p>should be one of:</p>

<ul>
<li>
<p> "functional": (default) The edge's indicate both physical and
functional associations.
</p>
</li>
<li>
<p> "physical": The edges indicate that two proteins have a physical
interaction or are parts of a complex.</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hide_node_labels</code></td>
<td>
<p>Logical: (Default = FALSE) Hide proteins names from
the image</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use_query_labels</code></td>
<td>
<p>Logical: (Default = FALSE) Use the names supplied
with the 'ids' argument as the nodes labels instead of STRING's default
ones.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hide_disconnected_nodes</code></td>
<td>
<p>Logical: (Default = FALSE) Hide proteins that
are not connected to any other proteins from the image</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hide_structure_pics</code></td>
<td>
<p>Logical: (Default = FALSE) Hide protein's
structure picture from inside the bubbles</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>flat_nodes</code></td>
<td>
<p>Logical: (Default = FALSE) Make the nodes design flat
instead of the default 3D design</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>node_labels_center</code></td>
<td>
<p>Logical: (Default = FALSE) Position the protein
names labels center aligned on the nodes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>node_labels_font_size</code></td>
<td>
<p>Numeric (Between 5 to 50, Default = 12) Font
size of the protein nodes labels</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A network images which can be PNG or SVG depending on the inputs.
</p>


<h3>Corresponding API Resources</h3>

<p>"POST https://string-db.org/api/{output-format}/network?identifiers=
{your_identifiers}&amp;{optional_parameters}"
</p>


<h3>References</h3>


<ul>
<li>
<p> Damian Szklarczyk, Rebecca Kirsch, Mikaela Koutrouli, Katerina
Nastou, Farrokh Mehryary, Radja Hachilif, Annika L Gable, Tao Fang,
Nadezhda T Doncheva, Sampo Pyysalo, Peer Bork, Lars J Jensen, Christian
von Mering, The STRING database in 2023: protein–protein association
networks and functional enrichment analyses for any sequenced genome of
interest, Nucleic Acids Research, Volume 51, Issue D1, 6 January 2023,
Pages D638–D646, https://doi.org/10.1093/nar/gkac1000
</p>
</li>
<li> <p><a href="https://string-db.org/help/api/">STRING API Documentation</a>
</p>
</li>
<li>
<p><a href="https://string-db.org/cgi/about?footer_active_subpage=references">
Citations note on STRING website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p><code>rba_string_map_ids</code>
</p>
<p>Other "STRING": 
<code>rba_string_annotations()</code>,
<code>rba_string_enrichment()</code>,
<code>rba_string_enrichment_ppi()</code>,
<code>rba_string_homology_inter()</code>,
<code>rba_string_homology_intra()</code>,
<code>rba_string_interaction_partners()</code>,
<code>rba_string_interactions_network()</code>,
<code>rba_string_map_ids()</code>,
<code>rba_string_version()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
rba_string_network_image(ids = c("9606.ENSP00000269305",
    "9606.ENSP00000398698",
    "9606.ENSP00000275493"),
    network_type = "functional",
    save_image = FALSE)

## End(Not run)
## Not run: 
rba_string_network_image(ids = c("TP53", "TNF", "EGFR"),
    species = 9606,
    save_image = TRUE)

## End(Not run)
## Not run: 
rba_string_network_image(ids = "9606.ENSP00000269305",
    image_format = "highres_image",
    save_image = file.path(getwd(), "TP53_network.png"))

## End(Not run)

</code></pre>


</div>