<div class="container">

<table style="width: 100%;"><tr>
<td>rba_uniprot_coordinates_location</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Search UniProt entries by taxonomy and genomic coordinates</h2>

<h3>Description</h3>

<p>For more information about how UniProt imports and calculates genomic
coordinates data, see:
<br> McGarvey, P. B., Nightingale, A., Luo, J., Huang, H., Martin, M. J.,
Wu, C., &amp; UniProt Consortium (2019). UniProt genomic mapping for
deciphering functional effects of missense variants. Human mutation,
40(6), 694–705. https://doi.org/10.1002/humu.23738
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_uniprot_coordinates_location(
  taxid,
  locations,
  in_range = TRUE,
  feature = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locations</code></td>
<td>
<p>genomic location formatted as: chromosome:start-end.
(e.g. "Y:17100001-19600000"). If you omit chromosome, it will be interpreted
as any chromosome (e.g. "1-10000").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>in_range</code></td>
<td>
<p>Only return proteins that are in range.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>feature</code></td>
<td>
<p>(logical) Get features?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a list containing UniProt proteins which match the supplied genomic
location and taxonomy ID.
</p>


<h3>Corresponding API Resources</h3>

<p>"GET https://ebi.ac.uk/proteins/api/coordinates
/{taxonomy}/{locations}/feature"
<br> "GET https://ebi.ac.uk/proteins/api/coordinates
/{taxonomy}/{locations}"
</p>


<h3>References</h3>


<ul>
<li>
<p> The UniProt Consortium, UniProt: the universal protein
knowledgebase in 2021, Nucleic Acids Research, Volume 49, Issue D1,
8 January 2021, Pages D480–D489, https://doi.org/10.1093/nar/gkaa1100
</p>
</li>
<li>
<p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Other "UniProt - Coordinates": 
<code>rba_uniprot_coordinates()</code>,
<code>rba_uniprot_coordinates_search()</code>,
<code>rba_uniprot_coordinates_sequence()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_uniprot_coordinates_location(taxid = 9606,
    locations = "Y:17100001-19600000", in_range = TRUE)


rba_uniprot_coordinates_location(taxid = 9606,
    locations = "20:39000001", in_range = FALSE)


</code></pre>


</div>