<div class="container">

<table style="width: 100%;"><tr>
<td>rba_uniprot_antigens_search</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Search Antigens in UniProt</h2>

<h3>Description</h3>

<p>UniProt maps Antigenic (Antibody-binding) features from different sources
to the proteins' sequences. Using this function, you can search for
Antigenic sequences that has been map to UniProt proteins. You may also
refine your search with modifiers such as score etc. See
"Arguments section" for more information.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rba_uniprot_antigens_search(
  accession = NULL,
  antigen_sequence = NULL,
  antigen_id = NULL,
  ensembl_id = NULL,
  match_score = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>antigen_sequence</code></td>
<td>
<p>Protein sequence in the antigenic site.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>antigen_id</code></td>
<td>
<p>Human Protein Atlas (HPA) antigen ID. You can supply up to
20 IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ensembl_id</code></td>
<td>
<p>Ensembl Stable Transcript ID. You can supply up to
20 IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>match_score</code></td>
<td>
<p>(Numeric) Minimum alignment score for the antigen
sequence and the target protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>rbioapi option(s). See <code>rba_options</code>'s
arguments manual for more information on available options.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
</p>


<h3>Value</h3>

<p>A list Where each element correspond to a UniProt protein (search
hit) and Antigenic features are organized under the "features" sub-list.
</p>


<h3>Corresponding API Resources</h3>

<p>"GET https://www.ebi.ac.uk/proteins/api/antigen"
</p>


<h3>References</h3>


<ul>
<li>
<p> The UniProt Consortium, UniProt: the universal protein
knowledgebase in 2021, Nucleic Acids Research, Volume 49, Issue D1,
8 January 2021, Pages D480–D489, https://doi.org/10.1093/nar/gkaa1100
</p>
</li>
<li>
<p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Other "UniProt - Antigen": 
<code>rba_uniprot_antigens()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
rba_uniprot_antigens_search(antigen_id = "HPA001060")


</code></pre>


</div>