<div class="container">

<table style="width: 100%;"><tr>
<td>RCircos.Gene.Name.Plot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Draw Gene Names on a Data Track
</h2>

<h3>Description</h3>

<p>Label gene names along chromosome ideogram or a data track. RCircos core components and graphic device must be initialized before drawing.
</p>


<h3>Usage</h3>

<pre><code class="language-R">RCircos.Gene.Name.Plot(gene.data=NULL, name.col=NULL,  
    track.num=NULL, side="in", inside.pos=NULL,  
    outside.pos=NULL, genomic.columns=3, is.sorted=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gene.data</code></td>
<td>

<p>A data frame. The first three columns should be chromosome name, start position, end position.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name.col</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the column in input data that contains gene names.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>track.num</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the plot track where the gene names are plotted.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>side</code></td>
<td>

<p>Character vector, either "in" or "out", representing the position related to chromosome ideogram.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track .
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position in each row. Must be either 3 or 2.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is.sorted</code></td>
<td>

<p>Logic, whether the data is sorted by chromosome names and start positions.
</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
library(RCircos)
data(UCSC.HG19.Human.CytoBandIdeogram)
data(RCircos.Gene.Label.Data)

cyto.info &lt;- UCSC.HG19.Human.CytoBandIdeogram
RCircos.Set.Core.Components(cyto.info,  
    chr.exclude=c("chrX", "chrY"),  
    num.inside=10, num.outside=0)

RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()

gene.data &lt;- RCircos.Gene.Label.Data;
RCircos.Gene.Name.Plot(gene.data, name.col=4,  
    track.num=2, side="in")
RCircos.Gene.Name.Plot(gene.data, name.col=4,  
    track.num=2, side="out",  
    inside.pos=2.5, outside.pos=3)

## End(Not run)
</code></pre>


</div>