<div class="container">

<table style="width: 100%;"><tr>
<td>vst</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate population-wise Vst</h2>

<h3>Description</h3>

<p>This function calculates Vst (variant fixation index) for populations given
a list of duplicated loci
</p>


<h3>Usage</h3>

<pre><code class="language-R">vst(AD, pops, id.list = NULL, qGraph = TRUE, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>AD</code></td>
<td>
<p>data frame of total allele depth values of (duplicated, if
<code>id.list</code> is not provided) SNPs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pops</code></td>
<td>
<p>character. A vector of population names for each individual.
Must be the same length as the number of samples in AD</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id.list</code></td>
<td>
<p>character. A vector of duplicated SNP IDs. Must match the IDs
in the AD data frame</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>qGraph</code></td>
<td>
<p>logical. Plot the network plot based on Vst values
(see details)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical. show progress</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional arguments passed to <code>qgraph</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Vst is calculated with the following equation
</p>
<p style="text-align: center;"><code class="reqn">V_{T} = \frac{ V_{S} }{V_{T}}</code>
</p>
<p> where VT is the variance of normalized
read depths among all individuals from the two populations and VS is the
average of the variance within each population, weighed for population size
(see reference for more details)
See <code>qgraph</code> help for details on qgraph output
</p>


<h3>Value</h3>

<p>Returns a matrix of pairwise Vst values for populations
</p>


<h3>Author(s)</h3>

<p>Piyal Karunarathne
</p>


<h3>References</h3>

<p>Redon, Richard, et al. Global variation in copy number in the human genome.
nature 444.7118 (2006)
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: data(alleleINF)
data(ADtable)
DD&lt;-dupGet(alleleINF)
ds&lt;-DD[DD$dup.stat=="deviant",]
ad&lt;-ADtable[match(paste0(ds$CHROM,".",ds$POS),paste0(ADtable$CHROM,".",ADtable$POS)),]
vst(ad,pops=substr(colnames(ad)[-c(1:4)],1,11))
## End(Not run)

</code></pre>


</div>