<div class="container">

<table style="width: 100%;"><tr>
<td>allele.freq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate allele frequency table for individuals or populations</h2>

<h3>Description</h3>

<p>Get alternative allele frequency across all individuals per SNP from the
genotype or allele depth tables
</p>


<h3>Usage</h3>

<pre><code class="language-R">allele.freq(gtt, f.typ = c("pop", "ind"), verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gtt</code></td>
<td>
<p>a list or data frame of genotype and/or allele depth table produced from <code>hetTgen</code> (or similar)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>f.typ</code></td>
<td>
<p>character. type of allele frequency to be calculated (individual <code>"ind"</code> or population <code>"pop"</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical. whether to show the progress of the analysis</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If the allele frequencies to be calculated for populations from both genotype table and the allele depth table, they must be provided in a list with element names <code>AD</code> for allele depth table and <code>GT</code> for the genotype table. See the examples.
</p>


<h3>Value</h3>

<p>Returns a data frame or a list (if both genotype and allele depth used)
of allele frequencies
</p>


<h3>Author(s)</h3>

<p>Piyal Karunarathne
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: vcf.file.path &lt;- paste0(path.package("rCNV"), "/example.raw.vcf.gz")
vcf &lt;- readVCF(vcf.file.path=vcf.file.path)
het.table&lt;-hetTgen(vcf,"GT")
ad.table&lt;-hetTgen(vcf,"AD")

# for individual based AF
frQ&lt;-allele.freq(het.table,f.typ="ind")

#for population-wise and both allele depth and genotype tables
frQ&lt;-allele.freq(list(AD=ad.table,GT=het.table),f.typ="pop")
## End(Not run)

</code></pre>


</div>