<div class="container">

<table style="width: 100%;"><tr>
<td>allele.info</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get allele information for duplicate detection</h2>

<h3>Description</h3>

<p>The function to calculate allele median ratios, proportion of heterozygotes
and allele probability values under different assumptions (see details),
and their chi-square significance values for duplicate detection
</p>


<h3>Usage</h3>

<pre><code class="language-R">allele.info(
  X,
  x.norm = NULL,
  Fis,
  method = c("MedR", "QN", "pca", "TMM", "TMMex"),
  logratioTrim = 0.3,
  sumTrim = 0.05,
  Weighting = TRUE,
  Acutoff = -1e+10,
  plot.allele.cov = TRUE,
  verbose = TRUE,
  parallel = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>allele depth table generated from the function
<code>hetTgen</code> (non-normalized)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x.norm</code></td>
<td>
<p>a data frame of normalized allele coverage, output of
<code>cpm.normal</code>. If not provided, calculated using <code>X</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Fis</code></td>
<td>
<p>numeric. Inbreeding coefficient calculated using <code>h.zygosity()</code> function</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>character. method to be used for normalization
(see <code>cpm.normal</code> details). Default <code>TMM</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logratioTrim</code></td>
<td>
<p>numeric. percentage value (0 - 1) of variation to be
trimmed in log transformation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sumTrim</code></td>
<td>
<p>numeric. amount of trim to use on the combined absolute
levels (“A” values) for method <code>TMM</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Weighting</code></td>
<td>
<p>logical, whether to compute (asymptotic binomial precision)
weights</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Acutoff</code></td>
<td>
<p>numeric, cutoff on “A” values to use before trimming</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.allele.cov</code></td>
<td>
<p>logical, plot comparative plots of allele depth
coverage in homozygotes and heterozygotes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical, whether to print progress</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel</code></td>
<td>
<p>logical. whether to parallelize the process</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments to be passed to <code>plot</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Allele information generated here are individual SNP based and presents the
proportion of heterozygotes, number of samples, and deviation of allele
detection from a 1:1 ratio of reference and alternative alleles.
The significance of the deviation is tested with Z-score test
<code class="reqn">Z = \frac{ \frac{N}{2}-N_A}{ \sigma_{x}}</code>,
and chi-square test (see references for more details on the method).
</p>


<h3>Value</h3>

<p>Returns a data frame of median allele ratio, proportion of
heterozygotes, number of heterozygotes, and allele probability at different
assumptions with their chi-square significance
</p>


<h3>Author(s)</h3>

<p>Piyal Karunarathne, Pascal Milesi, Klaus Schliep
</p>


<h3>References</h3>


<ul>
<li>
<p>McKinney, G. J., Waples, R. K., Seeb, L. W., &amp; Seeb, J. E. (2017).
Paralogs are revealed by proportion of heterozygotes and deviations in read
ratios in genotyping by sequencing data from natural populations.
Molecular Ecology Resources, 17(4)
</p>
</li>
<li>
<p>Karunarathne et al. 2022 (to be added)
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: data(ADtable)
hz&lt;-h.zygosity(vcf,verbose=FALSE)
Fis&lt;-mean(hz$Fis,na.rm = TRUE)
AI&lt;-allele.info(ADtable,x.norm=ADnorm,Fis=Fis)
## End(Not run)

</code></pre>


</div>