<div class="container">

<table style="width: 100%;"><tr>
<td>norm.fact</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate normalization factor for each sample</h2>

<h3>Description</h3>

<p>This function calculates the normalization factor for each sample using
different methods. See details.
</p>


<h3>Usage</h3>

<pre><code class="language-R">norm.fact(
  df,
  method = c("TMM", "TMMex", "MedR", "QN"),
  logratioTrim = 0.3,
  sumTrim = 0.05,
  Weighting = TRUE,
  Acutoff = -1e+10
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>
<p>a data frame or matrix of allele depth values
(total depth per snp per sample)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>character. method to be used (see details). Default <code>TMM</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logratioTrim</code></td>
<td>
<p>numeric. percentage value (0 - 1) of variation to be
trimmed in log transformation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sumTrim</code></td>
<td>
<p>numeric. amount of trim to use on the combined absolute
levels (“A” values) for method <code>TMM</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Weighting</code></td>
<td>
<p>logical, whether to compute (asymptotic binomial precision)
weights</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Acutoff</code></td>
<td>
<p>numeric, cutoff on “A” values to use before trimming</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Originally described for normalization of RNA sequences
(Robinson &amp; Oshlack 2010), this function computes normalization (scaling)
factors to convert observed library sizes into effective library sizes.
It uses the method trimmed means of M-values proposed by Robinson &amp;
Oshlack (2010). See the original publication and <code>edgeR</code> package
for more information.
The method <code>MedR</code> is median ratio normalization;
QN - quantile normalization (see  Maza, Elie, et al. 2013 for a
comparison of methods).
</p>


<h3>Value</h3>

<p>Returns a numerical vector of normalization factors for each sample
</p>


<h3>Author(s)</h3>

<p>Piyal Karunarathne
</p>


<h3>References</h3>


<ul>
<li>
<p>Robinson MD, and Oshlack A (2010). A scaling normalization method for
differential expression analysis of RNA-seq data. Genome Biology 11, R25
</p>
</li>
<li>
<p>Robinson MD, McCarthy DJ and Smyth GK (2010). edgeR: a Bioconductor
package for differential expression analysis of digital gene expression
data. Bioinformatics 26
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: vcf.file.path &lt;- paste0(path.package("rCNV"), "/example.raw.vcf.gz")
vcf &lt;- readVCF(vcf.file.path)
df&lt;-hetTgen(vcf,"AD-tot",verbose=FALSE)
norm.fact(df)
## End(Not run)


</code></pre>


</div>