<div class="container">

<table style="width: 100%;"><tr>
<td>ad.correct</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Correct allele depth values</h2>

<h3>Description</h3>

<p>A function to correct depth values with odd number of coverage values due to
sequencing anomalies or miss classification where genotype is homozygous and
depth values indicate heterozygosity.
The function adds a value of one to the allele with the lowest depth value
for when odd number anomalies or make the depth value zero for when
miss-classified. The genotype table must be provided for the latter.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ad.correct(
  het.table,
  gt.table = NULL,
  odd.correct = TRUE,
  verbose = TRUE,
  parallel = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>het.table</code></td>
<td>
<p>allele depth table generated from the function
<code>hetTgen</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gt.table</code></td>
<td>
<p>genotype table generated from the function hetTgen</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>odd.correct</code></td>
<td>
<p>logical, to correct for odd number anomalies in AD values.
default <code>TRUE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical. show progress. Default <code>TRUE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel</code></td>
<td>
<p>logical. whether to parallelize the process</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns the coverage corrected allele depth table similar to the
output of <code>hetTgen</code>
</p>


<h3>Author(s)</h3>

<p>Piyal Karunarathne
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: adc&lt;-ad.correct(ADtable)

</code></pre>


</div>