<div class="container">

<table style="width: 100%;"><tr>
<td>get.miss</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get missingness of individuals in raw vcf</h2>

<h3>Description</h3>

<p>A function to get the percentage of missing data of snps per SNP and per
sample
</p>


<h3>Usage</h3>

<pre><code class="language-R">get.miss(
  data,
  type = c("samples", "snps"),
  plot = TRUE,
  verbose = TRUE,
  parallel = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>a list containing imported vcf file using <code>readVCF</code> or
genotype table generated using <code>hetTgen</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>character.  Missing percentages per sample
“samples” or per SNP “snps”, default both</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>logical. Whether to plot the missingness density with ninety
five percent quantile</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical. Whether to show progress</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel</code></td>
<td>
<p>logical. whether to parallelize the process</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a data frame of allele depth or genotypes
</p>


<h3>Author(s)</h3>

<p>Piyal Karunarathne
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: vcf.file.path &lt;- paste0(path.package("rCNV"), "/example.raw.vcf.gz")
vcf &lt;- readVCF(vcf.file.path=vcf.file.path)
missing&lt;-get.miss(vcf,plot=TRUE)
## End(Not run)

</code></pre>


</div>