<div class="container">

<table style="width: 100%;"><tr>
<td>hetTgen</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate allele depth or genotype table</h2>

<h3>Description</h3>

<p>hetTgen extracts the read depth and coverage values for each snp for all
the individuals from a vcf file generated from readVCF (or GatK
VariantsToTable: see details)
</p>


<h3>Usage</h3>

<pre><code class="language-R">hetTgen(
  vcf,
  info.type = c("AD", "AD-tot", "GT", "GT-012", "GT-AB", "DP"),
  verbose = TRUE,
  parallel = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcf</code></td>
<td>
<p>an imported vcf file in a list using <code>readVCF</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>info.type</code></td>
<td>
<p>character. <code>AD</code>: allele depth value, <code>AD-tot</code>:total
allele depth, <code>DP</code>=unfiltered depth (sum), <code>GT</code>: genotype,
<code>GT-012</code>:genotype in 012 format, <code>GT-AB</code>:genotype in AB format.
Default <code>AD</code>,  See details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical. whether to show the progress of the analysis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel</code></td>
<td>
<p>logical. whether to parallelize the process</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If you generate the depth values for allele by sample using GatK
VariantsToTable option, use only -F CHROM -F POS -GF AD flags to generate
the table. Or keep only the CHROM, POS, ID, ALT, and individual AD columns.
For info.type <code>GT</code> option is provided to extract the genotypes of
individuals by snp.
</p>


<h3>Value</h3>

<p>Returns a data frame of allele depth, genotype of SNPs for all the
individuals extracted from a VCF file
</p>


<h3>Author(s)</h3>

<p>Piyal Karunarathne, Klaus Schliep
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: vcf.file.path &lt;- paste0(path.package("rCNV"), "/example.raw.vcf.gz")
vcf &lt;- readVCF(vcf.file.path=vcf.file.path)
het.table&lt;-hetTgen(vcf)
## End(Not run)

</code></pre>


</div>