<div class="container">

<table style="width: 100%;"><tr>
<td>gt.format</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Format genotype for BayEnv and BayPass</h2>

<h3>Description</h3>

<p>This function generates necessary genotype count formats for BayEnv and
BayPass with a subset of SNPs
</p>


<h3>Usage</h3>

<pre><code class="language-R">gt.format(
  gt,
  info,
  format = c("benv", "bpass"),
  snp.subset = NULL,
  parallel = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gt</code></td>
<td>
<p>multi-vector. an imported data.frame of genotypes or genotype
data frame generated by <code>hetTgen</code> or path to GT.FORMAT
file generated from VCFTools</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>info</code></td>
<td>
<p>a data frame containing sample and population information.
It must have “sample” and “population” columns</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>format</code></td>
<td>
<p>character. output format i.e., for BayPass or BayEnv</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.subset</code></td>
<td>
<p>numerical. number of randomly selected subsets of SNPs.
<code>default = NULL</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel</code></td>
<td>
<p>logical. whether to parallelize the process</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a list with formatted genotype data: <code>$bayenv</code> - snps
in horizontal format - for BayEnv (two lines per snp); <code>$baypass</code> - vertical format - for BayPass
(two column per snp); <code>$sub.bp</code> - subsets snps for BayPass <code>$sub.be</code> - subsets of snps for BayEnv
</p>


<h3>Author(s)</h3>

<p>Piyal Karunarathne
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: vcf.file.path &lt;- paste0(path.package("rCNV"), "/example.raw.vcf.gz")
vcf &lt;- readVCF(vcf.file.path=vcf.file.path)
het.table&lt;-hetTgen(vcf,"GT")
info&lt;-unique(substr(colnames(het.table)[-c(1:3)],1,8))
GT&lt;-gt.format(het.table,info)
## End(Not run)

</code></pre>


</div>