<div class="container">

<table style="width: 100%;"><tr>
<td>cpm.normal</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate normalized depth for alleles</h2>

<h3>Description</h3>

<p>This function outputs the normalized depth values separately for each allele,
calculated using normalization factor with trimmed mean of M-values of
sample libraries, median ratios normalization or quantile normalization,
See details.
</p>


<h3>Usage</h3>

<pre><code class="language-R">cpm.normal(
  het.table,
  method = c("MedR", "QN", "pca", "TMM", "TMMex"),
  logratioTrim = 0.3,
  sumTrim = 0.05,
  Weighting = TRUE,
  Acutoff = -1e+10,
  verbose = TRUE,
  plot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>het.table</code></td>
<td>
<p>allele depth table generated from the function
<code>hetTgen</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>character. method to be used (see details). Default <code>TMM</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logratioTrim</code></td>
<td>
<p>numeric. percentage value (0 - 1) of variation to be
trimmed in log transformation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sumTrim</code></td>
<td>
<p>numeric. amount of trim to use on the combined absolute
levels (“A” values) for method <code>TMM</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Weighting</code></td>
<td>
<p>logical, whether to compute (asymptotic binomial precision)
weights</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Acutoff</code></td>
<td>
<p>numeric, cutoff on “A” values to use before trimming
(only for TMM(ex))</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical. show progress</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>logical. Plot the boxplot of sample library sizes showing outliers</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function converts an observed depth value table to an
effective depth value table using several normalization methods;
</p>

<ol>
<li>
<p> TMM normalization (See the original publication for more information).
It is different from the function <code>normz</code> only in calculation of the
counts per million is for separate alleles instead of the total depth.
The <code>TMMex</code> method is an extension of the <code>TMM</code> method for
large data sets containing SNPs exceeding 10000
</p>
</li>
<li>
<p> The method <code>MedR</code> is median ratio normalization;
</p>
</li>
<li>
<p> QN - quantile normalization (see  Maza, Elie, et al. 2013 for a
comparison of methods).
</p>
</li>
<li>
<p> PCA - a modified Kaiser's Rule applied to depth values: Sample variation
of eigen values smaller than 0.7 are removed (i.e., the first eigen value &lt; 0.7)
to eliminate the effect of the library size of samples
</p>
</li>
</ol>
<h3>Value</h3>

<p>Returns a list with (AD), a data frame of normalized depth values
similar to the output of <code>hetTgen</code> function and
(outliers) a list of outlier sample names
</p>


<h3>Author(s)</h3>

<p>Piyal Karunarathne, Qiujie Zhou
</p>


<h3>References</h3>


<ul>
<li>
<p>Robinson MD, Oshlack A (2010). A scaling normalization method for
differential expression analysis of RNA-seq data. Genome Biology 11, R25
</p>
</li>
<li>
<p>Robinson MD, McCarthy DJ and Smyth GK (2010). edgeR: a Bioconductor
package for differential expression analysis of digital gene expression
data. Bioinformatics 26
</p>
</li>
<li>
<p>Maza, Elie, et al. "Comparison of normalization methods for
differential gene expression analysis in RNA-Seq experiments: a matter of
relative size of studied transcriptomes." Communicative &amp; integrative
biology 6.6 (2013): e25849
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: data(ADtable)
ADnormalized&lt;-cpm.normal(ADtable)
## End(Not run)


</code></pre>


</div>