<div class="container">

<table style="width: 100%;"><tr>
<td>h.zygosity</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Determine per sample heterozygosity and inbreeding coefficient</h2>

<h3>Description</h3>

<p>This function will calculate the heterozygosity on a per-sample basis from
vcf files (snps), and most importantly inbreeding coefficient which is used
to filter out the samples with bad mapping quality.
</p>


<h3>Usage</h3>

<pre><code class="language-R">h.zygosity(vcf, plot = FALSE, pops = NA, verbose = TRUE, parallel = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcf</code></td>
<td>
<p>an imported vcf file in in a list using
<code>readVCF</code> or a data frame of genotypes generated using
<code>hetTgen</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>logical. Whether to plot a boxplot of inbreeding coefficients
for populations. A list of populations must be provided</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pops</code></td>
<td>
<p>character. A list of population names with the same length and
order as the number of samples in the vcf</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical. Show progress</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel</code></td>
<td>
<p>logical. Parallelize the process</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a data frame of expected “E(Hom)” and observed
“O(Hom)” homozygotes with their inbreeding coefficients.
</p>


<h3>Author(s)</h3>

<p>Piyal Karunarathne, Pascal Milesi, Klaus Schliep
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: vcf.file.path &lt;- paste0(path.package("rCNV"), "/example.raw.vcf.gz")
vcf &lt;- readVCF(vcf.file.path=vcf.file.path)
pp&lt;-substr(colnames(vcf$vcf)[-c(1:9)],1,8)
hzygots&lt;-h.zygosity(vcf,plot=TRUE,pops=pp)
## End(Not run)

</code></pre>


</div>