<div class="container">

<table style="width: 100%;"><tr>
<td>alleleINF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Allele info example data</h2>

<h3>Description</h3>

<p>Semi-randomly generated data from the function dup.snp.info.
Data contains depth and proportion values of 2857 snps
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(alleleINF)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 2857 rows and 28 columns.
</p>


<h3>Source</h3>

<p><a href="https://zenodo.org/record/5025423#.Yj2XKRDMLyQ">Chinook Salmon sequence reads McKinney et al. 2017</a>
</p>


<h3>References</h3>


<ul>
<li>
<p>Larson, W. A., Seeb, L. W., Everett, M. V., Waples, R. K., Templin, W. D., &amp; Seeb, J. E. (2014). Genotyping by sequencing resolves #' shallow population structure to inform conservation of Chinook salmon (Oncorhynchus tshawytscha). Evolutionary Applications, 7(3)
</p>
</li>
<li>
<p>McKinney, G. J., Waples, R. K., Seeb, L. W., &amp; Seeb, J. E. (2017). Paralogs are revealed by proportion of heterozygotes and deviations in read ratios in genotyping by sequencing data from natural populations. Molecular Ecology Resources, 17(4)
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">data(alleleINF)
with(alleleINF,plot(medRatio~propHet))
</code></pre>


</div>