<div class="container">

<table style="width: 100%;"><tr>
<td>vcf.stat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get sequencing quality statistics of raw VCF files
(with GatK generated vcf files only)</h2>

<h3>Description</h3>

<p>This function will generate a table similar to VariantsToTable option in
GatK from raw vcf files for filtering purposes. The function will also
plot all the parameters (see details &amp; values).
</p>


<h3>Usage</h3>

<pre><code class="language-R">vcf.stat(vcf, plot = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcf</code></td>
<td>
<p>an imported vcf file in data.frame or matrix format using
<code>readVCF</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>logical. Whether to plot the (12) parameters</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>other arguments passed on to <code>plot</code>
(e.g. col,border)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For more details see instructions of GatK
</p>


<h3>Value</h3>

<p>Returns a data frame with quality parameters from the INFO. field of
the vcf
</p>

<ul>
<li>
<p>QUAL: The Phred-scaled probability that a REF/ALT polymorphism exists
at this site given sequencing data
</p>
</li>
<li>
<p>AC: Allele count
</p>
</li>
<li>
<p>AF: Allele frequency
</p>
</li>
<li>
<p>DP: unfiltered depth
</p>
</li>
<li>
<p>QD: QualByDepth - This is the variant confidence (from the QUAL
field) divided by the unfiltered depth of non-hom-ref samples
</p>
</li>
<li>
<p>FS: FisherStrand - This is the Phred scaled probability that there is
strand bias at the site
</p>
</li>
<li>
<p>SOR: StrandOddsRatio - This is another way to estimate strand bias
using a test similar to the symmetric odds ratio test
</p>
</li>
<li>
<p>MQ: RMSMappingQuality - This is the root mean square mapping quality
over all the reads at the site
</p>
</li>
<li>
<p>MQRankSum: MappingQualityRankSumTest - This is the u-based
z-approximation from the Rank Sum Test for mapping qualities
</p>
</li>
<li>
<p>ReadPosRankSum: ReadPosRankSumTest: This is the u-based
z-approximation from the Rank Sum Test for site position within reads
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Piyal Karunarathne
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: vcf.file.path &lt;- paste0(path.package("rCNV"), "/example.raw.vcf.gz")
vcf &lt;- readVCF(vcf.file.path=vcf.file.path)
statistics&lt;-vcf.stat(vcf,plot=TRUE)
## End(Not run)

</code></pre>


</div>