<div class="container">

<table style="width: 100%;"><tr>
<td>dup.validate</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Validate detected deviants/cnvs</h2>

<h3>Description</h3>

<p>This function will validate the detected duplicated-SNPs (deviants/cnvs) using a moving
window approach (see details)
</p>


<h3>Usage</h3>

<pre><code class="language-R">dup.validate(d.detect, window.size = 100, scaf.size = 10000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>d.detect</code></td>
<td>
<p>a data frame of detected duplicates or deviants from the outputs of <code>dupGet</code> or <code>cnv</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>window.size</code></td>
<td>
<p>numerical. a single value of the desired moving window
size (default <code>100</code> bp)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scaf.size</code></td>
<td>
<p>numerical. scaffold size to be checked. i.e. the chromosome/scaffolds will be split into equal pieces of this size
default=10000</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Loci/SNP positions correctly ordered according to a reference
sequence is necessary for this function to work properly. The list of deviants/cnvs provided in the <code>d.detect</code> will be split into pices of <code>scaf.size</code> and the number of deviants/cnvs will be counted along each split with a moving window of <code>window.size</code>. The resulting percentages of deviants/cnvs will be averaged for each scaf.size split; this is the <code>cnv.ratio</code> column in the output. Thus, ideally, the <code>cnv.ratio</code> is a measure of how confident the detected deviants/cnvs are in an actual putative duplicated region withing the given <code>scaf.size</code>. This ratio is sensitive to the picked window size and the scaf.size; as a rule of thumb, it is always good to use a known gene length as the scaf.size, if you need to check a specific gene for the validity of the detected duplicates.
Please also note that this function is still in its <code>beta-testing</code> phase and also under development for non-mapped reference sequences. Therefore, your feedback and suggestions will be highly appreciated.
</p>


<h3>Value</h3>

<p>A data frame of deviant/cnv ratios (column <code>cnv.ratio</code>) for a split of the chromosome/scaffold given by the <code>scaf.size</code>; this ratio is an average value of the percentage of deviants/cnvs present within the given <code>window.size</code> for each split (<code>chromosome/scaffold length/sacf.size</code>); the start and the end positions of each split is given in the <code>start</code> and <code>end</code> columns
</p>


<h3>Author(s)</h3>

<p>Piyal Karunarathne
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# suggestion to visualize dup.validate output

library(ggplot2)
library(dplyr)

dvs&lt;-dupGet(alleleINF,test=c("z.05","chi.05"))
dvd&lt;-dup.validate(dvs,window.size = 1000)

# Example data frame
df &lt;- data.frame(dvd[,3:5])
df$cnv.ratio&lt;-as.numeric(df$cnv.ratio)

# Calculate midpoints
df &lt;- df %&gt;%
  mutate(midpoint = (start + end) / 2)

ggplot() +
  # Horizontal segments for each start-end range
  geom_segment(data = df, aes(x = start, xend = end,
  y = cnv.ratio, yend = cnv.ratio), color = "blue") +
  # Midpoints line connecting midpoints of each range
  geom_path(data = df, aes(x = midpoint, y = cnv.ratio), color = "red") +
  geom_point(data = df, aes(x = midpoint, y = cnv.ratio), color = "red") +
  # Aesthetic adjustments
  theme_minimal() +
  labs(title = "CNV Ratio along a Continuous Axis with Midpoint Fluctuation",
      x = "Genomic Position",
       y = "CNV Ratio")

## End(Not run)


</code></pre>


</div>