<div class="container">

<table style="width: 100%;"><tr>
<td>plotVennDE</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot Venn diagram from multiple DE Analysis results</h2>

<h3>Description</h3>

<p>Plot a Venn diagram from multiple DE Analysis results.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotVennDE(
  DEResults,
  pThreshold = 0.05,
  useFDR = TRUE,
  stat = "logFC",
  statThreshold = 0,
  topToList = 10
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>DEResults</code></td>
<td>
<p>A list of data frames with the results of DE analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pThreshold</code></td>
<td>
<p>The p-value threshold to determine if a gene is differentially expressed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useFDR</code></td>
<td>
<p>Use the FDR adjusted p-value instead of the raw p-value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stat</code></td>
<td>
<p>The additional statistics column to use for filtering differentially expressed genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>statThreshold</code></td>
<td>
<p>The absolute value of the statistic threshold to use for filtering differentially expressed genes.
Default is 0, which means no filtering.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>topToList</code></td>
<td>
<p>The number of common DE genes that are used to annotate the plot</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
library(RCPA)
library(SummarizedExperiment)

affyDEExperiment &lt;- loadData("affyDEExperiment")
agilDEExperiment &lt;- loadData("agilDEExperiment")
RNASeqDEExperiment &lt;- loadData("RNASeqDEExperiment")

DEResults &lt;- list(
    "Affymetrix - GSE5281" = rowData(affyDEExperiment),
    "Agilent - GSE61196"   = rowData(agilDEExperiment),
    "RNASeq - GSE153873"   = rowData(RNASeqDEExperiment)
)

DEResultUps &lt;- lapply(DEResults, function(df) df[!is.na(df$logFC) &amp; df$logFC &gt; 0, ])

DEResultDowns &lt;- lapply(DEResults, function(df) df[!is.na(df$logFC) &amp; df$logFC &lt; 0, ])

if (require("ggvenn", quietly = TRUE)){
p1 &lt;- RCPA::plotVennDE(DEResults) + 
        ggplot2::ggtitle("All DE Genes")
p2 &lt;- RCPA::plotVennDE(DEResultUps) +
        ggplot2::ggtitle("Up-regulated DE Genes")
p3 &lt;- RCPA::plotVennDE(DEResultDowns) + 
        ggplot2::ggtitle("Down-regulated DE Genes")
}


</code></pre>


</div>