<div class="container">

<table style="width: 100%;"><tr>
<td>readLGF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Extracting LGF karyotype data from JSON files </h2>

<h3>Description</h3>

<p>A function to read binary karyotype data, stored in LGF format in JSON
files produced by the CytoGPS web site, into R for further analysis.
</p>


<h3>Usage</h3>

<pre><code class="language-R">readLGF(files = NULL, folder = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>files</code></td>
<td>
<p>The name of the JSON file (or a character vector of
such file names) from which you want to extract and format
data. If <code>NULL</code>, then it will extract all JSON files within
the folder path provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>folder</code></td>
<td>
<p>The specified directory/folder from which the user
wants to extract JSON files. If <code>NULL</code>, then the function
will look in the current working directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>A logical value; should the function keep you informed about what it is doing?</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>CytoGPS is an algorithm that converts conventional karyotypes from the
standard text-based notation (the International Standard for Human
Cytogenetic/Cytogenomic Nomenclature; ISCN) into binary vectors with
three bits (loss, gain, or fusion) per cytoband, which we call the LGF
model. The web site <a href="http://cytogps.org">http://cytogps.org</a> provides an
implementation that allows users to upload text files containing one
karyotype per line. It produces its output as a file in JavaScript
Object Notation (JSON).
</p>
<p>The <code>readLGF</code> function reads and parses these JSON files and
converts them into an R data structure. The <code>raw</code> component of
this structure contains binary matrices that can serve as input to the
Mercator package (see <code>Mercator-class</code>) for
unsupervised analyses. The <code>frequency</code> component summarizes the
fraction of input karyotype-clones with each abnormality, and can be
visualized with other tools in the <code>RCytoGPS</code> package.
</p>


<h3>Value</h3>

<p>A list containing five elements:
</p>
<table>
<tr style="vertical-align: top;">
<td>
<code>source</code>:</td>
<td>
<p>A character vector containing the names of the
JSON files from which data was read.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td>
<code>raw</code>:</td>
<td>
<p>A list of lists, one per JSON source file. Each
internal list contains two elements, <code>Status</code> and
<code>LGF</code>. <code>Status</code> is a data frame with one row per
karyotype in the input file, describing the results of CytoGPS
parsing and mapping. Results can be "<code>Success</code>",
"<code>Nonfixable grammar error</code>",
"<code>Validation error</code>",
"<code>Fixable grammar error and success</code>",  or
"<code>Fixable grammar error but containing validation error</code>".
<code>LGF</code> is a data frame where the columns are LGF-cytobands and the
rows are clones from the successfully processed input karyotypes;
each karyotype can have multiple clones. Entries are zero or one
indicating the absence or presence of an abnormality.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td>
<code>frequency</code>:</td>
<td>
<p>A data frame, where the rows are cytobands
and the columns are Loss, Gain, and Fusion, with three columns per
input file. Entries are the fraction of karyotype clones with
that abnormality.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td>
<code>size</code>:</td>
<td>
<p>An integer vector containing the total number of
clones per input file. These values can be used to turn fractions
back into counts.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td>
<code>CL</code>:</td>
<td>
<p>A data frame with one row per cytoband detailing
the chromosomal location and (grayscale) color of the band
produced by Giemsa staining.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Dwayne G. Tally <a href="mailto:dtally110@hotmail.com">dtally110@hotmail.com</a>
</p>


<h3>References</h3>

<p>Abrams ZB, Zhang L, Abruzzo LV, Heerema NA, Li S, Dillon T, Rodriguez R,
Coombes KR, Payne PRO. CytoGPS: a web-enabled karyotype analysis tool
for cytogenetics. Bioinformatics. 2019 Dec 15;35(24):5365-5366.
</p>
<p>Abrams ZB, Tally DG,  Zhang L, Coombes CE, Payne PRO, Abruzzo LV,
Coombes KR. Pattern recognition in lymphoid malignancies using CytoGPS
and Mercator. Under review.
</p>
<p>Abrams ZB, Tally DG, Coombes KR. RCytoGPS: An R Package for Analyzing
and Visualizing Cytogenetic Data. In preparation.
</p>


<h3>See Also</h3>

<p><code>Mercator-class</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">jsonDir &lt;- system.file("Examples/JSONfiles", package = "RCytoGPS")
x &lt;- readLGF(folder = jsonDir)

jsonFile &lt;- dir(jsonDir, pattern = "*.json")[1]
y &lt;- readLGF(jsonFile, jsonDir)

</code></pre>


</div>