<div class="container">

<table style="width: 100%;"><tr>
<td>kmer</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The Basic Kmer Descriptor</h2>

<h3>Description</h3>

<p>The Basic Kmer Descriptor
</p>


<h3>Usage</h3>

<pre><code class="language-R">kmer(x, k = 2, upto = FALSE, normalize = FALSE, reverse = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>the input data, which should be a list or file type.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>the k value of kmer, it should be an integer larger than 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>upto</code></td>
<td>
<p>generate all the kmers: 1mer, 2mer, ..., kmer. The output feature vector is
the combination of all these kmers. The default value of this parameter is False.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normalize</code></td>
<td>
<p>with this option, the final feature vector will be normalized based
on the total occurrences of all kmers. Therefore, the elements in the feature vectors
represent the frequencies of kmers. The default value of this parameter is False.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reverse</code></td>
<td>
<p>make reverse complements into a single feature, The default value of this parameter is False.
if reverse is True, this method returns the reverse compliment kmer feature vector.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function calculates the basic kmer descriptor
</p>


<h3>Value</h3>

<p>A vector
</p>


<h3>Note</h3>

<p>if the parameters normalize and upto are both True, and then the feature vector is
the combination of all these normalized kmers, e.g. the combination of normalized 1-kmer
and normalized 2-kmer when k=2, normalize=True, upto=True.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>References</h3>

<p>Noble W S, Kuehn S, Thurman R, et al. Predicting the in vivo signature of human gene regulatory sequences.
<em>Bioinformatics</em>, 2005, 21 Suppl 1, i338-343.
Lee D, Karchin R, Beer M A. Discriminative prediction of mammalian enhancers from
DNA sequence. <em>Genome research</em>. 2005, 21, 2167-2180.
</p>


<h3>See Also</h3>

<p>See <code>make_kmer_index</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">x = 'GACTGAACTGCACTTTGGTTTCATATTATTTGCTC'
kmer(x)
</code></pre>


</div>