<div class="container">

<table style="width: 100%;"><tr>
<td>extrDCC</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The Dinucleotide-based Cross Covariance Descriptor</h2>

<h3>Description</h3>

<p>The Dinucleotide-based Cross Covariance Descriptor
</p>


<h3>Usage</h3>

<pre><code class="language-R">extrDCC(x, index = c("Twist", "Tilt"), nlag = 2, normaliztion = FALSE,
  customprops = NULL, allprop = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>the input data, which should be a list or file type.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>index</code></td>
<td>
<p>the physicochemical indices, it should be a list and there are 38
different physicochemical indices (Table 1), which the users can choose.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nlag</code></td>
<td>
<p>an integer larger than or equal to 0 and less than or equal to L-2 (L means the length
of the shortest DNA sequence in the dataset). It represents the distance between two dinucleotides.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normaliztion</code></td>
<td>
<p>with this option, the final feature vector will be normalized based
on the total occurrences of all kmers. Therefore, the elements in the feature vectors
represent the frequencies of kmers. The default value of this parameter is False.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>customprops</code></td>
<td>
<p>the users can use their own indices to generate the feature vector. It should be a dict,
the key is dinucleotide (string), and its corresponding value is a list type.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allprop</code></td>
<td>
<p>all the 38 physicochemical indices will be
employed to generate the feature vector. Its default value is False.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function calculates the dinucleotide-based cross covariance descriptor
</p>


<h3>Value</h3>

<p>A vector
</p>


<h3>Note</h3>

<p>if the user defined physicochemical indices have not been normalized, it should be normalized.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>References</h3>

<p>Dong Q, Zhou S, Guan J. A new taxonomy-based protein fold recognition approach based on
autocross-covariance transformation. <em>Bioinformatics</em>, 2009, 25(20): 2655-2662.
</p>


<h3>See Also</h3>

<p>See <code>extrDAC</code> and <code>extrDACC</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">x = 'GACTGAACTGCACTTTGGTTTCATATTATTTGCTC'
extrDCC(x)
</code></pre>


</div>