<div class="container">

<table style="width: 100%;"><tr>
<td>parSeqSim</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Parallellized DNA/RNA Sequence Similarity Calculation based on Sequence Alignment</h2>

<h3>Description</h3>

<p>Parallellized DNA/RNA Sequence Similarity Calculation based on Sequence Alignment
</p>


<h3>Usage</h3>

<pre><code class="language-R">parSeqSim(dnalist, cores = 2, type = "local", submat = "BLOSUM62")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dnalist</code></td>
<td>
<p>A length <code>n</code> list containing <code>n</code> DNA/RNA sequences,
each component of the list is a character string, storing one DNA/RNA sequence.
Unknown sequences should be represented as <code>''</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cores</code></td>
<td>
<p>Integer. The number of CPU cores to use for parallel execution,
default is <code>2</code>. Users could use the <code>detectCores()</code> function
in the <code>parallel</code> package to see how many cores they could use.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Type of alignment, default is <code>'local'</code>,
could be <code>'global'</code> or <code>'local'</code>,
where <code>'global'</code> represents Needleman-Wunsch global alignment;
<code>'local'</code> represents Smith-Waterman local alignment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>submat</code></td>
<td>
<p>Substitution matrix, default is <code>'BLOSUM62'</code>, could be one of
<code>'BLOSUM45'</code>, <code>'BLOSUM50'</code>, <code>'BLOSUM62'</code>, <code>'BLOSUM80'</code>, <code>'BLOSUM100'</code>,
<code>'PAM30'</code>, <code>'PAM40'</code>, <code>'PAM70'</code>, <code>'PAM120'</code>, <code>'PAM250'</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function implemented the parallellized version for calculating
DNA/RNA sequence similarity based on sequence alignment.
</p>


<h3>Value</h3>

<p>A <code>n</code> x <code>n</code> similarity matrix.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu  &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code>twoSeqSim</code> for DNA/RNA sequence alignment
for two DNA/RNA sequences. See <code>parGOSim</code> for
DNA/RNA similarity calculation based on
Gene Ontology (GO) semantic similarity.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Be careful when testing this since it involves parallelisation
# and might produce unpredictable results in some environments

require(Biostrings)
require(foreach)
require(doParallel)

s1 = readFASTA(system.file('dnaseq/hs.fasta', package = 'rDNA'))[[1]]
s2 = readFASTA(system.file('dnaseq/hs.fasta', package = 'rDNA'))[[2]]
s3 = readFASTA(system.file('dnaseq/hs.fasta', package = 'rDNA'))[[3]]
s4 = readFASTA(system.file('dnaseq/hs.fasta', package = 'rDNA'))[[4]]
s5 = readFASTA(system.file('dnaseq/hs.fasta', package = 'rDNA'))[[5]]
plist = list(s1, s2, s3, s4, s5)
psimmat = parSeqSim(plist, cores = 2, type = 'local', submat = 'BLOSUM62')
print(psimmat)
</code></pre>


</div>