<div class="container">

<table style="width: 100%;"><tr>
<td>twoSeqSim</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>DNA/RNA Sequence Alignment for Two DNA/RNA Sequences</h2>

<h3>Description</h3>

<p>DNA/RNA Sequence Alignment for Two DNA/RNA Sequences
</p>


<h3>Usage</h3>

<pre><code class="language-R">twoSeqSim(seq1, seq2, type = "local", submat = "BLOSUM62")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seq1</code></td>
<td>
<p>A character string, containing one DNA/RNA sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq2</code></td>
<td>
<p>A character string, containing another DNA/RNA sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Type of alignment, default is <code>'local'</code>,
could be <code>'global'</code> or <code>'local'</code>,
where <code>'global'</code> represents Needleman-Wunsch global alignment;
<code>'local'</code> represents Smith-Waterman local alignment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>submat</code></td>
<td>
<p>Substitution matrix, default is <code>'BLOSUM62'</code>, could be one of
<code>'BLOSUM45'</code>, <code>'BLOSUM50'</code>, <code>'BLOSUM62'</code>, <code>'BLOSUM80'</code>, <code>'BLOSUM100'</code>,
<code>'PAM30'</code>, <code>'PAM40'</code>, <code>'PAM70'</code>, <code>'PAM120'</code>, <code>'PAM250'</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function implements the sequence alignment between two DNA/RNA sequences.
</p>


<h3>Value</h3>

<p>An Biostrings object containing the scores and other alignment information.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu  &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code>parSeqSim</code> for paralleled pairwise
DNA/RNA similarity calculation based on sequence alignment.
See <code>twoGOSim</code> for calculating the
GO semantic similarity between two groups of GO terms or two Entrez gene IDs.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Be careful when testing this since it involves sequence alignment
# and might produce unpredictable results in some environments

require(Biostrings)

s1 = readFASTA(system.file('dnaseq/hs.fasta', package = 'rDNA'))[[1]]
s2 = readFASTA(system.file('dnaseq/hs.fasta', package = 'rDNA'))[[2]]
seqalign = twoSeqSim(s1, s2)
summary(seqalign)
print(seqalign@score)
</code></pre>


</div>