<div class="container">

<table style="width: 100%;"><tr>
<td>RCLSMIX-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Class <code>"RCLSMIX"</code>
</h2>

<h3>Description</h3>

<p>Object of class <code>RCLSMIX</code>.</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("RCLSMIX", ...)</code>. Accessor methods for the slots are <code>a.o(x = NULL)</code>,
<code>a.Dataset(x = NULL)</code>, <code>a.s(x = NULL)</code>, <code>a.ntrain(x = NULL)</code>, <code>a.P(x = NULL)</code>, <code>a.ntest(x = NULL)</code>, <code>a.Zt(x = NULL)</code>,
<code>a.Zp(x = NULL)</code>, <code>a.CM(x = NULL)</code>, <code>a.Accuracy(x = NULL)</code>, <code>a.Error(x = NULL)</code>, <code>a.Precision(x = NULL)</code>, <code>a.Sensitivity(x = NULL)</code>,
<code>a.Specificity(x = NULL)</code> and <code>a.Chunks(x = NULL)</code>, where <code>x</code> stands for an object of class <code>RCLSMIX</code>.
</p>


<h3>Slots</h3>


<dl>
<dt>
<code>x</code>:</dt>
<dd>
<p>a list of objects of class <code>REBMIX</code> of length <code class="reqn">o</code> obtained by running <code>REBMIX</code> on <code class="reqn">g = 1, \ldots, s</code> train datasets <code class="reqn">Y_{\mathrm{train}g}</code> all of length <code class="reqn">n_{\mathrm{train}g}</code>.
For the train datasets the corresponding class membership <code class="reqn">\bm{\Omega}_{g}</code> is known. This yields
<code class="reqn">n_{\mathrm{train}} = \sum_{g = 1}^{s} n_{\mathrm{train}g}</code>, while <code class="reqn">Y_{\mathrm{train}q} \cap Y_{\mathrm{train}g} = \emptyset</code> for all <code class="reqn">q \neq g</code>.
Each object in the list corresponds to one chunk, e.g., <code class="reqn">(y_{1j}, y_{3j})^{\top}</code>.
</p>
</dd>
<dt>
<code>o</code>:</dt>
<dd>
<p>number of chunks <code class="reqn">o</code>. <code class="reqn">Y = \{\bm{y}_{j}; \ j = 1, \ldots, n\}</code> is an observed <code class="reqn">d</code>-dimensional dataset of size <code class="reqn">n</code> of vector observations <code class="reqn">\bm{y}_{j} = (y_{1j}, \ldots, y_{dj})^{\top}</code> and
is partitioned into train and test datasets. Vector observations <code class="reqn">\bm{y}_{j}</code> may further be split into <code class="reqn">o</code> chunks when running <code>REBMIX</code>, e.g.,
for <code class="reqn">d = 6</code> and <code class="reqn">o = 3</code> the set of chunks substituting <code class="reqn">\bm{y}_{j}</code> may be as follows <code class="reqn">(y_{1j}, y_{3j})^{\top}</code>, <code class="reqn">(y_{2j}, y_{4j}, y_{6j})^{\top}</code> and <code class="reqn">y_{5j}</code>.
</p>
</dd>
<dt>
<code>Dataset</code>:</dt>
<dd>
<p>a data frame containing test dataset <code class="reqn">Y_{\mathrm{test}}</code> of length <code class="reqn">n_{\mathrm{test}}</code>. For the test dataset the corresponding class membership <code class="reqn">\bm{\Omega}_{g}</code> is not known.
</p>
</dd>
<dt>
<code>s</code>:</dt>
<dd>
<p>finite set of size <code class="reqn">s</code> of classes <code class="reqn">\bm{\Omega} = \{\bm{\Omega}_{g}; \ g = 1, \ldots, s\}</code>.
</p>
</dd>
<dt>
<code>ntrain</code>:</dt>
<dd>
<p>a vector of length <code class="reqn">s</code> containing numbers of observations in train datasets <code class="reqn">Y_{\mathrm{train}g}</code>.
</p>
</dd>
<dt>
<code>P</code>:</dt>
<dd>
<p>a vector of length <code class="reqn">s</code> containing prior probabilities <code class="reqn">P(\bm{\Omega}_{g}) = \frac{n_{\mathrm{train}g}}{n_{\mathrm{train}}}</code>.
</p>
</dd>
<dt>
<code>ntest</code>:</dt>
<dd>
<p>number of observations in test dataset <code class="reqn">Y_{\mathrm{test}}</code>.
</p>
</dd>
<dt>
<code>Zt</code>:</dt>
<dd>
<p>a factor of true class membership <code class="reqn">\bm{\Omega}_{g}</code> for the test dataset.
</p>
</dd>
<dt>
<code>Zp</code>:</dt>
<dd>
<p>a factor of predictive class membership <code class="reqn">\bm{\Omega}_{g}</code> for the test dataset.
</p>
</dd>
<dt>
<code>CM</code>:</dt>
<dd>
<p>a table containing confusion matrix for multiclass classifier. It contains
number <code class="reqn">x_{qg}</code> of test observations with the true class <code class="reqn">q</code> that are classified into the class <code class="reqn">g</code>, where <code class="reqn">q, g = 1, \ldots, s</code>.
</p>
</dd>
<dt>
<code>Accuracy</code>:</dt>
<dd>
<p>proportion of all test observations that are classified correctly. <code class="reqn">\mathrm{Accuracy} = \frac{\sum_{g = 1}^{s} x_{gg}}{n_{\mathrm{test}}}</code>.
</p>
</dd>
<dt>
<code>Error</code>:</dt>
<dd>
<p>proportion of all test observations that are classified wrongly. <code class="reqn">\mathrm{Error} = 1 - \mathrm{Accuracy}</code>.
</p>
</dd>
<dt>
<code>Precision</code>:</dt>
<dd>
<p>a vector containing proportions of predictive observations in class <code class="reqn">g</code> that are
classified correctly into class <code class="reqn">g</code>. <code class="reqn">\mathrm{Precision}(g) = \frac{x_{gg}}{\sum_{q = 1}^{s} x_{qg}}</code>.
</p>
</dd>
<dt>
<code>Sensitivity</code>:</dt>
<dd>
<p>a vector containing proportions of test observations in class <code class="reqn">g</code> that are classified
correctly into class <code class="reqn">g</code>. <code class="reqn">\mathrm{Sensitivity}(g) = \frac{x_{gg}}{\sum_{q = 1}^{s} x_{gq}}</code>.
</p>
</dd>
<dt>
<code>Specificity</code>:</dt>
<dd>
<p>a vector containing proportions of test observations that are not in class <code class="reqn">g</code> and
are classified into the non <code class="reqn">g</code> class. <code class="reqn">\mathrm{Specificity}(g) = \frac{n_{\mathrm{test}} - \sum_{q = 1}^{s} x_{qg}}{n_{\mathrm{test}} - \sum_{q = 1}^{s} x_{gq}}</code>.
</p>
</dd>
<dt>
<code>Chunks</code>:</dt>
<dd>
<p>a vector containing selected chunks.
</p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>Marko Nagode</p>


<h3>References</h3>

<p>D. M. Dziuda. Data Mining for Genomics and Proteomics: Analysis of Gene and Protein
Expression Data. John Wiley &amp; Sons, New York, 2010.
</p>


</div>