<div class="container">

<table style="width: 100%;"><tr>
<td>discoversomaticInteractions</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>discoversomaticInteractions</h2>

<h3>Description</h3>

<p>Function adapted to maftools where given a .maf file, it graphs the somatic interactions between a group of genes, i.e., the combination of gene expression and mutation data to detect mutually exclusive or co-ocurring events.
</p>


<h3>Usage</h3>

<pre><code class="language-R">discoversomaticInteractions(
  maf,
  top = 25,
  genes = NULL,
  pvalue = c(0.05, 0.01),
  getMutexMethod = "ShiftedBinomial",
  getMutexMixed = TRUE,
  returnAll = TRUE,
  geneOrder = NULL,
  fontSize = 0.8,
  showSigSymbols = TRUE,
  showCounts = FALSE,
  countStats = "all",
  countType = "all",
  countsFontSize = 0.8,
  countsFontColor = "black",
  colPal = "BrBG",
  showSum = TRUE,
  colNC = 9,
  nShiftSymbols = 5,
  sigSymbolsSize = 2,
  sigSymbolsFontSize = 0.9,
  pvSymbols = c(46, 42),
  limitColorBreaks = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>maf</code></td>
<td>
<p>maf object generated by read.maf</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>top</code></td>
<td>
<p>check for interactions among top 'n' number of genes. Defaults to top 25. genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genes</code></td>
<td>
<p>List of genes among which interactions should be tested. If not provided, test will be performed between top 25 genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvalue</code></td>
<td>
<p>Default c(0.05, 0.01) p-value threshold. You can provide two values for upper and lower threshold.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>getMutexMethod</code></td>
<td>
<p>Method for the 'getMutex' function (by default "ShiftedBinomial")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>getMutexMixed</code></td>
<td>
<p>Mixed parameter for the 'getMutex' function (by default TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>returnAll</code></td>
<td>
<p>If TRUE returns test statistics for all pair of tested genes. Default FALSE, returns for only genes below pvalue threshold.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneOrder</code></td>
<td>
<p>Plot the results in given order. Default NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fontSize</code></td>
<td>
<p>cex for gene names. Default 0.8</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showSigSymbols</code></td>
<td>
<p>Default TRUE. Heighlight significant pairs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showCounts</code></td>
<td>
<p>Default FALSE. Include number of events in the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>countStats</code></td>
<td>
<p>Default 'all'. Can be 'all' or 'sig'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>countType</code></td>
<td>
<p>Default 'all'. Can be 'all', 'cooccur', 'mutexcl'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>countsFontSize</code></td>
<td>
<p>Default 0.8</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>countsFontColor</code></td>
<td>
<p>Default 'black'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colPal</code></td>
<td>
<p>colPalBrewer palettes. Default 'BrBG'. See RColorBrewer::display.brewer.all() for details</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showSum</code></td>
<td>
<p>show [sum] with gene names in plot, Default TRUE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colNC</code></td>
<td>
<p>Number of different colors in the palette, minimum 3, default 9</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nShiftSymbols</code></td>
<td>
<p>shift if positive shift SigSymbols by n to the left, default 5</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigSymbolsSize</code></td>
<td>
<p>size of symbols in the matrix and in legend. Default 2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigSymbolsFontSize</code></td>
<td>
<p>size of font in legends. Default 0.9</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvSymbols</code></td>
<td>
<p>vector of pch numbers for symbols of p-value for upper and lower thresholds c(upper, lower). Default c(46, 42)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>limitColorBreaks</code></td>
<td>
<p>limit color to extreme values. Default TRUE</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Internally, this function run the getMutex function. With the 'getMutexMethod' parameter user might select
the 'method' parameter of the getMutex function. For more details run '?getMutex'
</p>
<p>#' @return A list of data.tables and it will print a heatmap with the results.
</p>


<h3>References</h3>

<p>Mayakonda A, Lin DC, Assenov Y, Plass C, Koeffler HP. 2018. Maftools:
efficient and comprehensive analysis of somatic variants in cancer.
Genome Research. http://dx.doi.org/10.1101/gr.239244.118
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Not run: 

  #An example of how to perform the function,
  #using data from TCGA, Colon Adenocarcinoma in this case. 
  
  #coad.maf &lt;- GDCquery_Maf("COAD", pipelines = "muse") %&gt;% read.maf
  coad.maf &lt;- read.maf(GDCquery_Maf("COAD", pipelines = "muse"))
  discoversomaticInteractions(maf = coad.maf, top = 35, pvalue = c(1e-2, 2e-3))
  
## End(Not run)




</code></pre>


</div>