<div class="container">

<table style="width: 100%;"><tr>
<td>refdb_import_NCBI</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Download and import NCBI Nucleotide records</h2>

<h3>Description</h3>

<p>This function allows to search and download data
from the the NCBI Nucleotide database.
Additionally it uses the NCBI Taxonomy database
to get the sequence taxonomic classification.
</p>


<h3>Usage</h3>

<pre><code class="language-R">refdb_import_NCBI(
  query,
  full = FALSE,
  max_seq_length = 10000,
  seq_bin = 200,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>query</code></td>
<td>
<p>a character string with the query.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>full</code></td>
<td>
<p>a logical. If FALSE (the default), only a subset of
the most important fields is included in the result.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_seq_length</code></td>
<td>
<p>a numeric giving the maximum length of sequences
to retrieve. Useful to exclude complete genomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq_bin</code></td>
<td>
<p>number of sequences to download at once.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>print information in the console.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function uses several functions of the <span class="pkg">rentrez</span>
package to interface with the NCBI's EUtils API.*
</p>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>Errors</h3>

<p><code>Error in curl::curl_fetch_memory(url, handle = handle) :
transfer closed with outstanding read data remaining</code> <br>
This error seems to appear with long sequences.
You can try to decrease <code>max_seq_length</code> to exclude them.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
silo_ncbi &lt;- refdb_import_NCBI("Silo COI")

## End(Not run)

</code></pre>


</div>