<div class="container">

<table style="width: 100%;"><tr>
<td>refseq_CDSseq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Extract the CDS nucleotide sequence into a Biostrings object</h2>

<h3>Description</h3>

<p><code>refseq_CDSseq()</code> Parses a single/multiple transcript accessions (RefSeq format) and extract
the CDS nucleotide sequences into a <code>DNAStringSet</code> object.
</p>
<p>Depending on the function, available accessions in <code>refseqR</code> include RefSeq models with the prefixes XM_ (mRNA), XR_ (non-coding RNA), and XP_ (protein), as well as subsequently curated RefSeq records with NM_, NR_, or NP_ accession prefixes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">refseq_CDSseq(transcript)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>transcript</code></td>
<td>
<p>A character string of the single/multiple transcript id.</p>
</td>
</tr></table>
<h3>Value</h3>

<p>An object of <code>DNAStringSet</code> class.
</p>


<h3>Author(s)</h3>

<p>Jose V. Die
</p>


<h3>See Also</h3>

<p><code>refseq_CDScoords</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
transcript &lt;-  c("XM_004487701", "XM_004488493", "XM_004501904")
my_cds &lt;- refseq_CDSseq(transcript)
# Now, the `DNAStringSet` can easily used to make a fasta file :
# writeXStringSet(x= my_cds, filepath = "cds_result")

</code></pre>


</div>