<div class="container">

<table style="width: 100%;"><tr>
<td>regnet</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>fit a regression for given lambda with network-based regularization</h2>

<h3>Description</h3>

<p>Network-based penalization regression for given values of <code class="reqn">\lambda_{1}</code> and <code class="reqn">\lambda_{2}</code>.
Typical usage is to have the cv.regnet function compute the optimal lambdas, then provide them to the
regnet function. Users could also use MCP or Lasso.
</p>


<h3>Usage</h3>

<pre><code class="language-R">regnet(
  X,
  Y,
  response = c("binary", "continuous", "survival"),
  penalty = c("network", "mcp", "lasso"),
  lamb.1 = NULL,
  lamb.2 = NULL,
  r = NULL,
  clv = NULL,
  initiation = NULL,
  alpha.i = 1,
  robust = FALSE,
  debugging = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>a matrix of predictors without intercept. Each row should be an observation vector. A column of 1 will be added to the X matrix
by the program as the intercept.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>the response variable. For response="binary", Y should be a numeric vector with zeros and ones. For response="survival", Y should be a
two-column matrix with columns named 'time' and 'status'. The latter is a binary variable, with '1' indicating an event, and '0'
indicating censoring.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>response</code></td>
<td>
<p>the response type. regnet supports three types of response: "binary", "continuous" and "survival".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>penalty</code></td>
<td>
<p>the penalty type. regnet provides three choices for the penalty function: "network", "mcp" and "lasso".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lamb.1</code></td>
<td>
<p>the tuning parameter <code class="reqn">\lambda_{1}</code> that imposes sparsity.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lamb.2</code></td>
<td>
<p>the tuning parameter <code class="reqn">\lambda_{2}</code> that controls the smoothness among coefficient profiles. <code class="reqn">\lambda_{2}</code> is  needed
for network penalty.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r</code></td>
<td>
<p>the regularization parameter in MCP. For binary response, r should be larger than 4.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clv</code></td>
<td>
<p>a value or a vector, indexing variables that are not subject to penalty. clv only works for continuous and survival responses
for now, and will be ignored for other types of responses.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>initiation</code></td>
<td>
<p>the method for initiating the coefficient vector. The default method is elastic-net.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha.i</code></td>
<td>
<p>the elastic-net mixing parameter. The program can use the elastic-net for choosing initial values of
the coefficient vector. alpha.i is the elastic-net mixing parameter, with 0 <code class="reqn">\le</code> alpha.i <code class="reqn">\le</code> 1. alpha.i=1 is the
lasso penalty, and alpha.i=0 is the ridge penalty. If the user chooses a method other than elastic-net for initializing
coefficients, alpha.i will be ignored.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>robust</code></td>
<td>
<p>a logical flag. Whether or not to use robust methods. Robust methods are available for survival and continuous response.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>debugging</code></td>
<td>
<p>a logical flag. If TRUE, extra information will be returned.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The current version of regnet supports three types of responses: “binary”, "continuous" and “survival”.
</p>

<ul>
<li>
<p> regnet(…, response="binary", penalty="network") fits a network-based penalized logistic regression.
</p>
</li>
<li>
<p> regnet(…, response="continuous", penalty="network") fits a network-based least square regression.
</p>
</li>
<li>
<p> regnet(…, response="survival", penalty="network", robust=TRUE) fits a robust regularized AFT model using network penalty.
</p>
</li>
</ul>
<p>By default, regnet uses non-robust methods for all types of responses. To use robust methods, simply set robust=TRUE. It is recommended to use robust methods for survival response. Please see the references for more details about the models. Users could also use MCP or Lasso penalty.
</p>
<p>The coefficients are always estimated on a standardized X matrix. regnet standardizes each column of X to have unit variance
(using 1/n rather than 1/(n-1) formula). If the coefficients on the original scale are needed, the user can refit a standard model
using the subset of variables that have non-zero coefficients.
</p>


<h3>Value</h3>

<p>an object of class "regnet" is returned, which is a list with components:
</p>

<ul>
<li>
<p> coeff: a vector of estimated coefficients. Please note that, if there are variables not subject to penalty (indicated by clv),
the order of returned vector is c(Intercept, unpenalized coefficients of clv variables, penalized coefficients of other variables).
</p>
</li>
<li>
<p> Adj: a matrix of adjacency measures of the identified genetic variants. Identified genetic variants are those that have non-zero estimated coefficients.
</p>
</li>
</ul>
<h3>References</h3>

<p>Ren, J., He, T., Li, Y., Liu, S., Du, Y., Jiang, Y., and Wu, C. (2017).
Network-based regularization for high dimensional SNP data in the case-control study of
Type 2 diabetes. <em>BMC Genetics</em>, 18(1):44 <a href="https://doi.org/10.1186/s12863-017-0495-5">doi:10.1186/s12863-017-0495-5</a>
</p>
<p>Ren, J., Du, Y., Li, S., Ma, S., Jiang,Y. and Wu, C. (2019). Robust network-based regularization
and variable selection for high dimensional genomics data in cancer prognosis.
<em>Genet. Epidemiol.</em>, 43:276-291 <a href="https://doi.org/10.1002/gepi.22194">doi:10.1002/gepi.22194</a>
</p>


<h3>See Also</h3>

<p><code>cv.regnet</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Survival response
data(SurvExample)
X = rgn.surv$X
Y = rgn.surv$Y
clv = c(1:5) # variables 1 to 5 are clinical variables which we choose not to penalize.
penalty = "network"
fit = regnet(X, Y, "survival", penalty, rgn.surv$lamb1, rgn.surv$lamb2, clv=clv, robust=TRUE)
index = which(rgn.surv$beta != 0)
pos = which(fit$coeff != 0)
tp = length(intersect(index, pos))
fp = length(pos) - tp
list(tp=tp, fp=fp)

</code></pre>


</div>