<div class="container">

<table style="width: 100%;"><tr>
<td>haplohh2sweepfinder</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Translate object of <code>haplohh-class</code> into SweepFinder format</h2>

<h3>Description</h3>

<p>Extract allele frequencies of an object of class <code>haplohh-class</code>
and returns a table in SweepFinder input format.
</p>


<h3>Usage</h3>

<pre><code class="language-R">haplohh2sweepfinder(haplohh, polarized = TRUE, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>haplohh</code></td>
<td>
<p>object of class <code>haplohh-class</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>polarized</code></td>
<td>
<p>logical. If <code>TRUE</code> (default), flag "folded" is set to 0, otherwise to 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical. If <code>TRUE</code> (default), prints filter statements.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>SweepFinder and SweeD are two stand-alone programs which
implement the same method to detect selective sweeps using the
allele frequency at each site. This function calculates these frequencies
from a <code>haplohh-class</code> and returns a table which
can be saved into a file (with tabs as separators, without row names and quotes) that can
be used as input for the two programs.
</p>
<p>Sites with less than two haplotypes genotyped or with more than two alleles are removed.
If <code>polarized</code>, sites monomorphic for the ancestral allele are removed, too.
</p>


<h3>Value</h3>

<p>A dataframe with four columns:
</p>

<ul>
<li> <p><strong>position</strong> marker position
</p>
</li>
<li> <p><strong>x</strong> (absolute) frequency of the alternative (derived) variant
</p>
</li>
<li> <p><strong>n</strong> number of non-missing genotypes
</p>
</li>
<li> <p><strong>folded</strong> a flag marking polarization
</p>
</li>
</ul>
<h3>References</h3>

<p>DeGiorgio, M., and, Huber, CD and Hubisz, MJ and, Hellmann, I. and Nielsen, R. (2016)
SweepFinder2: increased robustness and flexibility. <em>Bioinformatics</em> <strong>32</strong>:1895-1897
</p>
<p>Pavlidis, P., D. Zivkovic, A. Stamatakis, and N. Alachiotis, (2013)
SweeD: likelihood-based detection of selective sweeps in thousands of genomes.
<em>Molecular Biology and Evolution</em> <strong>30</strong>: 2224-34.
</p>


<h3>See Also</h3>

<p><code>haplohh-class</code>, <code>data2haplohh</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#example
# sweepfinder example from vignette
make.example.files()
hh &lt;- data2haplohh("example_sweep_with_recombination.vcf")
haplohh2sweepfinder(hh)
remove.example.files()
</code></pre>


</div>