<div class="container">

<table style="width: 100%;"><tr>
<td>calc_ehh</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>EHH and iHH computation for a given focal marker</h2>

<h3>Description</h3>

<p>Compute Extended Haplotype Homozygosity (EHH) and integrated EHH (iHH) for a given focal marker.
</p>


<h3>Usage</h3>

<pre><code class="language-R">calc_ehh(
  haplohh,
  mrk,
  limhaplo = 2,
  limhomohaplo = 2,
  limehh = 0.05,
  include_zero_values = FALSE,
  include_nhaplo = FALSE,
  phased = TRUE,
  polarized = TRUE,
  scalegap = NA,
  maxgap = NA,
  discard_integration_at_border = TRUE,
  lower_y_bound = limehh,
  interpolate = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>haplohh</code></td>
<td>
<p>an object of class <code>haplohh</code> (see <code>data2haplohh</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrk</code></td>
<td>
<p>integer representing the number of the focal marker within the haplohh object
or string representing its ID/name.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>limhaplo</code></td>
<td>
<p>if there are less than <code>limhaplo</code> chromosomes that can be used for
the calculation of EHH, the calculation is stopped. The option is intended for the case of missing data,
which leads to the successive exclusion of haplotypes: the further away from the focal marker
the less haplotypes contribute to EHH.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>limhomohaplo</code></td>
<td>
<p>if there are less than <code>limhomohaplo</code> homozygous chromosomes, the
calculation is stopped. This option is intended for unphased data and should be invoked only
if relatively low frequency variants are not filtered subsequently (see main vignette and Klassmann et al. 2020).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>limehh</code></td>
<td>
<p>limit at which EHH stops to be evaluated</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>include_zero_values</code></td>
<td>
<p>logical. If <code>FALSE</code>, return values only for those positions where the calculation is
actually performed, i.e. until stopped by reaching either <code>limehh</code> or <code>limhaplo</code>. If <code>TRUE</code>, report EHH values for
all markers, the additional ones being zero.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>include_nhaplo</code></td>
<td>
<p>logical. If <code>TRUE</code>, report the number of evaluated haplotypes at each marker
(only informative, if missing data leads to a decrease of evaluated haplotypes).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phased</code></td>
<td>
<p>logical. If <code>TRUE</code> (default) chromosomes are expected to be phased. If <code>FALSE</code>, the haplotype data is assumed to
consist of pairwise ordered chromosomes belonging to diploid individuals.
EHH is then estimated over individuals which are homozygous at the focal marker.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>polarized</code></td>
<td>
<p>logical. <code>TRUE</code> by default. If <code>FALSE</code>, use major and minor allele instead of ancestral and derived. If there
are more than two alleles then the minor allele refers to the second-most frequent allele.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scalegap</code></td>
<td>
<p>scale or cap gaps larger than the specified size to the specified size (default=<code>NA</code>, i.e. no scaling).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxgap</code></td>
<td>
<p>maximum allowed gap in bp between two markers. If exceeded, further calculation of EHH is stopped at the gap
(default=<code>NA</code>, i.e no limitation).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>discard_integration_at_border</code></td>
<td>
<p>logical. If <code>TRUE</code> (default) and computation reaches first or last marker or a gap larger than <code>maxgap</code>,
iHH is set to <code>NA</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lower_y_bound</code></td>
<td>
<p>lower y boundary of the area to be integrated over (default: <code>limehh</code>). Can be set
to zero for compatibility with the program hapbin.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>interpolate</code></td>
<td>
<p>logical. Affects only IHH values. If <code>TRUE</code> (default), integration
is performed over a continuous EHH curve (values are interpolated linearly between consecutive markers),
otherwise the EHH curve decreases stepwise at markers.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Values for allele-specific Extended Haplotype Homozygosity (EHH) are computed
upstream and downstream of the focal marker for each of its alleles.
These values are integrated with respect to their genomic positions to yield
an 'integrated EHH' (iHH) value for each allele.
</p>


<h3>Value</h3>

<p>The returned value is a list containing the following elements:
</p>

<dl>
<dt>mrk.name</dt>
<dd>
<p>The name/identifier of the focal marker.</p>
</dd>
<dt>freq</dt>
<dd>
<p>A vector with the frequencies of the alleles of the focal marker.</p>
</dd>
<dt>ehh</dt>
<dd>
<p>A data frame with EHH values for each allele of the focal marker.</p>
</dd>
<dt>ihh</dt>
<dd>
<p>A vector with iHH (integrated EHH) values for each allele of the focal marker.</p>
</dd>
</dl>
<h3>References</h3>

<p>Gautier, M. and Naves, M. (2011). Footprints of selection in the ancestral admixture of a New World Creole cattle breed. <em>Molecular Ecology</em>, <strong>20</strong>, 3128-3143.
</p>
<p>Klassmann, A. and Gautier, M. (2020). Detecting selection using Extended Haplotype
Homozygosity-based statistics on unphased or unpolarized data (preprint). 
https://doi.org/10.22541/au.160405572.29972398/v1
</p>
<p>Sabeti, P.C. et al. (2002). Detecting recent positive selection in the human genome from haplotype structure. <em>Nature</em>, <strong>419</strong>, 832-837.
</p>
<p>Sabeti, P.C. et al. (2007). Genome-wide detection and characterization of positive selection in human populations. <em>Nature</em>, <strong>449</strong>, 913-918.
</p>
<p>Tang, K. and Thornton, K.R. and Stoneking, M. (2007). A New Approach for Using Genome Scans to Detect Recent Positive Selection in the Human Genome. <em>Plos Biology</em>, <strong>7</strong>, e171.
</p>
<p>Voight, B.F. and Kudaravalli, S. and Wen, X. and Pritchard, J.K. (2006). A map of recent positive selection in the human genome. <em>Plos Biology</em>, <strong>4</strong>, e72.
</p>


<h3>See Also</h3>

<p><code>data2haplohh</code>, <code>plot.ehh</code>, <code>calc_ehhs</code>, <code>scan_hh</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#example haplohh object (280 haplotypes, 1424 SNPs)
#see ?haplohh_cgu_bta12 for details
data(haplohh_cgu_bta12)
#computing EHH statistics for the marker "F1205400"
#which displays a strong signal of selection
ehh &lt;- calc_ehh(haplohh_cgu_bta12, mrk = "F1205400")
</code></pre>


</div>