<div class="container">

<table style="width: 100%;"><tr>
<td>plot.haplen</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot the length of extended haplotypes around a focal marker</h2>

<h3>Description</h3>

<p>Plot the length of extended haplotype around a focal marker.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'haplen'
plot(
  x,
  allele = NA,
  group_by_allele = TRUE,
  order_by_length = FALSE,
  col = c("blue", "red", "violet", "orange"),
  mrk.col = "gray",
  lwd = 1,
  hap.names = NULL,
  cex.lab = 1,
  family.lab = "",
  offset.lab = 0.5,
  pos.lab = "left",
  legend = NA,
  legend.xy.coords = "automatic",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class <code>haplen</code> generated by <code>calc_haplen</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele</code></td>
<td>
<p>if <code>NA</code> (default), haplotypes of all alleles are plotted,
otherwise for the specified alleles. Alleles must be specified by their
internal coding, i.e. '0' for ancestral resp. major allele, etc.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group_by_allele</code></td>
<td>
<p>logical. If <code>TRUE</code> (default), group chromosomes
by their allele at the focal marker; alleles are ordered by their internal coding
unless parameter <code>alleles</code> is specified. If <code>FALSE</code>, haplotypes are drawn by
their order in the input file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>order_by_length</code></td>
<td>
<p>if <code>TRUE</code>, chromosomes are ordered by their
shared haplotype length.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>color for each allele (as coded internally).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrk.col</code></td>
<td>
<p>color of the vertical line at the focal marker position.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lwd</code></td>
<td>
<p>line width.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hap.names</code></td>
<td>
<p>a vector containing the names of chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.lab</code></td>
<td>
<p>relative letter size of labels. See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>family.lab</code></td>
<td>
<p>font family for labels. See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>offset.lab</code></td>
<td>
<p>offset of labels. See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos.lab</code></td>
<td>
<p>position of haplotype labels. Either <code>"left"</code> , <code>"right"</code> or <code>"both"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend</code></td>
<td>
<p>legend text.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend.xy.coords</code></td>
<td>
<p>if <code>"automatic"</code> (default) places legend either top left or top right;
if <code>"none"</code>, no legend is drawn; otherwise argument is passed to <code>legend</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>other parameters to be passed to <code>plot.default</code>.</p>
</td>
</tr>
</table>
<h3>See Also</h3>

<p><code>calc_haplen</code>, <code>plot.furcation</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#example haplohh object (280 haplotypes, 1424 SNPs)
#see ?haplohh_cgu_bta12 for details
data(haplohh_cgu_bta12)
#plotting length of extended haplotypes for both ancestral and derived allele
#of the marker "F1205400"
#which displays a strong signal of selection
f &lt;- calc_furcation(haplohh_cgu_bta12, mrk = "F1205400")
h &lt;- calc_haplen(f)
plot(h)
plot(h, hap.names = hap.names(haplohh_cgu_bta12), cex.lab = 0.3)
</code></pre>


</div>