<div class="container">

<table style="width: 100%;"><tr>
<td>rehh-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>rehh: Searching for Footprints of Selection using 'Extended Haplotype
Homozygosity' Based Tests</h2>

<h3>Description</h3>

<p>Population genetic data such as 'Single Nucleotide
Polymorphisms' (SNPs) is often used to identify genomic regions
that have been under recent natural or artificial selection
and might provide clues about the molecular mechanisms of adaptation. 
One approach, the concept of an 'Extended Haplotype Homozygosity' (EHH), 
introduced by (Sabeti 2002) &lt;doi:10.1038/nature01140&gt;, has given rise to 
several statistics designed for whole genome scans. 
The package provides functions to compute three of these,
namely: 'iHS' (Voight 2006) &lt;doi:10.1371/journal.pbio.0040072&gt; for
detecting positive or 'Darwinian' selection within a single population as well as
'Rsb' (Tang 2007) &lt;doi:10.1371/journal.pbio.0050171&gt; and 
'XP-EHH' (Sabeti 2007) &lt;doi:10.1038/nature06250&gt;, targeted
at differential selection between two populations. 
Various plotting functions are included to facilitate
visualization and interpretation of these statistics.
</p>


<h3>Details</h3>

<p>See <code>vignette("rehh", package = "rehh")</code> for an overview of the package and
<code>vignette("examples", package = "rehh")</code> for a more detailed discussion of two small example
data sets.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Alexander Klassmann <a href="mailto:rehh@oneoverx.eu">rehh@oneoverx.eu</a>
</p>
<p>Authors:
</p>

<ul>
<li>
<p> Mathieu Gautier <a href="mailto:mathieu.gautier@inrae.fr">mathieu.gautier@inrae.fr</a>
</p>
</li>
<li>
<p> Renaud Vitalis
</p>
</li>
</ul>
<h3>References</h3>

<p>Gautier, M. and Naves, M. (2011). Footprints of selection in the ancestral admixture of a New World Creole cattle breed. <em>Molecular Ecology</em>, <strong>20</strong>, 3128-3143.
</p>
<p>Gautier M. and Vitalis R. (2012). rehh: An R package to detect footprints of selection in
genome-wide SNP data from haplotype structure. <em>Bioinformatics</em>, <strong>28</strong>(8), 1176-1177.
</p>
<p>Gautier M., Klassmann A., and Vitalis R. (2017). rehh 2.0: a reimplementation of the R package
rehh to detect positive selection from haplotype structure. <em>Molecular Ecology Resources</em>, <strong>17</strong>, 78-90.
</p>
<p>Klassmann, A. and Gautier, M. (2020). Detecting selection using Extended Haplotype
Homozygosity-based statistics on unphased or unpolarized data (preprint). 
https://doi.org/10.22541/au.160405572.29972398/v1
</p>
<p>Sabeti, P.C. et al. (2002). Detecting recent positive selection in the human genome from haplotype structure. <em>Nature</em>, <strong>419</strong>, 832-837.
</p>
<p>Sabeti, P.C. et al. (2007). Genome-wide detection and characterization of positive selection in human populations. <em>Nature</em>, <strong>449</strong>, 913-918.
</p>
<p>Tang, K. and Thornton, K.R. and Stoneking, M. (2007). A New Approach for Using Genome Scans to Detect Recent Positive Selection in the Human Genome. <em>Plos Biology</em>, <strong>7</strong>, e171.
</p>
<p>Voight, B.F. and Kudaravalli, S. and Wen, X. and Pritchard, J.K. (2006). A map of recent positive selection in the human genome. <em>Plos Biology</em>, <strong>4</strong>, e72.
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://CRAN.R-project.org/package=rehh">https://CRAN.R-project.org/package=rehh</a>
</p>
</li>
<li> <p><a href="https://gitlab.com/oneoverx/rehh">https://gitlab.com/oneoverx/rehh</a>
</p>
</li>
<li>
<p> Report bugs at <a href="https://gitlab.com/oneoverx/rehh/-/issues">https://gitlab.com/oneoverx/rehh/-/issues</a>
</p>
</li>
</ul>
</div>