<div class="container">

<table style="width: 100%;"><tr>
<td>plot.haplohh</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot the variants of a haplohh object</h2>

<h3>Description</h3>

<p>Plot the variants of a haplohh object. This method is intended
for visualization of very small data sets such as the examples
provided by the package.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'haplohh'
plot(
  x,
  mrk = NA,
  allele = NA,
  group_by_allele = FALSE,
  ignore.distances = FALSE,
  col = c("blue", "red", "violet", "orange"),
  linecol = "gray",
  mrk.col = "gray",
  pch = 19,
  cex = 1,
  lwd = 1,
  hap.names = NULL,
  mrk.names = NULL,
  cex.lab.hap = 0.8,
  cex.lab.mrk = 0.8,
  family.lab = "",
  offset.lab.hap = 0.5,
  offset.lab.mrk = 0.25,
  pos.lab.hap = "left",
  pos.lab.mrk = "top",
  srt.hap = 0,
  srt.mrk = 0,
  highlight.mrk = NULL,
  highlight.mrk.col = c("lightgray", "black", "darkgray"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class <code>haplo-hh</code> generated by <code>data2haplohh</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrk</code></td>
<td>
<p>the focal marker. Used only, if alleles are grouped or (de-)selected.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele</code></td>
<td>
<p>if <code>NA</code> (default), haplotypes of all alleles are plotted,
otherwise for the specified alleles. Alleles must be specified by their
internal coding, i.e. '0' for ancestral resp. major allele, etc.
Haplotypes with missing values at the focal marker can only be selected in
combination with genotyped alleles, e.g. as <code>c(1,NA)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group_by_allele</code></td>
<td>
<p>logical. If <code>TRUE</code>, group chromosomes
by their allele at the focal marker; alleles are ordered by their internal coding
unless parameter <code>alleles</code> is specified. If <code>FALSE</code> (default), haplotypes are drawn by
their order in the input file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ignore.distances</code></td>
<td>
<p>logical. If <code>TRUE</code>, markers are drawn equally-spaced.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>color for each allele (as coded internally).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>linecol</code></td>
<td>
<p>the color of the background lines. If more than one color is specified and sequences
sorted by the marker allele, the specified colors are used to distinguish the alleles; otherwise
consecutive sequences are set into the specified colors.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrk.col</code></td>
<td>
<p>color of the vertical line at the focal marker position.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pch</code></td>
<td>
<p>symbol used for markers. See <code>points</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex</code></td>
<td>
<p>relative size of marker symbol. See <code>points</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lwd</code></td>
<td>
<p>line width.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hap.names</code></td>
<td>
<p>a vector containing the names of chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrk.names</code></td>
<td>
<p>a vector containing the names of markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.lab.hap</code></td>
<td>
<p>relative letter size of haplotype labels. See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.lab.mrk</code></td>
<td>
<p>relative letter size of marker labels. See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>family.lab</code></td>
<td>
<p>font family for labels. See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>offset.lab.hap</code></td>
<td>
<p>offset of haplotype labels. See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>offset.lab.mrk</code></td>
<td>
<p>offset of marker labels. See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos.lab.hap</code></td>
<td>
<p>position of haplotype labels. Either <code>"left"</code> (default),
<code>"right"</code>, <code>"none"</code> or <code>"both"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos.lab.mrk</code></td>
<td>
<p>position of marker labels. Either <code>"top"</code> (default) or <code>"none"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>srt.hap</code></td>
<td>
<p>rotation of haplotype labels (see <code>par</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>srt.mrk</code></td>
<td>
<p>rotation of marker labels (see <code>par</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>highlight.mrk</code></td>
<td>
<p>vector of markers to be highlighted</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>highlight.mrk.col</code></td>
<td>
<p>color for each allele (as coded internally) at highlighted markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>other parameters to be passed to <code>plot.default</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Specifying a haplohh-object with more than 4096 haplotypes or
markers produces an error.
</p>


<h3>See Also</h3>

<p><code>calc_haplen</code>, <code>plot.furcation</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#example haplohh object
make.example.files()
hh &lt;- data2haplohh(hap_file = "example1.hap",
                   map_file = "example1.map",
                   allele_coding = "01")
plot(hh)
hh &lt;- data2haplohh(hap_file = "example2.hap",
                   map_file = "example2.map",
                   allele_coding = "01",
                   min_perc_geno.mrk = 50)
plot(hh)
remove.example.files()
</code></pre>


</div>