<div class="container">

<table style="width: 100%;"><tr>
<td>freqbinplot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot of unstandardized iHS within frequency bins</h2>

<h3>Description</h3>

<p>Plot of unstandardized iHS within frequency bins.
</p>


<h3>Usage</h3>

<pre><code class="language-R">freqbinplot(
  x,
  spectrum = FALSE,
  main = NA,
  xlab = "Derived allele frequency",
  ylab = NA,
  xlim = c(0, 1),
  ylim = NULL,
  pch = 20,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>data (output of function <code>ihh2ihs</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>spectrum</code></td>
<td>
<p>logical. If <code>TRUE</code>, plot frequency spectrum instead of iHS.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main</code></td>
<td>
<p>an overall title for the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>a title for the x axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>a title for the y axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlim</code></td>
<td>
<p>the x coordinate range of the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylim</code></td>
<td>
<p>the y coordinate range of the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pch</code></td>
<td>
<p>plotting 'character' see <code>points</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments to be passed to plot resp. points.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The plot shows the mean and the quantiles calculated by
function <code>ihh2ihs</code> for the unstandardized iHS in each frequency bin.
Note that the standardization of iHS is performed bin-wise in order
to reduce the frequency-dependence of
iHS values (expected under neutrality).
An implicit assumption of this procedure is that each bin is dominated
by neutral markers.
</p>


<h3>See Also</h3>

<p><code>ihh2ihs</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(rehh.data)
data(wgscan.cgu)
#results from a genome scan (44,057 SNPs)
#see ?wgscan.eut and ?wgscan.cgu for details
wgscan.cgu.ihs &lt;- ihh2ihs(wgscan.cgu)
freqbinplot(wgscan.cgu.ihs)
</code></pre>


</div>