<div class="container">

<table style="width: 100%;"><tr>
<td>plot.ehh</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot EHH around a focal marker</h2>

<h3>Description</h3>

<p>Plot curve of EHH values around a focal marker.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'ehh'
plot(
  x,
  ylim = c(0, 1),
  type = "l",
  main = paste0("EHH around '", x$mrk.name, "'"),
  xlab = "Position",
  ylab = "Extended Haplotype Homozygosity",
  col = c("blue", "red", "violet", "orange"),
  mrk.col = "gray",
  bty = "n",
  legend = NA,
  legend.xy.coords = "automatic",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>data (output of <code>calc_ehh</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylim</code></td>
<td>
<p>the y limits of the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>plot type (see codeplot.default and <code>matplot</code>). 
Type <code>"s"</code> or <code>"S"</code> yield both (the same) piecewise constant curve.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main</code></td>
<td>
<p>title for the plot (default <code>NA</code>, i.e. none).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>title for the x-axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>title for the y-axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>color for the ancestral and derived alleles (respectively) curves.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrk.col</code></td>
<td>
<p>color of the vertical line at the focal marker position.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bty</code></td>
<td>
<p>box type around plot (see <code>par</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend</code></td>
<td>
<p>legend text.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend.xy.coords</code></td>
<td>
<p>if <code>"automatic"</code> (default) places legend either top left or top right;
if <code>"none"</code>, no legend is drawn; otherwise the argument is passed to <code>legend</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments to be passed to function <code>matplot</code>.</p>
</td>
</tr>
</table>
<h3>See Also</h3>

<p><code>data2haplohh</code>, <code>calc_ehh</code>, <code>plot.ehhs</code>, <code>scan_hh</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#example haplohh object (280 haplotypes, 1424 SNPs)
#see ?haplohh_cgu_bta12 for details
data(haplohh_cgu_bta12)
#computing EHH statistics for the marker "F1205400"
#which displays a strong signal of selection
ehh &lt;- calc_ehh(haplohh_cgu_bta12, mrk = "F1205400")
plot(ehh)
</code></pre>


</div>