<div class="container">

<table style="width: 100%;"><tr>
<td>scan_hh</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute iHH, iES and inES over a whole chromosome</h2>

<h3>Description</h3>

<p>Compute integrated EHH (iHH), integrated EHHS (iES) and integrated normalized EHHS (inES)
for all markers of a chromosome (or linkage group).
</p>


<h3>Usage</h3>

<pre><code class="language-R">scan_hh(
  haplohh,
  limhaplo = 2,
  limhomohaplo = 2,
  limehh = 0.05,
  limehhs = 0.05,
  phased = TRUE,
  polarized = TRUE,
  scalegap = NA,
  maxgap = NA,
  discard_integration_at_border = TRUE,
  lower_ehh_y_bound = limehh,
  lower_ehhs_y_bound = limehhs,
  interpolate = TRUE,
  threads = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>haplohh</code></td>
<td>
<p>an object of class <code>haplohh</code> (see <code>data2haplohh</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>limhaplo</code></td>
<td>
<p>if there are less than <code>limhaplo</code> chromosomes that can be used for
the calculation of EHH(S), the calculation is stopped. The option is intended for the case of missing data,
which leads to the successive exclusion of haplotypes: the further away from the focal marker
the less haplotypes contribute to EHH(S).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>limhomohaplo</code></td>
<td>
<p>if there are less than <code>limhomohaplo</code> homozygous chromosomes, the
calculation is stopped. This option is intended for unphased data and should be invoked only
if relatively low frequency variants are not filtered subsequently (see main vignette and Klassmann et al. 2020).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>limehh</code></td>
<td>
<p>limit at which EHH stops to be evaluated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>limehhs</code></td>
<td>
<p>limit at which EHHS stops to be evaluated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phased</code></td>
<td>
<p>logical. If <code>TRUE</code> (default) chromosomes are expected to be phased. If <code>FALSE</code>, the haplotype data is assumed to
consist of pairwise ordered chromosomes belonging to diploid individuals.
EHH(S) is then estimated over individuals which are homozygous at the focal marker.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>polarized</code></td>
<td>
<p>logical. <code>TRUE</code> by default. If <code>FALSE</code>, use major and minor allele instead of ancestral and derived. If there
are more than two alleles then the minor allele refers to the second-most frequent allele.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scalegap</code></td>
<td>
<p>scale or cap gaps larger than the specified size to the specified size (default=<code>NA</code>, i.e. no scaling).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxgap</code></td>
<td>
<p>maximum allowed gap in bp between two markers. If exceeded, further calculation of EHH(S) is stopped at the gap
(default=<code>NA</code>, i.e no limitation).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>discard_integration_at_border</code></td>
<td>
<p>logical. If <code>TRUE</code> (default) and computation reaches first or last marker or a gap larger than <code>maxgap</code>,
iHH, iES and inES are set to <code>NA</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lower_ehh_y_bound</code></td>
<td>
<p>lower y boundary of the area to be integrated over (default: <code>limehh</code>). Can be set
to zero for compatibility with the program hapbin.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lower_ehhs_y_bound</code></td>
<td>
<p>lower y boundary of the area to be integrated (default: <code>limehhs</code>). Can be set
to zero for compatibility with the program hapbin.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>interpolate</code></td>
<td>
<p>logical. If <code>TRUE</code> (default), integration
is performed over a continuous EHH(S) curve (values are interpolated linearly between consecutive markers),
otherwise the EHH(S) curve decreases stepwise at markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threads</code></td>
<td>
<p>number of threads to parallelize computation</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Integrated EHH (iHH), integrated EHHS (iES) and integrated normalized EHHS (inES)
are computed for all markers of the chromosome (or linkage group). This function is several
times faster as a procedure calling in turn <code>calc_ehh</code> and <code>calc_ehhs</code>
for all markers. To perform a whole genome-scan this function needs
to be called for each chromosome and results concatenated.
</p>
<p>Note that setting <code>limehh</code> or <code>limehhs</code> to zero is likely to reduce power,
since even under neutrality a tiny fraction (&lt;&lt;0.05) of extremely long shared haplotypes is expected
which, if fully accounted for, would obfuscate the signal at selected sites.
</p>


<h3>Value</h3>

<p>The returned value is a dataframe with markers in rows and the following columns
</p>

<ol>
<li>
<p> chromosome name
</p>
</li>
<li>
<p> position in the chromosome
</p>
</li>
<li>
<p> sample frequency of the ancestral / major allele
</p>
</li>
<li>
<p> sample frequency of the second-most frequent remaining allele
</p>
</li>
<li>
<p> number of evaluated haplotypes at the focal marker for the ancestral / major allele
</p>
</li>
<li>
<p> number of evaluated haplotypes at the focal marker for the second-most frequent remaining allele
</p>
</li>
<li>
<p> iHH of the ancestral / major allele
</p>
</li>
<li>
<p> iHH of the second-most frequent remaining allele
</p>
</li>
<li>
<p> iES (used by Sabeti et al 2007)
</p>
</li>
<li>
<p> inES (used by Tang et al 2007)</p>
</li>
</ol>
<p>Note that in case of unphased data the evaluation is restricted to
haplotypes of homozygous individuals which reduces the power
to detect selection, particularly for iHS (for appropriate parameter setting
see the main vignette and Klassmann et al (2020)).
</p>


<h3>References</h3>

<p>Gautier, M. and Naves, M. (2011). Footprints of selection in the ancestral admixture of a New World Creole cattle breed. <em>Molecular Ecology</em>, <strong>20</strong>, 3128-3143.
</p>
<p>Klassmann, A. and Gautier, M. (2020). Detecting selection using Extended Haplotype
Homozygosity-based statistics on unphased or unpolarized data (preprint). 
https://doi.org/10.22541/au.160405572.29972398/v1
</p>
<p>Sabeti, P.C. et al. (2002). Detecting recent positive selection in the human genome from haplotype structure. <em>Nature</em>, <strong>419</strong>, 832-837.
</p>
<p>Sabeti, P.C. et al. (2007). Genome-wide detection and characterization of positive selection in human populations. <em>Nature</em>, <strong>449</strong>, 913-918.
</p>
<p>Tang, K. and Thornton, K.R. and Stoneking, M. (2007). A New Approach for Using Genome Scans to Detect Recent Positive Selection in the Human Genome. <em>Plos Biology</em>, <strong>7</strong>, e171.
</p>
<p>Voight, B.F. and Kudaravalli, S. and Wen, X. and Pritchard, J.K. (2006). A map of recent positive selection in the human genome. <em>Plos Biology</em>, <strong>4</strong>, e72.
</p>


<h3>See Also</h3>

<p><code>data2haplohh</code>, <code>calc_ehh</code>, <code>calc_ehhs</code>
<code>ihh2ihs</code>,<code>ines2rsb</code>, <code>ies2xpehh</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#example haplohh object (280 haplotypes, 1424 SNPs)
#see ?haplohh_cgu_bta12 for details
data(haplohh_cgu_bta12)
scan &lt;- scan_hh(haplohh_cgu_bta12)
</code></pre>


</div>