<div class="container">

<table style="width: 100%;"><tr>
<td>make.example.files</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Copy example input files into current working directory</h2>

<h3>Description</h3>

<p>This function copies the following example files to the current working directory:
</p>

<ul>
<li> <p><code>example1.hap</code> "example 1" haplotype file in "standard format"
</p>
</li>
<li> <p><code>example1.map</code> "example 1" marker information file
</p>
</li>
<li> <p><code>example1.vcf</code> "example 1" as vcf file
</p>
</li>
<li> <p><code>example2.hap</code> "example 2" haplotype file in "standard format"
</p>
</li>
<li> <p><code>example2.map</code> "example 2" marker information file
</p>
</li>
<li> <p><code>example2.vcf</code> "example 2" as vcf file
</p>
</li>
<li> <p><code>example_neutral.vcf</code> "example neutral evolution" as vcf file
</p>
</li>
<li> <p><code>example_sweep.vcf</code> "example for a selective sweep (without recombination)"
</p>
</li>
<li> <p><code>example_sweep_with_recombination.vcf</code> "example for a selective sweep with recombination
</p>
</li>
<li> <p><code>ms.out output</code> from a small simulation by the program 'ms'
</p>
</li>
<li> <p><code>bta12_cgu.hap</code> an haplotype file in "standard format"
</p>
</li>
<li> <p><code>bta12_cgu.thap</code> an haplotype file in "transposed format"
</p>
</li>
<li> <p><code>bta12_hapguess_switch.out</code> an haplotype file in fastphase output format
</p>
</li>
<li> <p><code>map.inp</code> a marker information file for all bta_cgu markers
</p>
</li>
</ul>
<p>Example 1 was used in (Gautier 2017) to explain the various EHH derived statistics calculated by this package.
Example 2 is an extension containing multi-allelic markers and missing values.
</p>
<p>Examples for neutral data and sweeps are discussed in a supplement of Klassmann (2020).
</p>
<p>The bta12 files contain data for 280 haplotypes, originating from 140 individuals belonging to the
Creole cattle breed from Guadeloupe, at 1.424 markers mapping to bovine chromosome 12 (BTA12) (Gautier 2011).
</p>


<h3>Usage</h3>

<pre><code class="language-R">make.example.files()
</code></pre>


<h3>References</h3>

<p>Gautier, M. and Naves, M. (2011). Footprints of selection in the ancestral admixture of a New World Creole cattle breed. <em>Molecular Ecology</em>, <strong>20</strong>, 3128-3143.
</p>
<p>Gautier, M., Klassmann,  A. and Vitalis, R. (2017). rehh 2.0: a reimplementation of the R package rehh to detect positive selection from haplotype structure. <em>Molecular Ecology Resources</em>, <strong>17</strong>, 78-90.
</p>
<p>Klassmann, A. and Gautier, M. (2020). Detecting selection using Extended Haplotype
Homozygosity-based statistics on unphased or unpolarized data (preprint). 
https://doi.org/10.22541/au.160405572.29972398/v1
</p>


<h3>See Also</h3>

<p><code>data2haplohh</code>, <code>remove.example.files</code>
</p>


</div>