<div class="container">

<table style="width: 100%;"><tr>
<td>rehh.data-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Data Only: Searching for Footprints of Selection using Haplotype Homozygosity Based Tests
Description: Contains example data for the 'rehh' package.
</h2>

<h3>Description</h3>

<p>Contains example data for the package rehh.
</p>


<h3>Details</h3>


<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> rehh.data</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 1.0.0</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;"> GPL(&gt;=2)</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p>Index:
</p>
<pre>
wgscan.cgu              Whole genome scan results for the CGU
                        (Creole from Guadeloupe island)
wgscan.eut              Whole genome scan results for a pool of
                        European taurine cattle
</pre>


<h3>References</h3>

<p>Gautier M., Klassmann A., and Vitalis R. (2016). rehh: An R package to detect footprints of selection in genome-wide SNP data from haplotype structure.
<em>Molecular Ecology Resources</em>, submitted
</p>
<p>Gautier M. and Vitalis R. (2012). rehh: An R package to detect footprints of 
selection in genome-wide SNP data from haplotype structure. <em>Bioinformatics</em>, <b>28</b>(8), 1176–1177.
</p>
<p>Gautier, M. and Naves, M. (2011). Footprints of selection in the ancestral admixture of a New World
Creole cattle breed. <em>Molecular Ecology</em>, <b>20</b>, 3128–3143.
</p>


</div>