<div class="container">

<table style="width: 100%;"><tr>
<td>furcation-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>An S4 class representing the complete furcation pattern around a focal marker.</h2>

<h3>Description</h3>

<p>An S4 class representing the complete furcation pattern around a focal marker.
</p>


<h3>Slots</h3>


<dl>
<dt><code>.Data</code></dt>
<dd>
<p>a list containing for each allele an object of <code>allelefurcation-class</code>.</p>
</dd>
<dt><code>mrk.name</code></dt>
<dd>
<p>the name/identifier of the focal marker.</p>
</dd>
<dt><code>position</code></dt>
<dd>
<p>the chromosomal position of the focal marker.</p>
</dd>
<dt><code>xlim</code></dt>
<dd>
<p>the range of marker positions.</p>
</dd>
<dt><code>nhap</code></dt>
<dd>
<p>the number of haplotypes in the sample.</p>
</dd>
</dl>
<h3>See Also</h3>

<p><code>calc_furcation</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># copy example files into working directory
make.example.files()
# read first example file
hh &lt;- data2haplohh(hap_file = "example1.hap", map_file = "example1.map", allele_coding = "01")
# remove example files
remove.example.files()
# calculate furcation structure around marker "rs6"
f &lt;- calc_furcation(hh, mrk = "rs6")
# extract left side tree of ancestral allele (which is coded by '0')
f[['0']]@left
# the tree consists of seven nodes, '1' being the root node
# nodes 2 and 3 have the root node as parent, etc.
# the first chromosome is attached as a label node to node 7, etc.
# For comparison, a plot of the complete furcation structure:
plot(f)
</code></pre>


</div>