<div class="container">

<table style="width: 100%;"><tr>
<td>plot.furcation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plots furcation trees around a focal marker</h2>

<h3>Description</h3>

<p>Plots furcation trees around a focal marker
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'furcation'
plot(
  x,
  allele = NA,
  col = c("blue", "red", "violet", "orange"),
  mrk.col = "gray",
  lwd = 0.1,
  hap.names = NULL,
  cex.lab = 1,
  family.lab = "",
  offset.lab = 0.5,
  legend = NA,
  legend.xy.coords = "automatic",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class furcation (see <code>calc_furcation</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele</code></td>
<td>
<p>If <code>NA</code> (default), furcation trees for all alleles of the focal marker are plotted,
otherwise for the specified alleles. Alleles must be specified by their
internal coding, i.e. '0' for ancestral resp. major allele, etc.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>color for each allele (as coded internally).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrk.col</code></td>
<td>
<p>color of the vertical line at the focal marker position.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lwd</code></td>
<td>
<p>controls the relative width of the diagram lines on the plot (default 0.1).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hap.names</code></td>
<td>
<p>a vector containing names of chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.lab</code></td>
<td>
<p>relative size of labels. See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>family.lab</code></td>
<td>
<p>font family for labels. See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>offset.lab</code></td>
<td>
<p>offset of labels. See <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend</code></td>
<td>
<p>legend text.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend.xy.coords</code></td>
<td>
<p>if <code>"automatic"</code> (default) places legend either top left or top right;
if <code>"none"</code>, no legend is drawn; otherwise argument is passed to <code>legend</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>other arguments to be passed to <code>plot.default</code>.</p>
</td>
</tr>
</table>
<h3>See Also</h3>

<p><code>plot.haplen</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#example haplohh object (280 haplotypes, 1424 SNPs)
#see ?haplohh_cgu_bta12 for details
data(haplohh_cgu_bta12)
#plotting furcation diagram for both ancestral and derived allele
#from the marker "F1205400"
#which display a strong signal of selection
f &lt;- calc_furcation(haplohh_cgu_bta12, mrk = "F1205400")
plot(f)
plot(f, xlim = c(2e+07,3.5e+07))
plot(f, xlim = c(2.7e+07,3.1e+07))
plot(f, xlim = c(2.7e+07,3.1e+07), hap.names = hap.names(haplohh_cgu_bta12), cex.lab=0.3)
</code></pre>


</div>