<div class="container">

<table style="width: 100%;"><tr>
<td>subset.haplohh</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Subsets object of <code>haplohh-class</code>
</h2>

<h3>Description</h3>

<p>Subsets the data of an object of class <code>haplohh-class</code>,
meeting certain conditions.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'haplohh'
subset(
  x,
  select.hap = NULL,
  select.mrk = NULL,
  min_perc_geno.hap = NA,
  min_perc_geno.mrk = 100,
  min_maf = NA,
  max_alleles = NA,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>object of class <code>haplohh-class</code> to be subset.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>select.hap</code></td>
<td>
<p>expression, indicating haplotypes to select.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>select.mrk</code></td>
<td>
<p>expression, indicating markers to select.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_perc_geno.hap</code></td>
<td>
<p>threshold on percentage of missing data for haplotypes
(haplotypes with less than <code>min_perc_geno.hap</code> percent of markers genotyped are discarded). Default is <code>NA</code>,
hence no constraint.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_perc_geno.mrk</code></td>
<td>
<p>threshold on percentage of missing data for markers (markers genotyped on less than
<code>min_perc_geno.mrk</code> percent of haplotypes are discarded). By default, <code>min_perc_geno.mrk=100</code>,
hence only fully genotyped markers are retained.
This value cannot be set to <code>NA</code> or zero.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_maf</code></td>
<td>
<p>threshold on the Minor Allele Frequency. Markers having a MAF lower than or equal to minmaf are discarded.
In case of multi-allelic markers the second-most frequent allele is referred to as minor allele.
Setting this value to zero eliminates monomorphic sites. Default is <code>NA</code>,
hence no constraint.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_alleles</code></td>
<td>
<p>threshold for the maximum number of different alleles at a site. Default is <code>NA</code>,
hence no restriction. In order to retain only bi-allelic markers, set this parameter to 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical. If <code>TRUE</code> (default), report verbose progress.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments are ignored.</p>
</td>
</tr>
</table>
<h3>See Also</h3>

<p><code>haplohh-class</code>, <code>data2haplohh</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#example haplohh object (280 haplotypes, 1424 SNPs)
#see ?haplohh_cgu_bta12 for details
data(haplohh_cgu_bta12)
#select subset of first 10 hyplotypes and first 5 markers
subset(haplohh_cgu_bta12, select.hap = 1:10, select.mrk = 1:5)
</code></pre>


</div>