<div class="container">

<table style="width: 100%;"><tr>
<td>ihh2ihs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute iHS</h2>

<h3>Description</h3>

<p>Compute iHS (standardized ratio of iHH values of two alleles).
</p>


<h3>Usage</h3>

<pre><code class="language-R">ihh2ihs(
  scan,
  freqbin = 0.025,
  min_maf = 0.05,
  min_nhaplo = NA,
  standardize = TRUE,
  include_freq = FALSE,
  right = FALSE,
  alpha = 0.05,
  p.side = NA,
  p.adjust.method = "none",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>scan</code></td>
<td>
<p>a data frame with chromosome name,
marker position, frequency of ancestral (resp. major) allele, frequency of derived (resp. minor)
allele, and iHH for both alleles, as obtained from function <code>scan_hh</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freqbin</code></td>
<td>
<p>size of the bins to standardize log(iHH_A/iHH_D). Markers are binned with
respect to the derived allele frequency at the focal marker. The bins are built from
<code>min_maf</code> to <code>1-min_maf</code> in steps of size <code>freqbin</code>. If set to 0, standardization
is performed considering each observed frequency as a discrete frequency
class (useful in case of a large number of markers and few different haplotypes).
If set to an integer of 1 or greater, a corresponding number of equally sized bins are created.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_maf</code></td>
<td>
<p>focal markers with a MAF (Minor Allele Frequency) lower than or equal to <code>min_maf</code>
are discarded from the analysis (default 0.05).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_nhaplo</code></td>
<td>
<p>focal markers with least one of the two compared alleles carried by fewer
than <code>min_nhaplo</code> haplotypes, are discarded (default <code>NA</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>standardize</code></td>
<td>
<p>logical. If <code>TRUE</code> (default), then standardize iHS, else report unstandardized iHS.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>include_freq</code></td>
<td>
<p>logical. If <code>TRUE</code> include columns with allele frequencies into result.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>right</code></td>
<td>
<p>logical. If <code>TRUE</code> the bin intervals are closed on the right (and open on the left).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>calculate quantiles <code>alpha/2</code> and <code>(1-alpha/2)</code> for unstandardized binned iHS.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.side</code></td>
<td>
<p>side to which refers the p-value. Default <code>NA</code>, meaning two-sided. Can be set
to <code>"left"</code> or <code>"right"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.adjust.method</code></td>
<td>
<p>method passed to function <code>p.adjust</code> to correct the p-value for
multiple testing. Default <code>"none"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical. If <code>TRUE</code> (default), report number of markers of the source data frame and result data frame.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Computes log ratio of iHH of two focal alleles as described in Voight et al. (2006). The standardization
is performed within each bins separately because of the frequency-dependence
of expected iHS values under neutrality. An implicit assumption of this
approach is that each bin is dominated by neutral markers.
</p>
<p>Since the standardized iHS values follow, if markers evolve predominantly neutrally, approximately
a standard Gaussian distribution, it is practical to assign to the values a p-value relative
to the null-hypothesis of neutral evolution. The parameter <code>p.side</code> determines
if the p-value is assigned to both sides of the distribution or to one side of interest.
</p>


<h3>Value</h3>

<p>The returned value is a list containing two elements
</p>

<dl>
<dt>ihs</dt>
<dd>
<p>a data frame with markers in rows and the columns for chromosome name, marker position,
iHS and, if standardized, p-value in a negative log10 scale. Optionally, allele frequencies are included.</p>
</dd>
<dt>frequency.class</dt>
<dd>
<p>a data frame with bins in rows and columns for
the number of markers, mean uniHS, standard deviation uniHS, lower quantile uniHS, upper quantile uniHS.</p>
</dd>
</dl>
<h3>References</h3>

<p>Gautier, M. and Naves, M. (2011). Footprints of selection in the ancestral admixture of a New World Creole cattle breed. <em>Molecular Ecology</em>, <strong>20</strong>, 3128-3143.
</p>
<p>Voight, B.F. and Kudaravalli, S. and Wen, X. and Pritchard, J.K. (2006). A map of recent positive selection in the human genome. <em>Plos Biology</em>, <strong>4</strong>, e72.
</p>


<h3>See Also</h3>

<p><code>scan_hh</code>, <code>distribplot</code>, <code>freqbinplot</code>, <code>manhattanplot</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(rehh.data)
data(wgscan.cgu)
#results from a genome scan (44,057 SNPs)
#see ?wgscan.eut and ?wgscan.cgu for details
wgscan.cgu.ihs &lt;- ihh2ihs(wgscan.cgu)
</code></pre>


</div>