<div class="container">

<table style="width: 100%;"><tr>
<td>calc_pairwise_haplen</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate pairwise shared haplotype length between all chromosomes</h2>

<h3>Description</h3>

<p>Calculate pairwise shared haplotype length between all chromosomes at a focal marker.
</p>


<h3>Usage</h3>

<pre><code class="language-R">calc_pairwise_haplen(
  haplohh,
  mrk,
  phased = TRUE,
  maxgap = NA,
  max_extend = NA,
  side = "both"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>haplohh</code></td>
<td>
<p>an object of class <code>haplohh</code> (see <code>data2haplohh</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrk</code></td>
<td>
<p>integer representing the number of the focal marker within the haplohh object
or string representing its ID/name.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phased</code></td>
<td>
<p>logical. If <code>TRUE</code> (default) chromosomes are expected to be phased. If <code>FALSE</code>, the haplotype data is assumed to
consist of pairwise ordered chromosomes belonging to diploid individuals and only the two chromosomes of
each individual are compared.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxgap</code></td>
<td>
<p>maximum allowed gap in bp between two markers. If exceeded, further calculation is stopped at the gap
(default=<code>NA</code>, i.e. no limitation).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_extend</code></td>
<td>
<p>maximum distance in bp to extend shared haplotypes away from the focal marker.
(default <code>NA</code>, i.e. no limitation).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>side</code></td>
<td>
<p>side to consider, either "left" (positions lower than focal position), "right" (positions higher than focal position)
or "both" (default).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function computes the length of shared haplotypes (stretches of identical sequence) around
the focal marker.
</p>
<p>Note that the function <code>calc_haplen</code> calculates for each chromosome
the boundaries of its longest shared haplotype; separately upstream and downstream of
the focal marker.
</p>


<h3>Value</h3>

<p>The returned value is a matrix with pairwise shared haplotype lengths.
</p>


<h3>See Also</h3>

<p><code>data2haplohh</code>, <code>scan_hh_full</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#example haplohh object (280 haplotypes, 1424 SNPs)
#see ?haplohh_cgu_bta12 for details
data(haplohh_cgu_bta12)
#computing shared haplotype lengths around the marker "F1205400"
#which displays a strong signal of selection
m &lt;- calc_pairwise_haplen(haplohh_cgu_bta12, mrk = "F1205400")
</code></pre>


</div>