<div class="container">

<table style="width: 100%;"><tr>
<td>as.newick</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert a furcation tree into Newick format</h2>

<h3>Description</h3>

<p>Convert a furcation tree into Newick format.
</p>


<h3>Usage</h3>

<pre><code class="language-R">as.newick(furcation, allele = 0, side, hap.names = seq_len(furcation@nhap))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>furcation</code></td>
<td>
<p>an object of <code>furcation-class</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele</code></td>
<td>
<p>the allele to be considered (default 0).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>side</code></td>
<td>
<p>side (either <code>"left"</code> or <code>"right"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hap.names</code></td>
<td>
<p>names/labels of chromosomes in haplotype data file.
Per default haplotypes are numbered by their order in the input file.</p>
</td>
</tr>
</table>
<h3>See Also</h3>

<p><code>ftree-class</code>, <code>calc_furcation</code>,
<code>plot.furcation</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#example haplohh object (280 haplotypes, 1424 SNPs)
#see ?haplohh_cgu_bta12 for details
data(haplohh_cgu_bta12)
#calculate furcation for the marker "F1205400"
#which displays a strong signal of selection
f &lt;- calc_furcation(haplohh_cgu_bta12, mrk = "F1205400")
#get left tree of ancestral allele (coded as '0')
as.newick(f, 0, "left")
</code></pre>


</div>