<div class="container">

<table style="width: 100%;"><tr>
<td>manhattanplot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Manhattan plot of iHS, XP-EHH or Rsb over a genome.</h2>

<h3>Description</h3>

<p>Manhattanplot of iHS, XP-EHH or Rsb over a genome.
</p>


<h3>Usage</h3>

<pre><code class="language-R">manhattanplot(
  data,
  pval = FALSE,
  threshold = c(-2, 2),
  chr.name = NA,
  cr = NULL,
  cr.col = "gray",
  cr.opacity = 0.5,
  cr.lab.cex = 0.6,
  cr.lab.offset = 0,
  cr.lab.pos = "top",
  mrk = NULL,
  mrk.cex = 1,
  mrk.col = "gray",
  mrk.pch = 1,
  mrk.lab.cex = 0.4,
  mrk.lab.pos = 4,
  ignore_sign = FALSE,
  cex = 0.5,
  las = 1,
  pch = 20,
  inset = 5e+06,
  resolution = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>output of either <code>ihh2ihs</code>, <code>ies2xpehh</code> or <code>ines2rsb</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pval</code></td>
<td>
<p>logical. If <code>TRUE</code>, the p-value is plotted, otherwise the score itself.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>a horizontal line is added at the corresponding value(s),
for instance to represent a significance threshold.
A single value (upper or lower threshold) or two values (upper and lower) can be specified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr.name</code></td>
<td>
<p>if <code>NA</code> (default), all chromosomes are plotted, otherwise only those specified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cr</code></td>
<td>
<p>highlight "candidate regions" specified by a data.frame with columns <code>CHR</code>, <code>START</code>
and <code>END</code> as obtained by the function <code>calc_candidate_regions</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cr.col</code></td>
<td>
<p>the color for highlighting</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cr.opacity</code></td>
<td>
<p>a value between 0 (invisible) and 1 (opaque).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cr.lab.cex</code></td>
<td>
<p>text size of candidate region labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cr.lab.offset</code></td>
<td>
<p>offset of candidate region labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cr.lab.pos</code></td>
<td>
<p>if <code>"top"</code> (default) or <code>"bottom"</code>, candidate regions are labeled by numbers; to turn off, use <code>"none"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrk</code></td>
<td>
<p>highlight marker specified by a data.frame containing the
colums <code>CHR</code> and <code>POSITION</code>. The row names of that data frame are taken as labels.
Alternatively a vector with marker IDs can be specified. In the latter case the ID is used as label.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrk.cex</code></td>
<td>
<p>size of marker label.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrk.col</code></td>
<td>
<p>color of the highlighted points.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrk.pch</code></td>
<td>
<p>type of the highlighted points.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrk.lab.cex</code></td>
<td>
<p>text size of marker label. If zero, no labels are printed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrk.lab.pos</code></td>
<td>
<p>a position specifier for the text.
Values of 1, 2, 3 and 4, respectively indicate positions below, to the left of, above and to the right of the highlighted marker.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ignore_sign</code></td>
<td>
<p>logical. If <code>TRUE</code>, absolute values are plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex</code></td>
<td>
<p>size of the points representing markers in the plot(s) (see <code>par</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>las</code></td>
<td>
<p>orientation of axis labels (see <code>par</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pch</code></td>
<td>
<p>type of the points representing markers in the plot(s) (see <code>points</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inset</code></td>
<td>
<p>inset (in bases) between chromosomes to avoid overlap of data points. Default: 5,000,000 bases.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resolution</code></td>
<td>
<p>Rasterize data points to the specified resolution and remove
duplicate points. Defaults to NULL, i.e. no rasterization. A typical value might be <code>c(1E5, 0.01)</code>,
meaning that resolution on the x-axis (chromosomal position) is 100000 and on the y-axis (score or p-value) is 0.01.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments to be passed to <code>plot.default</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The color of chromosomes is taken from the "Graphics Palette", see <code>palette</code>.
</p>
<p>If a single chromosome is plotted, a genomic region can be specified by
argument <code>xlim</code>.
</p>
<p>Other statistics can be plotted as well, although a warning is issued. They must be given by a data.frame
with columns CHR and POSITION and the statistic in the third column.
</p>


<h3>Value</h3>

<p>The function returns a plot.
</p>


<h3>See Also</h3>

<p><code>ihh2ihs</code>, <code>ies2xpehh</code>, <code>ines2rsb</code>, <code>calc_candidate_regions</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(rehh.data)
data(wgscan.cgu)
## results from a genome scan (44,057 SNPs)
## see ?wgscan.eut and ?wgscan.cgu for details
wgscan.ihs &lt;- ihh2ihs(wgscan.cgu)
manhattanplot(wgscan.ihs)
</code></pre>


</div>