<div class="container">

<table style="width: 100%;"><tr>
<td>wgscan.eut</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Whole genome scan results for a pool of European taurine cattle</h2>

<h3>Description</h3>

<p>A dataframe object with 44,057 rows (SNPs) and 7 columns: i) chromosome name (CHR), ii) position of the SNP in bp (POSITION), iii) Ancestral allele frequency (freq_A), iv) iHH for the ancestral allele (iHH_A), v) iHH for the derived allele (iHH_D), vi) iES using the estimator by Tang et al. (2007) (iES_Tang_et_al_2007), vii) iES using the estimator by Sabeti et al. (2007) (iES_Sabeti_et_al_2007).
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(wgscan.eut)</code></pre>


<h3>References</h3>

<p>Gautier, M. and Naves, M. (2011). Footprints of selection in the ancestral admixture of a New World
Creole cattle breed. <em>Molecular Ecology</em>, <b>20</b>, 3128â€“3143.
</p>


</div>