<div class="container">

<table style="width: 100%;"><tr>
<td>RelCoefLine</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Relatedness Coefficients Estimation for Lines

</h2>

<h3>Description</h3>

<p>This function performs Maximum Likelihood estimation for the relatedness coefficients between lines based on a bi-allelic genotype matrix.

</p>


<h3>Usage</h3>

<pre><code class="language-R">RelCoefLine(LineGenom = matrix(0,nrow=0,ncol=0), 
	    Freq = matrix(0,nrow=0,ncol=0), 
	    LinePop = rep(0,0), 
	    Combination = NULL, 
	    NbInit = 5, Prec = 10^(-4), NbCores = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>LineGenom</code></td>
<td>
<p>Genotype matrix of lines. Each line is described by 1 column. Each row corresponds to a marker. Entries of matrix Genotype should be either 0 or 1.

</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Freq</code></td>
<td>
<p>Allelic frequencies for allele 1 at each markers and for all populations (one column per population, one line per marker).

</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LinePop</code></td>
<td>
<p>A vector of numbers corresponding to population membership for the parental lines.

</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Combination</code></td>
<td>
<p>If provided, a list of vector with two components. The jth vector is composed with the number of the first hybrid and the number of the second hybrid of the jth couple to study.

</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>NbInit</code></td>
<td>
<p>Number of initial values for the EM algorithm.

</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Prec</code></td>
<td>
<p>Convergence precision parameter for the EM algorithm.

</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>NbCores</code></td>
<td>
<p>Number of cores used by the algorithm (Default is the number of cores available minus one). Only available for linux and Mac.

</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>By default, relatedness coefficients are displayed for all couple of genotyped lines. In that case the function returns a matrix corresponding to the Simple Relatedness Coefficient, i.e. the probability that each couple of lines are related. Element (i,j) of the matrix corresponds to the estimated relatedness coefficient for the couple of lines i and j. Alternatively, if a list of couples is specified with argument Combination, the function returns a list of coefficients (each coefficient corresponding to an relatedness coefficient). In that case element i of the list corresponds to the estimated relatedness coefficient of the ith couple specified in Combination.





</p>


<h3>Author(s)</h3>

<p>Fabien Laporte, 'UMR Genetique Quantitative et Evolution' INRA France.

</p>


<h3>Examples</h3>

<pre><code class="language-R">require('Relatedness')
data(Genotype)
data(Frequencies)
data(Cross)
SimpleRelatedness &lt;- RelCoefLine(LineGenom=Genotype,Freq=Frequencies,
LinePop=rep(1,8),NbCores=2)
print(SimpleRelatedness)
</code></pre>


</div>