<div class="container">

<table style="width: 100%;"><tr>
<td>16S</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Count Data for 16S rRNA Sequences</h2>

<h3>Description</h3>

<p>This data set contains count data for 16S ribosomal RNA (rRNA) sequences for
the two phylogenetic classes Alphaproteobacteria and Mollicutes. 
The counts for 30
sequences for each class were obtained by counting the occurrence of triplets
of nucleotides in windows of length 100 without any overlap. To
separate sequences a row of dummy count of <code>NA</code> is used.</p>


<h3>Usage</h3>

<pre><code class="language-R">data("16S")</code></pre>


<h3>Format</h3>

<p><code>Alphaproteobacteria16S</code> and <code>Mollicutes16S</code> are matrices
with about 449 rows and 64 (number of possible triplets) columns. 
</p>


<h3>Source</h3>

<p>The raw sequence information was obtained from the 
National center for biotechnology information (NCBI) website at
http://www.ncbi.nih.gov/
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("16S")

emm &lt;- EMM("Kullback", threshold=0.1)
build(emm, Mollicutes16S+1)

## start state for sequences have an initial state probability &gt;0
it &lt;- initial_transition(emm)
it[it&gt;0]
</code></pre>


</div>