<div class="container">

<table style="width: 100%;"><tr>
<td>MFAseq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Replication profile for wild type DS2 H.volcanii</h2>

<h3>Description</h3>

<p>Replication profile for H.volcanii wild isolate DS2.
Genomic DNA for deep sequencing was isolated from 100 ml culture
in stationary phase (A650 &gt; 1, DS2_stat sample) or 1 litre in
exponential phase (A650 0.1, DS2_exp sample). Unique
reads for the two samples were calculated in 1 kb genomic bins
using ASM2568v1 genome assembly. The ratio was created by
dividing 'score' values from replicating sample by non-
replicating sample 'score' values, adjusted by total number
of reads. The ratio values were further adjusted by multiplying
them by 1.12 to put the values onto biologically relevant
relative copy number scale from 1 to 2.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(MFAseq)
</code></pre>


<h3>Format</h3>

<p>data frame with 3887 rows and 7 variables:
</p>

<dl>
<dt>chrom</dt>
<dd>
<p>short chromosome name</p>
</dd>
<dt>chromStart</dt>
<dd>
<p>left chromosome coordinate</p>
</dd>
<dt>chromEnd</dt>
<dd>
<p>right chromosome coordinate</p>
</dd>
<dt>name.rep</dt>
<dd>
<p>replicating sample name</p>
</dd>
<dt>name.nonRep</dt>
<dd>
<p>non-replicating sample name</p>
</dd>
<dt>ratio</dt>
<dd>
<p>ratio value in the current bin</p>
</dd>
<dt>ratioFactor</dt>
<dd>
<p>adjustment factor used for the current ratio</p>
</dd>
</dl>
<h3>Source</h3>

<p>DS2_exp exponential phase sample: <a href="https://www.ncbi.nlm.nih.gov/sra/SRX202169">SRA</a>;
DS2_stat stationary sample: <a href="https://www.ncbi.nlm.nih.gov/sra/SRX202170">SRA</a>
</p>


<h3>References</h3>

<p>Hawkins et al. (2013) Nature 503(7477):544-547
(<a href="https://pubmed.ncbi.nlm.nih.gov/24185008">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(MFAseq)
</code></pre>


</div>