<div class="container">

<table style="width: 100%;"><tr>
<td>syncSeq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Replication profiles budding yeast arrest-release samples</h2>

<h3>Description</h3>

<p>Replication profiles of wild type S. cerevisiae arrest-release samples
(AUY077 strain). The cells were arrested in G1 with alpha-factor
followed by release using pronase. The samples were collected
before the release (aFactor) and various time intervals after
the release (25min,30min,35min,40min,45min,50min and 90min)
Extracted DNA was sequenced and mapped to sacCer3 genome. Unique
reads for replicating (post-release) and non-replicating (aFactor)
samples were calculated in 1 kb genomic bins. The ratios were
created by dividing 'score' values from replicating samples by
non-replicating sample 'score' values, adjusted by total number
of reads. The ratio values were further adjusted based on bulk
genome replication (as determined by flow cytometry), to put the
values onto biologically relevant relative copy number scale from
1 to 2. The relative copy number values were smoothed using cubic
spline.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(syncSeq)
</code></pre>


<h3>Format</h3>

<p>List containing two data frames
</p>

<dl>
<dt>data</dt>
<dd>
<p>syncSeq replication profiles data. Columns:
<strong>chrom</strong> (short chromosome name),
<strong>chromStart</strong> (left chromosome coordinate),
<strong>chromEnd</strong> (right chromosome coordinate),
<strong>name.rep</strong> (replicating sample name),
<strong>name.nonRep</strong> (non-replicating sample name),
<strong>ratio</strong> (ratio value in the current bin),
<strong>ratioFactor</strong> (adjustment factor used for the current ratio),
<strong>group</strong> (Group number of the current bin),
<strong>splineSmooth</strong> (Smoothed ratio value)
</p>
</dd>
<dt>guide</dt>
<dd>
<p>Guide dataframe for plotting the syncSeq data
<strong>order</strong> (Order to plot data in),
<strong>name.rep</strong> (Name of replicating sample),
<strong>name.nonRep</strong> (Name of non-replicating sample),
<strong>raw</strong> (Should raw data be plotted?),
<strong>smooth</strong> (Should smooth data be plotted?),
<strong>color</strong> (Color to plot the profile in)
</p>
</dd>
</dl>
<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE48212">GEO</a>
</p>


<h3>References</h3>

<p>MÃ¼ller et al. (2014) NAR 42(1):e3
(<a href="https://pubmed.ncbi.nlm.nih.gov/24089142">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(syncSeq)
</code></pre>


</div>