<div class="container">

<table style="width: 100%;"><tr>
<td>W303</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Sequence read coverage ratios for wild type S.cerevisiae W303</h2>

<h3>Description</h3>

<p>Sequence read coverage ratios for wild type sample
(T7107 strain). The cells were stained with DNA dye and sorted
based on DNA content into S or G2/M phase fractions. Extracted
DNA was sequenced and mapped to sacCer3 genome. Unique reads
for replicating (S) and non-replicating (G2/M) samples were
calculated in 1 kb genomic bins. The ratio was created by
dividing 'score' values from replicating sample by non-
replicating sample 'score' values, adjusted by total number
of reads.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(W303)
</code></pre>


<h3>Format</h3>

<p>data frame with 11350 rows and 7 variables:
</p>

<dl>
<dt>chrom</dt>
<dd>
<p>short chromosome name</p>
</dd>
<dt>chromStart</dt>
<dd>
<p>left chromosome coordinate</p>
</dd>
<dt>chromEnd</dt>
<dd>
<p>right chromosome coordinate</p>
</dd>
<dt>name.rep</dt>
<dd>
<p>replicating sample name</p>
</dd>
<dt>name.nonRep</dt>
<dd>
<p>non-replicating sample name</p>
</dd>
<dt>ratio</dt>
<dd>
<p>ratio value in the current bin</p>
</dd>
<dt>ratioFactor</dt>
<dd>
<p>adjustment factor used for the current ratio</p>
</dd>
</dl>
<h3>Source</h3>

<p>S phase sample: <a href="https://www.ncbi.nlm.nih.gov/sra/SRX204358">SRA</a>;
G2 sample: <a href="https://www.ncbi.nlm.nih.gov/sra/SRX204357">SRA</a>
</p>


<h3>References</h3>

<p>Natsume et al. (2013) Mol Cell 50(5):661-74
(<a href="https://pubmed.ncbi.nlm.nih.gov/23746350">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(W303)
</code></pre>


</div>