<div class="container">

<table style="width: 100%;"><tr>
<td>get_reporter_score</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate reporter score</h2>

<h3>Description</h3>

<p>Calculate reporter score
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_reporter_score(
  ko_stat,
  type = c("pathway", "module")[1],
  feature = "ko",
  threads = 1,
  modulelist = NULL,
  perm = 4999,
  verbose = TRUE,
  p.adjust.method2 = "BH",
  min_exist_KO = 3,
  max_exist_KO = 600
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ko_stat</code></td>
<td>
<p>ko_stat result from <code>pvalue2zs</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>'pathway' or 'module' for default KOlist for microbiome, 'CC', 'MF', 'BP', 'ALL' for default GOlist for homo sapiens. And org in listed in 'https://www.genome.jp/kegg/catalog/org_list.html' such as 'hsa' (if your kodf is come from a specific organism, you should specify type here).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>feature</code></td>
<td>
<p>one of 'ko', 'gene', 'compound'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threads</code></td>
<td>
<p>default 1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>modulelist</code></td>
<td>
<p>NULL or customized modulelist dataframe, must contain 'id','K_num','KOs','Description' columns. Take the 'KOlist' as example, use <code>custom_modulelist</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perm</code></td>
<td>
<p>permutation number, default: 4999.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.adjust.method2</code></td>
<td>
<p>p.adjust.method for the correction of ReporterScore, see <code>p.adjust</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_exist_KO</code></td>
<td>
<p>min exist KO number in a pathway (default, 3, when a pathway contains KOs less than 3, there will be no RS)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_exist_KO</code></td>
<td>
<p>max exist KO number in a pathway (default, 600, when a pathway contains KOs more than 600, there will be no RS)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>reporter_res data.frame
</p>


<h3>See Also</h3>

<p>Other GRSA: 
<code>combine_rs_res()</code>,
<code>ko.test()</code>,
<code>pvalue2zs()</code>,
<code>reporter_score()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data("KO_abundance_test")
ko_pvalue &lt;- ko.test(KO_abundance, "Group", metadata)
ko_stat &lt;- pvalue2zs(ko_pvalue, mode = "directed")
reporter_s1 &lt;- get_reporter_score(ko_stat, perm = 499)

</code></pre>


</div>