<div class="container">

<table style="width: 100%;"><tr>
<td>reporter_score</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>One step to get the reporter score of your KO abundance table.</h2>

<h3>Description</h3>

<p>One step to get the reporter score of your KO abundance table.
</p>


<h3>Usage</h3>

<pre><code class="language-R">reporter_score(
  kodf,
  group,
  metadata = NULL,
  method = "wilcox.test",
  pattern = NULL,
  p.adjust.method1 = "none",
  mode = c("directed", "mixed")[1],
  verbose = TRUE,
  feature = "ko",
  type = c("pathway", "module")[1],
  p.adjust.method2 = "BH",
  modulelist = NULL,
  threads = 1,
  perm = 4999,
  min_exist_KO = 3,
  max_exist_KO = 600
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>kodf</code></td>
<td>
<p>KO_abundance table, rowname are feature ids (e.g. K00001 if feature="ko"; PEX11A if feature="gene"; C00024 if feature="compound"), colnames are samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group</code></td>
<td>
<p>The comparison groups (at least two categories) in your data, one column name of metadata when metadata exist or a vector whose length equal to columns number of kodf. And you can use factor levels to change order.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>metadata</code></td>
<td>
<p>sample information data.frame contains group</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>the type of test. Default is 'wilcox.test'. Allowed values include:
</p>

<ul>
<li> <p><code>t.test</code> (parametric) and <code>wilcox.test</code> (non-parametric). Perform comparison between two groups of samples. If the grouping variable contains more than two levels, then a pairwise comparison is performed.
</p>
</li>
<li> <p><code>anova</code> (parametric) and <code>kruskal.test</code> (non-parametric). Perform one-way ANOVA test comparing multiple groups.
</p>
</li>
<li>
<p> 'pearson', 'kendall', or 'spearman' (correlation), see <code>cor</code>.</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pattern</code></td>
<td>
<p>a named vector matching the group, e.g. c('G1'=1,'G2'=3,'G3'=2), use the correlation analysis with specific pattern to calculate p-value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.adjust.method1</code></td>
<td>
<p>p.adjust.method for 'ko.test', see <code>p.adjust</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mode</code></td>
<td>
<p>'mixed' or 'directed' (default, only for two groups differential analysis or multi-groups correlation analysis.), see details in <code>pvalue2zs</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>feature</code></td>
<td>
<p>one of 'ko', 'gene', 'compound'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>'pathway' or 'module' for default KOlist for microbiome, 'CC', 'MF', 'BP', 'ALL' for default GOlist for homo sapiens. And org in listed in 'https://www.genome.jp/kegg/catalog/org_list.html' such as 'hsa' (if your kodf is come from a specific organism, you should specify type here).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.adjust.method2</code></td>
<td>
<p>p.adjust.method for the correction of ReporterScore, see <code>p.adjust</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>modulelist</code></td>
<td>
<p>NULL or customized modulelist dataframe, must contain 'id','K_num','KOs','Description' columns. Take the 'KOlist' as example, use <code>custom_modulelist</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threads</code></td>
<td>
<p>default 1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perm</code></td>
<td>
<p>permutation number, default: 4999.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_exist_KO</code></td>
<td>
<p>min exist KO number in a pathway (default, 3, when a pathway contains KOs less than 3, there will be no RS)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_exist_KO</code></td>
<td>
<p>max exist KO number in a pathway (default, 600, when a pathway contains KOs more than 600, there will be no RS)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>reporter_score object:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>kodf</code></td>
<td>
<p>your input KO_abundance table</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ko_stat</code></td>
<td>
<p>ko statistics result contains p.value and z_score</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reporter_s</code></td>
<td>
<p>the reporter score in each pathway</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>modulelist</code></td>
<td>
<p>default KOlist or customized modulelist dataframe</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group</code></td>
<td>
<p>The comparison groups in your data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>metadata</code></td>
<td>
<p>sample information dataframe contains group</p>
</td>
</tr>
</table>
<p>for the 'reporter_s' in result, whose columns represent:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>ID</code></td>
<td>
<p>pathway id</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Description</code></td>
<td>
<p>pathway description</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K_num</code></td>
<td>
<p>total number of KOs/genes in the pathway</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Exist_K_num</code></td>
<td>
<p>number of KOs/genes in your inputdata that exist in the pathway</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Significant_K_num</code></td>
<td>
<p>number of kos/genes in your inputdata that are significant in the pathway</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Z_score</code></td>
<td>
<p><code class="reqn">Z_{pathway}=\frac{1}{\sqrt{k}}\sum Z_{koi}</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BG_Mean</code></td>
<td>
<p>Background mean, <code class="reqn">\mu _k</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BG_Sd</code></td>
<td>
<p>Background standard deviation, <code class="reqn">\sigma _k</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ReporterScore</code></td>
<td>
<p>ReporterScore of the pathway, <code class="reqn">ReporterScore=(Z_{pathway}-\mu _k)/\sigma _k</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>
<p>p.value of the ReporterScore</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.adjust</code></td>
<td>
<p>adjusted p.value by p.adjust.method2</p>
</td>
</tr>
</table>
<h3>See Also</h3>

<p>Other GRSA: 
<code>combine_rs_res()</code>,
<code>get_reporter_score()</code>,
<code>ko.test()</code>,
<code>pvalue2zs()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">message("The following example require some time to run:")

data("KO_abundance_test")
reporter_score_res &lt;- reporter_score(KO_abundance, "Group", metadata,
  mode = "directed", perm = 499
)
head(reporter_score_res$reporter_s)
reporter_score_res2 &lt;- reporter_score(KO_abundance, "Group2", metadata,
  mode = "mixed",
  method = "kruskal.test", p.adjust.method1 = "none", perm = 499
)
reporter_score_res3 &lt;- reporter_score(KO_abundance, "Group2", metadata,
  mode = "directed",
  method = "pearson", pattern = c("G1" = 1, "G2" = 3, "G3" = 2), perm = 499
)

</code></pre>


</div>