<div class="container">

<table style="width: 100%;"><tr>
<td>reconstructGsMN</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reconstuction of the specific-organism genome-scale metabolic network</h2>

<h3>Description</h3>

<p>Reconstruction of genome-scale metabolic network (GsMN) whose 
nodes represents compounds and whose edges represents reactions.
</p>


<h3>Usage</h3>

<pre><code class="language-R">reconstructGsMN(metabolic.data, RefData = RefDbcache, threshold = 10,
  is.gaint = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>metabolic.data, </code></td>
<td>
<p>df or a character vector. More details see function 
<code>getOrgMetabolicData</code> and <code>details</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>RefData</code></td>
<td>
<p>The reference metabolic data. It does not need reference data 
While organism metabolic data was collected from KEGG database, and RefData 
is set to NULL. Otherwise, RefDbCache, an internal dataset in this package,
was taken as the Reference metabolic data for Genome scale metabolic reconstruction.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>numeric, Nodes belonging to components with fewer than the 
value of threshold nodes will be ignored. This is a good option for networks
that contain many small and trivial components. Default is 10.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is.gaint</code></td>
<td>
<p>logical, Ignore all nodes except those in the giant component:
selecting the only main largest component (connected set of nodes) of the
network. All smaller components will be ignored. This is a good option for
networks with a dominant component. Default is TRUE.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The input of this function can be of two forms. If organims is 
collected in KEGG database, it can be obtained with 
<code>getOrgMetabolicData</code> which is a data frame. Otherwise, 
<code>metabolic.data</code> could be a  character vecotr which contains the KEGG 
Orthology annotated information on this organism, e.g. we can download this 
KO annotation profile in the <a href="https://img.jgi.doe.gov">https://img.jgi.doe.gov</a> website for 
species detected in a human microbime which not contained in KEGG organism 
database. Several functions, such as <code>link{read.table}</code> and 
<code>read.delim</code> could help us to read KO annotation profile.
</p>


<h3>Value</h3>

<p>igraph object
</p>


<h3>See Also</h3>

<p><code>getOrgMetabolicData</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## not run (organism in KEGG)
## metabolic.data &lt;- getOrgMetabolicData("buc")
## g &lt;- reconstructGsMN(metabolic.data)

## species detected in a human microbiome
annodir &lt;- system.file("extdata","koanno.tab",package = "RevEcoR")
metabolic.data &lt;- read.delim2(file=annodir,stringsAsFactors=FALSE)
##load the reference metabolic data
data(RefDbcache)
g2 &lt;- reconstructGsMN(metabolic.data, RefData = RefDbcache)

</code></pre>


</div>