<div class="container">

<table style="width: 100%;"><tr>
<td>read.blast</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Read BLAST data
</h2>

<h3>Description</h3>

<p>Function to read BLAST data generated with standalone BLAST from NCBI.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read.blast(file, sep = "\t")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p> character: BLAST file to read in.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>
<p> the field separator character. Values on each line of the file are 
separated by this character. Default <code>"\t"</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function reads data which was generated with standalone BLAST from NCBI;
see ftp://ftp.ncbi.nih.gov/blast/executables/release/.
</p>
<p>Possible steps:<br>
1) Install NCBI BLAST<br>
2) Generate and import database(s)<br>
3) Apply BLAST with options <code>outfmt</code> and <code>out</code>; e.g.<br><code>blastn -query Testquery -db Testdatabase -outfmt 6 -out out.txt</code><br>
or<br><code>blastn -query Testquery -db Testdatabase -outfmt 10 -out out.csv</code><br>
One can also call BLAST from inside R by using function <code>system</code><br>
system("blastn -query Testquery -db Testdatabase -outfmt 6 -out out.txt")<br>
4) Read in the results<br><code>test.res &lt;- read.blast(file = "out.txt")</code><br>
or<br><code>test.res &lt;- read.blast(file = "out.csv", sep = ",")</code>
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> with variables
</p>

<dl>
<dt><code>query.id</code></dt>
<dd>
<p>character: sequence identifier. </p>
</dd>
<dt><code>subject.id</code></dt>
<dd>
<p>character: subject identifier. </p>
</dd>
<dt><code>identity</code></dt>
<dd>
<p>numeric: identity between sequences (in percent). </p>
</dd>
<dt><code>alignment.length</code></dt>
<dd>
<p>integer: number of nucleotides. </p>
</dd>
<dt><code>mismatches</code></dt>
<dd>
<p>integer: number of mismatches. </p>
</dd>
<dt><code>gap.opens</code></dt>
<dd>
<p>integer: number of gaps. </p>
</dd>
<dt><code>q.start</code></dt>
<dd>
<p>integer: query sequence start. </p>
</dd>
<dt><code>q.end</code></dt>
<dd>
<p>integer: query sequence end. </p>
</dd>
<dt><code>s.start</code></dt>
<dd>
<p>integer: subject sequence start. </p>
</dd>
<dt><code>s.end</code></dt>
<dd>
<p>integer: subject sequence end. </p>
</dd>
<dt><code>evalue</code></dt>
<dd>
<p>numeric: evalue. </p>
</dd>
<dt><code>bit.score</code></dt>
<dd>
<p>numeric: score value. </p>
</dd>
</dl>
<h3>Author(s)</h3>

 
<p>Fabienne Flessa <a href="mailto:Fabienne.Flessa@uni-bayreuth.de">Fabienne.Flessa@uni-bayreuth.de</a>,<br>
Alexandra Kehl <a href="mailto:Alexandra.Kehl@uni-tuebingen.de">Alexandra.Kehl@uni-tuebingen.de</a>,<br>
Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>References</h3>

<p>Standalone Blast download: <a href="https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/">https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/</a>
</p>
<p>Blast News: <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastNews">https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastNews</a>
</p>
<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>See Also</h3>

 <p><code>BLASTdata</code>, <code>simMatrix</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">Dir &lt;- system.file("extdata", package = "RFLPtools") # input directory 
filename &lt;- file.path(Dir, "BLASTexample.txt")
BLAST1 &lt;- read.blast(file = filename)
str(BLAST1)
</code></pre>


</div>