<div class="container">

<table style="width: 100%;"><tr>
<td>track.fasta-constructors</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Track constructors for FASTA files
</h2>

<h3>Description</h3>

<p>Produces <code>sliceable</code>-inheriting objects to query "in situ" FASTA files.
</p>
<p><code>track.fasta.multi</code> is designed to handle a single multi-FASTA file aggregating all the chromosomes of an organism. An index as generated by the HTSlib (formerly "SAMtools") <code>faidx</code> command is required.
</p>
<p><code>track.fasta.collection</code> is designed to handle a collection of standard FASTA files, one per chromosome.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  track.fasta.multi(fastaFile, indexFile, .name, .organism, .assembly,
    .parameters, warn = TRUE)
  track.fasta.collection(files, chromosomes, .name, .organism, .assembly,
    .parameters, warn = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>fastaFile</code></td>
<td>

<p>Single character value, the file name and path to a multi-FASTA file (.fa) to build a track upon.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>indexFile</code></td>
<td>

<p>Single character value, the file name and path to a multi-FASTA index file (.fai) corresponding to <code>fastaFile</code>. This index file is supposd to be produced by the <code>faidx</code> command from SAMtools.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.name</code></td>
<td>

<p>Single character value, to fill the <code>name</code> field inherited from <code>drawable</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.organism</code></td>
<td>

<p>Single character value, to fill the <code>organism</code> field of the object.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.assembly</code></td>
<td>

<p>Single character value, to fill the <code>assembly</code> field of the object.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.parameters</code></td>
<td>

<p>A <code>list</code> of drawing parameters, to fill the <code>parameters</code> field of the object.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>warn</code></td>
<td>

<p>Single logical value, to be passed to the appropriate <code>check</code> method. Other conversion warnings rely on this value too in <code>track.table</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>files</code></td>
<td>

<p>Character vector, file names and paths to multiple single-FASTA file (.fa) to build a track upon (one per chromosome).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosomes</code></td>
<td>

<p>Character vector, chromosomes names to attribute to each file in <code>files</code>.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Both functions suppose the FASTA files to respect the following :
</p>

<ul>
<li>
<p>They begin with a single line of comment, after a '&gt;' sign.
</p>
</li>
<li>
<p>All sequence lines have the same length (whatever it is).
</p>
</li>
<li>
<p>The line separator (\n, \r\n) is always the same in a file.
</p>
</li>
<li>
<p>No empty line.
</p>
</li>
</ul>
<p>Returned sequences may be wrong (without error or notice !) if any of these is not fullfilled. Standard sources (see below) usually enforce these conditions.
</p>


<h3>Value</h3>

<p>An object of class <code>track.fasta</code>.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>References</h3>

<p>Example of FASTA file collection (human assembly 'hg19'), from UCSC : <a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/chromFa.tar.gz">http://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/chromFa.tar.gz</a>
</p>
<p>Example of single multi-FASTA file (human assembly 'hg19'), from 1000 genomes : <a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/human_g1k_v37.fasta.gz">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/human_g1k_v37.fasta.gz</a>
</p>
<p>The <a href="http://www.htslib.org/doc/faidx.html">faidx</a> documentation, from the HTSlib project.
</p>


<h3>See Also</h3>

<p><code>track-constructors</code>, <code>Annotation</code>
</p>
<p><code>tk.browse</code>, <code>browsePlot</code>
</p>


</div>