<div class="container">

<table style="width: 100%;"><tr>
<td>track-constructors</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Track constructors
</h2>

<h3>Description</h3>

<p>Produces <code>track.table</code>-inheriting objects.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  track.table(..., .name, .parameters, .organism, .assembly, .chromosomes,
    .makeNames = FALSE, .orderCols = TRUE, warn = TRUE)
  track.bam(bamPath, baiPath, addChr, quiet = FALSE, .name, .organism,
    .assembly, .parameters, warn = TRUE)
  track.genes(...)
  track.bands(...)
  track.exons(...)
  track.CNV(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Arguments to be passed to the inherited constructor (<code>refTable</code> for <code>track.table</code>, <code>track.table</code> for the others). <code>track.table</code> has few more restrictions, see <code>track.table-class</code> for further details.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.name</code></td>
<td>

<p>Single character value, to fill the <code>name</code> field inherited from <code>drawable</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>warn</code></td>
<td>

<p>Single logical value, to be passed to the appropriate <code>check</code> method. Other conversion warnings rely on this value too in <code>track.table</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.parameters</code></td>
<td>

<p>A <code>list</code> of drawing parameters, to fill the <code>parameters</code> field of the object.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.organism</code></td>
<td>

<p>Single character value, to fill the <code>organism</code> field of the object.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.assembly</code></td>
<td>

<p>Single character value, to fill the <code>assembly</code> field of the object.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.chromosomes</code></td>
<td>

<p>Single character value, levels to use for the 'chrom' column if conversion to factor is needed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.makeNames</code></td>
<td>

<p>Single logical value, whether to compute the 'name' column with unique values or not. If <code>TRUE</code>, any existing 'name' column will be replaced.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.orderCols</code></td>
<td>

<p>Single logical value, whether to reorder the columns for more consistency between tracks or not.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bamPath</code></td>
<td>

<p>Single character value, the file name and path to a BAM file (.bam) to build a track around.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>baiPath</code></td>
<td>

<p>Single character value, the file name and path to the corresponding BAI file (.bai) to build a track around. If missing, a guess will be tried (adding '.bai' to <code>bamPath</code> or remplacing '.bam' by '.bai').
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addChr</code></td>
<td>

<p>Single logical value, whether to automatically add 'chr' ahead chromosome names when querying or not. If missing, a guess will be tried looking for chromosome names beginning by 'chr' in the BAM header declaration.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>

<p>Single logical value, whether to throw diagnostic messages during BAI parsing or not.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p><code>track.table</code> and <code>track.bam</code> inheriting objects.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code>track.table-class</code>, <code>track.bam-class</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">  # track.table from a data.frame
  df &lt;- data.frame(
    chrom=1, strand="+", start=1:5, end=2:6, name=letters[1:5],
    stringsAsFactors=FALSE
  )
  track.table(df)
  
  # track.table from vectors
  track.table(chrom=1, strand="+", start=1:5, end=2:6, name=letters[1:5])
  
  # track.bam
  track.bam(system.file("extdata/ATM.bam", package="Rgb"))
</code></pre>


</div>