<div class="container">

<table style="width: 100%;"><tr>
<td>draw.pileup</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Pileup plot of a BAM track
</h2>

<h3>Description</h3>

<p>This function draws a slice of a sequence pileup, highlighting polymorphisms.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  draw.pileup(slice, start, end, ylim = NA, bty = "o", label = TRUE, labelCex = 0.75,
    bases = c(A = "#44CC44", C = "#4444CC", G = "#FFCC00", T = "#CC4444"),
    maxRange = 500, cex.lab = 1, alphaOrder = 3, alphaMin = 0.1, fg = "#000000", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>slice</code></td>
<td>

<p>An integer matrix of read counts, with nucleotides in rows and positions in columns. Both dimensions must be named.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start</code></td>
<td>

<p>Single integer value, the left boundary of the window, in base pairs.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>end</code></td>
<td>

<p>Single integer value, the right boundary of the window, in base pairs.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylim</code></td>
<td>

<p>See <code>plot.default</code>. <code>NA</code> will compute a dynamic ylim fiting the data.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bty</code></td>
<td>

<p>A character string which determined the type of box which is drawn about plots. If bty is one of "o" (the default), "l", "7", "c", "u", or "]" the resulting box resembles the corresponding upper case letter. A value of "n" suppresses the box. See <code>par</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>

<p>Single logical value, whether to print nucleotide on bars or not.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labelCex</code></td>
<td>

<p>Single numeric value, character expansion factor for labels.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bases</code></td>
<td>

<p>Named character vector, defining the color to use for each nucleotide.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxRange</code></td>
<td>

<p>Single integer value, nothing will be ploted if the plot window is wider by this value (in bases).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.lab</code></td>
<td>

<p>The relative character size of x and y axis labels (default: 1). See <code>par</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alphaOrder</code></td>
<td>

<p>Single numeric value, the order of the formula controlling the transparency. Increase this value to increase sensitivity to rare variants.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alphaMin</code></td>
<td>

<p>Single numeric value, the minimal intensity in the formula controlling the transparency (between 0 and 1). Perfectly homozyguous positions will typically use this intensity of color.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fg</code></td>
<td>

<p>Single character value, defining the color of the foreground (axes, labels...) as an english name or a hexadecimal code. Similar to <code>par</code>'s argument but not relying on it.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Further arguments to be passed to <code>draw.bg</code>.
</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code>track.bam-class</code>
</p>
<p><code>draw.bg</code>, <code>draw.boxes</code>, <code>draw.density</code>, <code>draw.hist</code>, <code>draw.points</code>, <code>draw.seq</code>, <code>draw.steps</code>
</p>


</div>