<div class="container">

<table style="width: 100%;"><tr>
<td>Annotation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Annotation track constructors
</h2>

<h3>Description</h3>

<p>These functions constructs <code>track.table</code> inheriting objects from free annotation files.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  track.table.GTF(file, name = NA, attr = "split", features = "exon", quiet = FALSE, 
    .chromosomes, ...)
  track.exons.CCDS(file, name = "CCDS exons", ...)
  track.exons.GENCODE(file, name = "GENCODE exons",
    extra = c("gene_id", "gene_name", "exon_id"), ...)
  track.CNV.DGV(file, name = "DGV CNV", ...)
  track.genes.NCBI(file, name = "NCBI genes", selection, ...)
  track.bands.UCSC(file, name = "UCSC bands", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>

<p>Single character value, the path to the raw file to parse. See the 'Details' section below.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>

<p>Single character value, the <code>name</code> field for the <code>track.table</code> object. For <code>track.table.GTF</code>, use <code>NA</code> to refer to the "source" column content (if it contains a unique value for all rows).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>attr</code></td>
<td>

<p>To be passed to <code>read.gtf</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>features</code></td>
<td>

<p>To be passed to <code>read.gtf</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>

<p>To be passed to <code>read.gtf</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.chromosomes</code></td>
<td>

<p>To be passed to the <code>track.table</code> constructor.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Further arguments are passed to the class constructor, as a result most of the handled arguments are <code>track.table</code> arguments. Consider notably <code>.organism</code> and <code>.assembly</code> for track annotation.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>extra</code></td>
<td>

<p>Character vector, names of optional columns to keep from the GENCODE GTF file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>selection</code></td>
<td>

<p>Character vector, filter to apply on the "group_label" column for NCBI genes. Raises an error with the possible values when missing.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>track.table.GTF</code> imports a "Gene Feature Transfert" file, as proposed by the UCSC Table Browser at <a href="http://www.genome.ucsc.edu/cgi-bin/hgTables">http://www.genome.ucsc.edu/cgi-bin/hgTables</a>) for a large amount of species. See the <code>read.gtf</code> manual for further details.
</p>
<p><code>track.exons.CCDS</code> contains various transcripts from the "Consensus Coding DNA Sequence" project, currently only available for mouse and human (see the NCBI data repository at <a href="https://ftp.ncbi.nlm.nih.gov/pub/CCDS/">https://ftp.ncbi.nlm.nih.gov/pub/CCDS/</a>, and look for a file named "CCDS_current.txt").
</p>
<p><code>track.exons.GENCODE</code> contains various transcripts from the GENCODE project, currently only available for mouse and human (see the dedicated website at <a href="https://www.gencodegenes.org/">https://www.gencodegenes.org/</a>). This function is intended to run on the GTF version of the data.
</p>
<p><code>track.CNV.DGV</code> parses constitutive copy number variations from the current version of the Database of Genomic Variants, downloadable from <a href="http://dgv.tcag.ca/dgv/app/downloads">http://dgv.tcag.ca/dgv/app/downloads</a> using "DGV Variants" links. The whole database is dedicated to the human specy only.
</p>
<p><code>track.genes.NCBI</code> parses the gene list from the MapView project of the NCBI, for one of many species available at <a href="https://ftp.ncbi.nih.gov/genomes/MapView/">https://ftp.ncbi.nih.gov/genomes/MapView/</a>. Select your specy of interest, then browse "sequence", "current" and "initial_release" (if the directories are available, they are not for certain species). Download the file named "seq_gene.md.gz". As many assemblies are included in the file, a first call to the function without "selection" is required, to list the available values. A second call with the appropriate assembly name will produce the desired track file.
</p>
<p><code>track.bands.UCSC</code> produces a track of cytogenetic banding, as made available by the UCSC for many species at <a href="http://hgdownload.cse.ucsc.edu/downloads.html">http://hgdownload.cse.ucsc.edu/downloads.html</a>. Select the specy and assembly version that suits your needs, and look for a file named "cytoBand.txt.gz" in the "Annotation database" section.
</p>


<h3>Value</h3>

<p>Return a <code>track.table</code>-inheriting object (of class <code>track.exons</code>, <code>track.CNV</code>, <code>track.genes</code> or <code>track.bands</code>).
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>References</h3>

<p>Example of <code>track.exons.CCDS</code> raw file (current human assembly) : <a href="https://ftp.ncbi.nlm.nih.gov/pub/CCDS/current_human/CCDS.current.txt">https://ftp.ncbi.nlm.nih.gov/pub/CCDS/current_human/CCDS.current.txt</a>
</p>
<p>Example of <code>track.exons.GENCODE</code> raw file (human assembly 'GRCh38') : <a href="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_29/gencode.v29.annotation.gtf.gz">ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_29/gencode.v29.annotation.gtf.gz</a>
</p>
<p>Example of <code>track.CNV.DGV</code> raw file (human assembly 'hg19') : <a href="http://dgv.tcag.ca/dgv/docs/GRCh37_hg19_variants_2013-05-31.txt">http://dgv.tcag.ca/dgv/docs/GRCh37_hg19_variants_2013-05-31.txt</a>
</p>
<p>Example of <code>track.genes.NCBI</code> raw file (human assembly 'GRCh37') : <a href="https://ftp.ncbi.nih.gov/genomes/MapView/Homo_sapiens/sequence/BUILD.37.3/initial_release/seq_gene.md.gz">https://ftp.ncbi.nih.gov/genomes/MapView/Homo_sapiens/sequence/BUILD.37.3/initial_release/seq_gene.md.gz</a>
</p>
<p>Example of <code>track.bands.UCSC</code> raw file (human assembly 'hg19') : <a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/cytoBand.txt.gz">http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/cytoBand.txt.gz</a>
</p>


<h3>See Also</h3>

<p><code>track.fasta-constructors</code>, <code>Annotation</code>
</p>
<p><code>track.table-class</code>, <code>track.exons-class</code>, <code>track.CNV-class</code>, <code>track.genes-class</code> or <code>track.bands-class</code>
</p>
<p><code>tk.browse</code>, <code>browsePlot</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">  # From the "How-to" vignette, section "Custom annotation tracks"
  file &lt;- system.file("extdata/Cosmic_ATM.gtf.gz", package="Rgb")
  tt &lt;- track.table.GTF(file)
</code></pre>


</div>