<div class="container">

<table style="width: 100%;"><tr>
<td>saf_from_gff</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Creates a SAF file from a GTF/GFF3 features for the given pairs of blocks and features</h2>

<h3>Description</h3>

<p>This function creates a SAF file from a GTF/GFF3 features for the given blocks
and features
</p>


<h3>Usage</h3>

<pre><code class="language-R">saf_from_gff(
  inFile,
  outFile,
  fileType = c("AUTO", "GFF3", "GTF"),
  forceOverwrite = FALSE,
  features = c("gene &gt; exon"),
  sep = "&gt;"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>inFile</code></td>
<td>
<p>Path to the input GFF file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outFile</code></td>
<td>
<p>Path to the output SAF file, if not provided the output path will be the input path with the suffix ".feature1-block1.feature2-block2(...).saf"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fileType</code></td>
<td>
<p>Version of the input file (GTF/GFF3). Default AUTO: determined from the file name.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>forceOverwrite</code></td>
<td>
<p>If output file exists, overwrite the existing file. (default FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>features</code></td>
<td>
<p>Vector of pairs of features/blocks, separated by '&gt;' (see sep argument). In the case of features without defined blocks, only the feature is needed (see example)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>
<p>Separator of each "feature" and "block" provided in the feature argument (default '&gt;')</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Path to the generated SAF file
</p>


<h3>Examples</h3>

<pre><code class="language-R">test_gff3&lt;-system.file("extdata", "AthSmall.gff3", package="Rgff")
## Default usage, extract gene features by exon blocks 
saf_from_gff(test_gff3)
## Define only feature without block to count reads within the whole genomic locus 
saf_from_gff(test_gff3, features=c("gene"))
## Define multiple features for counting readsoverlapping only in exonic regions 
saf_from_gff(test_gff3, features=c("gene &gt; exon", "ncRNA_gene &gt; exon"))
</code></pre>


</div>