<div class="container">

<table style="width: 100%;"><tr>
<td>saf_from_gff3</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Creates a SAF file from a GFF3 for the given pairs of blocks and features,
note that groupBy and block vectors must have the same length</h2>

<h3>Description</h3>

<p>This function creates a SAF file from a GFF3 for the given blocks
and features
</p>


<h3>Usage</h3>

<pre><code class="language-R">saf_from_gff3(
  gffFile,
  outFile,
  forceOverwrite = FALSE,
  features = c("gene &gt; exon"),
  sep = "&gt;"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gffFile</code></td>
<td>
<p>Path to the input GFF file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outFile</code></td>
<td>
<p>Path to the output SAF file,  if not provided the output path will be the input path (without extension) with the suffix ".groupBy1-block1.groupBy2-block2(...).saf"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>forceOverwrite</code></td>
<td>
<p>If output file exists, overwrite the existing file. (default FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>features</code></td>
<td>
<p>Vector of pairs of features, separated by '&gt;' (see sep argument), to be used respectively as "group by" and "block"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>
<p>Separator of each "group by" and "block" provided in the feature argument (default '&gt;')</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Path to the generated SAF file
</p>


</div>