<div class="container">

<table style="width: 100%;"><tr>
<td>check_gff</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Test consistency and order of a GFF file</h2>

<h3>Description</h3>

<p>This function tests the consistency and order of a GFF file.
</p>


<h3>Usage</h3>

<pre><code class="language-R">check_gff(inFile, fileType = c("AUTO", "GFF3", "GTF"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>inFile</code></td>
<td>
<p>Path to the input GFF file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fileType</code></td>
<td>
<p>Version of the input file (GTF/GFF3). Default AUTO: determined from the file name.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The following list indicates the code and description of the issues detected in GFF3 files
</p>

<dl>
<dt>NCOLUMNS_EXCEEDED</dt>
<dd>
<p>Input file contains lines with more than 9 fields</p>
</dd>
<dt>NCOLUMNS_INFERIOR</dt>
<dd>
<p>Input file contains lines with less than 9 fields</p>
</dd>
<dt>TOO_MANY_FEATURE_TYPES</dt>
<dd>
<p>Input file contains too many (more than 100) different feature types</p>
</dd>
<dt>NO_IDs</dt>
<dd>
<p>ID attribute not found in any feature</p>
</dd>
<dt>DUPLICATED_IDs</dt>
<dd>
<p>There are duplicated IDs</p>
</dd>
<dt>ID_IN_MULTIPLE_CHR</dt>
<dd>
<p>The same ID has been found in more than one chromosome</p>
</dd>
<dt>NO_PARENTs</dt>
<dd>
<p>Parent attribute not found in any feature</p>
</dd>
<dt>MISSING_PARENT_IDs</dt>
<dd>
<p>There are  missing Parent IDs</p>
</dd>
<dt>PARENT_IN DIFFERENT CHR</dt>
<dd>
<p>There are features whose Parent is located in a different chromosome</p>
</dd>
<dt>PARENT_DEFINED_BEFORE_ID</dt>
<dd>
<p>Feature ids referenced in Parent attribute before being defined as ID</p>
</dd>
<dt>NOT_GROUPED_BY_CHR</dt>
<dd>
<p>Features are not grouped by chromosome</p>
</dd>
<dt>NOT_SORTED_BY_COORDINATE</dt>
<dd>
<p>Features are not sorted by start coordinate</p>
</dd>
<dt>NOT_VALID_WARNING</dt>
<dd>
<p>File cannot be recognized as valid  GFF3. Parsing warnings.</p>
</dd>
<dt>NOT_VALID_ERROR</dt>
<dd>
<p>File cannot be recognized as valid GFF3. Parsing errors.</p>
</dd>
</dl>
<p>The following list indicates the code and description of the issues detected in GTF files
</p>

<dl>
<dt>NCOLUMNS_EXCEEDED</dt>
<dd>
<p>Input file contains lines with more than 9 fields</p>
</dd>
<dt>NCOLUMNS_INFERIOR</dt>
<dd>
<p>Input file contains lines with less than 9 fields</p>
</dd>
<dt>TOO_MANY_FEATURE_TYPES</dt>
<dd>
<p>Input file contains too many (more than 100) different feature types</p>
</dd>
<dt>NO_GENE_ID_ATTRIBUTE</dt>
<dd>
<p>gene_id attribute not found in any feature</p>
</dd>
<dt>MISSING_GENE_IDs</dt>
<dd>
<p>There are features without gene_id attribute</p>
</dd>
<dt>NO_GENE_FEATURES</dt>
<dd>
<p>Gene features are not included in this GTF file</p>
</dd>
<dt>DUPLICATED_GENE_IDs</dt>
<dd>
<p>There are duplicated gene_ids </p>
</dd>
<dt>GENE_ID_IN_MULTIPLE_CHR</dt>
<dd>
<p>The same gene_id has been found in more than one chromosome</p>
</dd>
<dt>NO_TRANSCRIPT_ID_ATTRIBUTE</dt>
<dd>
<p>transcript_id attribute not found in any feature There are no elements with transcript_id attribute</p>
</dd>
<dt>MISSING_TRANSCRIPT_IDs</dt>
<dd>
<p>There are  features without transcript_id attribute</p>
</dd>
<dt>NO_TRANSCRIPT_FEATURES</dt>
<dd>
<p>Transcript features are not included in this GTF file</p>
</dd>
<dt>DUPLICATED_TRANSCRIPT_IDs</dt>
<dd>
<p>There are  duplicated transcript_ids </p>
</dd>
<dt>TRANSCRIPT_ID_IN_MULTIPLE_CHR</dt>
<dd>
<p>The same transcript_id has been found in more than one chromosome</p>
</dd>
<dt>DUPLICATED_GENE_AND_TRANSCRIPT_IDs</dt>
<dd>
<p>Same id has been defined as gene_id and transcript_id</p>
</dd>
<dt>NOT_GROUPED_BY_CHR</dt>
<dd>
<p>Features are not grouped by chromosome</p>
</dd>
<dt>NOT_SORTED_BY_COORDINATE</dt>
<dd>
<p>Features are not sorted by start coordinate</p>
</dd>
<dt>NOT_VALID_WARNING</dt>
<dd>
<p>File cannot be recognized as valid GTF. Parsing warnings.</p>
</dd>
<dt>NOT_VALID_ERROR</dt>
<dd>
<p>File cannot be recognized as valid GTF. Parsing errors.</p>
</dd>
</dl>
<h3>Value</h3>

<p>A data frame of detected issues, including a short code name, a description and estimated severity each. In no issues are detected the function will return an empty data frame.
</p>


<h3>Examples</h3>

<pre><code class="language-R">test_gff3&lt;-system.file("extdata", "eden.gff3", package="Rgff")
check_gff(test_gff3)
</code></pre>


</div>