<div class="container">

<table style="width: 100%;"><tr>
<td>hierBAPS</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>hierBAPS</h2>

<h3>Description</h3>

<p>Runs the hierBAPS algorithm of Cheng et al. 2013
</p>


<h3>Usage</h3>

<pre><code class="language-R">hierBAPS(
  snp.matrix,
  max.depth = 2,
  n.pops = floor(nrow(snp.matrix)/5),
  quiet = FALSE,
  n.extra.rounds = 0,
  assignment.probs = FALSE,
  n.cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>snp.matrix</code></td>
<td>
<p>Character matrix of aligned sequences produced by load_fasta.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.depth</code></td>
<td>
<p>Maximum depth of hierarchical search (default = 2).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.pops</code></td>
<td>
<p>Maximum number of populations in the data (default = number of isolates/5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>Whether to suppress progress information (default=FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.extra.rounds</code></td>
<td>
<p>The number of additional rounds to perform after the default hierBAPS
settings (default=0). If set to Inf it will run until a local optimum is reached
(this might take a long time).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assignment.probs</code></td>
<td>
<p>whether or not to calculate the assignment probabilities to each cluster (default=FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.cores</code></td>
<td>
<p>The number of cores to use.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list containing a dataframe indicating an assignment of each sequence
to hierarchical clusters as well as the log marginal likelihoods for each level.
</p>


<h3>Author(s)</h3>

<p>Gerry Tonkin-Hill
</p>


<h3>References</h3>

<p>Cheng, Lu, Thomas R. Connor, Jukka Sirén, David M. Aanensen, and Jukka Corander. 2013. “Hierarchical and Spatially Explicit Clustering of DNA Sequences with BAPS Software.” Molecular Biology and Evolution 30 (5): 1224–28.
</p>


<h3>Examples</h3>

<pre><code class="language-R">snp.matrix &lt;- load_fasta(system.file("extdata", "small_seqs.fa", package = "rhierbaps"))
hb &lt;- hierBAPS(snp.matrix, max.depth=2, n.pops=20, quiet=FALSE)


snp.matrix &lt;- load_fasta(system.file("extdata", "seqs.fa", package = "rhierbaps"))
system.time({hb &lt;- hierBAPS(snp.matrix, max.depth=2, n.pops=20, quiet=FALSE)})


</code></pre>


</div>