<div class="container">

<table style="width: 100%;"><tr>
<td>identityCoefs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Omnibus function for identity coefficients</h2>

<h3>Description</h3>

<p>This function calculates the pairwise identity coefficients described by
Jacquard (1974). Unlike the previous <code>condensedIdentity()</code> (which will
continue to exist), this function also computes the 15 <em>detailed</em> identity
coefficients. The implementation supports pedigrees with inbred founders, and
X-chromosomal coefficients.
</p>


<h3>Usage</h3>

<pre><code class="language-R">identityCoefs(
  x,
  ids = labels(x),
  detailed = FALSE,
  Xchrom = FALSE,
  self = FALSE,
  simplify = TRUE,
  method = c("auto", "K", "WL", "LS", "GC", "idcoefs", "identity", "merlin"),
  verbose = FALSE,
  ...
)

detailed2condensed(d)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A pedigree in the form of a <code>pedtools::ped</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p>A vector of two ID labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>detailed</code></td>
<td>
<p>A logical. If FALSE (default), the 9 condensed coefficients
are computed; otherwise the 15 detailed identity coefficients.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Xchrom</code></td>
<td>
<p>A logical, by default FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>self</code></td>
<td>
<p>A logical indicating if self-relationships (i.e., between a
pedigree member and itself) should be included. FALSE by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simplify</code></td>
<td>
<p>Simplify the output (to a numeric of length 9) if <code>ids</code> has
length 2. Default: TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Either "auto", "K", "WL", "LS", "GC", "idcoefs", "identity" or
"merlin". By default ("auto") a suitable algorithm is chosen automatically.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>A logical.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Further arguments.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d</code></td>
<td>
<p>Either a numeric vector of length 15, or a data frame with 17
columns.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Both the condensed and detailed coefficients are given in the orders used by
Jacquard (1974). The function <code>detailed2condensed()</code> converts from detailed
coefficients (d1, ..., d15) to condensed ones (D1, ..., D9) using the
following relations:
</p>

<ul>
<li>
<p> D1 = d1
</p>
</li>
<li>
<p> D2 = d6
</p>
</li>
<li>
<p> D3 = d2 + d3
</p>
</li>
<li>
<p> D4 = d7
</p>
</li>
<li>
<p> D5 = d4 + d5
</p>
</li>
<li>
<p> D6 = d8
</p>
</li>
<li>
<p> D7 = d9 + d12
</p>
</li>
<li>
<p> D8 = d10 + d11 + d13 + d14
</p>
</li>
<li>
<p> D9 = d15
</p>
</li>
</ul>
<h4>Algorithms for computing identity coefficients</h4>

<p>The following is a brief overview of various algorithms for computing
(single-locus) condensed and/or detailed identity coefficients. This topic is
closely linked to that of <em>generalised kinship coefficients</em>, which is
further described in the documentation of <code>gKinship()</code>.
</p>
<p>For each algorithm below, it is indicated in brackets how to enforce it in
<code>identityCoefs()</code>.
</p>

<ul>
<li>
<p> Karigl (1981) gave the first recursive algorithm for the 9 condensed
identity coefficients. [<code>method = "K"</code>]
</p>
</li>
<li>
<p> Weeks &amp; Lange (1988) suggested a broader and more natural generalisation of
kinship coefficients, leading to a slightly different algorithm for condensed
coefficients. [<code>method = "WL"</code>]
</p>
</li>
<li>
<p> Lange &amp; Sinsheimer (1992) described an even further generalisation of
kinship coefficients, allowing a mix of deterministic and random sampling of
alleles. They used this to give (i) an alternative algorithm for the 9
condensed identity coefficients, and (ii) an algorithm for the 15 detailed
coefficients. [<code>method = "LS"</code>]
</p>
</li>
<li>
<p> The C program <code>IdCoefs</code> (version 2.1.1) by Mark Abney (2009) uses a graph
model to obtain very fast computation of condensed identity coefficients.
This requires <code>IdCoefs</code> to be installed on the computer (see link under
References) and available on the system search path. The function then writes
the necessary files to disk and calls <code>IdCoefs</code> via <code>system()</code>. [<code>method = "idcoefs"</code>]
</p>
</li>
<li>
<p> The R package <code>identity</code> provides an R interface for <code>IdCoefs</code>, avoiding
calls to <code>system()</code>. [<code>method = "identity"</code>]
</p>
</li>
<li>
<p> The MERLIN software (Abecasis et al, 2002) offers an option "â€“extended"
for computing detailed identity coefficients. This option requires MERLIN to
be installed on the system. The function then writes the necessary files to
disk and calls MERLIN via <code>system()</code>. If <code>detailed = FALSE</code>, the coefficients
are transformed with <code>detailed2condensed()</code> before returning. Note: MERLIN
rounds all numbers to 3 decimal places. Since this rounding is done on the
detailed coefficients, rounding errors may happen when converting to the
condensed ones. [<code>method = "merlin"</code>]
</p>
</li>
</ul>
<h3>Value</h3>

<p>A data frame with L + 2 columns, where L is either 9 or 15 (if
<code>detailed = TRUE</code>).
</p>
<p>If <code>simplify = TRUE</code> and <code>length(ids) = 2</code>: A numeric vector of length <code>L</code>.
</p>


<h3>References</h3>


<ul>
<li>
<p> Jacquard, A. (1974). The Genetic Structure of Populations. Springer.
</p>
</li>
<li>
<p> Karigl, G. (1981). A recursive algorithm for the calculation of identity
coefficients. Ann. Hum. Genet.
</p>
</li>
<li>
<p> Weeks, D.E. &amp; Lange, K. (1988). The affected-pedigree-member method of
linkage analysis. Am. J. Hum. Genet
</p>
</li>
<li>
<p> Lange, K. &amp; Sinsheimer, J.s. (1992). Calculation of genetic identity
coefficients. Ann. Hum. Genet.
</p>
</li>
<li>
<p> Abney, M. (2009). A graphical algorithm for fast computation of identity
coefficients and generalized kinship coefficients. Bioinformatics, 25,
1561-1563. <a href="https://home.uchicago.edu/~abney/abney_web/Software.html">https://home.uchicago.edu/~abney/abney_web/Software.html</a>
</p>
</li>
</ul>
<h3>See Also</h3>

<p><code>condensedIdentity()</code>, <code>gKinship()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">x = fullSibMating(1)

### Condensed coefficients
j1 = identityCoefs(x, method = "K")
j2 = identityCoefs(x, method = "WL")
j3 = identityCoefs(x, method = "LS")
j4 = identityCoefs(x, method = "GC")
j5 = condensedIdentity(x, ids = 1:6) # legacy version

stopifnot(all.equal(j1,j2), all.equal(j1,j3), all.equal(j1,j4), all.equal(j1,j5))

### Detailed coefficients
jdet1 = identityCoefs(x, detailed = TRUE, method = "LS")
jdet2 = identityCoefs(x, detailed = TRUE, method = "GC")

stopifnot(all.equal(jdet1,jdet2))

### X-chromosomal coefficients
jx1 = identityCoefs(x, Xchrom = TRUE, method = "K")
jx2 = identityCoefs(x, Xchrom = TRUE, method = "GC")
jx3 = condensedIdentityX(x, ids = 1:6)  # legacy version

stopifnot(all.equal(jx1,jx2), all.equal(jx1,jx3))

### Detailed X-chromosomal coefficients
jdx = identityCoefs(x, detailed = TRUE, Xchrom = TRUE, method = "GC")

stopifnot(all.equal(detailed2condensed(jdx), jx1))

</code></pre>


</div>