<div class="container">

<table style="width: 100%;"><tr>
<td>linearRidgeGenotypesPredict</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Predict phenotypes from genome-wide SNP data based on a file of coefficients
</h2>

<h3>Description</h3>

<p>Predict phenotypes from genome-wide SNP data based on a file of
coefficients. Genotypes and fitted coefficients are provided as
filenames, allowing the computation of fitted probabilities when SNP
data are too large to be read into R.
</p>


<h3>Usage</h3>

<pre><code class="language-R">linearRidgeGenotypesPredict(genotypesfilename, betafilename, phenotypesfilename = NULL,
verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genotypesfilename</code></td>
<td>

<p>character string: path to file containing SNP genotypes coded 0, 1,
2. See <code>Input file formats</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>betafilename</code></td>
<td>

<p>character string: path to file containing fitted coefficients. See <code>Input file formats</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phenotypesfilename</code></td>
<td>

<p>(optional) character string: path to file in which to write out the
predicted phenotypes. See <code>Output file formats</code>. Whether or not this argument
is supplied, the fitted coefficients are also returned by the function.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>Logical: If <code>TRUE</code>, additional information is printed to the R
outupt as the code runs. Defaults to <code>FALSE</code>.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A vector of fitted values, the same length as the number of
individuals whose data are in <code>genotypesfilename</code>. If
<code>phenotypesfilename</code> is supplied, the fitted values are also
written there.
</p>


<h3>Input file formats</h3>


<dl>
<dt>genotypesfilename:</dt>
<dd>
<p>A header row, plus one row for each
individual, one SNP per column. The header row contains SNP
names. SNPs are coded as 0, 1, 2 for minor allele count. Missing
values are not accommodated. </p>
</dd>
<dt>betafilename:</dt>
<dd>
<p>Two columns: First column is SNP names in same order as in <code>genotypesfilename</code>, second column is fitted coefficients. If the coefficients include an intercept then the first row of <code>betafilename</code> should contain it with the name Intercept in the first column. An Intercept thus labelled will be used appropriately in predicting the phenotypes. SNP names must match those in <code>genotypesfilename</code>. 
The format of <code>betafilename</code> is
that of the output of <code>linearRidgeGenotypes</code>, meaning
<code>linearRidgeGenotypesPredict</code> can be used to predict using
coefficients fitted using <code>linearRidgeGenotypes</code> (see the example).
</p>
</dd>
</dl>
<h3>Output file format</h3>

<p>Whether or not <code>phenotypesfilename</code> is provided, predicted phenotypes are returned to the R workshpace. If <code>phenotypesfilename</code> is provided, predicted phenotypes are written to the file specified (in addition).
</p>

<dl>
<dt>phenotypesfilename:</dt>
<dd>
<p>One column, containing predicted phenotypes, one individual per row.</p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>Erika Cule
</p>


<h3>References</h3>

<p>A semi-automatic method to guide the choice of ridge parameter in ridge regression. Cule, E. and De Iorio, M. (2012) arXiv:1205.0686v1 [stat.AP] 
</p>


<h3>See Also</h3>

<p><code>linearRidgeGenotypes</code> for model
fitting. <code>logisticRidgeGenotypes</code> and
<code>logisticRidgeGenotypesPredict</code> for corresponding functions
to fit and predict on SNP data with binary outcomes.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
genotypesfile &lt;- system.file("extdata","GenCont_genotypes.txt",package = "ridge")
phenotypesfile &lt;- system.file("extdata","GenCont_phenotypes.txt",package = "ridge")
betafile &lt;- tempfile(pattern = "beta", fileext = ".dat")
beta_linearRidgeGenotypes &lt;- linearRidgeGenotypes(genotypesfilename = genotypesfile,
                                                      phenotypesfilename = phenotypesfile,
                                                      betafilename = betafile)
pred_phen_geno &lt;- linearRidgeGenotypesPredict(genotypesfilename = genotypesfile,
                                                    betafilename = betafile)
## compare to output of linearRidge
data(GenCont) ## Same data as in GenCont_genotypes.txt and GenCont_phenotypes.txt
beta_linearRidge &lt;- linearRidge(Phenotypes ~ ., data = as.data.frame(GenCont))
pred_phen &lt;- predict(beta_linearRidge)
print(cbind(pred_phen_geno, pred_phen))
## Delete the temporary betafile
unlink(betafile)

## End(Not run)
</code></pre>


</div>