<div class="container">

<table style="width: 100%;"><tr>
<td>linearRidgeGenotypes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Fits linear ridge regression models for genome-wide SNP data.
</h2>

<h3>Description</h3>

<p>Fits linear ridge regression models for genome-wide SNP data. The SNP
genotypes are not read into R but file names are passed the code
directly, enabling the analysis of genome-wide scale SNP data sets.
</p>


<h3>Usage</h3>

<pre><code class="language-R">linearRidgeGenotypes(genotypesfilename, phenotypesfilename, lambda = -1, 
thinfilename = NULL, betafilename = NULL, approxfilename = NULL,
permfilename = NULL, intercept =
TRUE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genotypesfilename</code></td>
<td>

<p>character string: path to file containing SNP genotypes coded 0, 1,
2. See <code>Input file formats</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phenotypesfilename</code></td>
<td>

<p>character string: path to file containing phenotypes. See <code>Input file formats</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambda</code></td>
<td>

<p>(optional) shrinkage parameter. If not provided, the default denotes
automatic choice of the shrinkage parameter using the method of Cule &amp;
De Iorio (2012). 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>thinfilename</code></td>
<td>

<p>(optional) character string: path to file containing three columns: SNP name, chromosme and SNP psotion. See <code>Input file formats</code>. (See <code>details</code>.)
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>betafilename</code></td>
<td>

<p>(optional) character string: path to file where the output will be written. See <code>Output file formats</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>approxfilename</code></td>
<td>

<p>(optional) character string: path to fine where the approximate test p-values will be written. 
Approximate p-values are not computed unless this argument is given. Approximate p-values
are computed using the method of Cule et al (2011). See <code>Output file formats</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>permfilename</code></td>
<td>

<p>(optional) character string: path to file where the permutation test
p-values will be written.
Permutation test p-values are not computed unless this argument is
given. (See warning). See <code>Output file formats</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>intercept</code></td>
<td>

<p>Logical: Should the ridge regression model be fitted with an
intercept? (Defaults to <code>TRUE</code>)
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>Logical: If <code>TRUE</code>, additional information is printed to the R
output as the code runs. Defaults to <code>FALSE</code>.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If a file <code>thin</code> is supplied, and the shrinkage parameter
<code>lambda</code> is being computed automatically based on the data, then
this file is used to thin the SNP data by SNP position. If this file
is not supplied, SNPs are thinned automatically based on number of SNPs.
</p>


<h3>Value</h3>

<p>The vector of fitted ridge regression coefficients.
If <code>betafilename</code> is given, the fitted coefficients are written to this
file as well as being returned.
If <code>approxfilename</code> and/or <code>permfilename</code> are given, results of approximate
test p-values and/or permutation test p-values are written to the files
given in their arguments.
</p>


<h3>Input file formats</h3>


<dl>
<dt>genotypesfilename:</dt>
<dd>
<p>A header row, plus one row for each
individual, one SNP per column. The header row contains SNP
names. SNPs are coded as 0, 1, 2 for minor allele count. Missing
values are not accommodated. Invariant SNPs in the data cause an
error, please remove these from the file before calling the function.</p>
</dd>
<dt>phenofilename:</dt>
<dd>
<p>A single column of phenotypes with the individuals in the same order as those in the file <code>genotypesfilename</code>.</p>
</dd>
<dt>thin:</dt>
<dd>
<p>(optional) Three columns and the same number of rows as there are SNPs in the file <code>genotypesfilename</code>, one row per SNP. First column: SNP names (must match names in <code>genotypesfilename</code>); second column: chromosome;  third column: SNP position in BP.</p>
</dd>
</dl>
<h3>Output file formats</h3>

<p>All output files are optional. Whether or not <code>betafilename</code> is provided, fitted coefficients are returned to the R workshpace. If <code>betafilename</code> is provided, fitted coefficients are written to the file specified (in addition).
</p>

<dl>
<dt>betafilename:</dt>
<dd>
<p>Two columns: First column is SNP names in same order as in <code>genotypesfilename</code>, second column is fitted coefficients. If <code>intercept = TRUE</code> (the default) then the first row is the fitted intercept (with the name Intercept in the first column).</p>
</dd>
<dt>approxfilename:</dt>
<dd>
<p>Two columns: First column is SNP names in same order as in <code>genotypesfilename</code>, second column is approximate p-values.</p>
</dd>
<dt>permfilename:</dt>
<dd>
<p>Two columns: First column is SNP names in same order as in <code>genotypesfilename</code>, second column is permutation p-values.</p>
</dd>
</dl>
<h3>Warning </h3>

<p>When data are large, the permutation test p-values
may take a very long time to compute. It is recommended not to request
permutation test p-values (using the argument <code>permfilename</code>)
when data are large.</p>


<h3>Author(s)</h3>

<p>Erika Cule
</p>


<h3>References</h3>

<p>Significance testing in ridge regression for genetic data. Cule, E. et
al (2011) BMC Bioinformatics, 12:372
A semi-automatic method to guide the choice of ridge parameter in
ridge regression. Cule, E. and De Iorio, M. (2012) arXiv:1205.0686v1 [stat.AP]
</p>


<h3>See Also</h3>

<p><code>linearRidge</code> for fitting linear ridge regression models
when the data are small enough to be read into R.
<code>logisticRidge</code> and <code>logisticRidgeGenotypes</code> for fitting logistic ridge
regression models.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
    genotypesfile &lt;- system.file("extdata","GenCont_genotypes.txt",package = "ridge")
    phenotypesfile &lt;- system.file("extdata","GenCont_phenotypes.txt",package = "ridge")
    beta_linearRidgeGenotypes &lt;- linearRidgeGenotypes(genotypesfilename = genotypesfile,
phenotypesfilename = phenotypesfile)
    ## compare to output of linearRidge
    data(GenCont) ## Same data as in GenCont_genotypes.txt and GenCont_phenotypes.txt
    beta_linearRidge &lt;- linearRidge(Phenotypes ~ ., data = as.data.frame(GenCont))
    cbind(round(coef(beta_linearRidge), 6), beta_linearRidgeGenotypes)

## End(Not run)
  </code></pre>


</div>