<div class="container">

<table style="width: 100%;"><tr>
<td>getAssemblyFromFasta</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create an assembly from a FASTA file for its use as parameter of <code>genomemap_rjs</code> or <code>manhattan_rjs</code>.</h2>

<h3>Description</h3>

<p><code>getAssemblyFromFasta</code> create assemblies from a FASTA file for their use as parameter of <code>genomemap_rjs</code> or <code>manhattan_rjs</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getAssemblyFromFasta(fasta)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>fasta</code></td>
<td>
<p>a "character" string representing the input Fasta file to be added in the genome browser.</p>
</td>
</tr></table>
<h3>Value</h3>

<p>a data frame in BED format ready to use as an assembly in <code>genomemap_rjs</code> or <code>manhattan_rjs</code>.
</p>


<h3>Note</h3>

<p>FASTA format is a text-based format for representing either nucleotide sequences or peptide sequences, in which nucleotides or amino acids are represented using single-letter codes.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The ‘RJSplot’ Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code>genomemap_rjs</code>.
</p>
<p><code>manhattan_rjs</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Live examples and full tutorial on https://rjsplot.usal.es
</code></pre>


</div>