<div class="container">

<table style="width: 100%;"><tr>
<td>genomemap_rjs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create an interative genome map.</h2>

<h3>Description</h3>

<p><code>genomemap_rjs</code> creates an interactive genome map.
</p>


<h3>Usage</h3>

<pre><code class="language-R">genomemap_rjs(assembly, track = NULL, plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>assembly</code></td>
<td>
<p>a genome assembly data frame. RJSplot provides human assemblies (NCBI36, GRCh37, GRCh38), human assemblies with cytobands (GRCh37.bands or GRCh38.bands), or methods for creating your own assembly from a FASTA file <code>getAssemblyFromFasta</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>track</code></td>
<td>
<p>a data frame with values to represent on the genome map (chromosome, position, value).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dir</code></td>
<td>
<p>a "character" string representing the directory where the graph will be saved.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with Firefox. If you want to open this file stored in your local machine with other Web browser, please visit the help section on the RJSplot Web site (<a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>).
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The ‘RJSplot’ Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code>dendrogram_rjs</code>, <code>densityplot_rjs</code>, <code>genomemap_rjs</code>, <code>heatmap_rjs</code>, <code>manhattan_rjs</code>, <code>network_rjs</code>, <code>scatterplot_rjs</code>, <code>symheatmap_rjs</code>, <code>wordcloud_rjs</code>, <code>boxplot_rjs</code>, <code>bubbles_rjs</code>, <code>hiveplot_rjs</code>, <code>piechart_rjs</code>, <code>barplot_rjs</code>, <code>tables_rjs</code>, <code>surface3d_rjs</code>, <code>scatter3d_rjs</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Create an interactive Genome map in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es

if(interactive()){
# Create test data
chr &lt;- character()
pos &lt;- numeric()

for(i in 1:nrow(GRCh38)){
  chr &lt;- c(chr,as.character(rep(GRCh38[i,"chr"],100)))
  pos &lt;- c(pos,sample(GRCh38[i,"start"]:GRCh38[i,"end"],100))
}

value &lt;- round(rexp(length(pos)),2)

# Create a genome map
track &lt;- data.frame(chr,pos,pos+1,NA,value)
genomemap_rjs(GRCh38.bands, track)
}
</code></pre>


</div>