<div class="container">

<table style="width: 100%;"><tr>
<td>manhattan_rjs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create an interactive manhattan plot</h2>

<h3>Description</h3>

<p><code>manhattan_rjs</code> creates an interactive Manhattan plot.
</p>


<h3>Usage</h3>

<pre><code class="language-R">manhattan_rjs(data, assembly, axisv, distv, cell = 1e+06,
  outliers = TRUE, ylab = "score", yscale = NULL,
  cex = 1, plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>a data frame with four columns (identifier, chromosome/scaffold, position, score).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assembly</code></td>
<td>
<p>a data frame with an assembly to be used. It should be one of the following human assemblies data frames provided by RJSplot: <code>NCBI36</code>, <code>GRCh37</code>, <code>GRCh38</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>axisv</code></td>
<td>
<p>value for the coordinate of the main horizontal axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>distv</code></td>
<td>
<p>distance between the main horizontal axis and the secondary axes (which stablishes the limit for the outlier representation).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cell</code></td>
<td>
<p>segment size which is represented.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outliers</code></td>
<td>
<p>logical value to enable (<code>TRUE</code>) or disable (<code>FALSE</code>) outliers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>a title for the y axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>yscale</code></td>
<td>
<p>a vector with two values which specifies the minimun and maximun limits in the representation. By default maximun and minimun scores are taken as the limits.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex</code></td>
<td>
<p>number indicating the amount by which plotting text and symbols should be scaled relative to the default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dir</code></td>
<td>
<p>a "character" string representing the directory where the graph will be saved.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The ‘RJSplot’ Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code>dendrogram_rjs</code>, <code>densityplot_rjs</code>, <code>genomemap_rjs</code>, <code>heatmap_rjs</code>, <code>manhattan_rjs</code>, <code>network_rjs</code>, <code>scatterplot_rjs</code>, <code>symheatmap_rjs</code>, <code>wordcloud_rjs</code>, <code>boxplot_rjs</code>, <code>bubbles_rjs</code>, <code>hiveplot_rjs</code>, <code>piechart_rjs</code>, <code>barplot_rjs</code>, <code>tables_rjs</code>, <code>surface3d_rjs</code>, <code>scatter3d_rjs</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Create an interactive Manhattan plot in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es

if(interactive()){
# Create test data
chr &lt;- character()
pos &lt;- numeric()

for(i in 1:nrow(GRCh38)){
  chr &lt;- c(chr,as.character(rep(GRCh38[i,"chr"],100)))
  pos &lt;- c(pos,sample(GRCh38[i,"start"]:GRCh38[i,"end"],100))
}

value &lt;- round(rexp(length(pos)),2)

# Create a manhattan plot
data &lt;- data.frame(paste0("ProbeSet_",seq_along(pos)),chr,pos,value)
manhattan_rjs(data, GRCh38, 0, 1, 0, TRUE, "log2Ratio")
}

</code></pre>


</div>