<div class="container">

<table style="width: 100%;"><tr>
<td>online_iNMF-deprecated</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>[Deprecated] Perform online iNMF on scaled datasets</h2>

<h3>Description</h3>

<p><b>Please turn to <code>runOnlineINMF</code> or
<code>runIntegration</code></b>.
</p>
<p>Perform online integrative non-negative matrix factorization to represent
multiple single-cell datasets in terms of H, W, and V matrices. It optimizes
the iNMF objective function using online learning (non-negative least squares
for H matrix, hierarchical alternating least squares for W and V matrices),
where the number of factors is set by k. The function allows online learning
in 3 scenarios: (1) fully observed datasets; (2) iterative refinement using
continually arriving datasets; and (3) projection of new datasets without
updating the existing factorization. All three scenarios require fixed memory
independent of the number of cells.
</p>
<p>For each dataset, this factorization produces an H matrix (cells by k), a V
matrix (k by genes), and a shared W matrix (k by genes). The H matrices
represent the cell factor loadings. W is identical among all datasets, as it
represents the shared components of the metagenes across datasets. The V
matrices represent the dataset-specific components of the metagenes.
</p>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p><code>liger</code> object with data stored in HDF5 files. Should
normalize, select genes, and scale before calling.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X_new</code></td>
<td>
<p>List of new datasets for scenario 2 or scenario 3. Each list
element should be the name of an HDF5 file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>projection</code></td>
<td>
<p>Perform data integration by shared metagene (W) projection
(scenario 3). (default FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>W.init</code></td>
<td>
<p>Optional initialization for W. (default NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>V.init</code></td>
<td>
<p>Optional initialization for V (default NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>H.init</code></td>
<td>
<p>Optional initialization for H (default NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>A.init</code></td>
<td>
<p>Optional initialization for A (default NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B.init</code></td>
<td>
<p>Optional initialization for B (default NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>Inner dimension of factorizationâ€“number of metagenes (default 20).
A value in the range 20-50 works well for most analyses.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambda</code></td>
<td>
<p>Regularization parameter. Larger values penalize
dataset-specific effects more
strongly (ie. alignment should increase as lambda increases). We recommend
always using the default value except
possibly for analyses with relatively small differences (biological
replicates, male/female comparisons, etc.)
in which case a lower value such as 1.0 may improve reconstruction quality.
(default 5.0).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.epochs</code></td>
<td>
<p>Maximum number of epochs (complete passes through the
data). (default 5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>miniBatch_max_iters</code></td>
<td>
<p>Maximum number of block coordinate descent (HALS
algorithm) iterations to perform for each update of W and V (default 1).
Changing this parameter is not  recommended.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>miniBatch_size</code></td>
<td>
<p>Total number of cells in each minibatch (default 5000).
This is a reasonable default, but a smaller value such as 1000 may be
necessary for analyzing very small datasets. In general, minibatch size
should be no larger than the number of cells in the smallest dataset.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>h5_chunk_size</code></td>
<td>
<p>Chunk size of input hdf5 files (default 1000). The chunk
size should be no larger than the batch size.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>Random seed to allow reproducible results (default 123).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Print progress bar/messages (TRUE by default)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p><code>liger</code> object with H, W, V, A and B slots set.
</p>


</div>