<div class="container">

<table style="width: 100%;"><tr>
<td>read10X</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Load in data from 10X</h2>

<h3>Description</h3>

<p>Enables easy loading of sparse data matrices provided by 10X genomics.
</p>
<p><code>read10X</code> works generally for 10X cellranger pipelines including:
CellRanger &lt; 3.0 &amp; &gt;= 3.0 and CellRanger-ARC.
</p>
<p><code>read10XRNA</code> invokes <code>read10X</code> and takes the "Gene Expression" out,
so that the result can directly be used to construct a liger
object. See Examples for demonstration.
</p>
<p><code>read10XATAC</code> works for both cellRanger-ARC and cellRanger-ATAC
pipelines but needs user arguments for correct recognition. Similarly, the
returned value can directly be used for constructing a liger
object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read10X(
  path,
  sampleNames = NULL,
  useFiltered = NULL,
  reference = NULL,
  geneCol = 2,
  cellCol = 1,
  returnList = FALSE,
  verbose = getOption("ligerVerbose", TRUE),
  sample.dirs = path,
  sample.names = sampleNames,
  use.filtered = useFiltered,
  data.type = NULL,
  merge = NULL,
  num.cells = NULL,
  min.umis = NULL
)

read10XRNA(
  path,
  sampleNames = NULL,
  useFiltered = NULL,
  reference = NULL,
  returnList = FALSE,
  ...
)

read10XATAC(
  path,
  sampleNames = NULL,
  useFiltered = NULL,
  pipeline = c("atac", "arc"),
  arcFeatureType = "Peaks",
  returnList = FALSE,
  geneCol = 2,
  cellCol = 1,
  verbose = getOption("ligerVerbose", TRUE)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>path</code></td>
<td>
<p>[A.] A Directory containing the matrix.mtx, genes.tsv (or
features.tsv), and barcodes.tsv files provided by 10X. A vector, a named
vector, a list or a named list can be given in order to load several data
directories. [B.] The 10X root directory where subdirectories of per-sample
output folders can be found. Sample names will by default take the name of
the vector, list or subfolders.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sampleNames</code></td>
<td>
<p>A vector of names to override the detected or set sample
names for what is given to <code>path</code>. Default <code>NULL</code>. If no name
detected at all and multiple samples are given, will name them by numbers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useFiltered</code></td>
<td>
<p>Logical, if <code>path</code> is given as case B, whether to use
the filtered feature barcode matrix instead of raw (unfiltered). Default
<code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reference</code></td>
<td>
<p>In case of specifying a CellRanger&lt;3 root folder to
<code>path</code>, import the matrix from the output using which reference. Only
needed when multiple references present. Default <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneCol</code></td>
<td>
<p>Specify which column of genes.tsv or features.tsv to use for
gene names. Default <code>2</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cellCol</code></td>
<td>
<p>Specify which column of barcodes.tsv to use for cell names.
Default <code>1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>returnList</code></td>
<td>
<p>Logical, whether to still return a structured list instead
of a single matrix object, in the case where only one sample and only one
feature type can be found. Otherwise will always return a list. Default
<code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical. Whether to show information of the progress. Default
<code>getOption("ligerVerbose")</code> or <code>TRUE</code> if users have not set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.dirs, sample.names, use.filtered</code></td>
<td>
<p>These arguments are renamed and
will be deprecated in the future. Please see usage for corresponding
arguments.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data.type, merge, num.cells, min.umis</code></td>
<td>
<p>These arguments are defuncted
because the functionality can/should be fulfilled with other functions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Arguments passed to <code>read10X</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pipeline</code></td>
<td>
<p>Which cellRanger pipeline type to find the ATAC data. Choose
<code>"atac"</code> to read the peak matrix from cellranger-atac pipeline output
folder(s), or <code>"arc"</code> to split the ATAC feature subset out from the
multiomic cellranger-arc pipeline output folder(s). Default <code>"atac"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>arcFeatureType</code></td>
<td>
<p>When <code>pipeline = "arc"</code>, which feature type is
for the ATAC data of interests. Default <code>"Peaks"</code>. Other possible
feature types can be <code>"Chromatin Accessibility"</code>. Error message will
show available options if argument specification cannot be found.</p>
</td>
</tr>
</table>
<h3>Value</h3>


<ul>
<li>
<p>When only one sample is given or detected, and only one feature type
is detected or using CellRanger &lt; 3.0, and <code>returnList = FALSE</code>, a
sparse matrix object (dgCMatrix class) will be returned.
</p>
</li>
<li>
<p>When using <code>read10XRNA</code> or <code>read10XATAC</code>, which are modality
specific, returns a list named by samples, and each element is the
corresponding sparse matrix object (dgCMatrix class).
</p>
</li>
<li>
<p><code>read10X</code> generally returns a list named by samples. Each sample
element will be another list named by feature types even if only one feature
type is detected (or using CellRanger &lt; 3.0) for data structure consistency.
The feature type "Gene Expression" always comes as the first type if
available.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# For output from CellRanger &lt; 3.0
dir &lt;- 'path/to/data/directory'
list.files(dir) # Should show barcodes.tsv, genes.tsv, and matrix.mtx
mat &lt;- read10X(dir)
class(mat) # Should show dgCMatrix

# For root directory from CellRanger &lt; 3.0
dir &lt;- 'path/to/root'
list.dirs(dir) # Should show sample names
matList &lt;- read10X(dir)
names(matList) # Should show the sample names
class(matList[[1]][["Gene Expression"]]) # Should show dgCMatrix

# For output from CellRanger &gt;= 3.0 with multiple data types
dir &lt;- 'path/to/data/directory'
list.files(dir) # Should show barcodes.tsv.gz, features.tsv.gz, and matrix.mtx.gz
matList &lt;- read10X(dir, sampleNames = "tissue1")
names(matList) # Shoud show "tissue1"
names(matList$tissue1) # Should show feature types, e.g. "Gene Expression" and etc.

# For root directory from CellRanger &gt;= 3.0 with multiple data types
dir &lt;- 'path/to/root'
list.dirs(dir) # Should show sample names, e.g. "rep1", "rep2", "rep3"
matList &lt;- read10X(dir)
names(matList) # Should show the sample names: "rep1", "rep2", "rep3"
names(matList$rep1) # Should show the avalable feature types for rep1

## End(Not run)
## Not run: 
# For creating LIGER object from root directory of CellRanger &gt;= 3.0
dir &lt;- 'path/to/root'
list.dirs(dir) # Should show sample names, e.g. "rep1", "rep2", "rep3"
matList &lt;- read10XRNA(dir)
names(matList) # Should show the sample names: "rep1", "rep2", "rep3"
sapply(matList, class) # Should show matrix class all are "dgCMatrix"
lig &lt;- createLigerObject(matList)

## End(Not run)
</code></pre>


</div>