<div class="container">

<table style="width: 100%;"><tr>
<td>runIntegration</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Integrate scaled datasets with iNMF or variant methods</h2>

<h3>Description</h3>

<p>LIGER provides dataset integration methods based on iNMF (integrative
Non-negative Matrix Factorization [1]) and its variants (online iNMF [2] and
UINMF [3]). This function wraps <code>runINMF</code>,
<code>runOnlineINMF</code> and <code>runUINMF</code>, of which the help
pages have more detailed description.
</p>


<h3>Usage</h3>

<pre><code class="language-R">runIntegration(
  object,
  k = 20,
  lambda = 5,
  method = c("iNMF", "onlineINMF", "UINMF"),
  ...
)

## S3 method for class 'liger'
runIntegration(
  object,
  k = 20,
  lambda = 5,
  method = c("iNMF", "onlineINMF", "UINMF"),
  seed = 1,
  verbose = getOption("ligerVerbose", TRUE),
  ...
)

## S3 method for class 'Seurat'
runIntegration(
  object,
  k = 20,
  lambda = 5,
  method = c("iNMF", "onlineINMF"),
  datasetVar = "orig.ident",
  useLayer = "ligerScaleData",
  assay = NULL,
  seed = 1,
  verbose = getOption("ligerVerbose", TRUE),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>A liger object or a Seurat object with
non-negative scaled data of variable features (Done with
<code>scaleNotCenter</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>Inner dimension of factorization (number of factors). Generally, a
higher <code>k</code> will be needed for datasets with more sub-structure. Default
<code>20</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambda</code></td>
<td>
<p>Regularization parameter. Larger values penalize
dataset-specific effects more strongly (i.e. alignment should increase as
<code>lambda</code> increases). Default <code>5</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>iNMF variant algorithm to use for integration. Choose from
<code>"iNMF"</code>, <code>"onlineINMF"</code>, <code>"UINMF"</code>. Default <code>"iNMF"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Arguments passed to other methods and wrapped functions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>Random seed to allow reproducible results. Default <code>1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical. Whether to show information of the progress. Default
<code>getOption("ligerVerbose")</code> or <code>TRUE</code> if users have not set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>datasetVar</code></td>
<td>
<p>Metadata variable name that stores the dataset source
annotation. Default <code>"orig.ident"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useLayer</code></td>
<td>
<p>For Seurat&gt;=4.9.9, the name of layer to retrieve input
non-negative scaled data. Default <code>"ligerScaleData"</code>. For older Seurat,
always retrieve from <code>scale.data</code> slot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assay</code></td>
<td>
<p>Name of assay to use. Default <code>NULL</code> uses current active
assay.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Updated input object. For detail, please refer to the refered method
linked in Description.
</p>


<h3>References</h3>


<ol>
<li>
<p>Joshua D. Welch and et al., Single-Cell Multi-omic Integration Compares
and Contrasts Features of Brain Cell Identity, Cell, 2019
</p>
</li>
<li>
<p>Chao Gao and et al., Iterative single-cell multi-omic integration using
online learning, Nat Biotechnol., 2021
</p>
</li>
<li>
<p>April R. Kriebel and Joshua D. Welch, UINMF performs mosaic integration
of single-cell multi-omic datasets using nonnegative matrix factorization,
Nat. Comm., 2022
</p>
</li>
</ol>
<h3>Examples</h3>

<pre><code class="language-R">pbmc &lt;- normalize(pbmc)
pbmc &lt;- selectGenes(pbmc)
pbmc &lt;- scaleNotCenter(pbmc)
if (requireNamespace("RcppPlanc", quietly = TRUE)) {
    pbmc &lt;- runIntegration(pbmc)
}
</code></pre>


</div>