<div class="container">

<table style="width: 100%;"><tr>
<td>plotGeneHeatmap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot Heatmap of Gene Expression or Factor Loading</h2>

<h3>Description</h3>

<p>Plot Heatmap of Gene Expression or Factor Loading
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotGeneHeatmap(
  object,
  features,
  cellIdx = NULL,
  slot = c("normData", "rawData", "scaleData", "scaleUnsharedData"),
  useCellMeta = NULL,
  cellAnnotation = NULL,
  featureAnnotation = NULL,
  cellSplitBy = NULL,
  featureSplitBy = NULL,
  viridisOption = "C",
  ...
)

plotFactorHeatmap(
  object,
  factors = NULL,
  cellIdx = NULL,
  slot = c("H.norm", "H"),
  useCellMeta = NULL,
  cellAnnotation = NULL,
  factorAnnotation = NULL,
  cellSplitBy = NULL,
  factorSplitBy = NULL,
  trim = c(0, 0.03),
  viridisOption = "D",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>A liger object, with data to be plot available.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>features, factors</code></td>
<td>
<p>Character vector of genes of interests or numeric
index of factor to be involved. <code>features</code> is required, while
<code>factors</code> is by default all the factors (reads object recorded k value
in <code>uns</code> slot).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cellIdx</code></td>
<td>
<p>Valid index to subscribe cells to be included. See
<code>subsetLiger</code>. Default <code>NULL</code> use all cells.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>slot</code></td>
<td>
<p>Use the chosen matrix for heatmap. For <code>plotGeneHeatmap</code>,
default <code>"normData"</code>, alternatively <code>"rawData"</code>,
<code>"scaleData"</code> or <code>"scaleUnsharedData"</code>. For
<code>plotFactorHeatmap</code>, default <code>"H.norm"</code>, alternatively <code>"H"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useCellMeta</code></td>
<td>
<p>Character vector of available variable names in
<code>cellMeta</code>, variables will be added as annotation to the heatmap.
Default <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cellAnnotation</code></td>
<td>
<p>data.frame object for using external annotation, with
each column a variable and each row is a cell. Row names of this data.frame
will be used for matching cells involved in heatmap. For cells not found in
this data.frame, <code>NA</code>s will be added with warning. Default <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>featureAnnotation, factorAnnotation</code></td>
<td>
<p>Similar as <code>cellAnnotation</code>,
while each row would be a gene or factor, respectively. Default <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cellSplitBy</code></td>
<td>
<p>Character vector of variable names available in annotation
given by <code>useCellMeta</code> and <code>cellAnnotation</code>. This slices the
heatmap by specified variables. Default <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>featureSplitBy, factorSplitBy</code></td>
<td>
<p>Similar as <code>cellSplitBy</code>. Default
<code>NULL</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>viridisOption</code></td>
<td>
<p>See <code>option</code> argument of
<code>viridis</code>. Default <code>"C"</code> (plasma) for
<code>plotGeneHeatmap</code> and <code>"D"</code> (viridis) for <code>plotFactorHeatmap</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional arguments passed to general function
<code>.plotHeatmap</code> and <code>Heatmap</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trim</code></td>
<td>
<p>Numeric vector of two numbers. Higher value limits the maximum
value and lower value limits the minimum value. Default <code>c(0, 0.03)</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p><code>HeatmapList-class</code> object
</p>


<h3>Examples</h3>

<pre><code class="language-R">
plotGeneHeatmap(pbmcPlot, varFeatures(pbmcPlot))
plotGeneHeatmap(pbmcPlot, varFeatures(pbmcPlot),
                useCellMeta = c("leiden_cluster", "dataset"),
                cellSplitBy = "leiden_cluster")

plotFactorHeatmap(pbmcPlot)
plotFactorHeatmap(pbmcPlot, cellIdx = pbmcPlot$leiden_cluster %in% 1:3,
                  useCellMeta = c("leiden_cluster", "dataset"),
                  cellSplitBy = "leiden_cluster")

</code></pre>


</div>