<div class="container">

<table style="width: 100%;"><tr>
<td>MVP.Data.VCF2MVP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>MVP.Data.VCF2MVP: To transform vcf data to MVP package
Author: Haohao Zhang
Build date: Sep 12, 2018</h2>

<h3>Description</h3>

<p>Accept the | or / separated markers, any variant sites that are not 0 or 1 will be considered NA.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MVP.Data.VCF2MVP(
  vcf_file,
  out = "mvp",
  maxLine = 10000,
  type.geno = "char",
  threads = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcf_file</code></td>
<td>
<p>Genotype in VCF format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out</code></td>
<td>
<p>the name of output file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxLine</code></td>
<td>
<p>the number of markers handled at a time, smaller value would reduce the memory cost</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type.geno</code></td>
<td>
<p>the type of genotype elements</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threads</code></td>
<td>
<p>number of thread for transforming</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>whether to print the reminder</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>number of individuals and markers.
Output files:
genotype.desc, genotype.bin: genotype file in bigmemory format
phenotype.phe: ordered phenotype file, same taxa order with genotype file
map.map: SNP information
</p>


<h3>Examples</h3>

<pre><code class="language-R">
vcfPath &lt;- system.file("extdata", "01_vcf", "mvp.vcf", package = "rMVP")

MVP.Data.VCF2MVP(vcfPath, tempfile("outfile"), threads=1)


</code></pre>


</div>