<div class="container">

<table style="width: 100%;"><tr>
<td>MVP.MLM</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>To perform GWAS with GLM and MLM model and get the P value of SNPs</h2>

<h3>Description</h3>

<p>Build date: Aug 30, 2016
Last update: Aug 30, 2016
</p>


<h3>Usage</h3>

<pre><code class="language-R">MVP.MLM(
  phe,
  geno,
  K = NULL,
  eigenK = NULL,
  CV = NULL,
  geno_ind_idx = NULL,
  REML = NULL,
  cpu = 1,
  vc.method = c("BRENT", "EMMA", "HE"),
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>phe</code></td>
<td>
<p>phenotype, n * 2 matrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>genotype, m * n, m is marker size, n is population size</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>Kinship, Covariance matrix(n * n) for random effects; must be positive semi-definite</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eigenK</code></td>
<td>
<p>list of eigen Kinship</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CV</code></td>
<td>
<p>covariates</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno_ind_idx</code></td>
<td>
<p>the index of effective genotyped individuals</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>REML</code></td>
<td>
<p>a list that contains ve and vg</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cpu</code></td>
<td>
<p>number of cpus used for parallel computation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vc.method</code></td>
<td>
<p>the methods for estimating variance component("emma" or "he" or "brent")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>whether to print detail.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>results: a m * 2 matrix, the first column is the SNP effect, the second column is the P values
</p>


<h3>Author(s)</h3>

<p>Lilin Yin and Xiaolei Liu
</p>


<h3>Examples</h3>

<pre><code class="language-R">
phePath &lt;- system.file("extdata", "07_other", "mvp.phe", package = "rMVP")
phenotype &lt;- read.table(phePath, header=TRUE)
idx &lt;- !is.na(phenotype[, 2])
phenotype &lt;- phenotype[idx, ]
print(dim(phenotype))
genoPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.desc", package = "rMVP")
genotype &lt;- attach.big.matrix(genoPath)
genotype &lt;- deepcopy(genotype, cols=idx)
print(dim(genotype))
K &lt;- MVP.K.VanRaden(genotype, cpu=1)

mlm &lt;- MVP.MLM(phe=phenotype, geno=genotype, K=K, cpu=1)
str(mlm)


</code></pre>


</div>