<div class="container">

<table style="width: 100%;"><tr>
<td>MVP.FarmCPU</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Perform GWAS using FarmCPU method</h2>

<h3>Description</h3>

<p>Date build: Febuary 24, 2013
Last update: May 25, 2017
Requirement: Y, GD, and CV should have same taxa order. GD and GM should have the same order on SNPs
</p>


<h3>Usage</h3>

<pre><code class="language-R">MVP.FarmCPU(
  phe,
  geno,
  map,
  CV = NULL,
  geno_ind_idx = NULL,
  P = NULL,
  method.sub = "reward",
  method.sub.final = "reward",
  method.bin = c("EMMA", "static", "FaST-LMM"),
  bin.size = c(5e+05, 5e+06, 5e+07),
  bin.selection = seq(10, 100, 10),
  memo = "MVP.FarmCPU",
  Prior = NULL,
  ncpus = 2,
  maxLoop = 10,
  threshold.output = 0.01,
  converge = 1,
  iteration.output = FALSE,
  p.threshold = NA,
  QTN.threshold = 0.01,
  bound = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>phe</code></td>
<td>
<p>phenotype, n by t matrix, n is sample size, t is number of phenotypes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>genotype, m by n matrix, m is marker size, n is sample size. This is Pure Genotype Data Matrix(GD). THERE IS NO COLUMN FOR TAXA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>SNP map information, m by 3 matrix, m is marker size, the three columns are SNP_ID, Chr, and Pos</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CV</code></td>
<td>
<p>covariates, n by c matrix, n is sample size, c is number of covariates</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno_ind_idx</code></td>
<td>
<p>the index of effective genotyped individuals</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>P</code></td>
<td>
<p>start p values for all SNPs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method.sub</code></td>
<td>
<p>method used in substitution process, five options: 'penalty', 'reward', 'mean', 'median', or 'onsite'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method.sub.final</code></td>
<td>
<p>method used in substitution process, five options: 'penalty', 'reward', 'mean', 'median', or 'onsite'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method.bin</code></td>
<td>
<p>method for selecting the most appropriate bins, three options: 'static', 'EMMA' or 'FaST-LMM'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bin.size</code></td>
<td>
<p>bin sizes for all iterations, a vector, the bin size is always from large to small</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bin.selection</code></td>
<td>
<p>number of selected bins in each iteration, a vector</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>memo</code></td>
<td>
<p>a marker on output file name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Prior</code></td>
<td>
<p>prior information, four columns, which are SNP_ID, Chr, Pos, P-value</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncpus</code></td>
<td>
<p>number of threads used for parallele computation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxLoop</code></td>
<td>
<p>maximum number of iterations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold.output</code></td>
<td>
<p>only the GWAS results with p-values lower than threshold.output will be output</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>converge</code></td>
<td>
<p>a number, 0 to 1, if selected pseudo QTNs in the last and the second last iterations have a certain probality (the probability is converge) of overlap, the loop will stop</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iteration.output</code></td>
<td>
<p>whether to output results of all iterations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.threshold</code></td>
<td>
<p>if all p values generated in the first iteration are bigger than p.threshold, FarmCPU stops</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>QTN.threshold</code></td>
<td>
<p>in second and later iterations, only SNPs with lower p-values than QTN.threshold have chances to be selected as pseudo QTNs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bound</code></td>
<td>
<p>maximum number of SNPs selected as pseudo QTNs in each iteration</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>whether to print detail.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a m by 4 results matrix, m is marker size, the four columns are SNP_ID, Chr, Pos, and p-value
</p>


<h3>Author(s)</h3>

<p>Xiaolei Liu and Zhiwu Zhang
</p>


<h3>Examples</h3>

<pre><code class="language-R">
phePath &lt;- system.file("extdata", "07_other", "mvp.phe", package = "rMVP")
phenotype &lt;- read.table(phePath, header=TRUE)
idx &lt;- !is.na(phenotype[, 2])
phenotype &lt;- phenotype[idx, ]
print(dim(phenotype))
genoPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.desc", package = "rMVP")
genotype &lt;- attach.big.matrix(genoPath)
genotype &lt;- deepcopy(genotype, cols=idx)
print(dim(genotype))
mapPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.map", package = "rMVP")
map &lt;- read.table(mapPath , head = TRUE)

farmcpu &lt;- MVP.FarmCPU(phe=phenotype,geno=genotype,map=map,maxLoop=2,method.bin="static")
str(farmcpu)


</code></pre>


</div>