<div class="container">

<table style="width: 100%;"><tr>
<td>MVP.GLM</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>To perform GWAS with GLM and MLM model and get the P value of SNPs</h2>

<h3>Description</h3>

<p>Build date: Aug 30, 2016
Last update: May 25, 2017
</p>


<h3>Usage</h3>

<pre><code class="language-R">MVP.GLM(phe, geno, CV = NULL, geno_ind_idx = NULL, cpu = 1, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>phe</code></td>
<td>
<p>phenotype, n * 2 matrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>Genotype in numeric format, pure 0, 1, 2 matrix; m * n, m is marker size, n is population size</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CV</code></td>
<td>
<p>Covariance, design matrix(n * x) for the fixed effects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno_ind_idx</code></td>
<td>
<p>the index of effective genotyped individuals</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cpu</code></td>
<td>
<p>number of cpus used for parallel computation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>whether to print detail.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>m * 2 matrix, the first column is the SNP effect, the second column is the P values
</p>


<h3>Author(s)</h3>

<p>Lilin Yin and Xiaolei Liu
</p>


<h3>Examples</h3>

<pre><code class="language-R">
phePath &lt;- system.file("extdata", "07_other", "mvp.phe", package = "rMVP")
phenotype &lt;- read.table(phePath, header=TRUE)
idx &lt;- !is.na(phenotype[, 2])
phenotype &lt;- phenotype[idx, ]
print(dim(phenotype))
genoPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.desc", package = "rMVP")
genotype &lt;- attach.big.matrix(genoPath)
genotype &lt;- deepcopy(genotype, cols=idx)
print(dim(genotype))

glm &lt;- MVP.GLM(phe=phenotype, geno=genotype, cpu=1)
str(glm)

</code></pre>


</div>