<div class="container">

<table style="width: 100%;"><tr>
<td>MVP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>MVP, Memory-efficient, Visualization-enhanced, Parallel-accelerated</h2>

<h3>Description</h3>

<p>Object 1: To perform GWAS using General Linear Model (GLM), Mixed Linear Model (MLM), and FarmCPU model
Object 2: To calculate kinship among individuals using Varaden method
Object 3: Estimate variance components using EMMA, FaST-LMM, and HE regression
Object 4: Generate high-quality figures
</p>


<h3>Usage</h3>

<pre><code class="language-R">MVP(
  phe,
  geno,
  map,
  K = NULL,
  nPC.GLM = NULL,
  nPC.MLM = NULL,
  nPC.FarmCPU = NULL,
  CV.GLM = NULL,
  CV.MLM = NULL,
  CV.FarmCPU = NULL,
  REML = NULL,
  maxLine = 10000,
  ncpus = detectCores(logical = FALSE),
  vc.method = c("BRENT", "EMMA", "HE"),
  method = c("GLM", "MLM", "FarmCPU"),
  p.threshold = NA,
  QTN.threshold = 0.01,
  method.bin = "static",
  bin.size = c(5e+05, 5e+06, 5e+07),
  bin.selection = seq(10, 100, 10),
  maxLoop = 10,
  permutation.threshold = FALSE,
  permutation.rep = 100,
  memo = NULL,
  outpath = getwd(),
  col = c("#4197d8", "#f8c120", "#413496", "#495226", "#d60b6f", "#e66519", "#d581b7",
    "#83d3ad", "#7c162c", "#26755d"),
  file.output = TRUE,
  file.type = "jpg",
  dpi = 300,
  threshold = 0.05,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>phe</code></td>
<td>
<p>phenotype, n * 2 matrix, n is sample size</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>Genotype in bigmatrix format; m * n, m is marker size, n is sample size</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>SNP map information, SNP name, Chr, Pos</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>Kinship, Covariance matrix(n * n) for random effects, must be positive semi-definite</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nPC.GLM</code></td>
<td>
<p>number of PCs added as fixed effects in GLM</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nPC.MLM</code></td>
<td>
<p>number of PCs added as fixed effects in MLM</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nPC.FarmCPU</code></td>
<td>
<p>number of PCs added as fixed effects in FarmCPU</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CV.GLM</code></td>
<td>
<p>covariates added in GLM</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CV.MLM</code></td>
<td>
<p>covariates added in MLM</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CV.FarmCPU</code></td>
<td>
<p>covariates added in FarmCPU</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>REML</code></td>
<td>
<p>a list contains ve and vg</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxLine</code></td>
<td>
<p>the number of markers handled at a time, smaller value would reduce the memory cost</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncpus</code></td>
<td>
<p>number of cpus used for parallel</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vc.method</code></td>
<td>
<p>methods for estimating variance component("EMMA" or "HE" or "BRENT")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>the GWAS model, "GLM", "MLM", and "FarmCPU", models can be selected simutaneously, i.e. c("GLM", "MLM", "FarmCPU")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.threshold</code></td>
<td>
<p>if all p values generated in the first iteration are bigger than p.threshold, FarmCPU stops</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>QTN.threshold</code></td>
<td>
<p>in second and later iterations, only SNPs with lower p-values than QTN.threshold have chances to be selected as pseudo QTNs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method.bin</code></td>
<td>
<p>'static' or 'FaST-LMM'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bin.size</code></td>
<td>
<p>window size in genome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bin.selection</code></td>
<td>
<p>a vector, how many windows selected</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxLoop</code></td>
<td>
<p>maximum number of iterations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>permutation.threshold</code></td>
<td>
<p>if use a permutation cutoff or not (bonferroni cutoff)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>permutation.rep</code></td>
<td>
<p>number of permutation replicates</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>memo</code></td>
<td>
<p>Character. A text marker on output files</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outpath</code></td>
<td>
<p>Effective only when file.output = TRUE, determines the path of the output file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>for color of points in each chromosome on manhattan plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file.output</code></td>
<td>
<p>whether to output files or not</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file.type</code></td>
<td>
<p>figure formats, "jpg", "tiff"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dpi</code></td>
<td>
<p>resolution for output figures</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>a cutoff line on manhattan plot, 0.05/marker size</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>whether to print detail.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Build date: Aug 30, 2017
Last update: Dec 14, 2018
</p>


<h3>Value</h3>

<p>a m * 2 matrix, the first column is the SNP effect, the second column is the P values
Output: MVP.return$map - SNP map information, SNP name, Chr, Pos
Output: MVP.return$glm.results - p-values obtained by GLM method
Output: MVP.return$mlm.results - p-values obtained by MLM method
Output: MVP.return$farmcpu.results - p-values obtained by FarmCPU method
</p>


<h3>Author(s)</h3>

<p>Lilin Yin, Haohao Zhang, and Xiaolei Liu
</p>


<h3>Examples</h3>

<pre><code class="language-R">
phePath &lt;- system.file("extdata", "07_other", "mvp.phe", package = "rMVP")
phenotype &lt;- read.table(phePath, header=TRUE)
print(dim(phenotype))
genoPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.desc", package = "rMVP")
genotype &lt;- attach.big.matrix(genoPath)
print(dim(genotype))
mapPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.map", package = "rMVP")
map &lt;- read.table(mapPath , head = TRUE)

opts &lt;- options(rMVP.OutputLog2File = FALSE)

mvp &lt;- MVP(phe=phenotype, geno=genotype, map=map, maxLoop=3,
  method=c("GLM", "MLM", "FarmCPU"), file.output=FALSE, ncpus=1)
str(mvp)

options(opts)

</code></pre>


</div>