<div class="container">

<table style="width: 100%;"><tr>
<td>MVP.Data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>MVP.Data: To prepare data for MVP package
Author: Xiaolei Liu, Lilin Yin and Haohao Zhang
Build date: Aug 30, 2016
Last update: Sep 12, 2018</h2>

<h3>Description</h3>

<p>MVP.Data: To prepare data for MVP package
Author: Xiaolei Liu, Lilin Yin and Haohao Zhang
Build date: Aug 30, 2016
Last update: Sep 12, 2018
</p>


<h3>Usage</h3>

<pre><code class="language-R">MVP.Data(
  fileMVP = NULL,
  fileVCF = NULL,
  fileHMP = NULL,
  fileBed = NULL,
  fileNum = NULL,
  fileMap = NULL,
  filePhe = NULL,
  fileInd = NULL,
  fileKin = NULL,
  filePC = NULL,
  out = "mvp",
  sep.num = "\t",
  auto_transpose = TRUE,
  sep.map = "\t",
  sep.phe = "\t",
  sep.kin = "\t",
  sep.pc = "\t",
  type.geno = "char",
  pheno_cols = NULL,
  SNP.impute = "Major",
  maxLine = 10000,
  pcs.keep = 5,
  verbose = TRUE,
  ncpus = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>fileMVP</code></td>
<td>
<p>Genotype in MVP format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fileVCF</code></td>
<td>
<p>Genotype in VCF format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fileHMP</code></td>
<td>
<p>Genotype in hapmap format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fileBed</code></td>
<td>
<p>Genotype in PLINK binary format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fileNum</code></td>
<td>
<p>Genotype in numeric format; pure 0, 1, 2 matrix; m * n, m is marker size, n is sample size</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fileMap</code></td>
<td>
<p>SNP map information, there are three columns, including SNP_ID, Chromosome, and Position</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filePhe</code></td>
<td>
<p>Phenotype, the first column is taxa name, the subsequent columns are traits</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fileInd</code></td>
<td>
<p>Individual name file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fileKin</code></td>
<td>
<p>Kinship that represents relationship among individuals, n * n matrix, n is sample size</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filePC</code></td>
<td>
<p>Principal components, n*npc, n is sample size, npc is number of top columns of principal components</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out</code></td>
<td>
<p>prefix of output file name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep.num</code></td>
<td>
<p>seperator for numeric file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>auto_transpose</code></td>
<td>
<p>Whether to automatically transpose numeric genotypes,
the default is True, which will identify the most one of the rows or 
columns as a marker, If set to False, the row represents the marker and
the column represents the individual.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep.map</code></td>
<td>
<p>seperator for map file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep.phe</code></td>
<td>
<p>seperator for phenotype file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep.kin</code></td>
<td>
<p>seperator for Kinship file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep.pc</code></td>
<td>
<p>seperator for PC file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type.geno</code></td>
<td>
<p>type parameter in bigmemory, genotype data. The default is char, it is highly recommended *NOT* to modify this parameter.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno_cols</code></td>
<td>
<p>Extract which columns of the phenotype file (including individual IDs)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SNP.impute</code></td>
<td>
<p>"Left", "Middle", "Right", or NULL for skip impute.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxLine</code></td>
<td>
<p>the number of markers handled at a time, smaller value would reduce the memory cost</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pcs.keep</code></td>
<td>
<p>how many PCs to keep</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>whether to print detail.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncpus</code></td>
<td>
<p>The number of threads used, if NULL, (logical core number - 1) is automatically used</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Compatible with DEPRECATED parameters.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>NULL
Output files:
genotype.desc, genotype.bin: genotype file in bigmemory format
phenotype.phe: ordered phenotype file, same taxa order with genotype file
map.map: SNP information
k.desc, k.bin: Kinship matrix in bigmemory format
pc.desc, pc.bin: PC matrix in bigmemory format
Requirement: fileHMP, fileBed, and fileNum can not input at the same time
</p>


<h3>Examples</h3>

<pre><code class="language-R">
bfilePath &lt;- file.path(system.file("extdata", "02_bfile", package = "rMVP"), "mvp")
opts &lt;- options(rMVP.OutputLog2File = FALSE)

MVP.Data(fileBed=bfilePath, out=tempfile("outfile"), ncpus=1)

options(opts)

</code></pre>


</div>