<div class="container">

<table style="width: 100%;"><tr>
<td>GLMnetwork</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Infer a network from RNA-seq expression.</h2>

<h3>Description</h3>

<p><code>GLMnetwork</code> infers a network from RNA-seq expression with the
log-linear Poisson graphical model of (Allen and Liu, 2012).
</p>


<h3>Usage</h3>

<pre><code class="language-R">GLMnetwork(counts, lambdas = NULL, normalize = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>counts</code></td>
<td>
<p>a n x p matrix of RNA-seq expression (numeric matrix or data
frame)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambdas</code></td>
<td>
<p>a sequence of decreasing positive numbers to control the
regularization (numeric vector). Default to <code>NULL</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normalize</code></td>
<td>
<p>logical value to normalize predictors in the log-linear
Poisson graphical model. If <code>TRUE</code>, log normalization and scaling are
performed prior the model is fit. Default to <code>TRUE</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>When input <code>lambdas</code> are null the default sequence of
<code>glmnet</code> for the first model (the one with the first
column of <code>count</code> as the target) is used.
</p>


<h3>Value</h3>

<p>S3 object of class <code>GLMnetwork</code>: a list consisting of
</p>
<table>
<tr style="vertical-align: top;">
<td><code>lambda</code></td>
<td>
<p> regularization parameters used for LLGM path(vector)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path</code></td>
<td>
<p> a list having the same length than <code>lambda</code>. It
contains the estimated coefficients (in a matrix) along the path</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Alyssa Imbert, <a href="mailto:alyssa.imbert@gmail.com">alyssa.imbert@gmail.com</a>
</p>
<p>Nathalie Vialaneix, <a href="mailto:nathalie.vialaneix@inrae.fr">nathalie.vialaneix@inrae.fr</a>
</p>


<h3>References</h3>

<p>Allen, G. and Liu, Z. (2012) A log-linear model for inferring
genetic networks from high-throughput sequencing data. In <em>Proceedings
of IEEE International Conference on Bioinformatics and Biomedecine (BIBM)</em>.
</p>


<h3>See Also</h3>

<p><code>stabilitySelection</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(lung)
lambdas &lt;- 4 * 10^(seq(0, -2, length = 10))
ref_lung &lt;- GLMnetwork(lung, lambdas = lambdas)

</code></pre>


</div>