<div class="container">

<table style="width: 100%;"><tr>
<td>imputedGLMnetwork</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Multiple hot-deck imputation and network inference from RNA-seq data.</h2>

<h3>Description</h3>

<p><code>imputedGLMnetwork</code> performs a multiple hot-deck imputation and infers a
network for each imputed dataset with a log-linear Poisson graphical model
(LLGM).
</p>


<h3>Usage</h3>

<pre><code class="language-R">imputedGLMnetwork(X, Y, sigma, m = 50, lambdas = NULL, B = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>n x p numeric matrix containing RNA-seq expression with missing rows
(numeric matrix or data frame)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>auxiliary dataset (n' x q numeric matrix or data frame)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma</code></td>
<td>
<p>affinity threshold for donor pool</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>
<p>number of replicates in multiple imputation (integer). Default to 50</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambdas</code></td>
<td>
<p>a sequence of decreasing positive numbers to control the
regularization (numeric vector). Default to <code>NULL</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B</code></td>
<td>
<p>number of iterations for stability selection. Default to 20</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>When input <code>lambdas</code> are null the default sequence of
<code>glmnet</code> for the first model (the one with the first
column of <code>count</code> as the target) is used. A common default sequence is
generated for all imputed datasets using this method.
</p>


<h3>Value</h3>

<p>S3 object of class <code>HDpath</code>: a list consisting of
</p>
<table>
<tr style="vertical-align: top;">
<td><code>path</code></td>
<td>
<p> a list of <code>m</code> data frames, each containing the
adjacency matrix of the inferred network obtained from the corresonding
imputed dataset. The regularization parameter is selected by StARS</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>efreq</code></td>
<td>
<p> a numeric matrix of size p x p,  which indicates the
number of times an edge has been predicted among the <code>m</code> inferred
networks</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Alyssa Imbert, <a href="mailto:alyssa.imbert@gmail.com">alyssa.imbert@gmail.com</a>
</p>
<p>Nathalie Vialaneix, <a href="mailto:nathalie.vialaneix@inrae.fr">nathalie.vialaneix@inrae.fr</a>
</p>


<h3>References</h3>

<p>Imbert, A., Valsesia, A., Le Gall, C., Armenise, C., Lefebvre,
G. Gourraud, P.A., Viguerie, N. and Villa-Vialaneix, N. (2018) Multiple
hot-deck imputation for network inference from RNA sequencing data.
<em>Bioinformatics</em>. <a href="https://doi.org/10.1093/bioinformatics/btx819">doi:10.1093/bioinformatics/btx819</a>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(lung)
data(thyroid)
nobs &lt;- nrow(lung)
miss_ind &lt;- sample(1:nobs, round(0.2 * nobs), replace = FALSE)
lung[miss_ind, ] &lt;- NA
lung &lt;- na.omit(lung)
lambdas &lt;- 4 * 10^(seq(0, -2, length = 10))
## Not run: 
lung_hdmi &lt;- imputedGLMnetwork(lung, thyroid, sigma = 2, lambdas = lambdas,
                               m = 10, B = 5)

## End(Not run)

</code></pre>


</div>