<div class="container">

<table style="width: 100%;"><tr>
<td>plot_chromosome</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot gene expression along a chromosome</h2>

<h3>Description</h3>

<p>Plot gene expression along a chromosome
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_chromosome(vsd, chr, scale = FALSE, trunc_val = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vsd</code></td>
<td>
<p>An object generated by <code>DESeq2::vst()</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>A string denoting a chromosome as annotated by ENSEMBL, e.g.
'1', '2', 'X', 'Y', 'MT'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale</code></td>
<td>
<p>Whether to scale the columns of the heatmap</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trunc_val</code></td>
<td>
<p>Truncate the expression matrix to this value prior to plotting. This is useful
if some very high expression values dominate the heatmap. By default, the heatmap is truncated
to expression values at most 3 standard deviations from the mean.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A Heatmap-class object of the <code>ComplexHeatmap</code> package that contains the heatmap of expression values.
</p>


<h3>Examples</h3>

<pre><code class="language-R">library("DESeq2")
chr1 &lt;- T47D[which(mcols(T47D)$chromosome=="1"),]
vsd &lt;- vst(chr1)
plot_chromosome(vsd, chr="1")

</code></pre>


</div>