<div class="container">

<table style="width: 100%;"><tr>
<td>gaston2neiGWAS</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert gaston's bed.matrix data to rNeighborGWAS genotype data.</h2>

<h3>Description</h3>

<p>A function convert a bed.matrix dataset to rNeighborGWAS genotype data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gaston2neiGWAS(x)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A bed.matrix created using the <code>gaston</code> package (Perdry &amp; Dandine-Roulland 2020).</p>
</td>
</tr></table>
<h3>Details</h3>

<p>This function converts genotype data into -1, 0, or 1 digit as the rNeighborGWAS format. Zero indicates heterozygotes.
</p>


<h3>Value</h3>

<p>A list including an individual x marker matrix, a data.frame including chromosome numbers in the first column, and SNP positions in the second column, and a numeric vector including phenotypes for individuals.
</p>

<ul>
<li>
<p><code>geno</code> Individual x marker matrix
</p>
</li>
<li>
<p><code>gmap</code> Data.frame including chromosome numbers in the first column, and SNP positions in the second column
</p>
</li>
<li>
<p><code>pheno</code> Numeric vector including phenotypes for individuals
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>References</h3>

<p>Perdry H, Dandine-Roulland C. (2020) gaston: Genetic Data Handling (QC, GRM, LD, PCA) &amp; Linear Mixed Models. https://CRAN.R-project.org/package=gaston
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("TTN", package="gaston")
x &lt;- gaston::as.bed.matrix(TTN.gen, TTN.fam, TTN.bim)
g &lt;- gaston2neiGWAS(x)
</code></pre>


</div>