<div class="container">

<table style="width: 100%;"><tr>
<td>int_neighbor</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Testing marker-by-marker epistasis in neighbor QTL effects</h2>

<h3>Description</h3>

<p>A function to test interaction terms between one focal marker and the other markers across a genome.
</p>


<h3>Usage</h3>

<pre><code class="language-R">int_neighbor(
  genoprobs,
  pheno,
  smap,
  scale,
  addcovar = NULL,
  addQTL,
  intQTL,
  grouping = rep(1, nrow(smap)),
  response = c("quantitative", "binary"),
  contrasts = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genoprobs</code></td>
<td>
<p>Conditional genotype probabilities as taken from <code>qtl::calc.genoprob()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p>A vector of individual phenotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>smap</code></td>
<td>
<p>A matrix showing a spatial map for individuals. The first and second column include spatial positions along an x-axis and y-axis, respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale</code></td>
<td>
<p>A numeric scalar indicating the maximum spatial distance between a focal individual and neighbors to define neighbor effects.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addcovar</code></td>
<td>
<p>An optional matrix including additional non-genetic covariates. It contains no. of individuals x no. of covariates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addQTL</code></td>
<td>
<p>A vector containing marker names that are considered covariates. This argument is necessary for <code>int_neighbor()</code>, and must match the marker names of <code>gmap</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>intQTL</code></td>
<td>
<p>A name of a focal marker to be tested for its epistasis with the other markers in neighbor effects. The marker name must be included by <code>addQTL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grouping</code></td>
<td>
<p>An optional integer vector assigning each individual to a group. This argument can be used when <code>smap</code> contains different experimental replicates. Default setting means that all individuals are belong to a single group.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>response</code></td>
<td>
<p>An optional argument to select trait types. The <code>"quantitative"</code> or <code>"binary"</code> calls the <code>"gaussian"</code> or <code>"binomial"</code> family in <code>glm()</code>, respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>contrasts</code></td>
<td>
<p>An optional vector composed of three TRUE/FALSE values, which represents the presence/absence of specific genotypes as c(TRUE/FALSE, TRUE/FALSE, TRUE/FALSE) = AA, AB, BB. If <code>NULL</code>, it is compiled from <code>genoprobs</code> automatically.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This is an optimal function to test two-way interactions between the main neighbor effect of a focal marker given by <code>intQTL</code> and the others.
All the main neighbor effects are first estimated using <code>eff_neighbor()</code>, and then a two-way interaction term between the focal marker effect and its counterpart was considered an additional explanatory variable.
LOD score was compared between models with or without the two-way interaction.
</p>


<h3>Value</h3>

<p>A matrix of LOD scores for neighbor epistasis effects, with the chromosome numbers and positions. The row names correspond to marker names.
</p>

<ul>
<li>
<p><code>chr</code> Chromosome number
</p>
</li>
<li>
<p><code>pos</code> Marker position
</p>
</li>
<li>
<p><code>LOD_int</code> LOD score for epistasis in neighbor effects between a focal and the other markers
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>See Also</h3>

<p><code>scan_neighbor</code> <code>eff_neighbor</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(1234)
test_map &lt;- qtl::sim.map(len=rep(20,5),n.mar=3,include.x=FALSE)
test_cross &lt;- qtl::sim.cross(test_map,n.ind=50)
test_smap &lt;- cbind(runif(50,1,100),runif(50,1,100))
test_genoprobs &lt;- qtl::calc.genoprob(test_cross,step=2)

test_int &lt;- int_neighbor(genoprobs=test_genoprobs,
                         pheno=test_cross$pheno$phenotype,
                         smap=test_smap,scale=20,
                         addQTL=c("c1_D1M1","c1_D1M2"),intQTL="c1_D1M1"
                         )
plot_nei(test_int, type="int")
</code></pre>


</div>