<div class="container">

<table style="width: 100%;"><tr>
<td>eff_neighbor</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimation of self and neighbor QTL effects across a genome</h2>

<h3>Description</h3>

<p>A function to estimate additive and dominance deviation for self and neighbor QTL effects by a simple regression.
</p>


<h3>Usage</h3>

<pre><code class="language-R">eff_neighbor(
  genoprobs,
  pheno,
  smap,
  scale,
  addcovar = NULL,
  addQTL = NULL,
  grouping = rep(1, nrow(smap)),
  response = c("quantitative", "binary"),
  fig = TRUE,
  contrasts = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genoprobs</code></td>
<td>
<p>Conditional genotype probabilities as taken from <code>qtl::calc.genoprob()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p>A vector of individual phenotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>smap</code></td>
<td>
<p>A matrix showing a spatial map for individuals. The first and second column include spatial position along an x-axis and y-axis, respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale</code></td>
<td>
<p>A numeric scalar indicating the maximum spatial distance between a focal individual and neighbors to define neighbor effects.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addcovar</code></td>
<td>
<p>An optional matrix including additional non-genetic covariates. It contains no. of individuals x no. of covariates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addQTL</code></td>
<td>
<p>An optional vector containing marker names that are considered covariates. Namely, this option allows composite interval mapping (Jansen 1993).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grouping</code></td>
<td>
<p>An optional integer vector assigning each individual to a group. This argument can be used when <code>smap</code> contains different experimental replicates. Default setting means that all individuals are belong to a single group.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>response</code></td>
<td>
<p>An optional argument to select trait types. The <code>"quantitative"</code> or <code>"binary"</code> calls the <code>"gaussian"</code> or <code>"binomial"</code> family in <code>glm()</code>, respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fig</code></td>
<td>
<p>TRUE/FALSE to plot the effects or not.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>contrasts</code></td>
<td>
<p>An optional vector composed of three TRUE/FALSE values, which represents the presence/absence of specific genotypes as c(TRUE/FALSE, TRUE/FALSE, TRUE/FALSE) = AA, AB, BB. If <code>NULL</code>, it is compiled from <code>genoprobs</code> automatically.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Similar to Haley-Knott regression (Haley &amp; Knott 1992), the additive and dominance deviations are approximated by a regression of trait values on conditional genotype probabilities.
The self QTL effects <code>a1</code> and <code>d1</code> are estimated in the same way as the <code>qtl</code> package performs the Haley-Knott regression.
If <code>contrasts = c(TRUE, TRUE, TRUE)</code>, neighbor QTL effects <code>a1</code> and <code>d1</code> are estimated using a quadratic regression; otherwise, the additive neighbor effects are estimated using a linear regression.
See also Sato, Takeda &amp; Nagano (2021) for the rationale behind the approximation.
</p>


<h3>Value</h3>

<p>A matrix of estimated additive and dominance deviation for self and neighbor effects, with the chromosome numbers and positions. The row names correspond to marker names.
</p>

<ul>
<li>
<p><code>chr</code> Chromosome number
</p>
</li>
<li>
<p><code>pos</code> Marker position
</p>
</li>
<li>
<p><code>a1</code> Additive deviation for self effects
</p>
</li>
<li>
<p><code>d1</code> Dominance deviation for self effects
</p>
</li>
<li>
<p><code>a2</code> Additive deviation for neighbor effects
</p>
</li>
<li>
<p><code>d2</code> Dominance deviation for neighbor effects
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>References</h3>


<ul>
<li>
<p> Haley CS, Knott SA (1992) A simple regression method for mapping quantitative trait loci in line crosses using flanking markers. Heredity 69:315-324.
</p>
</li>
<li>
<p> Jansen RC (1993) Interval mapping of multiple quantitative trait loci. Genetics 135:205-211.
</p>
</li>
<li>
<p> Sato Y, Takeda K, Nagano AJ (2021) Neighbor QTL: an interval mapping method for quantitative trait loci underlying plant neighborhood effects. G3; Genes|Genomes|Genetics 11:jkab017.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">set.seed(1234)
test_map &lt;- qtl::sim.map(len=rep(20,5),n.mar=3,include.x=FALSE)
test_cross &lt;- qtl::sim.cross(test_map,n.ind=50)
test_smap &lt;- cbind(runif(50,1,100),runif(50,1,100))
test_genoprobs &lt;- qtl::calc.genoprob(test_cross,step=2)

test_eff &lt;- eff_neighbor(genoprobs=test_genoprobs,
                         pheno=test_cross$pheno$phenotype,
                         smap=test_smap, scale=20, fig=TRUE
                         )
</code></pre>


</div>