<div class="container">

<table style="width: 100%;"><tr>
<td>rNeighborQTL-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>rNeighbor QTL: quantitative trait loci mapping for neighbor effects</h2>

<h3>Description</h3>

<p>The "rNeighborQTL" package provides a set of functions to perform a genome scan of neighbor QTL effects.
Theoretical background is described in Sato, Takeda &amp; Nagano (2021).
See <code>vignette("rNeighborQTL")</code> for how to use this package.
</p>


<h3>Details</h3>

<p>Neighbor QTL is an extended method of single-marker regression.
Taking conditional genotype probabilities from <code>r/qtl</code> package, the <code>rNeighborQTL</code> package performs an interval mapping of neighbor genotypic identity.
Alternative to QTL mapping, <code>rNeighborGWAS</code> package is also available for genome-wide association studies (GWAS) via CRAN at <a href="https://cran.r-project.org/package=rNeighborGWAS">https://cran.r-project.org/package=rNeighborGWAS</a>.
</p>


<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>References</h3>


<ul>
<li>
<p> Sato Y, Yamamoto E, Shimizu KK, Nagano AJ (2021) Neighbor GWAS: incorporating neighbor genotypic identity into genome-wide association studies of field herbivory. Heredity 126:597â€“614.
</p>
</li>
<li>
<p> Sato Y, Takeda K, Nagano AJ (2021) Neighbor QTL: an interval mapping method for quantitative trait loci underlying plant neighborhood effects. G3; Genes|Genomes|Genetics 11:jkab017.
</p>
</li>
</ul>
</div>