<div class="container">

<table style="width: 100%;"><tr>
<td>RNentropy-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Entropy Based Method for the Detection of Significant Variation in Gene Expression Data
</h2>

<h3>Description</h3>

<p>An implementation of a method based on information theory devised 
    for the identification of genes showing a significant variation of expression
    across multiple conditions. Given expression estimates from any number of 
    RNA-Seq samples and conditions it identifies genes or transcripts with a 
    significant variation of expression across all the conditions studied, 
    together with the samples in which they are over- or under-expressed. 
    Zambelli et al. (2018) &lt;doi:10.1093/nar/gky055&gt;.
</p>


<h3>Author(s)</h3>

<p>NA
</p>
<p>Maintainer: Federico Zambelli &lt;federico.zambelli@unimi.it&gt;
</p>


<h3>References</h3>

<p>doi = 10.1093/nar/gky055
doi = 10.1007/978-1-0716-1307-8_6
</p>


<h3>Examples</h3>

<pre><code class="language-R">#load expression values and experiment design
data("RN_Brain_Example_tpm", "RN_Brain_Example_design")
#compute statistics and p-values (considering only a subset of genes due to
#examples running time limit of CRAN). 
Results &lt;- RN_calc(RN_Brain_Example_tpm[1:10000,], RN_Brain_Example_design)
#select only genes with significant changes of expression
Results &lt;- RN_select(Results)
#Compute the Point Mutual information Matrix
Results &lt;- RN_pmi(Results)

#load expression values and experiment design
data("RN_BarresLab_FPKM", "RN_BarresLab_design")
#compute statistics and p-values (considering only a subset of genes due to
#examples running time limit of CRAN)
Results_B &lt;- RN_calc(RN_BarresLab_FPKM[1:10000,], RN_BarresLab_design)
#select only genes with significant changes of expression
Results_B &lt;- RN_select(Results_B)
#Compute the Point Mutual information matrix
Results_B &lt;- RN_pmi(Results_B)
</code></pre>


</div>