<div class="container">

<table style="width: 100%;"><tr>
<td>showdown</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimator Showdown by Monte-Carlo Study.</h2>

<h3>Description</h3>

<p>The function <code>showdown</code> can be used to perform Monte-Carlo studies 
comparing a competitor with rmx estimators in case of normal location and scale. 
In addition, maximum likelihood (ML) estimators (mean and sd) and median and 
MAD are computed. The comparison is based on the empirical MSE.
</p>


<h3>Usage</h3>

<pre><code class="language-R">showdown(n, M, eps, contD, seed = 123, estfun, estMean, estSd,
         eps.lower = 0, eps.upper = 0.05, steps = 3L, fsCor = TRUE, 
         plot1 = FALSE, plot2 = FALSE, plot3 = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>integer; sample size, should be at least 3.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>M</code></td>
<td>
<p>integer; Monte-Carlo replications.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eps</code></td>
<td>
<p>amount of contamination in [0, 0.5].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>contD</code></td>
<td>
<p>object of class <code>"UnivariateDistribution"</code>; contaminating distribution.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>random seed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estfun</code></td>
<td>
<p>function to compute location and scale estimator; see details below.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estMean</code></td>
<td>
<p>function to compute location estimator; see details below.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estSd</code></td>
<td>
<p>function to compute scale estimator; see details below.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eps.lower</code></td>
<td>
<p>used by rmx estimator.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eps.upper</code></td>
<td>
<p>used by rmx estimator.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>steps</code></td>
<td>
<p>integer; steps used for estimator construction.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fsCor</code></td>
<td>
<p>logical; use finite-sample correction.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot1</code></td>
<td>
<p>logical; plot cdf of ideal and real distribution.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot2</code></td>
<td>
<p>logical; plot 20 (or M if M &lt; 20) randomly selected samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot3</code></td>
<td>
<p>logical; generate boxplots of the results.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Normal location and scale with mean = 0 and sd = 1 is used as ideal model (without
restriction due to equivariance).
</p>
<p>Since there is no estimator which yields reliable results if 50 percent or more of the
observations are contaminated, we use a modification where we re-simulate all samples
including at least 50 percent contaminated data.
</p>
<p>If <code>estfun</code> is specified it has to compute and return a location and scale estimate
(vector of length 2). One can also specify the location and scale estimator separately 
by using <code>estMean</code> and <code>estSd</code> where <code>estMean</code> computes and returns
the location estimate and <code>estSd</code> the scale estimate.
</p>
<p>We use funtion <code>rowRoblox</code> for the computation of the rmx estimator.
</p>


<h3>Value</h3>

<p>Data.frame including empirical MSE (standardized by sample size n) and
relMSE with respect to the rmx estimator.
</p>


<h3>Author(s)</h3>

<p>Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a></p>


<h3>References</h3>

<p>M. Kohl (2005). Numerical Contributions to the Asymptotic Theory of Robustness.
Dissertation. University of Bayreuth. <a href="https://epub.uni-bayreuth.de/id/eprint/839/2/DissMKohl.pdf">https://epub.uni-bayreuth.de/id/eprint/839/2/DissMKohl.pdf</a>.
</p>
<p>H. Rieder (1994): Robust Asymptotic Statistics. Springer. <a href="https://doi.org/10.1007/978-1-4684-0624-5">doi:10.1007/978-1-4684-0624-5</a>
</p>
<p>H. Rieder, M. Kohl, and P. Ruckdeschel (2008). The Costs of Not Knowing the Radius.
Statistical Methods and Applications  <em>17</em>(1): 13-40. <a href="https://doi.org/10.1007/s10260-007-0047-7">doi:10.1007/s10260-007-0047-7</a>
</p>
<p>M. Kohl, P. Ruckdeschel, and H. Rieder (2010). Infinitesimally Robust Estimation in 
General Smoothly Parametrized Models. Statistical Methods and Applications <em>19</em>(3): 333-354.
<a href="https://doi.org/10.1007/s10260-010-0133-0">doi:10.1007/s10260-010-0133-0</a>.
</p>
<p>M. Kohl and H.P. Deigner (2010). Preprocessing of gene expression data by 
optimally robust estimators. BMC Bioinformatics <em>11</em>, 583.
<a href="https://doi.org/10.1186/1471-2105-11-583">doi:10.1186/1471-2105-11-583</a>.
</p>


<h3>See Also</h3>

<p><code>rowRoblox</code></p>


<h3>Examples</h3>

<pre><code class="language-R">library(MASS)
## compare with Huber's Proposal 2
showdown(n = 20, M = 100, eps = 0.02, contD = Norm(mean = 3, sd = 3), 
         estfun = function(x){ unlist(hubers(x)) },
         plot1 = TRUE, plot2 = TRUE, plot3 = TRUE)

## compare with Huber M estimator with MAD scale
showdown(n = 20, M = 100, eps = 0.02, contD = Norm(mean = 3, sd = 3), 
         estfun = function(x){ unlist(huber(x)) },
         plot1 = TRUE, plot2 = TRUE, plot3 = TRUE)
</code></pre>


</div>