<div class="container">

<table style="width: 100%;"><tr>
<td>nci60</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>NCI-60 cancer cell panel</h2>

<h3>Description</h3>

<p>The data set is a pre-processed version of the NCI-60 cancer cell panel as
used in Alfons, Croux &amp; Gelper (2013).  One observation was removed since
all values in the gene expression data were missing.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data("nci60")
</code></pre>


<h3>Format</h3>

<p>Protein and gene expression data on 59 observations are stored in two
separate matrices:
</p>

<dl>
<dt><code>protein</code></dt>
<dd>
<p>a matrix containing protein expressions based on
antibodies (162 columns), acquired via reverse-phase protein lysate
arrays and log2 transformed.</p>
</dd>
<dt><code>gene</code></dt>
<dd>
<p>a matrix containing gene expression data (22283
columns), obtained with an Affymetrix HG-U133A chip and normalized
with the GCRMA method.</p>
</dd>
</dl>
<p>In addition, meta information on the proteins, genes, and cancer cell lines
is stored in three separate data frames:
</p>

<dl>
<dt><code>proteinInfo</code></dt>
<dd>
<p>a data frame with 162 rows and the following 4
columns: <code>Experiment</code> (the name of the experiment for collecting
the data), <code>Probe</code> (the name of the individual probe), <code>Symbol</code>
(the symbol of the protein in Human Genome Organisation (HUGO)
nomenclature), and <code>ID</code> (identifier of the protein per the National
Center for Biotechnology Information (NCBI) Entrez database).  The rows of
this data frame correspond to the columns of the matrix <code>protein</code>.</p>
</dd>
<dt><code>geneInfo</code></dt>
<dd>
<p>a data frame with 22283 rows and the following 4
columns: <code>Experiment</code> (the name of the experiment for collecting
the data), <code>Probe</code> (the name of the individual probe), <code>Symbol</code>
(the symbol of the gene in Human Genome Organisation (HUGO) nomenclature),
and <code>ID</code> (identifier of the gene per the National Center for
Biotechnology Information (NCBI) Entrez database).  The rows of this
data frame correspond to the columns of the matrix <code>gene</code>.</p>
</dd>
<dt><code>cellLineInfo</code></dt>
<dd>
<p>a data frame with 59 rows and 15 columns
containing various information on the cancer cell lines, such as tissue of
origin and histology, or age and sex of the patient. The rows of this data
frame correspond to the rows of the matrices <code>protein</code> and
<code>gene</code>.</p>
</dd>
</dl>
<h3>Source</h3>

<p>The original data were downloaded from
<a href="https://discover.nci.nih.gov/cellminer/">https://discover.nci.nih.gov/cellminer/</a> on 2012-01-27.
</p>
<p>The exact version of the data used in Alfons, Croux &amp; Gelper (2013) can be
obtained from <a href="https://github.com/aalfons/nci60">https://github.com/aalfons/nci60</a>, together with the
script for pre-processing.  The data in package <span class="pkg">robustHD</span> differ in
that the matrix of the gene expressions is called <code>gene</code> and that they
include the three data frames with meta information on proteins, genes, and
cancer cell lines.
</p>


<h3>References</h3>

<p>Reinhold, W.C., Sunshine, M., Liu, H., Varma, S., Kohn, K.W., Morris, J.,
Doroshow, J. and Pommier, Y. (2012) CellMiner: A Web-Based Suite of Genomic
and Pharmacologic Tools to Explore Transcript and Drug Patterns in the
NCI-60 Cell Line Set. <em>Cancer Research</em>, <b>72</b>(14), 3499–3511.
<a href="https://doi.org/10.1158/0008-5472.CAN-12-1370">doi:10.1158/0008-5472.CAN-12-1370</a>
</p>
<p>Alfons, A., Croux, C. and Gelper, S. (2013) Sparse least trimmed squares
regression for analyzing high-dimensional large data sets. <em>The Annals
of Applied Statistics</em>, <b>7</b>(1), 226–248. <a href="https://doi.org/10.1214/12-AOAS575">doi:10.1214/12-AOAS575</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# load data
data("nci60")
# define response variable
y &lt;- protein[, 92]
# screen most correlated predictor variables
correlations &lt;- apply(gene, 2, corHuber, y)
keep &lt;- partialOrder(abs(correlations), 100, decreasing = TRUE)
X &lt;- gene[, keep]

</code></pre>


</div>