<div class="container">

<table style="width: 100%;"><tr>
<td>TipInstability</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Tip instability</h2>

<h3>Description</h3>

<p><code>TipInstability()</code> calculates the instability of each leaf in a tree.
Unstable leaves are likely to display roguish behaviour.
</p>


<h3>Usage</h3>

<pre><code class="language-R">TipInstability(
  trees,
  log = TRUE,
  average = "mean",
  deviation = "sd",
  checkTips = TRUE,
  parallel = FALSE
)

ColByStability(
  trees,
  log = TRUE,
  average = "mean",
  deviation = "sd",
  pal = hcl.colors(131, "inferno")[1:101]
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>trees</code></td>
<td>
<p>List of trees to analyse.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log</code></td>
<td>
<p>Logical specifying whether to log-transform distances when
calculating leaf stability.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>average</code></td>
<td>
<p>Character specifying whether to use <code>"mean"</code> or <code>"median"</code>
tip distances to calculate leaf stability.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>deviation</code></td>
<td>
<p>Character specifying whether to use <code>"sd"</code> or <code>"mad"</code> to
calculate leaf stability.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>checkTips</code></td>
<td>
<p>Logical specifying whether to check that tips are numbered
consistently.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel</code></td>
<td>
<p>Logical specifying whether parallel execution should take
place in C++.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pal</code></td>
<td>
<p>A vector listing a sequence of colours to be used for plotting.
The earliest entries will be assigned to the most stable tips.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Smith (2022) defines the <em>instability</em> of a pair
of leaves as the median absolute divergence in the graph geodesic
(the number of edges in the shortest path between the leaves) across all
trees, normalized against the mean graph geodesic.
The instability of a single leaf is the mean instability of all pairs that
include that leaf; higher values characterise leaves whose position is more
variable between trees.
</p>
<p>Other concepts of leaf instability include
</p>

<ul>
<li>
<p> The "taxonomic instability index", as implemented in Mesquite:
described by Thomson and Shaffer (2010) as
<code class="reqn">\sum\limits_{(x, y), j \neq i}{\frac{|D~ijx~ - D~ijy~|}{(D~ijx~ - D~ijy~)^2}}</code>,
where <code class="reqn">D~ijx~</code> is the patristic distance (i.e. length of edges)
between leaves <code class="reqn">i</code> and <code class="reqn">j</code> in tree <code class="reqn">x</code>.
</p>
</li>
<li>
<p> the average stability of triplets (i.e. quartets including the root) that
include the leaf (Thorley and Wilkinson 1999), implemented in "Phyutility"
(Smith and Dunn 2008); and related to "positional congruence"
measures (Estabrook 1992; Pol and Escapa 2009).
</p>
</li>
</ul>
<h3>Value</h3>

<p><code>ColByStability()</code> returns a named character vector that assigns a
colour to each leaf in <code>trees</code> according to their stability.
</p>


<h3>Author(s)</h3>

<p><a href="https://smithlabdurham.github.io/">Martin R. Smith</a>
(<a href="mailto:martin.smith@durham.ac.uk">martin.smith@durham.ac.uk</a>)
</p>


<h3>References</h3>

<p>Estabrook GF (1992).
“Evaluating undirected positional congruence of individual taxa between two estimates of the phylogenetic tree for a group of taxa.”
<em>Systematic Biology</em>, <b>41</b>(2), 172–177.
<a href="https://doi.org/10.1093/sysbio/41.2.172">doi:10.1093/sysbio/41.2.172</a>.<br><br> Pol D, Escapa IH (2009).
“Unstable taxa in cladistic analysis: identification and the assessment of relevant characters.”
<em>Cladistics</em>, <b>25</b>(5), 515–527.
<a href="https://doi.org/10.1111/j.1096-0031.2009.00258.x">doi:10.1111/j.1096-0031.2009.00258.x</a>.<br><br> Smith MR (2022).
“Using information theory to detect rogue taxa and improve consensus trees.”
<em>Systematic Biology</em>, <b>71</b>(5), 986–1008.
<a href="https://doi.org/10.1093/sysbio/syab099">doi:10.1093/sysbio/syab099</a>.<br><br> Smith SA, Dunn CW (2008).
“Phyutility: a phyloinformatics tool for trees, alignments and molecular data.”
<em>Bioinformatics</em>, <b>24</b>(5), 715–716.
<a href="https://doi.org/10.1093/bioinformatics/btm619">doi:10.1093/bioinformatics/btm619</a>.<br><br> Thomson RC, Shaffer HB (2010).
“Sparse supermatrices for phylogenetic inference: taxonomy, alignment, rogue taxa, and the phylogeny of living turtles.”
<em>Systematic Biology</em>, <b>59</b>(1), 42–58.
<a href="https://doi.org/10.1093/sysbio/syp075">doi:10.1093/sysbio/syp075</a>.<br><br> Thorley JL, Wilkinson M (1999).
“Testing the phylogenetic stability of early tetrapods.”
<em>Journal of Theoretical Biology</em>, <b>200</b>(3), 343–344.
<a href="https://doi.org/10.1006/jtbi.1999.0999">doi:10.1006/jtbi.1999.0999</a>.
</p>


<h3>See Also</h3>

<p>Other tip instability functions: 
<code>TipVolatility()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">library("TreeTools", quietly = TRUE)

# Generate some trees with a rogue taxon
trees &lt;- AddTipEverywhere(BalancedTree(8), "Rogue")[3:6]

# Display the strict consensus
plot(consensus(trees), tip.col = ColByStability(trees))

# Add a legend for the colour scale used
PlotTools::SpectrumLegend(
  "bottomleft", bty = "n", # No box
  legend = c("Unstable", "", "Stable"),
  palette = hcl.colors(131, "inferno")[1:101]
)

# Calculate leaf stability
instab &lt;- TipInstability(trees, log = FALSE, ave = "mean", dev = "mad")

# Plot a consensus that omits the least stable leaves
plot(ConsensusWithout(trees, names(instab[instab &gt; 0.2])))
</code></pre>


</div>