<div class="container">

<table style="width: 100%;"><tr>
<td>diffuseMat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generic diffusion function</h2>

<h3>Description</h3>

<p>Generic diffusion function using automated
individualised sigma calculation
</p>


<h3>Usage</h3>

<pre><code class="language-R">diffuseMat(data, ndims = 20, nsig = 5, removeFirst = TRUE,
  useARPACK = TRUE, distfun = NULL, sigmas = NULL, sqdistmat = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>Matrix of data with genes in rows and cells in columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ndims</code></td>
<td>
<p>Number of dimensions to return</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nsig</code></td>
<td>
<p>For automatic sigma calculation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>removeFirst</code></td>
<td>
<p>Default TRUE. Removes the first eigenvector</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useARPACK</code></td>
<td>
<p>Default TRUE. Uses Arnoldi algorithm for eignvector calculations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>distfun</code></td>
<td>
<p>A different distance function that returns the <strong>squared</strong>
distance</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigmas</code></td>
<td>
<p>Manually provide sigma</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sqdistmat</code></td>
<td>
<p><strong>Squared</strong> distance matrix.
Give your own squared distance matrix.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Generic diffusion function using automated individualised sigma calculation.
</p>
<p>A Gaussian kernel is applied to the chosen distance metric producing
an <code class="reqn">n \times n</code> square unnormalised symmetric transition matrix, <code class="reqn">A</code>.
Let <code class="reqn">D</code> be an <code class="reqn">n \times n</code> diagonal matrix with row(column) sums of
<code class="reqn">A</code> as entries. The density corrected transition matrix will now
be:
</p>
<p style="text-align: center;"><code class="reqn">D^{-1} A D^{-1}</code>
</p>

<p>and can be normalised:
</p>
<p style="text-align: center;"><code class="reqn">B^{-1} D^{-1} A D^{-1}</code>
</p>

<p>where <code class="reqn">B</code> is an <code class="reqn">n \times n</code> diagonal matrix with row sums of
the density corrected transition matrix as entries. The eigen decomposition of
this matrix can be simplified by solving the symmetric system:
</p>
<p style="text-align: center;"><code class="reqn">B^{-\frac{1}{2}} D^{-1} A D^{-1} B^{-\frac{1}{2}} R^\prime = %
      R^\prime \lambda^\prime</code>
</p>

<p>where <code class="reqn">R^\prime</code> is a matrix of the right eigenvectors that solve
the system and <code class="reqn">\lambda^\prime</code> is the corresponding eigenvalue
diagonal matrix. Now the solution of:
</p>
<p style="text-align: center;"><code class="reqn">B^{-1} D^{-1} A D^{-1} R = R \lambda</code>
</p>

<p>in terms of <code class="reqn">R^\prime</code> and <code class="reqn">B^{-\frac{1}{2}}</code> is:
</p>
<p style="text-align: center;"><code class="reqn">B^{-1} D^{-1} A D^{-1} B^{-\frac{1}{2}} R^\prime = %
      B^{-\frac{1}{2}} R^\prime \lambda^\prime</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">R = B^{-\frac{1}{2}} R^\prime</code>
</p>

<p>This <code class="reqn">R</code> without the first eigen vector is returned as the diffusion map.
</p>


<h3>Value</h3>

<p>List output containing:
</p>

<table>
<tr>
<td style="text-align: right;">
  <em>values</em> </td>
<td style="text-align: left;"> Eigenvalues, excluding the first eigenvalue, which should
                      always be 1.</td>
</tr>
<tr>
<td style="text-align: right;">
   <em>vectors</em> </td>
<td style="text-align: left;"> Matrix of eigen vectors in columns, first eigen vector
                       removed.</td>
</tr>
<tr>
<td style="text-align: right;">
   <em>nconv</em> </td>
<td style="text-align: left;"> Number of eigen vectors/values that converged.</td>
</tr>
<tr>
<td style="text-align: right;">
   <em>niter</em> </td>
<td style="text-align: left;"> Iterations taken for Arnoldi algorithm to converge.</td>
</tr>
<tr>
<td style="text-align: right;">
   <em>nops</em> </td>
<td style="text-align: left;">  Number of operations. </td>
</tr>
<tr>
<td style="text-align: right;">
   <em>val0</em> </td>
<td style="text-align: left;"> 1st eigen value - should be 1. If not be suspicious!</td>
</tr>
<tr>
<td style="text-align: right;">
   <em>vec0</em> </td>
<td style="text-align: left;"> 1st eigen vector - should be <code class="reqn">n^{-\frac{1}{2}}</code>,
                     where n is the number of cells/samples.</td>
</tr>
<tr>
<td style="text-align: right;">
   <em>usedARPACK</em> </td>
<td style="text-align: left;"> Predicates use of ARPACK for spectral decomposition.</td>
</tr>
<tr>
<td style="text-align: right;">
   <em>distfun</em> </td>
<td style="text-align: left;"> Function used to calculate the squared distance.</td>
</tr>
<tr>
<td style="text-align: right;">
   <em>nn</em> </td>
<td style="text-align: left;"> Number of nearest neighbours used for calculating <code>sigmas</code>.</td>
</tr>
<tr>
<td style="text-align: right;">
   <em>d2</em> </td>
<td style="text-align: left;"> Matrix of squared distances, returned from <code>distfun</code>.</td>
</tr>
<tr>
<td style="text-align: right;">
   <em>sigmas</em> </td>
<td style="text-align: left;"> Vector of sigmas. Same length as number of cells if individual</td>
</tr>
<tr>
<td style="text-align: right;">
                 </td>
<td style="text-align: left;"> sigmas were calculated, otherwise a scalar if was supplied.</td>
</tr>
<tr>
<td style="text-align: right;">
   <em>gaussian</em> </td>
<td style="text-align: left;"> Unnormalised transition matrix after applying Gaussian.</td>
</tr>
<tr>
<td style="text-align: right;">
   <em>markov</em>  </td>
<td style="text-align: left;"> Normalised <code>gaussian</code> matrix.</td>
</tr>
<tr>
<td style="text-align: right;">
   <em>densityCorrected</em> </td>
<td style="text-align: left;"> Matrix after applying density correction to 
                                <code>markov</code>.</td>
</tr>
<tr>
<td style="text-align: right;">
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Wajid Jawaid
</p>


<h3>References</h3>

<p>Haghverdi, L., Buettner, F., Theis, F.J., 2015. Diffusion maps for high-dimensional single-cell analysis of differentiation data. Bioinformatics 31, 2989–2998.
</p>
<p>Haghverdi, L., Büttner, M., Wolf, F.A., Buettner, F., Theis, F.J., 2016. Diffusion pseudotime robustly reconstructs lineage branching. Nat Meth 13, 845–848.
</p>
<p>Angerer, P., Haghverdi, L., Büttner, M., Theis, F.J., Marr, C., Buettner, F., 2016. destiny: diffusion maps for large-scale single-cell data in R. Bioinformatics 32, 1241–1243.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
xx &lt;- diffuseMat(x)

## End(Not run)
</code></pre>


</div>