<div class="container">

<table style="width: 100%;"><tr>
<td>phylosignal_sub_network</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Compute clade-specific phylogenetic signals in a bipartite interaction network
</h2>

<h3>Description</h3>

<p>This function computes the clade-specific phylogenetic signals in species interactions. For each node of tree A having a certain number of descending species, it computes the phylogenetic signal in the resulting sub-network by performing a Mantel test between the phylogenetic distances and the ecological distances for the given sub-clade of tree A. 
Mantel tests can be computed using quantified or binary networks, with the Jaccard, Bray-Curtis, or UniFrac ecological distances.
</p>


<h3>Usage</h3>

<pre><code class="language-R">phylosignal_sub_network(network, tree_A, tree_B = NULL, 
method = "Jaccard_weighted", nperm = 1000, 
correlation = "Pearson", minimum = 10, degree = FALSE, 
permutation = "shuffle")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>network</code></td>
<td>

<p>a matrix representing the bipartite interaction network with species from guild A in columns and species from guild B in rows. Row names (resp. columns names) must correspond to the tip labels of tree B (resp. tree A).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree_A</code></td>
<td>

<p>a phylogenetic tree of guild A (the columns of the interaction network). It must be an object of class "phylo".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree_B</code></td>
<td>

<p>(optional) a phylogenetic tree of guild B (the rows of the interaction network). It must be an object of class "phylo".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>

<p>indicates which method is used to compute the phylogenetic signal in species interactions using Mantel tests. You can choose "Jaccard_weighted" (default) for computing ecological distances using Jaccard dissimilarities (or "Jaccard_binary" to not take into account the abundances of the interactions), "Bray-Curtis" for computing the Bray-Curtis dissimilarity, or "GUniFrac" for computing the weighted (or generalized) UniFrac distances ("UniFrac_unweighted" to not take into account the interaction abundances).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>correlation</code></td>
<td>

<p>indicates which correlation (R) must be used in the Mantel test, among Pearson (default), Spearman, and Kendall correlations.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nperm</code></td>
<td>

<p>a number of permutations to evaluate the significance of the Mantel test. By default, it equals 10,000, but this can be very long for the Kendall correlation.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>permutation</code></td>
<td>

<p>(optional) indicates which permutations must be performed to evaluate the significance of the Mantel correlation: either "shuffle" (by default - i.e. random shufflying of the distance matrix) or "nbpartners" (i.e. keeping constant the number of partners per species and shuffling at random their identity).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minimum</code></td>
<td>

<p>indicates the minimal number of descending species for a node in tree A to compute its clade-specific phylogenetic signal. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>degree</code></td>
<td>

<p>if degree=TRUE, Mantel tests testing for phylogenetic signal in the number of partners are additionally performed in each sub-clade.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>See the tutorial on GitHub (https://github.com/BPerezLamarque/Phylosignal_network). 
</p>


<h3>Value</h3>

<p>For Mantel tests, the function outputs a table where each line corresponds to a tested clade and which contains at least 8 columns: the name of the node ("node"), the number of species in the sub-clade A ("nb_A"), the number of species in guild B associated with the sub-clade A ("nb_B"), the Mantel correlation for guild A ("mantel_cor"), its associated upper p-value ("pvalue_upper"), its associated lower p-value ("pvalue_lower"), and the associated Bonferroni corrected p-values ("pvalue_upper_corrected" and "pvalue_lower_corrected"). 
</p>
<p>"mantel_cor" indicates the strength of the phylogenetic signal in the sub-clade A. The upper p-value "pvalue_upper" indicates the significance of the phylogenetic signal in the sub-clade A. The lower p-value "pvalue_lower" indicates the significance of the anti-phylogenetic signal in the sub-clade A. Both Bonferroni p-values are corrected using the number of tested nodes. 
For instance, if "pvalue_upper_corrected"&lt;0.05 for a given node, there is a significant phylogenetic signal in the corresponding sub-clade of A.
</p>
<p>If degree=TRUE, it also indicates in each sub-clade, the phylogenetic signal in the number of partners ("degree_mantel_cor") and its significance with or without the Bonferroni correction ("degree_pvalue_upper", "degree_pvalue_lower" and "degree_pvalue_upper_corrected", "degree_pvalue_lower_corrected")
</p>


<h3>Author(s)</h3>

<p>Benoît Perez-Lamarque</p>


<h3>References</h3>

<p>Perez-Lamarque B, Maliet O, Pichon B, Selosse M-A, Martos F, Morlon H. 2022. Do closely related species interact with similar partners? Testing for phylogenetic signal in bipartite interaction networks. bioRxiv, 2021.08.30.458192, ver. 6 peer- reviewed and recommended by Peer Community in Evolutionary Biology. https://doi.org/10.1101/2021.08.30.458192
</p>
<p>Goslee, S.C. &amp; Urban, D.L. (2007). The ecodist package for dissimilarity-based analysis of ecological data. J. Stat. Softw., 22, 1–19.
</p>
<p>Chen, J., Bittinger, K., Charlson, E.S., Hoffmann, C., Lewis, J., Wu, G.D., et al. (2012). Associating microbiome composition with environmental covariates using generalized UniFrac distances. Bioinformatics, 28, 2106–2113.
</p>


<h3>See Also</h3>

<p><code>phylosignal_sub_network</code>
<code>plot_phylosignal_sub_network</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Load the data
data(mycorrhizal_network)

network &lt;- mycorrhizal_network[[1]] # interaction matrix 
tree_orchids &lt;- mycorrhizal_network[[2]] # phylogenetic tree (phylo object)
tree_fungi &lt;- mycorrhizal_network[[3]] # phylogenetic tree (phylo object)


if(test){

# Clade-specific phylogenetic signal in species interactions in guild A 
# (do closely related species interact with similar partners in sub-clades of guild A?)

results_clade_A &lt;- phylosignal_sub_network(network, tree_A = tree_orchids, tree_B = tree_fungi,
method = "GUniFrac", correlation = "Pearson", degree = TRUE)
plot_phylosignal_sub_network(tree_A = tree_orchids, results_clade_A, network)

# Clade-specific phylogenetic signal in species interactions in guild B 
# (do closely related species interact with similar partners in sub-clades of guild B?)

results_clade_B &lt;- phylosignal_sub_network(t(network), tree_A = tree_fungi, tree_B = tree_orchids, 
method = "GUniFrac", correlation = "Pearson", degree = TRUE)
plot_phylosignal_sub_network(tree_A = tree_fungi, results_clade_B, t(network))
}

</code></pre>


</div>