<div class="container">

<table style="width: 100%;"><tr>
<td>plot_phylosignal_sub_network</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Plot clade-specific phylogenetic signals in a bipartite interaction network
</h2>

<h3>Description</h3>

<p>This function plots the clade-specific phylogenetic signals in species interactions. For each node of tree A having a certain number of descending species, it represents the phylogenetic signal in the resulting sub-network by performing a Mantel test between the phylogenetic distances and the ecological distances for the given sub-clade of tree A. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_phylosignal_sub_network(tree_A, results_sub_clades, network, legend=TRUE, 
show.tip.label=FALSE, where="bottomleft")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>tree_A</code></td>
<td>

<p>a phylogenetic tree of guild A (the columns of the interaction network). It must be an object of class "phylo".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>results_sub_clades</code></td>
<td>

<p>output of the function phylosignal_sub_network.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>network</code></td>
<td>

<p>a matrix representing the bipartite interaction network with species from guild A in columns and species from guild B in rows. Row names (resp. columns names) must correspond to the tip labels of tree B (resp. tree A).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend</code></td>
<td>

<p>indicates whether the legend should be plotted. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.tip.label</code></td>
<td>

<p>indicates whether the tip labels should be plotted. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>where</code></td>
<td>

<p>indicates where to put the legend (default is "bottomleft").
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>See the tutorial on GitHub (https://github.com/BPerezLamarque/Phylosignal_network). 
</p>


<h3>Value</h3>

<p>A phylogenetic tree with nodes colored according to the clade-specific phylogenetic signals. Blue nodes are not significant (Bonferonni correction), whereas orange-red nodes present significant phylogenetic signals and their color indicates the strength of the signal (correlation R of the Mantel test).
</p>


<h3>Author(s)</h3>

<p>Benoît Perez-Lamarque</p>


<h3>References</h3>

<p>Perez-Lamarque B, Maliet O, Pichon B, Selosse M-A, Martos F, Morlon H. 2022. Do closely related species interact with similar partners? Testing for phylogenetic signal in bipartite interaction networks. bioRxiv, 2021.08.30.458192, ver. 6 peer-reviewed and recommended by Peer Community in Evolutionary Biology. https://doi.org/10.1101/2021.08.30.458192
</p>
<p>Goslee, S.C. &amp; Urban, D.L. (2007). The ecodist package for dissimilarity-based analysis of ecological data. J. Stat. Softw., 22, 1–19.
</p>
<p>Chen, J., Bittinger, K., Charlson, E.S., Hoffmann, C., Lewis, J., Wu, G.D., et al. (2012). Associating microbiome composition with environmental covariates using generalized UniFrac distances. Bioinformatics, 28, 2106–2113.
</p>


<h3>See Also</h3>

<p><code>phylosignal_network</code>
<code>phylosignal_sub_network</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Load the data
data(mycorrhizal_network)

network &lt;- mycorrhizal_network[[1]] # interaction matrix 
tree_orchids &lt;- mycorrhizal_network[[2]] # phylogenetic tree (phylo object)
tree_fungi &lt;- mycorrhizal_network[[3]] # phylogenetic tree (phylo object)


if(test){

# Clade-specific phylogenetic signal in species interactions in guild A 
# (do closely related species interact with similar partners in sub-clades of guild A?)

results_clade_A &lt;- phylosignal_sub_network(network, tree_A = tree_orchids, tree_B = tree_fungi,
method = "GUniFrac", correlation = "Pearson")
plot_phylosignal_sub_network(tree_A = tree_orchids, results_clade_A, network)

# Clade-specific phylogenetic signal in species interactions in guild B 
# (do closely related species interact with similar partners in sub-clades of guild B?)

results_clade_B &lt;- phylosignal_sub_network(t(network), tree_A = tree_fungi, tree_B = tree_orchids,
method = "GUniFrac", correlation = "Pearson")
plot_phylosignal_sub_network(tree_A = tree_fungi, results_clade_B, t(network))
}
</code></pre>


</div>