<div class="container">

<table style="width: 100%;"><tr>
<td>kinship.BLUP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Genomic prediction by kinship-BLUP (deprecated)
</h2>

<h3>Description</h3>

<p>***This function has been superseded by <code>kin.blup</code>; please refer to its help page.
</p>


<h3>Usage</h3>

<pre><code class="language-R">kinship.BLUP(y, G.train, G.pred=NULL, X=NULL, Z.train=NULL, 
     K.method="RR", n.profile=10, mixed.method="REML", n.core=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>

<p>Vector (<code class="reqn">n.obs \times 1</code>) of observations.  Missing values (NA) are omitted.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>G.train</code></td>
<td>

<p>Matrix (<code class="reqn">n.train \times m</code>) of unphased genotypes for the training population: <code class="reqn">n.train</code> lines with <code class="reqn">m</code> bi-allelic markers.
Genotypes should be coded as {-1,0,1}; fractional (imputed) and missing (NA) alleles are allowed.  
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>G.pred</code></td>
<td>

<p>Matrix (<code class="reqn">n.pred \times m</code>) of unphased genotypes for the prediction population: <code class="reqn">n.pred</code> lines with <code class="reqn">m</code> bi-allelic markers.
Genotypes should be coded as {-1,0,1}; fractional (imputed) and missing (NA) alleles are allowed.  
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>

<p>Design matrix (<code class="reqn">n.obs \times p</code>) of fixed effects.  If not passed, a vector of 1's is used 
to model the intercept.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Z.train</code></td>
<td>

<p>0-1 matrix (<code class="reqn">n.obs \times n.train</code>) relating observations to lines in the training set.  If not passed
the identity matrix is used.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K.method</code></td>
<td>

<p>"RR" (default) is ridge regression, for which K is the realized additive relationship matrix computed with <code>A.mat</code>.  The option "GAUSS" is a Gaussian kernel (<code class="reqn">K = e^{-D^2/\theta^2}</code>) and "EXP" is an exponential kernel (<code class="reqn">K = e^{-D/\theta}</code>), where Euclidean distances <code class="reqn">D</code> are computed with <code>dist</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.profile</code></td>
<td>

<p>For K.method = "GAUSS" or "EXP", the number of points to use in the log-likelihood profile for the scale parameter <code class="reqn">\theta</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mixed.method</code></td>
<td>

<p>Either "REML" (default) or "ML".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.core</code></td>
<td>

<p>Setting n.core &gt; 1 will enable parallel execution of the Gaussian kernel computation (use only at UNIX command line).   
</p>
</td>
</tr>
</table>
<h3>Value</h3>


<dl>
<dt>$g.train</dt>
<dd>
<p>BLUP solution for the training set</p>
</dd>
<dt>$g.pred</dt>
<dd>
<p>BLUP solution for the prediction set (when G.pred != NULL)</p>
</dd>
<dt>$beta</dt>
<dd>
<p>ML estimate of fixed effects</p>
</dd>
</dl>
<p>For GAUSS or EXP, function also returns
</p>

<dl>
<dt>$profile</dt>
<dd>
<p>log-likelihood profile for the scale parameter</p>
</dd>
</dl>
<h3>References</h3>

<p>Endelman, J.B. 2011. Ridge regression and other kernels for genomic selection with R package rrBLUP. Plant Genome 4:250-255.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#random population of 200 lines with 1000 markers
G &lt;- matrix(rep(0,200*1000),200,1000)
for (i in 1:200) {
  G[i,] &lt;- ifelse(runif(1000)&lt;0.5,-1,1)
}

#random phenotypes
g &lt;- as.vector(crossprod(t(G),rnorm(1000)))
h2 &lt;- 0.5 
y &lt;- g + rnorm(200,mean=0,sd=sqrt((1-h2)/h2*var(g)))

#split in half for training and prediction
train &lt;- 1:100
pred &lt;- 101:200
ans &lt;- kinship.BLUP(y=y[train],G.train=G[train,],G.pred=G[pred,],K.method="GAUSS")

#correlation accuracy
r.gy &lt;- cor(ans$g.pred,y[pred])
</code></pre>


</div>