<div class="container">

<table style="width: 100%;"><tr>
<td>A.mat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Additive relationship matrix</h2>

<h3>Description</h3>

<p>Calculates the realized additive relationship matrix
</p>


<h3>Usage</h3>

<pre><code class="language-R">A.mat(
  X,
  min.MAF = NULL,
  max.missing = NULL,
  impute.method = "mean",
  tol = 0.02,
  n.core = 1,
  shrink = FALSE,
  return.imputed = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>matrix (<code class="reqn">n \times m</code>) of unphased genotypes for <code class="reqn">n</code> lines and <code class="reqn">m</code> biallelic markers, coded as {-1,0,1}. Fractional (imputed) and missing values (NA) are allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.MAF</code></td>
<td>
<p>Minimum minor allele frequency. The A matrix is not sensitive to rare alleles, so by default only monomorphic markers are removed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.missing</code></td>
<td>
<p>Maximum proportion of missing data; default removes completely missing markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>impute.method</code></td>
<td>
<p>There are two options. The default is "mean", which imputes with the mean for each marker.  The "EM" option imputes with an EM algorithm (see details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tol</code></td>
<td>
<p>Specifies the convergence criterion for the EM algorithm (see details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.core</code></td>
<td>
<p>Specifies the number of cores to use for parallel execution of the EM algorithm</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shrink</code></td>
<td>
<p>set shrink=FALSE to disable shrinkage estimation.  See Details for how to enable shrinkage estimation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.imputed</code></td>
<td>
<p>When TRUE, the imputed marker matrix is returned.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>At high marker density, the relationship matrix is estimated as <code class="reqn">A=W W'/c</code>, where <code class="reqn">W_{ik} = X_{ik} + 1 - 2 p_k</code> and <code class="reqn">p_k</code> is the frequency of the 1 allele at marker k.  By using a normalization constant of <code class="reqn">c = 2 \sum_k {p_k (1-p_k)}</code>, the mean of the diagonal elements is <code class="reqn">1 + f</code> (Endelman and Jannink 2012).
The EM imputation algorithm is based on the multivariate normal distribution and was designed for use with GBS (genotyping-by-sequencing) markers, which tend to be high density but with lots of missing data.  Details are given in Poland et al. (2012). The EM algorithm stops at iteration <code class="reqn">t</code> when the RMS error = <code class="reqn">n^{-1} \|A_{t} - A_{t-1}\|_2</code> &lt; tol.  
Shrinkage estimation can improve the accuracy of genome-wide marker-assisted selection, particularly at low marker density (Endelman and Jannink 2012). The shrinkage intensity ranges from 0 (no shrinkage) to 1 (<code class="reqn">A=(1+f)I</code>). Two algorithms for estimating the shrinkage intensity are available.  The first is the method described in Endelman and Jannink (2012) and is specified by <code>shrink=list(method="EJ")</code>.  The second involves designating a random sample of the markers as simulated QTL and then regressing the A matrix based on the QTL against the A matrix based on the remaining markers (Yang et al. 2010; Mueller et al. 2015). The regression method is specified by <code>shrink=list(method="REG",n.qtl=100,n.iter=5)</code>, where the parameters <code>n.qtl</code> and <code>n.iter</code> can be varied to adjust the number of simulated QTL and number of iterations, respectively.  
The shrinkage and EM-imputation options are designed for opposite scenarios (low vs. high density) and cannot be used simultaneously. 
When the EM algorithm is used, the imputed alleles can lie outside the interval [-1,1].  Polymorphic markers that do not meet the min.MAF and max.missing criteria are not imputed.
</p>


<h3>Value</h3>

<p>If return.imputed = FALSE, the <code class="reqn">n \times n</code> additive relationship matrix is returned. If return.imputed = TRUE, the function returns a list containing
</p>

<dl>
<dt>$A</dt>
<dd>
<p>the A matrix</p>
</dd>
<dt>$imputed</dt>
<dd>
<p>the imputed marker matrix</p>
</dd>
</dl>
<h3>References</h3>

<p>Endelman, J.B., and J.-L. Jannink. 2012. Shrinkage estimation of the realized relationship matrix. G3:Genes, Genomes, Genetics. 2:1405-1413. &lt;doi:10.1534/g3.112.004259&gt;
</p>
<p>Mueller et al. 2015. Shrinkage estimation of the genomic relationship matrix can improve genomic estimated breeding values in the training set. Theor Appl Genet 128:693-703. &lt;doi:10.1007/s00122-015-2464-6&gt;
</p>
<p>Poland, J., J. Endelman et al. 2012. Genomic selection in wheat breeding using genotyping-by-sequencing. Plant Genome 5:103-113. &lt;doi:10.3835/plantgenome2012.06.0006&gt;
</p>
<p>Yang et al. 2010. Common SNPs explain a large proportion of the heritability for human height. Nat. Genetics 42:565-569. &lt;doi:10.1038/ng.608&gt;
</p>


</div>