<div class="container">

<table style="width: 100%;"><tr>
<td>grm.pair</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>GRM for a pair of individuals.</h2>

<h3>Description</h3>

<p><code>grm.pair</code> computes relatedness estimates between two individuals.
</p>


<h3>Usage</h3>

<pre><code class="language-R">grm.pair(geno1, geno2, freq, method = "twostep", weights = NULL,
  init.est = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>geno1, geno2</code></td>
<td>
<p>numeric vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freq</code></td>
<td>
<p>numeric vector, values between 0 and 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>string.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p>numeric vector, values between 0 and 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>init.est</code></td>
<td>
<p>numeric.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>geno1</code> and <code>geno2</code> are vectors of counts of reference alleles. <code>freq</code> is the vector of reference allele frequencies.
</p>
<p>The default <code>method</code> is "twostep", other options include "classic", "robust" and "general". When using the default "twostep" method, user can supply an initial estimate through <code>init.est</code> to bypass the first step. When "general" is selected, <code>weights</code> must also be specified. The difference between the two-step GRM, classic GRM and robust GRM is discussed in Wang et al.  (2017).
</p>


<h3>Value</h3>

<p>An estimate of realized relatedness.
</p>


<h3>References</h3>

<p>Wang et al. (2017) Genetics 205:1063-1078, <a href="https://www.ncbi.nlm.nih.gov/pubmed/28100587">https://www.ncbi.nlm.nih.gov/pubmed/28100587</a>.
</p>


<h3>See Also</h3>

<p><code>grm.matrix</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># simulate genotypes for a full sib pair
pedigree = as.character(rep(1, 4))
member = as.character(c(11, 12, 21, 22))
sex = as.numeric(c(1, 2, 1, 2))
father = as.character(c(NA, NA, 11, 11))
mother = as.character(c(NA, NA, 12, 12))
pedinfo = data.frame(pedigree, member, sex, father, mother, stringsAsFactors = FALSE)
set.seed(1)
inher = sim.recomb(pedinfo, 3500) # on a hypothetical chromosome

nsnp = 100000
marker = seq(0,3500,length.out=nsnp)
freq = runif(nsnp, 0.05, 0.95)
haplo = sim.haplotype(freq, 4)
geno = populate.snp(inher, haplo, marker, output.allele = FALSE)

# simulation truth
ibd.proportion(inher,3,4)

# different GRM estimates
grm.pair(geno[3,], geno[4,], freq, method = "twostep")
grm.pair(geno[3,], geno[4,], freq, method = "classic")
grm.pair(geno[3,], geno[4,], freq, method = "robust")
grm.pair(geno[3,], geno[4,], freq, method = "general", weights = sample(freq, nsnp)/sum(freq))

# compute the relatedness matrix
grm.matrix(geno, freq)
grm.matrix(geno, freq, method = "robust")
</code></pre>


</div>