<div class="container">

<table style="width: 100%;"><tr>
<td>ld.weights</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>LD weights</h2>

<h3>Description</h3>

<p><code>ld.weights</code> computes LD weights for all markers, which is subsequently used to compute LD weighted GRM.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ld.weights(data, input.genotype = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>numeric matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>input.genotype</code></td>
<td>
<p>logical.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>data</code> can either be the subject by marker numeric genotype matrix (with 0, 1 or 2 coding), or the matrix of marker genotypic correlations. The default option is to input genotype matrix.
</p>


<h3>Value</h3>

<p>A numeric vector of weights. Note that the sum of weights is not constrained to be 1. They should be scaled appropriately before computing the LD weighted GRM.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># simulate genotypes of 500 individuals at 100 markers
nsnp = 100 # number of SNPs
freq = runif(nsnp, 0.05, 0.95)
nhaplo = 1000 # number of founder haplotypes
haplo.mat = sim.haplotype(freq, nhaplo)
geno.mat = t(sapply(c(1:500), function(x) 4 - haplo.mat[2*x-1,] - haplo.mat[2*x,]))

# compute unconstrained LD weights
ld.weights(geno.mat)
</code></pre>


</div>