<div class="container">

<table style="width: 100%;"><tr>
<td>read.plink.binary</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read PLINK binary file.</h2>

<h3>Description</h3>

<p><code>read.plink.binary</code> reads PLINK binary <code>.bed</code> file and the corresponding <code>.bim</code> and <code>.fam</code> file.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read.plink.binary(bed, bim = NULL, fam = NULL, na.strings = c("0", "-9"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>bed, bim, fam</code></td>
<td>
<p>PLINK files with appropriate extensions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.strings</code></td>
<td>
<p>string vector, text entries to be treated as NA's.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>When the three files have the same name, only the <code>.bed</code> file needs to be specified.
</p>


<h3>Value</h3>

<p>A list of three elements: <code>genotype</code>, <code>fam</code> and <code>map</code>.
To be consistent with PLINK .bed file, <code>genotype</code> is a n_subject by n_marker matrix of counts of reference alleles. Missing values are -9.
<code>fam</code> is a dataframe that contains the first six columns of a PLINK <code>.ped</code> file.
<code>map</code> is a dataframe that contains the four columns of a PLINK <code>.map</code> file, with two additional columns: <code>allele_1</code> for the reference allele type, <code>allele_2</code> for the alternate allele type.
</p>


</div>