<div class="container">

<table style="width: 100%;"><tr>
<td>populate.snp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Populate SNPs.</h2>

<h3>Description</h3>

<p><code>populate.snp</code> assigns alleles to markers, given inheritance information and founder haplotypes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">populate.snp(inheritance, haplotype, marker, member.index = NULL,
  output.allele = TRUE, output.haplotype = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>inheritance</code></td>
<td>
<p>list of numeric matrices.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>haplotype</code></td>
<td>
<p>numeric matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker</code></td>
<td>
<p>numeric vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>member.index</code></td>
<td>
<p>integer vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output.allele</code></td>
<td>
<p>logical.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output.haplotype</code></td>
<td>
<p>logical.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>inheritance</code> is a list of matrices produced by, e.g., <code>sim.recomb</code>. Each matrix contains a column of founder genome labels and a column of recombination breakpoints for the corresponding meiosis.
</p>
<p><code>haplotype</code> is a numeric matrix. The matrix is number of haplotypes by number of markers in dimension. Standard coding in this package is 1 for reference allele and 2 for alternate allele. This coding is required when <code>output.allele = FALSE</code>. Input data with different coding of alleles can be recoded using <code>recode.snpdata</code>. Number of haplotypes cannot be fewer than the number of founder genome labels in <code>inheritance</code>. The haplotypes will be assigned to each founder genome label in given order.
</p>
<p><code>marker</code> is a vector of marker genetic positions in Haldane centiMorgan in ascending order. Range of marker positions cannot exceed range covered by <code>inheritance</code>.
</p>
<p><code>member.index</code> contains indices of members in the pedigree that we wish to output data. Default value is <code>FALSE</code>, in which case marker data on everyone will be produced. <code>get.pedindex</code> can help find indices given member ID.
</p>
<p><code>output.allele</code> determines if one or two numbers will be used to represent data at each marker. Default is <code>TRUE</code>, in which case marker data is represented by two ordered (paternal first) alleles. Otherwise marker data is represented by a single number (0, 1 or 2) of reference alleles.
</p>
<p><code>output.haplotype</code> determines if haplotype data are separate in output. It is only used when <code>output.allele = TRUE</code>. Default value is <code>FALSE</code>, in which case each row in the output matrix represents ordered genotypes from all markers of the same individual. Otherwise each row in the output matrix represents a parental haplotype.
</p>


<h3>Value</h3>

<p>A matrix of genotypic/haplotypic data. The matrix is in individual major, where marker data for each individual/meiosis are found on the same row. Exact format of the matrix depends on various input arguments.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># a simple pedigree with sibling marriage
pedigree = as.character(rep(1, 5))
member = as.character(c(11, 12, 21, 22, 31))
sex = as.numeric(c(1, 2, 1, 2, 1))
father = as.character(c(NA, NA, 11, 11, 21))
mother = as.character(c(NA, NA, 12, 12, 22))
pedinfo = data.frame(pedigree, member, sex, father, mother, stringsAsFactors = FALSE)

L = 100.0 # segment length
nsnp = 10 # number of SNPs
nhaplo = 4 # number of founder haplotypes
inher = sim.recomb(pedinfo, L)
haplo = matrix(c(3,4,4,4), nhaplo, nsnp)
marker = sort(runif(nsnp, 0, L))

# output genotype data for the 4th and 5th member
# of pedigree, genotype data displayed as two alleles
populate.snp(inher, haplo, marker, c(4, 5))

# output haplotype data for the 4th and 5th member of pedigree
populate.snp(inher, haplo, marker,c(4, 5), output.haplotype = TRUE)

# output genotype data for all members, genotype data
# displayed as counts of reference alleles
geno = recode.snpdata(haplo, input.haplotype = TRUE, output.haplotype = TRUE)[[1]]
populate.snp(inher, geno, marker, output.allele = FALSE)
</code></pre>


</div>