<div class="container">

<table style="width: 100%;"><tr>
<td>write.ibdhaplo</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Write IBDHAPLO</h2>

<h3>Description</h3>

<p><code>write.ibdhaplo</code> prepares the marker data file for running IBDHAPLO.
</p>


<h3>Usage</h3>

<pre><code class="language-R">write.ibdhaplo(marker, freq, data, member, input.allele = TRUE,
  input.haplotype = FALSE, outfile = tempfile("ibdhaplo", fileext = ".txt"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>marker</code></td>
<td>
<p>numeric vector, marker genetic positions in cM.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freq</code></td>
<td>
<p>numeric vector, marker reference allele frequencies.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>numeric matrix, genetic marker data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>member</code></td>
<td>
<p>string vector, member ID.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>input.allele</code></td>
<td>
<p>logical, default TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>input.haplotype</code></td>
<td>
<p>logical, default FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile</code></td>
<td>
<p>string, output file name.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The input marker data needs to be subject/haplotype by marker/allele. For example, suppose <code>data</code> is a 4x10 matrix, use <code>input.allele = FALSE</code> if <code>data</code> contains counts of reference alleles of 4 individuals at 10 markers; use <code>input.haplotype = TRUE</code> if <code>data</code> contains allelic types of 4 haplotypes at 10 markers; use default options if <code>data</code> contains allelic types of 4 individuals at 5 markers.
</p>


<h3>References</h3>

<p>MORGAN Tutorial, <a href="https://www.stat.washington.edu/thompson/Genepi/MORGAN/Morgan.shtml">https://www.stat.washington.edu/thompson/Genepi/MORGAN/Morgan.shtml</a>.
</p>
<p>Brown et al. (2012) Genetics 190:1447-1460, <a href="https://www.ncbi.nlm.nih.gov/pubmed/22298700">https://www.ncbi.nlm.nih.gov/pubmed/22298700</a>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
nsnp = 7 # number of SNPs
freq = runif(nsnp, 0.05, 0.95)
nhaplo = 4 # number of founder haplotypes
haplotype = sim.haplotype(freq, nhaplo)
marker = sort(runif(7,0,100))
write.ibdhaplo(marker, freq, haplotype, member = c("ind1", "ind2"), 
input.haplotype = TRUE)

## End(Not run)
</code></pre>


</div>