<div class="container">

<table style="width: 100%;"><tr>
<td>grm.matrix</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>GRM for multiple individuals</h2>

<h3>Description</h3>

<p><code>grm.matrix</code> computes relatedness estimates between every pairs of individuals.
</p>


<h3>Usage</h3>

<pre><code class="language-R">grm.matrix(genotype, freq, method = "twostep", weights = NULL,
  init.est = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genotype</code></td>
<td>
<p>numeric matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freq</code></td>
<td>
<p>numeric vector, values between 0 and 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>string.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p>numeric vector, values between 0 and 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>init.est</code></td>
<td>
<p>numeric.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>genotype</code> is the matrix of counts of reference alleles. Rows represents subjects and columns represents SNP markers. <code>freq</code> is the vector of reference allele frequencies.
</p>
<p>The default <code>method</code> is "twostep", other options include "classic", "robust" and "general". When using the default "twostep" method, user can supply an initial estimate through <code>init.est</code> to bypass the first step. When "general" is selected, <code>weights</code> must also be specified. The difference between the two-step GRM, classic GRM and robust GRM is discussed in Wang et al.  (2017).
</p>


<h3>References</h3>

<p>Wang et al. (2017) Genetics 205:1063-1078, <a href="https://www.ncbi.nlm.nih.gov/pubmed/28100587">https://www.ncbi.nlm.nih.gov/pubmed/28100587</a>.
</p>


<h3>See Also</h3>

<p><code>grm.pair</code>.
</p>


</div>