<div class="container">

<table style="width: 100%;"><tr>
<td>recode.snpdata</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Recode SNP marker data.</h2>

<h3>Description</h3>

<p><code>recode.snpdata</code> recodes SNP marker data for use with other functions in this package.
</p>


<h3>Usage</h3>

<pre><code class="language-R">recode.snpdata(data, snp.major = FALSE, ma.ref = FALSE,
  input.haplotype = FALSE, output.allele = TRUE, output.haplotype = FALSE,
  na.string = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>numeric matrix or dataframe.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.major</code></td>
<td>
<p>logical.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ma.ref</code></td>
<td>
<p>logical.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>input.haplotype</code></td>
<td>
<p>logical.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output.allele</code></td>
<td>
<p>logical.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output.haplotype</code></td>
<td>
<p>logical.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.string</code></td>
<td>
<p>numeric or character vector.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The standard marker data used by other functions of this package takes one of three forms: (a) subjects by row, counts of reference alleles by column; (b) subjects by row, allelic types (2 per marker) by column; (c) haplotypes (2 per subject) by row, allelic types by column. Reference alleles are coded 1, alternate alleles are coded 2.
</p>
<p>By default, <code>snp.major = FALSE</code>, set it to <code>TRUE</code> if input matrix has SNPs by row and allelic types (2 per subject) by column. <code>ma.ref = FALSE</code>, set it to <code>TRUE</code> if the minor allele is to be the reference allele. <code>input.haplotype = FALSE</code>, set it to <code>TRUE</code> if input matrix has haplotypes (2 per subject) by row and allelic types by column. <code>output.allele = TRUE</code>, set it to <code>FALSE</code> if counts of reference alleles is the desired output format. <code>output.haplotype = FALSE</code>, set it to <code>TRUE</code> if recoded marker data by haplotype is the desired output format. <code>input.haplotype</code> is only invoked when <code>snp.major = FALSE</code>. <code>output.haplotype</code> is only invoked when <code>output.allele = TRUE</code>.
</p>


<h3>Value</h3>

<p>A list of two elements. First element named <code>data</code> is a matrix of recoded marker data in specified format. Second element is a dataframe named <code>alleles</code> that specifies reference/alternate alleles at all markers.
</p>


<h3>Examples</h3>

<pre><code class="language-R">test.dat = matrix(c(3,4,4,3), 4, 10)

# treat test.dat as 4 input haplotypes of two subjects at 10 SNP markers,
# output recoded data as haplotypes
recode.snpdata(test.dat, input.haplotype = TRUE, output.haplotype = TRUE)

# treat test.dat as 4 input haplotypes of two subjects at 10 SNP markers,
# output recoded data as counts of reference alleles
recode.snpdata(test.dat, input.haplotype = TRUE, output.allele = FALSE)
#'
# treat test.dat as allelic types at 5 SNPs of 4 subjects,
# output recoded data as haplotypes
recode.snpdata(test.dat, output.haplotype = TRUE)
</code></pre>


</div>