<div class="container">

<table style="width: 100%;"><tr>
<td>threeFACTORplot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Bar plot of the relative gene expression (<code class="reqn">\Delta C_T</code> method) from the <code>qpcrANOVARE</code> output of a a three-factorial experiment data</h2>

<h3>Description</h3>

<p>Bar plot of the relative expression (<code class="reqn">\Delta C_T</code> method) of a gene along with the confidence interval and significance
</p>


<h3>Usage</h3>

<pre><code class="language-R">threeFACTORplot(
  res,
  arrangement = c(1, 2, 3),
  bar.width = 0.5,
  fill = "Reds",
  xlab = "none",
  ylab = "Relative Expression",
  errorbar,
  y.axis.adjust = 0.5,
  y.axis.by = 2,
  letter.position.adjust = 0.3,
  legend.title = "Legend Title",
  legend.position = c(0.4, 0.8),
  fontsize = 12,
  fontsizePvalue = 5,
  show.letters = TRUE,
  axis.text.x.angle = 0,
  axis.text.x.hjust = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>res</code></td>
<td>
<p>the FC data frame created by <code>qpcrANOVARE(x)$Result</code> function on a three factorial data such as <code>data_3factor</code> example data frame.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>arrangement</code></td>
<td>
<p>order based on the columns in the output table (e.g. c(2,3,1) or c(1,3,2)) affecting factor arrangement of the output graph.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bar.width</code></td>
<td>
<p>a positive number determining bar width.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fill</code></td>
<td>
<p>a color vector specifying the fill color for the columns of the bar plot. One of the palettes in <code>display.brewer.all</code> (e.g. "Reds" or "Blues", ...) can be applied.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>the title of the x axis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>the title of the y axis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>errorbar</code></td>
<td>
<p>Type of error bar, can be <code>se</code> or <code>ci</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y.axis.adjust</code></td>
<td>
<p>a negative or positive number for reducing or increasing the length of the y axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y.axis.by</code></td>
<td>
<p>determines y axis step length</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>letter.position.adjust</code></td>
<td>
<p>adjust the distance between the grouping letters to the error bars</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend.title</code></td>
<td>
<p>legend title</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend.position</code></td>
<td>
<p>a two digit vector specifying the legend position.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fontsize</code></td>
<td>
<p>all fonts size of the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fontsizePvalue</code></td>
<td>
<p>font size of the pvalue labels</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.letters</code></td>
<td>
<p>a logical variable. If TRUE, mean grouping letters are added to the bars.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>axis.text.x.angle</code></td>
<td>
<p>angle of x axis text</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>axis.text.x.hjust</code></td>
<td>
<p>horizontal justification of x axis text</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The <code>threeFACTORplot</code> function generates the bar plot of the average fold change for target genes along with the significance, standard error (se) and the 95% confidence interval (ci).
</p>


<h3>Value</h3>

<p>Bar plot of the average fold change for target genes along with the standard error or 95% confidence interval as error bars.
</p>


<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>Examples</h3>

<pre><code class="language-R">
#' # See a sample data frame
data_3factor

# Before plotting, the result needs to be extracted as below:
res &lt;- qpcrANOVARE(data_3factor, numberOfrefGenes = 1, block = NULL)$Result
res

# Arrange the first three colunms of the result table.
# This determines the columns order and shapes the plot output.
threeFACTORplot(res, arrangement = c(3, 1, 2), errorbar = "se",
    xlab = "condition")


threeFACTORplot(res, arrangement = c(1, 2, 3), bar.width = 0.5, fill = "Greys", 
xlab = "Genotype", ylab = "Relative Expression", errorbar = "se")


# Reordering factor levels to a desired order.
res$Conc &lt;- factor(res$Conc, levels = c("L","M","H"))
res$Type &lt;- factor(res$Type, levels = c("S","R"))

# Producing the plot
threeFACTORplot(res, arrangement = c(2, 3, 1), bar.width = 0.5, 
fill = "Reds", xlab = "Drought", ylab = "Relative Expression", 
errorbar = "se", legend.title = "Genotype", legend.position = c(0.2, 0.8))


# When using ci as error, increase the 
# y.axis.adjust value to see the plot correctly!
threeFACTORplot(res, arrangement = c(2, 3, 1), bar.width = 0.8, fill = "Greens", 
xlab = "Drought", ylab = "Relative Expression", errorbar = "ci", 
y.axis.adjust = 1, y.axis.by = 2, letter.position.adjust = 0.6, 
legend.title = "Genotype", fontsize = 12, legend.position = c(0.2, 0.8), 
show.letters = TRUE)



</code></pre>


</div>