<div class="container">

<table style="width: 100%;"><tr>
<td>allelic_list</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create genotype lists for each locus</h2>

<h3>Description</h3>

<p>Uses the allele counts from <code>a_freq_list</code> and the cleaned short-format output of
<code>tcf2long</code> to generate a nested list of individual genotypes for each locus
</p>


<h3>Usage</h3>

<pre><code class="language-R">allelic_list(cs, ac, samp_type = "both")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cs</code></td>
<td>
<p>a clean short genetic data matrix; the second element of the
output from <code>tcf2long</code>. Must have a column of individual identifiers, named "indiv"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ac</code></td>
<td>
<p>allele counts from a_freq_list</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>samp_type</code></td>
<td>
<p>choose which sample types of individuals to include in output:
"mixture", "both", or "reference"</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p><code>allelic_list</code> returns a two-component nested list, with data stored as character
names of alleles ($chr) or as integer indices for the alleles ($int). Both forms contain lists
representing to loci, with two component vectors corresponding to gene copies a and b.
</p>


<h3>Examples</h3>

<pre><code class="language-R">example(a_freq_list)
ale_cs &lt;- ale_long$clean_short
# Get the vectors of gene copies a and b for all loci in integer index form
ale_alle_list &lt;- allelic_list(ale_cs, ale_ac)$int

</code></pre>


</div>