<div class="container">

<table style="width: 100%;"><tr>
<td>gprob_sim_gc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate genotype log-likelihoods from a population by gene copy</h2>

<h3>Description</h3>

<p>Takes a list of parameters from a genetic dataset, and returns a genotype log-likelihood
matrix for individuals simulated by gene copy from the specified collections
</p>


<h3>Usage</h3>

<pre><code class="language-R">gprob_sim_gc(par_list, sim_colls)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>par_list</code></td>
<td>
<p>genetic data converted to the param_list format by <code>tcf2param_list</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sim_colls</code></td>
<td>
<p>a vector of indices for the collections desired for simulation;
each element of the list corresponds to an individual</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>In simulation by gene copy, the genotype at a locus for any individual is the result
of two random draws from the allele count matrix of that locus. Draws within an individual
are performed without replacement, but allele counts are replaced between individuals.
</p>


<h3>Value</h3>

<p><code>gprob_sim</code> returns a matrix of the summed log-likelihoods
for all loci of a simulated population mixture; columns represent individuals,
with each row containing their log-likelihood of belonging to the collection
of the same index, given the selection of two independent gene copies from the
desired collection of origin's reference allele frequencies
</p>


<h3>Examples</h3>

<pre><code class="language-R">example(tcf2param_list)
sim_colls &lt;- sample(ale_par_list$C, 1070, replace = TRUE)
ale_sim_gprobs_gc &lt;- gprob_sim_gc(ale_par_list, sim_colls)
</code></pre>


</div>