<div class="container">

<table style="width: 100%;"><tr>
<td>plot_cutoff</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Check number of DE genes at different cutoff combinations</h2>

<h3>Description</h3>

<p>This function processes summary statistics table generated by differential expression analysis
like <code>limma</code> or <code>DESeq2</code> to evaluate the number of differntially expressed genes with different FDR and
fold change cutoff.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_cutoff(
  data = data,
  comp.names = NULL,
  FCflag = "logFC",
  FDRflag = "adj.P.Val",
  FCmin = 1.2,
  FCmax = 2,
  FCstep = 0.1,
  p.min = 0,
  p.max = 0.2,
  p.step = 0.01,
  plot.save.to = NULL,
  gen.3d.plot = TRUE,
  gen.plot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>Summary statistics table or a list of summary statistics tables from limma or DEseq2, where each row is a gene.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>comp.names</code></td>
<td>
<p>A character vector that contains the comparison names which correspond to the same order as <code>data</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FCflag</code></td>
<td>
<p>The column name of the log2FC in the summary statistics table. Default = "logFC".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FDRflag</code></td>
<td>
<p>The column name of the False Discovery Rate (FDR) in the summary statistics table. Default = "adj.P.Val".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FCmin</code></td>
<td>
<p>The minimum starting fold change cutoff to be checked, so the minimum fold change cutoff to be evaluated
will be FCmin + FCstep, FCmin default = 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FCmax</code></td>
<td>
<p>The maximum fold change cutoff to be checked, default = 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FCstep</code></td>
<td>
<p>The step from the minimum to maximum fold change cutoff, one step increase at a time, default = 0.01.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.min</code></td>
<td>
<p>The minimum starting FDR cutoff to be checked, so the minimum fold change cutoff to be evaluated
will be p.min + p.step, p.min default = 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.max</code></td>
<td>
<p>The maximum FDR cutoff to be checked, default = 0.2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.step</code></td>
<td>
<p>The step from the minimum to maximum fold change cutoff, one step increase at a time, default = 0.005.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.save.to</code></td>
<td>
<p>The address where to save the plot from simplified cutoff combination with FDR of 0.01, 0.05, 0.1, and 0.2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gen.3d.plot</code></td>
<td>
<p>Whether generate a 3d plotly object to visualize the result, only applys to single dataframe input, default = F.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gen.plot</code></td>
<td>
<p>Whether generate a plot to visualize the result, default = T.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function takes the summary statistics and returns a list which contains 3 objects: a table which describes the number
of DE genes with different cutoff combinations of FDR and fold change, a ggplot object which depicts a simplified version of
cutoff selection combination, and a plotly 3d visulization object which depicts a high resolution of cutoff combinations.
The default range of the fold change is from 1 to 2, and p value is from 0 to 0.2, with the step of 0.01
for FC and 0.005 for FDR.
</p>


<h3>Value</h3>

<p>If the input <code>data</code> is a data list, then a multi-facet ggplot plot object which contains each
of the summary statistics table will be returned; otherwise, if the input <code>data</code> is a data frame, then the function will return a list which contains 3 elements:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>df.sub</code></td>
<td>
<p>A dataframe, which contains the number of genes(3rd column) with FDR (1st column), Fold Change (2nd column)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot3d</code></td>
<td>
<p>A plotly object to show the 3d illustration of all possible cutoff selectiosn and
the number of DE genes in the 3d surface</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gp</code></td>
<td>
<p>A ggplot object to show the simplified cutoff combination result</p>
</td>
</tr>
</table>
<h3>References</h3>

<p>Xingpeng Li &amp; Olya Besedina, RVA - RNAseq Visualization Automation tool.
</p>


<h3>Examples</h3>

<pre><code class="language-R">plot_cutoff(Sample_summary_statistics_table)

plot_cutoff(data = list(Sample_summary_statistics_table, Sample_summary_statistics_table1),
           comp.names = c("A", "B"))



</code></pre>


</div>