<div class="container">

<table style="width: 100%;"><tr>
<td>plot_gene</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot gene expression</h2>

<h3>Description</h3>

<p>This is the function to process the gene count table to show gene expression variations over time or across groups.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_gene(
  data = ~dat,
  anno = ~meta,
  gene.names = c("AAAS", "A2ML1", "AADACL3"),
  ct.table.id.type = "ENSEMBL",
  gene.id.type = "SYMBOL",
  treatment = "Treatment",
  sample.id = "sample_id",
  time = "day",
  log.option = TRUE,
  plot.save.to = NULL,
  input.type = "count"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>Count table in the format of dataframe with gene id as row.names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anno</code></td>
<td>
<p>Annotation table that provides design information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.names</code></td>
<td>
<p>Genes to be visualized, in the format of character vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ct.table.id.type</code></td>
<td>
<p>The gene id format in <code>data</code> should be one of: ACCNUM, ALIAS, ENSEMBL, ENSEMBLPROT,
ENSEMBLTRANS, ENTREZID, ENZYME, EVIDENCE, EVIDENCEALL, GENENAME, GO, GOALL, IPI, MAP, OMIM,
ONTOLOGY, ONTOLOGYALL, PATH, PFAM, PMID, PROSITE, REFSEQ, SYMBOL, UCSCKG, UNIGENE, UNIPROT.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.id.type</code></td>
<td>
<p>The gene id format of <code>gene.names</code>, should be one of: ACCNUM, ALIAS, ENSEMBL, ENSEMBLPROT,
ENSEMBLTRANS, ENTREZID, ENZYME, EVIDENCE, EVIDENCEALL, GENENAME, GO, GOALL, IPI, MAP, OMIM,
ONTOLOGY, ONTOLOGYALL, PATH, PFAM, PMID, PROSITE, REFSEQ, SYMBOL, UCSCKG, UNIGENE, UNIPROT.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>treatment</code></td>
<td>
<p>The column name to specify treatment groups.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.id</code></td>
<td>
<p>The column name to specify sample IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>time</code></td>
<td>
<p>The column name to specify different time points.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log.option</code></td>
<td>
<p>Logical option, whether to log2 transform the CPM as y-axis. Default = True.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.save.to</code></td>
<td>
<p>The address to save the plot from simplified cutoff combination with FDR of 0.01, 0.05, 0.1, and 0.2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>input.type</code></td>
<td>
<p>One of <code>count</code> or <code>cpm</code> indicating what the input data type is. If <code>count</code>, the CPM of the input data will be
calculated using <code>edgeR::cpm()</code>. Default = <code>count</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function takes the gene counts and returns a ggplot that shows gene expression variation over time or group.
</p>


<h3>Value</h3>

<p>The function returns a ggplot object.
</p>


<h3>References</h3>

<p>Xingpeng Li,Tatiana Gelaf Romer &amp; Aliyah Olaniyan, RVA - RNAseq Visualization Automation tool.
</p>


<h3>Examples</h3>

<pre><code class="language-R">plot_gene(data = count_table,
anno = sample_annotation)

</code></pre>


</div>