<div class="container">

<table style="width: 100%;"><tr>
<td>plot_pathway</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Pathway analysis and visualization</h2>

<h3>Description</h3>

<p>This is the function to do pathway enrichment analysis (and visualization) with rWikipathways (also KEGG, REACTOME &amp; Hallmark) from a summary statistics table generated by
differential expression analysis like <code>limma</code> or <code>DESeq2</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_pathway(
  data = ~df,
  comp.names = NULL,
  gene.id.type = "ENSEMBL",
  FC.cutoff = 1.2,
  FDR.cutoff = 0.05,
  FCflag = "logFC",
  FDRflag = "adj.P.Val",
  Fisher.cutoff = 0.1,
  Fisher.up.cutoff = 0.1,
  Fisher.down.cutoff = 0.1,
  plot.save.to = NULL,
  pathway.db = "rWikiPathways",
  customized.pathways = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>A summary statistics table (data.frame) or <code>data.list</code> generated by DE analysis software like limma or DEseq2, where rownames are gene id.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>comp.names</code></td>
<td>
<p>A character vector containing the comparison names corresponding to the same order of the <code>data.list</code>. Default = NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.id.type</code></td>
<td>
<p>The gene id format in <code>data</code> should be one of: ACCNUM, ALIAS, ENSEMBL, ENSEMBLPROT,
ENSEMBLTRANS, ENTREZID, ENZYME, EVIDENCE, EVIDENCEALL, GENENAME, GO, GOALL, IPI, MAP, OMIM,
ONTOLOGY, ONTOLOGYALL, PATH, PFAM, PMID, PROSITE, REFSEQ, SYMBOL, UCSCKG, UNIGENE, UNIPROT.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FC.cutoff</code></td>
<td>
<p>The fold change cutoff (numeric) selected to subset summary statistics table. Default = 1.5.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FDR.cutoff</code></td>
<td>
<p>The FDR cutoff selected (numeric) to subset summary statistics table. Default = 0.05.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FCflag</code></td>
<td>
<p>The column name (character) of fold change information, assuming the FC is log2 transformed. Default = "logFC".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FDRflag</code></td>
<td>
<p>The column name (character) of adjusted p value or FDR. Default = "adj.P.Val".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Fisher.cutoff</code></td>
<td>
<p>The FDR cutoff selected (numeric) for the pathway enrichment analysis' Fisher's exact test with all determined
Differentially Expressed (DE) genes by <code>FC.cutoff</code> and <code>FDR.cutoff</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Fisher.up.cutoff</code></td>
<td>
<p>The FDR cutoff selected (numeric) for the pathway enrichment analysis' Fisher's exact test with the upregulated gene set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Fisher.down.cutoff</code></td>
<td>
<p>The FDR cutoff selected (numeric) for the pathway enrichment analysis' Fisher's exact test with the downregulated gene set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.save.to</code></td>
<td>
<p>The address to save the plot from simplified cutoff combination with FDR of 0.01, 0.05, 0.1, and 0.2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pathway.db</code></td>
<td>
<p>The databse to be used for encrichment analysis. Can be one of the following, "rWikiPathways", "KEGG", "REACTOME", "Hallmark","rWikiPathways_aug_2020".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>customized.pathways</code></td>
<td>
<p>the customized pathways in the format of two column dataframe (column name as "gs_name" and "entrez_gene") to be used in analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>pass on variables</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function takes the summary statistics table and use user selected parameter based on check.cutoff to do pathway enrichment analysis
</p>


<h3>Value</h3>

<p>The function returns a list of 5 objects:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>1</code></td>
<td>
<p>result table from directional pathway enrichment analysis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>2</code></td>
<td>
<p>result table from non-directional pathway enrichment analysis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>3</code></td>
<td>
<p>plot from directional pathway enrichment analysis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>4</code></td>
<td>
<p>plot from non-directional pathway enrichment analysis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>5</code></td>
<td>
<p>plot combining both directional and non-directional plot</p>
</td>
</tr>
</table>
<h3>References</h3>

<p>Xingpeng Li &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>


<h3>Examples</h3>

<pre><code class="language-R">result &lt;- plot_pathway(data = Sample_summary_statistics_table,
gene.id.type = "ENSEMBL",
FC.cutoff = 1.5,
p.cutoff = 0.05,
pathway.db = "rWikiPathways_aug_2020"
)
</code></pre>


</div>