<div class="container">

<table style="width: 100%;"><tr>
<td>rhapPDT</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calling haplotype-based and variant-based pedigree disequilibrium test for rare variants in pedigrees.</h2>

<h3>Description</h3>

<p>To detecting rare variants for binary traits using general pedigrees, the pedigree disequilibrium tests are proposed by collapsing rare haplotypes/variants with/without weights.</p>


<h3>Usage</h3>

<pre><code class="language-R"> rhapPDT(ped, map, aff=2, unaff=1, mu=1.04,
merlinFN.prefix="merlin", nperm=1000, trace=TRUE) 
</code></pre>


<h3>Arguments</h3>

  
<table>
<tr style="vertical-align: top;">
<td><code>ped</code></td>
<td>
<p>input data, has same format with PLINK but having column names. The PED file is a white-space (space or tab) delimited file, and the first six columns are mandatory: 
FID: Family ID;
IID: Individual ID;
FA: Paternal ID;
MO: Maternal ID;
SEX: Sex (1=male; 2=female; other=unknown);
PHENO: Phenotype; 
Genotypes (column 7 onwards) should also be white-space delimited; they are coded as 0, 1 and 2, indicating the number of coding allele, and NA is for missing genotype. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>input data, has same format with MAP file required by MERLIN. The MAP file is a white-space (space or tab) delimited file with 3 columns as follows,  
CHROMOSOME: chromosome (1-22, X, Y or 0 if unplaced)  
MARKER: marker name in PED file that is usually rs# or snp identifier         
POSITION: Genetic distance (morgans)
The data file and map file can include different sets of markers, but markers that are absent from the map file will be ignored by MERLIN. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aff</code></td>
<td>
<p>indicates the values that represents affected status in "PHENO" column of PED data; default is 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>unaff</code></td>
<td>
<p>indicates the values that represents unaffected status in "PHENO" column of PED data; default is 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu</code></td>
<td>
<p>indicates mu value that defines causal in the training data; default is 1.04.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>merlinFN.prefix</code></td>
<td>
<p> Requests that output file of MERLIN names should be derived from outFN.prefix. For example, when it is set to be "merlin" as default, estimated haplotypes should be stored in a file called merlin.chr.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nperm</code></td>
<td>
<p>indicates the times of permutation; default is 1000.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trace</code></td>
<td>
<p> Indicates whether or not the intermediate outcomes should be printed; default is FALSE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

 
<table>
<tr style="vertical-align: top;">
<td><code>hPDT_v0</code></td>
<td>
<p> P value of unweighted haplotype PDT test statistic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hPDT_v1</code></td>
<td>
<p> P value of weighted haplotype PDT test statistic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rvPDT_v0</code></td>
<td>
<p> P value of unweighted rvPDT test statistic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rvPDT_v1</code></td>
<td>
<p> P value of weighted rvPDT test statistic.</p>
</td>
</tr>
</table>
<h3>References</h3>

    
<p>Guo W , Shugart YY, Does Haplotype-based Collapsing Tests Gain More Power than Variant-based Collapsing Tests for Detecting Rare Variants in Pedigrees  
(manuscript). 
</p>


<h3>Examples</h3>

<pre><code class="language-R">#ped&lt;-read.table("MLIP.ped",head=1,stringsAsFactors=FALSE)
#map&lt;-read.table("MLIP.map",head=1,stringsAsFactors=FALSE)  
#test&lt;-rhapPDT(ped, map, trace=TRUE) 
#test 
#$hPDT_v0
#[1] 0.4231359

#$hPDT_v1
#[1] 0.1481145

#$rvPDT_v0
#[1] 0.03237073

#$rvPDT_v1
#[1] 0.162997
</code></pre>


</div>