<div class="container">

<table style="width: 100%;"><tr>
<td>whap.prehap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Prepare haplotype pairs for hPDT tests in pedigree data.</h2>

<h3>Description</h3>

<p>Before running hPDT test, haplotype pairs are inferred by calling MERLIN in linux for all pedigree members, and then perpare some internal statistics. 
Require the R package of "gregmisc" and MERLIN software.
</p>


<h3>Usage</h3>

<pre><code class="language-R"> 
whap.prehap(ped,map, merlinDir="", outFN.prefix="merlin",aff=2,trace=FALSE)  
</code></pre>


<h3>Arguments</h3>

  
<table>
<tr style="vertical-align: top;">
<td><code>ped</code></td>
<td>
<p>input data, has same format with PLINK but having column names. The PED file is a white-space (space or tab) delimited file, and the first six columns are mandatory: 
FID: Family ID;
IID: Individual ID;
FA: Paternal ID;
MO: Maternal ID;
SEX: Sex (1=male; 2=female; other=unknown);
PHENO: Phenotype; 
Genotypes (column 7 onwards) should also be white-space delimited; they are coded as 0, 1 and 2, indicating the number of coding allele, and NA is for missing genotype. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>input data, has same format with MAP file required by MERLIN. The MAP file is a white-space (space or tab) delimited file with 3 columns as follows,  
CHROMOSOME: chromosome (1-22, X, Y or 0 if unplaced)  
MARKER: marker name in PED file that is usually rs# or snp identifier         
POSITION: Genetic distance (morgans)
The data file and map file can include different sets of markers, but markers that are absent from the map file will be ignored by MERLIN. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>merlinDir</code></td>
<td>
<p>indicates the directory of Merlin, for example, merlinDir="./Merlin/"; use the default="" when Merlin is in current directory or your bin directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outFN.prefix</code></td>
<td>
<p>Requests that output file of MERLIN names should be derived from outFN.prefix. For example, when it is set to be "merlin" as default, estimated haplotypes should be stored in a file called merlin.chr.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aff</code></td>
<td>
<p>indicates the values that represents affected status in ped data; default is 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trace</code></td>
<td>
<p>indicates whether or not the intermediate outcomes should be printed; default is FALSE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>SNPname</code></td>
<td>
<p> SNP names of testing.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hapData</code></td>
<td>
<p> Haplotype data for each individuals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freq</code></td>
<td>
<p> Estimated frequencies of haplotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trans</code></td>
<td>
<p> Transmission matrix of haplotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hapScore</code></td>
<td>
<p> Score matrix of haplotypes.</p>
</td>
</tr>
</table>
<h3>References</h3>

    
<p>Guo W , Shugart YY, Does Haplotype-based Collapsing Tests Gain More Power than Variant-based Collapsing Tests for Detecting Rare Variants in Pedigrees  
(manuscript). 
</p>


</div>