<div class="container">

<table style="width: 100%;"><tr>
<td>rvPDT.test.permu</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Variants-based pedigree disequilibrium test for rare variants in pedigrees.</h2>

<h3>Description</h3>

<p>To detecting rare variants for binary traits using general pedigrees, the pedigree disequilibrium tests are extended by collapsing rare variants with/without weights.</p>


<h3>Usage</h3>

<pre><code class="language-R">rvPDT.test.permu(ped, aff=2,unaff=1, snpCol, hfreq=NULL,
useFamWeight=TRUE, nperm=1000,trace=FALSE)    
</code></pre>


<h3>Arguments</h3>

   
<table>
<tr style="vertical-align: top;">
<td><code>ped</code></td>
<td>
<p>input data, has same format with PLINK but having column names. The PED file is a white-space (space or tab) delimited file: the first six columns are mandatory: 
FID: Family ID;
IID: Individual ID;
FA: Paternal ID;
MO: Maternal ID;
SEX: Sex (1=male; 2=female; other=unknown);
PHENO: Phenotype; 
Genotypes (column 7 onwards) should also be white-space delimited; they are coded as 0, 1 and 2, indicating the number of coding allele, and NA is for missing genotype. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aff</code></td>
<td>
<p>indicates the values that represents affected status in ped data; default is 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>unaff</code></td>
<td>
<p>indicates the values that represents unaffected status in ped data; default is 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpCol</code></td>
<td>
<p>indicates the columns of variants in ped data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hfreq</code></td>
<td>
<p>indicates the frequencies of variants that used in calculating weights; when it is NULL, the frequencies are estimated by ped data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useFamWeight</code></td>
<td>
<p>indicates whether the family weights need to be used in the test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nperm</code></td>
<td>
<p>indicates the times of permutation; default is 1000.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trace</code></td>
<td>
<p>indicates wether or not the intermediate outcomes should be printed; default is FALSE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>TDT</code></td>
<td>
<p> Transmission/disequilibrium matrix for each pedigrees.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Sib</code></td>
<td>
<p> Discordant sib pairs matrix for each pedigrees.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>PDT</code></td>
<td>
<p> Pedigree disequilibrium matrix for each pedigrees, which is the sum of TDT and Sib.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>W</code></td>
<td>
<p> Weights used in Weighted rvPDT test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test.v1</code></td>
<td>
<p> Weighted rvPDT test statistic with weights W.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test.v0</code></td>
<td>
<p> Unweighted rvPDT test statistic with weights=1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvalue.v1</code></td>
<td>
<p> P value of weighted rvPDT test statistic (test.v1).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvalue.v0</code></td>
<td>
<p> P value of unweighted rvPDT test statistic (test.v0).</p>
</td>
</tr>
</table>
<h3>References</h3>

    
<p>Guo W , Shugart YY, Does Haplotype-based Collapsing Tests Gain More Power than Variant-based Collapsing Tests for Detecting Rare Variants in Pedigrees  
(manuscript). 
</p>


</div>