<div class="container">

<table style="width: 100%;"><tr>
<td>process</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Optimization wrapper, maximizing either the joint model of total (TReC) and allele specific (ASE) counts or just TReC</h2>

<h3>Description</h3>

<p>Performs optimization of one of four combinations: joint TReC and ASE or just TReC for autosome or X chromosome and tests with lrt test several hypotheses:
additive, parent of origin, dominance, consistency of TreC and ASE additive effect, ASE only additive effect, 
sex, sex specific additive, dominance deviation for males.
</p>


<h3>Usage</h3>

<pre><code class="language-R">process(rc, hessian=FALSE)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>rc</code></td>
<td>
<p>an object of class readCounts.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hessian</code></td>
<td>
<p>a flag whether Hessian matrix for these genes should be calculated, by default set to FALSE</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a list of following matrices (if there is only one sex, only the relevant tests and matrices are outputed) :
</p>
<table>
<tr style="vertical-align: top;">
<td><code>pvals</code></td>
<td>
<p>matrix of p-values from description for each gene corresponding row</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coef.full</code></td>
<td>
<p>matrix of full model fit coefficients, -log(likelihood at these coefficients),phi, theta (2 overdispersion parameters used)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coef.add</code></td>
<td>
<p>matrix of additive restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coef.poo</code></td>
<td>
<p>matrix of parent of origin restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coef.dom</code></td>
<td>
<p>matrix of dominance restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coef.same</code></td>
<td>
<p>matrix of TReC=ASE additive restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coef.ase.add</code></td>
<td>
<p>matrix of ASE additive restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coef.sex</code></td>
<td>
<p>matrix of sex restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coef.sex.add</code></td>
<td>
<p>matrix of sex specific additive restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coef.dev.dom</code></td>
<td>
<p>matrix of dominance deviation for male restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>errorlist</code></td>
<td>
<p>a list of errors</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hess.lst</code></td>
<td>
<p>a list of heassian matrices, if parameter <b>hessian</b> is set to TRUE</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p> Vasyl Zhabotynsky vasyl@unc.edu </p>


<h3>See Also</h3>

<p><code>get.tausB</code>,<code>nLogLik</code>, <code>data.X</code>, <code>data.A</code>, <code>rcA</code>, <code>readCounts</code>.</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# fitting X chromosome data example, for a full model, i.e. assuming we have allele specific reads:
trecase.A.out = process(rc=rcA)
names(trecase.A.out)
trecase.A.out$pval

#alternatively for X chromosome:
trecase.X.out = process(rc=rcX)
names(trecase.X.out)
trecase.X.out$pval

## End(Not run)</code></pre>


</div>