<div class="container">

<table style="width: 100%;"><tr>
<td>simRX</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Produce simulated counts</h2>

<h3>Description</h3>

<p>This function is producing simulated counts for the joint model with Negative-Binomial distribution for TReC and Beta-Binomial for ASE counts. The simulated dataset should be reformatted to readCounts format to be used for optimization.
</p>


<h3>Usage</h3>

<pre><code class="language-R">simRX(b0f, b0m, b1f, b1m, beta_sex, beta_dom, beta_k=1, phi=1, theta=1, n=6, 
      mean.base.cnt=50, range.base.cnt=60, perc.ase=.35, n.simu=1E4, 
      is.X=FALSE, tauB=NULL, seed=NULL)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>b0f</code></td>
<td>
<p>a female additive strain effect</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b0m</code></td>
<td>
<p>a male additive strain effect</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b1f</code></td>
<td>
<p>a female parent of origin effect</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b1m</code></td>
<td>
<p>a male parent of origin effect</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta_sex</code></td>
<td>
<p>a sex effect</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta_dom</code></td>
<td>
<p>a dominance effect</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta_k</code></td>
<td>
<p>an effect associated with the library size kappas</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phi</code></td>
<td>
<p>a Negative-Binomial overdispersion, default value is 1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theta</code></td>
<td>
<p>a Beta-Binomial overdispersion, default value is 1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>a vector defining number of mice in each cross, default value is 6</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mean.base.cnt</code></td>
<td>
<p>a target expected number of counts for the base group (with no effects), default value is 50</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>range.base.cnt</code></td>
<td>
<p>a range in which the expected number of counts  for the base group will vary, default value is 60</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perc.ase</code></td>
<td>
<p>a percent reads that are allele-specific, default value is 35%</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.simu</code></td>
<td>
<p>a number of simulations, default value is 1E4</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is.X</code></td>
<td>
<p>a flag if the value to be simulated is X for chromosome (otherwise autosome), default value is FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tauB</code></td>
<td>
<p>a value describing allelic imbalance - <em>Xce</em> effect for the cross, default value is NULL, in which case 50% will be simulated</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>a random seed to be set, no set by default.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>output - 3 matrices with one row - one gene, one column - one mouse: 
</p>
<table>
<tr style="vertical-align: top;">
<td><code>index</code></td>
<td>
<p>vector defining the cross of the mouse, female - AB=1, BA=2, AA=3, BB=4, and male - AB=5, BA=6, AA=7, BB=8.
If mice are of only one sex, AB=1, BA=2, AA=3, BB=4.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>A matrix of total read counts</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>A matrix of allele specific counts</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n0B</code></td>
<td>
<p>A matrix of allele specific counts associated with allele B</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kappas</code></td>
<td>
<p>Offset parameter, given as overall TReC for the mouse.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tausB</code></td>
<td>
<p>In case of the simulating X chromosome the provided <em>Xce</em> effect is returned: expression of allele B relative to the overall allele specific count for each mouse. </p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p> Vasyl Zhabotynsky vasyl@unc.edu </p>


<h3>See Also</h3>

<p><code>process</code>,<code>readCounts</code>.</p>


<h3>Examples</h3>

<pre><code class="language-R"># simulating autosomal data:
dat.A = simRX(b0f=.5, b0m=.6, b1f=.3, b1m=.4, beta_sex=.1, beta_dom=.1, n.simu=1E1)
names(dat.A)
# simulating autosomal data:
dat.X = simRX(b0f=.5, b0m=.6, b1f=.3, b1m=.4, beta_sex=.1, beta_dom=.1, n.simu=1E1, 
              is.X=TRUE, tauB=.3)
names(dat.X)
</code></pre>


</div>