<div class="container">

<table style="width: 100%;"><tr>
<td>data.A</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Sample data example for autosomal genes</h2>

<h3>Description</h3>

<p>This data set provides with example of experimental data for a subset of autosomal genes.
The full model requires a combination of total read counts (<b>y</b>) - all the reads belonging for a gene, 
and finding out which of these reads we can specifically attribute to allele A or allele B - allele specific counts (<b>n</b>),
separately the reads attributed specifically to allele B (<b>n0B</b>).
Also, it includes the other data pieces to fit the model: <b>kappas</b> - total number of counts for each mouse, on log scale, 
index - specifying which cross each mouse belongs to, and geneids - Ensembl ids of genes.
They, as well as the datasets simulated with <b>simRX</b> can be fitted using <b>proc.trecase.A</b> or <b>proc.trec.A</b>.
</p>


<h3>Value</h3>

  
<table>
<tr style="vertical-align: top;">
<td><code>index</code></td>
<td>
<p>vector defining the cross of the mouse, female - AB=1, BA=2, AA=3, BB=4, and male - AB=5, BA=6, AA=7, BB=8.
If mice are of only one sex, AB=1, BA=2, AA=3, BB=4.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>matrix of TReC counts. Note, the expected input assumes that inbred mice will be in the last columns of the table, after the last F1 mouse.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>matrix of ASE counts for corresponding F1 mouse (classes 1, 2, 5, 6) for corresponding genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n0B</code></td>
<td>
<p>matrix of ASE counts belonging for allele B, for correponding genes and mice as in <b>n</b>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kappas</code></td>
<td>
<p>A parameter, specifying log(overall TReC) for each mouse.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneids</code></td>
<td>
<p>ids of genes, if not provided, rownames of the matrix <b>y</b> will be used</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p> Vasyl Zhabotynsky vasyl@unc.edu </p>


<h3>See Also</h3>

<p><code>process</code>,<code>readCounts</code>.</p>


<h3>Examples</h3>

<pre><code class="language-R"># see total read counts (TReC) for first 2  autosomal genes of a data example:
data.A$y[1:2,]
</code></pre>


</div>